---
title: "TE calculation"
author: "Kate"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r libraries}
library(here)
library(janitor)
library(tximport)
library(tidyverse)
```

```{r colors}
a209_col = "#009E73"
si3e_col = "#56B4E9"
si3d_col = "#E69F00"
```

### Import salmon count files

```{r load metadata}

metadata <- read_csv(here("accessories","metadata_NM.csv"), skip = 0) %>% 
  clean_names()

metadata <- metadata %>% 
  mutate(library = ifelse(library_prep_kit=="Qiagen_miRNA", "riboseq", "rnaseq"))

#ribosome lysate = ribosome seq
#rRNA depletion = RNA seq

```


```{r import gene info}

#import key on RNA type, remove premRNA
geneInfo <- read_csv(here("accessories","gencode.v26.primary.info.csv.gz"), col_names = F) %>%
  filter(!str_detect(X2, "pre_"))

colnames(geneInfo) <- c("gene_id","transcript_id","biotype","symbol")

#import transcripts for gene to transcript mapping
tx2gene <- geneInfo[,c(2,1)]
gene_symbol <- geneInfo %>% dplyr::select(gene_id, symbol) %>% unique()

```

```{r import salmon files}

#name files to import
myquantfiles <- paste(here(), "/salmon/",
                      metadata$sample_id,
                      "/quant.sf.gz",
                      sep = "")

names(myquantfiles) <- paste(metadata$genotype, metadata$treatment1, metadata$treatment2, metadata$library, sep = "_")

#import salmon files
myTxi <- tximport(myquantfiles, type = "salmon", tx2gene = tx2gene)

# visualize distribution of counts
hist_count <- hist(log2(rowSums(myTxi$counts)), breaks = 50)
#determine threshold

#distribution of abundance histogram
hist_abundance <- hist(log2(rowSums(myTxi$abundance)), breaks = 50)

#@NEEL i am using a cutoff of 10 here. previously i used a cutoff of 7. Numbers will be slightly different. what do you think.
#only keeping rows with counts  log2> 7 across all samples - want around 10-15k genes - want genes that are highly expressed
keepGenes <- rownames(myTxi$counts[log2(rowSums(myTxi$counts)) > 10 ,])

#removes noise and keeps only highly expressed genes
# results show bimodal distribution
#1 RNA per cell = 1 TPM

```

QC is in import_qc file

```{r keep only protein coding genes}

pc <- geneInfo %>%
  filter(biotype =="protein_coding") %>%
  pull(gene_id) %>%
  unique()

pc_keep <- pc[pc %in% keepGenes]

# remove non-polyadenylated mRNAs
nonpolyA <- readxl::read_xlsx(path = here("accessories","polyAminus.xlsx"), sheet = 1) %>% 
   pull(Gene)

nonpolyA_ID <- geneInfo %>%
  filter(symbol %in% nonpolyA) %>% 
  pull(gene_id) %>% 
  unique()

# we want to remove any genes that match these annotation categories # before we do DE analysis
blacklist_smRNA <- c("snoRNA","miRNA","snRNA","rRNA","Mt_tRNA","scaRNA","Mt_tRNA","Mt_rRNA")

blacklist_smRNA_ID <- geneInfo %>%
  filter(biotype %in% blacklist_smRNA) %>% 
  pull(gene_id) %>% 
  unique()

genes_to_kill <- unique(c(blacklist_smRNA_ID,nonpolyA_ID))
```


```{r filter and extract counts}

allCounts <- myTxi$counts %>%
  as.data.frame() %>% 
  rownames_to_column(var = "gene") %>% 
  filter(gene %in% pc_keep) %>%
  filter(!gene %in% genes_to_kill) %>%
  column_to_rownames(var = "gene") %>% 
  as.matrix() %>%
  round(.,0)

allCountsNorm <- edgeR::cpm(allCounts[,order(colnames(allCounts))])

riboN <- log2(1+ as.matrix(allCountsNorm[,grep(pattern = "ribo",colnames(allCountsNorm))]))
rnaN <- log2(1 + as.matrix(allCountsNorm[,grep(pattern = "rna",colnames(allCountsNorm))])) 

TE_original <- riboN - rnaN

hist(TE_original)

#center data --

TE <- (riboN - rnaN) + 0.7616566

hist(TE)

colnames(TE) <- gsub(pattern = "ribo", replacement = "TE", x = colnames(TE))

gghistogram(as.data.frame(TE_original) %>% melt(), x="value")

median(as.data.frame(TE_original) %>% melt() %>% pull(value))
# this is why im using 0.761! -- all changes are relative

gghistogram(as.data.frame(TE) %>% melt(), x="value")
ribo_counts <- as.matrix(allCounts[,grep(pattern = "ribo",colnames(allCounts))])
rna_counts <- as.matrix(allCounts[,grep(pattern = "rna",colnames(allCounts))])

```

### Organize metadata
Remove outlier, determine factor levels, rename variables

```{r}
#rename variables remove outlier
metadata <- metadata %>% 
  filter(sample_id != "NM2023_0113") %>%
  dplyr::rename("treatment" = "treatment1") %>% 
  dplyr::rename("rep" = "treatment2") %>% 
  dplyr::rename("condition" = "genotype")

#edit factor levels
metadata$library <- relevel(x = factor(metadata$library), ref = "rnaseq")
metadata$library <- recode(metadata$library, rnaseq = "rna", riboseq = "ribo")

metadata$treatment <- relevel(x = factor(metadata$treatment), ref = "sictrl")
metadata$condition <- relevel(x = factor(metadata$condition), ref = "Normoxic")
metadata$condition <- recode(metadata$condition, Normoxic = "normoxia", Hypoxic = "hypoxia")
metadata$rep <- recode(metadata$rep, RepA = "a", RepB = "b", RepC = "c")

metadata <- metadata %>%
dplyr::select(sample_id, library, treatment, condition, rep) 

# write metadata for use in future analyses

write_csv(metadata, here("calculate_te/metadata.csv"))
```

# Calculate TE

I need TE, rna, and ribo across the board in one dataset and for each condition
TE
plus TE ribo
plus TE rna
plus symbol
```{r}

full_te_list <- TE %>% 
  as.data.frame() %>%
  rownames_to_column("gene_id") %>% 
  left_join(gene_symbol, by="gene_id") %>% 
  select(symbol, everything())


full_rna_list <- rnaN %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id")


full_ribo_list <- riboN %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id")

full_te_list <- full_te_list %>% 
  left_join(full_rna_list, by="gene_id")

complete_list <- full_te_list %>% 
  left_join(full_ribo_list, by= "gene_id")

#export complete list

write_csv(complete_list, here("counts/complete_list.csv"))


```

Example- ATF4
```{r}
atf4_list <- complete_list %>% 
  filter(symbol=="ATF4") %>% 
  melt()


ggplot(atf4_list, aes(x=variable, y=value))+
  geom_point()
```




# Normoxia - siRNA knockdowns

```{r}



```

