---
title: "EIF3E and EIF3D-mediated translational regulation in normoxia"
author: "`r Sys.Date()`"
date: ""
output: html_document
editor_options: 
  chunk_output_type: console
---

# Figure 1: Overlap between functions of eIF3e and eIF3d in normoxia

```{r setup, echo=FALSE}
knitr::opts_chunk$set(echo = FALSE, message=FALSE, warning=FALSE)
```

```{r install packages, echo=FALSE}

library(here)
library(janitor)
library(tximport)
library(tidyverse)
library(EnhancedVolcano)
library(msigdbr)
library(viridis)
library(gridExtra)
library(ggthemes)
library(ggrepel)
library(scales)
library(DESeq2)
library(limma)
library(clusterProfiler)
library(plotly)
library(ggokabeito)
library(ggvenn)
library(ggpubr)
library(DT)
library(edgeR)
library(ggprism)
library(ggExtra)
library(reshape2)

```

```{r}
a209_col = "#009E73"
si3e_col = "#56B4E9"
si3d_col = "#E69F00"

theme(axis.title.y=element_text(size=11),
      axis.title.x=element_text(size=11))

```

## Load and organize metadata
```{r metadata}
# metadata <- read_csv(here("accessories","metadata_NM.csv"), skip = 0) %>% 
#   clean_names() %>% 
#   mutate(library = ifelse(library_prep_kit=="Qiagen_miRNA", "riboseq", "rnaseq"))
# 
# metadata <- metadata %>% 
#   filter(sample_id != "NM2023_0113") %>%
#   dplyr::rename("treatment" = "treatment1") %>% 
#   dplyr::rename("rep" = "treatment2") %>% 
#   dplyr::rename("condition" = "genotype")
# 
# metadata$library <- relevel(x = factor(metadata$library), ref = "rnaseq")
# metadata$library <- recode(metadata$library, rnaseq = "rna", riboseq = "ribo")
# 
# metadata$treatment <- relevel(x = factor(metadata$treatment), ref = "sictrl")
# metadata$condition <- relevel(x = factor(metadata$condition), ref = "Normoxic")
# metadata$condition <- recode(metadata$condition, Normoxic = "normoxia", Hypoxic = "hypoxia")
# metadata$rep <- recode(metadata$rep, RepA = "a", RepB = "b", RepC = "c")
# 
# metadata <- metadata %>%
# dplyr::select(sample_id, library, treatment, condition, rep) 
# 
# # write metadata for use in future analyses
# 
# write_csv(metadata, here("counts/fig1/metadata.csv"))
#   
```

## load metadata
```{r}
metadata <- read_csv(here("calculate_te/metadata.csv"))
```

## filter metadata for samples of interest

```{r}
metadata_norm <- metadata %>% 
  filter(treatment %in% c("sictrl","si3d","si3e")) %>%
  filter(condition == "normoxia")
```

## add gene id to symbol key

```{r}
geneInfo <- read_csv(here("accessories","gencode.v26.primary.info.csv.gz"), col_names = F) %>%
  filter(!str_detect(X2, "pre_"))

colnames(geneInfo) <- c("gene_id","transcript_id","biotype","symbol")

#map transcripts to gene
tx2gene <- geneInfo[,c(2,1)]
gene_symbol <- geneInfo %>% dplyr::select(gene_id, symbol) %>% unique()

```




## load salmon files - remove
```{r import and filter, message=FALSE, warning=FALSE}
# create the tx2 gene file for tximport
geneInfo <- read_csv(here("accessories","gencode.v26.primary.info.csv.gz"), col_names = F) %>%
  filter(!str_detect(X2, "pre_"))

colnames(geneInfo) <- c("gene_id","transcript_id","biotype","symbol")

#map transcripts to gene
tx2gene <- geneInfo[,c(2,1)]
gene_symbol <- geneInfo %>% dplyr::select(gene_id, symbol) %>% unique()

#load quant files based on metadata names
myquantfiles <- paste(here(), "/salmon/",
                      metadata_norm$sample_id,
                      "/quant.sf.gz",
                      sep = "")

names(myquantfiles) <- paste(metadata_norm$condition,
                             metadata_norm$treatment,
                             metadata_norm$library,
                             metadata_norm$rep,
                             sep = "_")


myTxi <- tximport(myquantfiles, type = "salmon", tx2gene = tx2gene)



#only keeping rows with enough counts - want around 10-15k genes

# hist(log2(rowSums(myTxi$counts)), breaks = 40)
keepGenes <- rownames(myTxi$counts[log2(rowSums(myTxi$counts)) > 7,])

pc <-geneInfo %>%
  filter(biotype =="protein_coding") %>%
  pull(gene_id) %>%
  unique()
pc_keep <- pc[pc %in% keepGenes]


# remove non-polyadenylated mRNAs
nonpolyA <- readxl::read_xlsx(path = here("accessories","polyAminus.xlsx"), sheet = 1) %>% 
   pull(Gene)

nonpolyA_ID <- geneInfo %>%
  filter(symbol %in% nonpolyA) %>% 
  pull(gene_id) %>% 
  unique()


# we want to remove any genes that match these annotation categories # before we do DE analysis
blacklist_smRNA <- c("snoRNA","miRNA","snRNA","rRNA","Mt_tRNA","scaRNA","Mt_tRNA","Mt_rRNA")

# there are 5,355 genes we want to exclude
blacklist_smRNA_ID <-geneInfo %>%
  filter(biotype %in% blacklist_smRNA) %>% 
  pull(gene_id) %>% 
  unique()


genes_to_kill <- unique(c(blacklist_smRNA_ID,nonpolyA_ID))


```

## extract rna, ribo, and te counts - remove
remove
```{r deltaTE, message=FALSE, warning=FALSE}
# separate out RNA counts versus ribo count
# from https://currentprotocols.onlinelibrary.wiley.com/doi/full/10.1002/cpmb.108

allCounts <- myTxi$counts %>%
  as.data.frame() %>% 
  rownames_to_column(var = "gene") %>% 
  filter(gene %in% pc_keep) %>%
  filter(!gene %in% genes_to_kill) %>%
  column_to_rownames(var = "gene") %>% 
  as.matrix() %>%
  round(.,0)

allCountsNorm <- edgeR::cpm(allCounts[,order(colnames(allCounts))])

riboN <- log2(
  1+ as.matrix(allCountsNorm[,grep(pattern = "ribo",colnames(allCountsNorm))])
) 
rnaN <- log2( 1 + as.matrix(allCountsNorm[,grep(pattern = "rna",colnames(allCountsNorm))])
) 

# as.data.frame(riboN)%>% melt() %>% view()

TE_original <- riboN - rnaN

# @kate wondering why you are adding 0.828 to TE  --- this centers TE at 0. is this important?

TE <- (riboN - rnaN) + 0.828 
colnames(TE) <- gsub(pattern = "ribo", replacement = "TE", x = colnames(TE))

gghistogram(as.data.frame(TE_original) %>% melt(), x="value")

median(as.data.frame(TE_original) %>% melt() %>% pull(value))
# this is why im using 0.828! BUT i still don't know why that is necessary.

gghistogram(as.data.frame(TE) %>% melt(), x="value")





ribo_counts <- as.matrix(allCounts[,grep(pattern = "ribo",colnames(allCounts))])
rna_counts <- as.matrix(allCounts[,grep(pattern = "rna",colnames(allCounts))])

# remove 3d libs
counts_3e <- allCounts[,!grepl("si3d",colnames(allCounts))]
metadata_norm_3e <- metadata_norm %>% filter(treatment != "si3d") %>% droplevels()

# remove 3e libs
counts_3d <- allCounts[,!grepl("si3e",colnames(allCounts))]
metadata_norm_3d <- metadata_norm %>% filter(treatment != "si3e") %>% droplevels()

ddsMat_3d <- DESeqDataSetFromMatrix(
  countData = counts_3d,
  colData = metadata_norm_3d,
  design = ~treatment + library + treatment:library
  )

ddsMat_3d <- DESeq(ddsMat_3d)
resultsNames(ddsMat_3d)
res_te_3d <- results(ddsMat_3d, name=resultsNames(ddsMat_3d)[4], cooksCutoff=F, independentFiltering=F)


res_te_3d <- lfcShrink(dds = ddsMat_3d, coef = resultsNames(ddsMat_3d)[4], res = res_te_3d) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>%
  filter(!symbol %in% nonpolyA) 


ddsMat_3e <- DESeqDataSetFromMatrix(
  countData = counts_3e,
  colData = metadata_norm_3e,
  design = ~treatment + library + treatment:library
  )

ddsMat_3e <- DESeq(ddsMat_3e)

res_te_3e <- results(ddsMat_3e,
                name=resultsNames(ddsMat_3e)[4],
                cooksCutoff=F, independentFiltering=F)

res_te_3e <- lfcShrink(
  dds = ddsMat_3e,
  coef = resultsNames(ddsMat_3e)[4],
  res = res_te_3e) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>%
  filter(!symbol %in% nonpolyA)  
```

## Separate ribo counts, rna counts, and te counts
```{r}
raw_counts <- read_csv(here("calculate_te/raw_filtered_allCounts.csv")) %>% 
  column_to_rownames("gene_id")


raw_counts_edit <- raw_counts %>%
  dplyr::select(!matches("DMSO|a209"))

ribo_counts <- as.matrix(raw_counts_edit[,grep(pattern = "ribo",colnames(raw_counts_edit))])
rna_counts <- as.matrix(raw_counts_edit[,grep(pattern = "rna",colnames(raw_counts_edit))])


rna_ribo_counts <- as.matrix(raw_counts_edit)


```

## Delta TE eIF3d KD vs. sicontrol (normoxia only here and everything below)

```{r}
#organize counts
# rna_ribo_3d <- rna_ribo_counts[,!grepl("si3e",colnames(rna_ribo_counts))]

#organize metadata
metadata_norm_3d <- metadata_norm %>% 
  filter(treatment== "sictrl" | treatment=="si3d") %>% 
  filter(condition=="normoxia") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_norm_3d$library <- relevel(factor(metadata_norm_3d$library), ref = "rna")
metadata_norm_3d$treatment <- relevel(factor(metadata_norm_3d$treatment), ref = "sictrl")

levels(metadata_norm_3d$library)
levels(metadata_norm_3d$treatment)

names <- as.character(metadata_norm_3d$name)

comparison_3d <- rna_ribo_counts[, colnames(rna_ribo_counts) %in% names]


ddsMat_3d <- DESeqDataSetFromMatrix(
  countData = comparison_3d,
  colData = metadata_norm_3d,
  design = ~treatment + library + treatment:library
  )

ddsMat_3d <- DESeq(ddsMat_3d)
resultsNames(ddsMat_3d)
res_te_3d <- results(ddsMat_3d, name=resultsNames(ddsMat_3d)[4], cooksCutoff=F, independentFiltering=F)


res_te_3d <- lfcShrink(dds = ddsMat_3d, coef = resultsNames(ddsMat_3d)[4], res = res_te_3d) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)
  # filter(!symbol %in% nonpolyA)  - check if this is necessary, should already be filtered

write_csv(res_te_3d, here("counts/fig1/normoxia_sictrl_3d.csv"))
```

## Delta TE eIF3e KD vs. sicontrol (normoxia only here and everything below)

```{r}
#organize counts
# rna_ribo_3e <- rna_ribo_counts[,!grepl("si3e",colnames(rna_ribo_counts))]

#organize metadata
metadata_norm_3e <- metadata_norm %>% 
  filter(treatment== "sictrl" | treatment=="si3e") %>% 
  filter(condition=="normoxia") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_norm_3e$library <- relevel(factor(metadata_norm_3e$library), ref = "rna")
metadata_norm_3e$treatment <- relevel(factor(metadata_norm_3e$treatment), ref = "sictrl")

levels(metadata_norm_3e$library)
levels(metadata_norm_3e$treatment)

names <- as.character(metadata_norm_3e$name)

comparison_3e <- rna_ribo_counts[, colnames(rna_ribo_counts) %in% names]


ddsMat_3e <- DESeqDataSetFromMatrix(
  countData = comparison_3e,
  colData = metadata_norm_3e,
  design = ~treatment + library + treatment:library
  )

ddsMat_3e <- DESeq(ddsMat_3e)
resultsNames(ddsMat_3e)
res_te_3e <- results(ddsMat_3e, name=resultsNames(ddsMat_3e)[4], cooksCutoff=F, independentFiltering=F)


res_te_3e <- lfcShrink(dds = ddsMat_3e, coef = resultsNames(ddsMat_3e)[4], res = res_te_3e) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)
  # filter(!symbol %in% nonpolyA) 


write_csv(res_te_3e, here("counts/fig1/normoxia_sictrl_3e.csv"))



```

## Import TE 
Calculated previously in calculate_te.Rmd
allgene TE has normalized counts
```{r}
all_gene_te <- read_csv(here("calculate_te/complete_te_list.csv"))

all_gene_te_long <- all_gene_te %>% 
  melt() %>% 
  separate(variable, into = c("condition", "treatment", "rep"))

slimmed_te <- all_gene_te_long %>% 
  filter(condition=="normoxia") %>% 
  filter(treatment != "a209") %>% 
  filter(treatment != "DMSO")


```

## Look at TE for example genes and how they change with 3e and 3d KD

```{r}

d_genes <- c("CLDN9", "CLIC1", "PRDM4")
e_genes <- c("APLP2", "PABPC1", "TOR1AIP2")
common <- c("ATF4", "EIF4EBP2", "GSK3B", "MED13L", "PCBP2", "XBP1")

```

Plot genes - common
```{r}

slimmed_te$treatment <- relevel(factor(slimmed_te$treatment), ref = "sictrl")

common_te <- slimmed_te %>% 
  filter(symbol %in% common) %>% 
  filter(type=="TE")


common_plot <-
ggstripchart(data = common_te, x = "treatment", y = "value", color = "treatment", facet.by = "symbol", scales="free_y", palette = c("black",si3d_col,si3e_col), position = position_jitterdodge(1.2)) + 
  ylab(expression("log"[2]*" TE (ribo/rna)"))
  
common_plot

ggsave(width = 5.5, height = 3.75, filename = here("plots/fig1/commongenes.pdf"))


```

Plot d only

```{r}
d_te <- slimmed_te %>% 
  filter(symbol %in% d_genes) %>% 
  filter(type=="TE")


d_plot <-
ggstripchart(data = d_te, x = "treatment", y = "value", color = "treatment", facet.by = "symbol", scales="free_y", palette = c("black",si3d_col,si3e_col), position = position_jitterdodge(1.2)) + 
  ylab(expression("log"[2]*" TE (ribo/rna)"))
  
d_plot

ggsave(width = 5.5, height = 2.5, filename = here("plots/fig1/dgenes.pdf"))

```


Plot e only

```{r}
e_te <- slimmed_te %>% 
  filter(symbol %in% e_genes) %>% 
  filter(type=="TE")


e_plot <-
ggstripchart(data = e_te, x = "treatment", y = "value", color = "treatment", facet.by = "symbol", scales="free_y", palette = c("black",si3d_col,si3e_col), position = position_jitterdodge(1.2)) + 
  ylab(expression("log"[2]*" TE (ribo/rna)"))
  
e_plot

ggsave(width = 5.5, height = 2.5, filename = here("plots/fig1/egenes.pdf"))

```



Visualize e and d specifically: controls

```{r}
e_and_d <-slimmed_te %>% 
  filter(symbol=="EIF3E" | symbol=="EIF3D")

ggstripchart(data = e_and_d, x = "treatment", y = "value", color = "treatment", facet.by = "symbol", scales="free_y", palette = c("black",si3d_col,si3e_col), position = position_jitterdodge(1.2))+
  facet_wrap(~symbol+type)


```




## Venn diagram of 3e/d gene overlaps in normoxia
```{r compare genes}
te_3d_up <- res_te_3d %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

te_3d_down <- res_te_3d %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)

te_3e_up <- res_te_3e %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

te_3e_down <- res_te_3e %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)


norm_te_up <-
  list(
    te_3d_up = res_te_3d %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id),
  te_3e_up = res_te_3e %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)
  )

norm_te_down <-
  list(
    te_3d_down = res_te_3d %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id),
  te_3e_down = res_te_3e %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)
  )


ggvenn(norm_te_up, auto_scale = T, fill_color = c(si3d_col,si3e_col), text_size = 2)

ggsave(width = 2, height = 2, filename = here("plots/fig1/vennup.pdf"))


ggvenn(norm_te_down, auto_scale = T, fill_color = c(si3d_col,si3e_col), text_size = 2)

ggsave(width = 2, height = 2, filename = here("plots/fig1/venndown.pdf"))


# stopifnot(
# identical(res_te_3d$gene_id,res_te_3e$gene_id))
```

### Density scatter plot
```{r}
norm_sig_te <- data.frame(
  gene_id = res_te_3d$gene_id,
  symbol = res_te_3d$symbol,
  te_3d = res_te_3d$log2FoldChange,
  te_3e = res_te_3e$log2FoldChange
) %>%
  filter(gene_id %in% c(te_3e_up,
                        te_3e_down,
                        te_3d_up,
                        te_3d_down)) 

scatter <-
ggscatter(data = norm_sig_te,
          x = "te_3d",
          y = "te_3e",
          xlab= "si-eIF3d TE",
          ylab="si-eIF3e TE",
          alpha = 0.25,
          title = "TE changes with 3d KD vs 3e KD", 
          xlim=c(-2,2), ylim = c(-2,2),
  add = "reg.line")+
   stat_cor(
   aes(label =
         paste(
           after_stat(rr.label),
           ..p.label..,
           sep = "~`,`~")
       ) + theme_prism()
   )+
  stat_regline_equation(
    aes(label =  paste(..eq.label.., after_stat(rr.label), sep = "~~~~")),
    label.x = -1.5, label.y = 1.5
  ) +
  geom_vline(xintercept=0, linetype='dotted') +
  geom_hline(yintercept=0, linetype='dotted') +
  theme_prism() +
  guides(color = FALSE)

scatter

#ggplotly(scat, tooltip="all")

ggsave(width = 5, height = 5, filename = here("plots/fig1/normoxia_quadrants.pdf"))

```


## Pathway analysis setup
```{r prep GSEA analysis}

eif3d_te <- res_te_3d$log2FoldChange
names(eif3d_te) <- res_te_3d$symbol

eif3d_te <- sort(eif3d_te, decreasing = TRUE)

eif3d_te <- eif3d_te[!duplicated(names(eif3d_te))]

eif3e_te <- res_te_3e$log2FoldChange
names(eif3e_te) <- res_te_3e$symbol

eif3e_te <- sort(eif3e_te, decreasing = TRUE)

eif3e_te <- eif3e_te[!duplicated(names(eif3e_te))]


m_df_h <- msigdbr(species = "Homo sapiens") %>%
  filter(gs_cat == "H") %>% 
  dplyr::select(gs_name, gene_symbol)

m_df_c2 <- msigdbr(species = "Homo sapiens") %>%
  filter(gs_cat == "C2" & gs_subcat != "CGP") %>% 
  dplyr::select(gs_name, gene_symbol)

```

### Hallmark pathways
```{r GSEA Hallmarks}

eif3d_te_h <- GSEA(
  geneList = eif3d_te,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )

eif3e_te_h <- GSEA(
  geneList = eif3e_te,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )

norm_te_h <- dplyr::full_join(
  eif3d_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3d_", .), is.numeric),
  eif3e_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3d_pvalue < 0.05 | eif3e_pvalue < 0.05)  %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = eif3e_NES - eif3d_NES)


norm_te_h$ID <- gsub(pattern = "HALLMARK_", replacement = "", x = norm_te_h$ID)

norm_te_h <- norm_te_h %>% 
  mutate(color = ifelse(ID=="HYPOXIA", "red", "black"))

ggscatter(data = norm_te_h,
          x = "eif3d_NES",
          y = "eif3e_NES",
          color = "color", palette = c("black", "red"),
          xlab = "si-eIF3d NES",
          ylab = "si-eIF3e NES") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5, 2.5) +
  xlim(-2.5, 2.5) +
  theme_prism(base_size = 12) +
  guides(color = "none") +
  geom_text_repel(aes(label = ID), size = 3)

ggsave(width = 4, height = 4, filename = here("plots/fig1/normoxia_halmarks.pdf"))

norm_h_table <- norm_te_h %>% 
  mutate(direction = ifelse(eif3d_NES < 0 & eif3e_NES < 0, "down_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES > 0, "up_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES < 0, "up_3d_down_3e",
                            ifelse(eif3d_NES < 0 & eif3e_NES > 0, "up_3e_down_3d", "NA")))))

write_csv(norm_h_table, here("counts/fig1/norm_h_table.csv"))

```

## Visualize a different way. - bars

```{r}


norm_te_h_3d <-
  eif3d_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>% 
  mutate(sample="eIF3d")

norm_te_h_3e <-
  eif3e_te_h@result %>% 
  dplyr::select(ID,NES,pvalue,qvalue) %>% 
  mutate(sample="eIF3e")

norm_hallmark <- rbind(norm_te_h_3d, norm_te_h_3e) %>% 
  mutate(significant = case_when(
    qvalue < 0.05 ~"*",
    .default=""
  ))

pathways_to_include <- norm_hallmark %>% 
  filter(significant=="*")

pathways_to_include <- pathways_to_include$ID

norm_hallmark %>%
  filter(ID %in% pathways_to_include) %>%
  ggplot(aes(x = ID, y = NES, fill = sample)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.9)) +
  geom_text(aes(label = significant), 
            position = position_dodge(width = 0.9), 
            hjust = -0.2) +  # Adjust hjust for better alignment
  theme_cowplot() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
  coord_flip()






norm_te_h <- dplyr::full_join(
  eif3d_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3d_", .), is.numeric),
  eif3e_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3d_pvalue < 0.05 | eif3e_pvalue < 0.05)  %>%
  dplyr::arrange(-eif3e_NES) 


ribo_counts <- as.matrix(raw_counts_edit[,grep(pattern = "ribo",colnames(raw_counts_edit))])
rna_counts <- as.matrix(raw_counts_edit[,grep(pattern = "rna",colnames(raw_counts_edit))])

long_hall <-
norm_te_h %>% 
  melt()



ggbarplot(data = norm_te_h,
          x = "eif3d_NES",
          y = "eif3e_NES",
          color = "color", palette = c("black", "red"),
          xlab = "si-eIF3d NES",
          ylab = "si-eIF3e NES") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5, 2.5) +
  xlim(-2.5, 2.5) +
  theme_prism(base_size = 12) +
  guides(color = "none") +
  geom_text_repel(aes(label = ID), size = 3)
```








### KEGG pathways

```{r GSEA KEGG}
eif3d_te_kegg <- GSEA(
  geneList = eif3d_te,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^KEGG', gs_name))
  )

eif3e_te_kegg <- GSEA(
  geneList = eif3e_te,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^KEGG', gs_name))
  )

norm_te_kegg <- dplyr::full_join(
  eif3d_te_kegg@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3d_", .), is.numeric),
  eif3e_te_kegg@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3d_qvalue < 0.05 | eif3e_qvalue < 0.05) %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = eif3e_NES - eif3d_NES)

norm_te_kegg$ID <- gsub(pattern = "KEGG_", replacement = "", x = norm_te_kegg$ID)

ggscatter(data = norm_te_kegg,
          x = "eif3d_NES",
          y = "eif3e_NES",
          label = "ID",
          repel = TRUE,
          title = "Kegg - Normoxia",
          font.label = 7) +
  scale_color_continuous(type = "viridis") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5,2.5) +
  xlim(-2.5,2.5) + theme_prism()


DT::datatable(norm_te_kegg) %>%
  formatRound(which(sapply(norm_te_kegg,is.numeric)), digits = 3) 


ggsave(width = 4, height = 4, filename = here("plots/fig1/normoxia_kegg.pdf"))

# @kate -> change this (and any other case of nested ifelse) to using case_when
norm_kegg_table <- norm_te_kegg %>% 
  mutate(direction = ifelse(eif3d_NES < 0 & eif3e_NES < 0, "down_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES > 0, "up_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES < 0, "up_3d_down_3e",
                            ifelse(eif3d_NES < 0 & eif3e_NES > 0, "up_3e_down_3d", "NA")))))

write_csv(norm_kegg_table, here("counts/fig1/norm_kegg_table.csv"))







```


### REACTOME pathways

```{r GSEA REACTOME}
eif3d_te_reactome <- GSEA(
  geneList = eif3d_te,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^REACTOME', gs_name))
  )

eif3e_te_reactome <- GSEA(
  geneList = eif3e_te,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^REACTOME', gs_name))
  )

norm_te_react <- dplyr::full_join(
  eif3d_te_reactome@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3d_", .), is.numeric),
  eif3e_te_reactome@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3d_qvalue < 0.05 | eif3e_qvalue < 0.05)  %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = eif3e_NES - eif3d_NES)

norm_te_react$ID <- gsub(pattern = "REACTOME_", replacement = "", x = norm_te_react$ID)

ggscatter(data = norm_te_react,
          x = "eif3d_NES",
          y = "eif3e_NES",
          label = "ID",
          title = "Reactome - Normoxia",
          repel = TRUE,
          font.label = 6) +
  scale_color_continuous(type = "viridis") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5,2.5) +
  xlim(-2.5,2.5)


DT::datatable(norm_te_react) %>%
  formatRound(which(sapply(norm_te_react,is.numeric)), digits = 3) 


ggsave(width = 4, height = 4, filename = here("plots/fig1/normoxia_reactome.pdf"))


norm_react_table <- norm_te_react %>% 
  mutate(direction = ifelse(eif3d_NES < 0 & eif3e_NES < 0, "down_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES > 0, "up_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES < 0, "up_3d_down_3e",
                            ifelse(eif3d_NES < 0 & eif3e_NES > 0, "up_3e_down_3d", "NA")))))

write_csv(norm_react_table, here("counts/fig1/norm_react_table.csv"))

```
