---
title: "EIF3E and EIF3D-mediated translational regulation in normoxia"
author: "`r Sys.Date()`"
date: ""
output: html_document
editor_options: 
  chunk_output_type: console
---

# Figure 1: Overlap between functions of eIF3e and eIF3d in normoxia

```{r setup, echo=FALSE}
knitr::opts_chunk$set(echo = FALSE, message=FALSE, warning=FALSE)
```

```{r install packages, echo=FALSE}

library(here)
library(janitor)
library(tximport)
library(tidyverse)
library(EnhancedVolcano)
library(msigdbr)
library(viridis)
library(gridExtra)
library(ggthemes)
library(ggrepel)
library(scales)
library(DESeq2)
library(limma)
library(clusterProfiler)
library(plotly)
library(ggokabeito)
library(ggvenn)
library(ggpubr)
library(DT)
library(edgeR)
library(ggprism)
library(ggExtra)

```

```{r}
a209_col = "#009E73"
si3e_col = "#56B4E9"
si3d_col = "#E69F00"

theme(axis.title.y=element_text(size=11),
      axis.title.x=element_text(size=11))

```

## Load and organize metadata
```{r metadata}
metadata <- read_csv(here("accessories","metadata_NM.csv"), skip = 0) %>% 
  clean_names() %>% 
  mutate(library = ifelse(library_prep_kit=="Qiagen_miRNA", "riboseq", "rnaseq"))

metadata <- metadata %>% 
  filter(sample_id != "NM2023_0113") %>%
  dplyr::rename("treatment" = "treatment1") %>% 
  dplyr::rename("rep" = "treatment2") %>% 
  dplyr::rename("condition" = "genotype")

metadata$library <- relevel(x = factor(metadata$library), ref = "rnaseq")
metadata$library <- recode(metadata$library, rnaseq = "rna", riboseq = "ribo")

metadata$treatment <- relevel(x = factor(metadata$treatment), ref = "sictrl")
metadata$condition <- relevel(x = factor(metadata$condition), ref = "Normoxic")
metadata$condition <- recode(metadata$condition, Normoxic = "normoxia", Hypoxic = "hypoxia")
metadata$rep <- recode(metadata$rep, RepA = "a", RepB = "b", RepC = "c")

metadata <- metadata %>%
dplyr::select(sample_id, library, treatment, condition, rep) 

# write metadata for use in future analyses

write_csv(metadata, here("counts/fig1/metadata.csv"))
  
```

## filter metadata for samples of interest

```{r}
metadata_norm <- metadata %>% 
  filter(treatment %in% c("sictrl","si3d","si3e")) %>%
  filter(condition == "normoxia") %>%
  droplevels()
```

## load salmon files
```{r import and filter, message=FALSE, warning=FALSE}
# create the tx2 gene file for tximport
geneInfo <- read_csv(here("accessories","gencode.v26.primary.info.csv.gz"), col_names = F) %>%
  filter(!str_detect(X2, "pre_"))

colnames(geneInfo) <- c("gene_id","transcript_id","biotype","symbol")

#map transcripts to gene
tx2gene <- geneInfo[,c(2,1)]
gene_symbol <- geneInfo %>% dplyr::select(gene_id, symbol) %>% unique()

#load quant files based on metadata names
myquantfiles <- paste(here(), "/salmon/",
                      metadata_norm$sample_id,
                      "/quant.sf.gz",
                      sep = "")

names(myquantfiles) <- paste(metadata_norm$condition,
                             metadata_norm$treatment,
                             metadata_norm$library,
                             metadata_norm$rep,
                             sep = "_")


myTxi <- tximport(myquantfiles, type = "salmon", tx2gene = tx2gene)



#only keeping rows with enough counts - want around 10-15k genes

# hist(log2(rowSums(myTxi$counts)), breaks = 40)
keepGenes <- rownames(myTxi$counts[log2(rowSums(myTxi$counts)) > 7,])

pc <-geneInfo %>%
  filter(biotype =="protein_coding") %>%
  pull(gene_id) %>%
  unique()
pc_keep <- pc[pc %in% keepGenes]


# remove non-polyadenylated mRNAs
nonpolyA <- readxl::read_xlsx(path = here("accessories","polyAminus.xlsx"), sheet = 1) %>% 
   pull(Gene)

nonpolyA_ID <- geneInfo %>%
  filter(symbol %in% nonpolyA) %>% 
  pull(gene_id) %>% 
  unique()


# we want to remove any genes that match these annotation categories # before we do DE analysis
blacklist_smRNA <- c("snoRNA","miRNA","snRNA","rRNA","Mt_tRNA","scaRNA","Mt_tRNA","Mt_rRNA")

# there are 5,355 genes we want to exclude
blacklist_smRNA_ID <-geneInfo %>%
  filter(biotype %in% blacklist_smRNA) %>% 
  pull(gene_id) %>% 
  unique()


genes_to_kill <- unique(c(blacklist_smRNA_ID,nonpolyA_ID))


```

## extract rna, ribo, and te counts

```{r deltaTE, message=FALSE, warning=FALSE}
# separate out RNA counts versus ribo count
# from https://currentprotocols.onlinelibrary.wiley.com/doi/full/10.1002/cpmb.108

allCounts <- myTxi$counts %>%
  as.data.frame() %>% 
  rownames_to_column(var = "gene") %>% 
  filter(gene %in% pc_keep) %>%
  filter(!gene %in% genes_to_kill) %>%
  column_to_rownames(var = "gene") %>% 
  as.matrix() %>%
  round(.,0)

allCountsNorm <- edgeR::cpm(allCounts[,order(colnames(allCounts))])

riboN <- log2(
  1+ as.matrix(allCountsNorm[,grep(pattern = "ribo",colnames(allCountsNorm))])
) 
rnaN <- log2( 1 + as.matrix(allCountsNorm[,grep(pattern = "rna",colnames(allCountsNorm))])
) 
# @kate wondering why you are adding 0.828 to TE  
TE <- (riboN - rnaN) + 0.828 
colnames(TE) <- gsub(pattern = "ribo", replacement = "TE", x = colnames(TE))

ribo_counts <- as.matrix(allCounts[,grep(pattern = "ribo",colnames(allCounts))])
rna_counts <- as.matrix(allCounts[,grep(pattern = "rna",colnames(allCounts))])

# remove 3d libs
counts_3e <- allCounts[,!grepl("si3d",colnames(allCounts))]
metadata_norm_3e <- metadata_norm %>% filter(treatment != "si3d") %>% droplevels()

# remove 3e libs
counts_3d <- allCounts[,!grepl("si3e",colnames(allCounts))]
metadata_norm_3d <- metadata_norm %>% filter(treatment != "si3e") %>% droplevels()

ddsMat_3d <- DESeqDataSetFromMatrix(
  countData = counts_3d,
  colData = metadata_norm_3d,
  design = ~treatment + library + treatment:library
  )

ddsMat_3d <- DESeq(ddsMat_3d)
resultsNames(ddsMat_3d)
res_te_3d <- results(ddsMat_3d, name=resultsNames(ddsMat_3d)[4], cooksCutoff=F, independentFiltering=F)


res_te_3d <- lfcShrink(dds = ddsMat_3d, coef = resultsNames(ddsMat_3d)[4], res = res_te_3d) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>%
  filter(!symbol %in% nonpolyA) 


ddsMat_3e <- DESeqDataSetFromMatrix(
  countData = counts_3e,
  colData = metadata_norm_3e,
  design = ~treatment + library + treatment:library
  )

ddsMat_3e <- DESeq(ddsMat_3e)

res_te_3e <- results(ddsMat_3e,
                name=resultsNames(ddsMat_3e)[4],
                cooksCutoff=F, independentFiltering=F)

res_te_3e <- lfcShrink(
  dds = ddsMat_3e,
  coef = resultsNames(ddsMat_3e)[4],
  res = res_te_3e) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>%
  filter(!symbol %in% nonpolyA)  
```
Exporting differential TE counts in figure 6

## Visualize e and d specific targets
```{r}
#import gene names
te_norm <- TE %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id") %>% 
  left_join(geneInfo, by="gene_id") 

# @kate -> geneInfo has transcript level information that you do NOT want because it messes up the join and creates > 100K rows! You only need unique(geneInfo[,c(1,4)]) rather than geneInfo. Everything below that uses `te_norm` will be messed up. NOTE -> when you make this change you will need to look at joins in your code below to make sure it still works -> where you use unique(geneInfo[,c(1,4)])


res_3d <- res_te_3d %>% 
  dplyr::select(gene_id, log2FoldChange, padj) %>% 
  rename(log2FC_d = log2FoldChange) %>% 
    rename(padj_d = padj)

res_3e <- res_te_3e %>% 
  dplyr::select(gene_id, log2FoldChange, padj) %>% 
    rename(log2FC_e = log2FoldChange) %>% 
    rename(padj_e = padj)

res_te_total <- left_join(res_3d, res_3e, by="gene_id") %>% 
  left_join(geneInfo, by="gene_id")

res_te_total <- res_te_total %>% 
  filter(padj_d <0.05 | padj_e <0.05)


no_d_change <- res_te_total %>% 
  filter(log2FC_d < 0.01 & log2FC_d > -0.01)

no_e_change <- res_te_total %>% 
  filter(log2FC_e < 0.01 & log2FC_e > -0.01)
```

### Exclusively changed with 3eKD
```{r}

e_only_genes <- no_d_change$gene_id

e_only <- TE[e_only_genes,] %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_id") %>%
  left_join(., unique(geneInfo[,c(1,4)])) %>%
  reshape2::melt(value.name = "TE", variable.name = "info") %>%
  separate(col = info, into = c("tx","sirna","type","rep"))

e_only$sirna <- relevel(x = factor(e_only$sirna), ref = "sictrl")

e_only <- e_only %>% 
  filter(symbol != "NA")

# @kate -> if this is for the manuscript, then restrict to the ones we care about. 
ggstripchart(data = e_only, x = "sirna", y = "TE", color = "sirna", facet.by = "symbol", shape = "sirna", scales="free_y", palette = c("black",si3d_col,si3e_col), position = position_jitterdodge(1.2)) + 
  ylab(expression("log"[2]*" TE (ribo/rna)")) 

```

### Exclusively changed with 3dKD

```{r}

d_only_genes <- no_e_change$gene_id

d_only <- TE[d_only_genes,] %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_id") %>%
  left_join(., unique(geneInfo[,c(1,4)])) %>%
  reshape2::melt(value.name = "TE", variable.name = "info") %>%
  separate(col = info, into = c("tx","sirna","type","rep"))

d_only$sirna <- relevel(x = factor(d_only$sirna), ref = "sictrl")

d_only <- d_only %>% 
  filter(symbol != "NA")

# @kate -> if this is for the manuscript, then restrict to the ones we care about.
ggstripchart(data = d_only, x = "sirna", y = "TE", color = "sirna", facet.by = "symbol", shape = "sirna", scales="free_y", palette = c("black",si3d_col,si3e_col), position = position_jitterdodge(1.2)) + 
  ylab(expression("log"[2]*" TE (ribo/rna)")) 

```


Sample of genes:
```{r}
d_genes <- c("CLDN9", "CLIC1", "PRDM4")
e_genes <- c("APLP2", "PABPC1", "TOR1AIP2")
common <- c("ATF4", "EIF4EBP2", "GSK3B", "MED13L", "PCBP2", "XBP1")

genes_of_interest <- c("CLDN9", "CLIC1", "PRDM4", "APLP2", "PABPC1", "TOR1AIP2", "ATF4", "EIF4EBP2", "GSK3B", "MED13L", "PCBP2", "XBP1")
```

```{r}

plot_chart <- res_te_total %>% 
  mutate(category = ifelse(symbol %in% d_genes == TRUE, "eIF3d-specific", 
                           ifelse(symbol %in% e_genes == TRUE, "eIF3e-specific",
                                  ifelse(symbol %in% common == TRUE, "eIF3d and eIF3e", "NA")))) %>% 
  filter(category != "NA")

chart_genes <- plot_chart %>% 
  dplyr::select(gene_id, category)

paper_genes <- chart_genes$gene_id %>% unique()

chart <- TE[paper_genes,] %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_id") %>%
  left_join(., unique(geneInfo[,c(1,4)])) %>%
  reshape2::melt(value.name = "TE", variable.name = "info") %>%
  separate(col = info, into = c("tx","sirna","type","rep"))

chart <- chart %>% 
left_join(chart_genes)

chart$sirna <- relevel(x = factor(chart$sirna), ref = "sictrl")

chart <- chart %>% 
  filter(category != "NA") %>% 
  filter(symbol != "NA")

```

Sample of 3e and 3d shared genes
```{r}

chart_genes <- plot_chart %>% 
  dplyr::select(gene_id, category) %>% unique()

common <- chart_genes %>% 
  filter(category=="eIF3d and eIF3e")

common_genes <- common$gene_id


common_only <- TE[common_genes,] %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_id") %>%
  left_join(., unique(geneInfo[,c(1,4)])) %>%
  reshape2::melt(value.name = "TE", variable.name = "info") %>%
  separate(col = info, into = c("tx","sirna","type","rep"))

common_only$sirna <- relevel(x = factor(common_only$sirna), ref = "sictrl")

common_only <- common_only %>% 
  filter(symbol != "NA")

# @kate - i highly recommend having a more specific name than plot
plot<-
ggstripchart(data = common_only, x = "sirna", y = "TE", color = "sirna", facet.by = "symbol", shape = "sirna", scales="free_y", palette = c("black",si3d_col,si3e_col), position = position_jitterdodge(1.2)) + 
  ylab(expression("log"[2]*" TE (ribo/rna)"))


plot <-
ggstripchart(data = common_only, 
             x = "sirna", 
             y = "TE", 
             color = "sirna", 
             facet.by = "symbol", 
             shape = "sirna", 
             scales = "free_y", 
             palette = c("black", si3d_col, si3e_col), 
             position = position_jitterdodge(1.2)) + 
  ylab(expression("log"[2]*" TE (ribo/rna)"))
  
# @kate -> not sure you want a title here as this can be done in affy  
  ggpar(plot, title= "eIF3e and eIF3d regulated genes")+theme(
    axis.title = element_text(size = 10),   
    axis.text = element_text(size = 8),    
    strip.text = element_text(size = 10)   
  )
 

ggsave(width = 4, height = 3.25, filename = here("plots/fig1/commongenes.pdf"))


```

Sample of 3d specific genes
```{r}

d_spec <- chart_genes %>% 
  filter(category=="eIF3d-specific")

d_spec_genes <- d_spec$gene_id


d_spec_only <- TE[d_spec_genes,] %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_id") %>%
  left_join(., unique(geneInfo[,c(1,4)])) %>%
  reshape2::melt(value.name = "TE", variable.name = "info") %>%
  separate(col = info, into = c("tx","sirna","type","rep"))

d_spec_only$sirna <- relevel(x = factor(d_spec_only$sirna), ref = "sictrl")

d_spec_only <- d_spec_only %>% 
  filter(symbol != "NA")

plot<-
ggstripchart(data = d_spec_only, x = "sirna", y = "TE", color = "sirna", facet.by = "symbol", shape = "sirna", scales="free_y", palette = c("black",si3d_col,si3e_col), position = position_jitterdodge(1.2)) + 
  ylab(expression("log"[2]*" TE (ribo/rna)"))
  
  
  plot <- plot + theme(legend.position = "none")

ggpar(plot, title= "eIF3d-only regulated genes")+
  theme(
    axis.title = element_text(size = 10),   
    axis.text = element_text(size = 8),    
    strip.text = element_text(size = 10)   
  )
ggsave(width = 4, height = 1.7, filename = here("plots/fig1/dgenes.pdf"))


```

Example of 3e specific genes
```{r}

e_spec <- chart_genes %>% 
  filter(category=="eIF3e-specific")

e_spec_genes <- e_spec$gene_id


e_spec_only <- TE[e_spec_genes,] %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_id") %>%
  left_join(., unique(geneInfo[,c(1,4)])) %>%
  reshape2::melt(value.name = "TE", variable.name = "info") %>%
  separate(col = info, into = c("tx","sirna","type","rep"))

e_spec_only$sirna <- relevel(x = factor(e_spec_only$sirna), ref = "sictrl")

e_spec_only <- e_spec_only %>% 
  filter(symbol != "NA")

plot<-
ggstripchart(data = e_spec_only, x = "sirna", y = "TE", color = "sirna", facet.by = "symbol", shape = "sirna", scales="free_y", palette = c("black",si3d_col,si3e_col), position = position_jitterdodge(1.2)) + 
  ylab(expression("log"[2]*" TE (ribo/rna)"))
  
  plot <-  plot + theme(legend.position = "none")

  ggpar(plot, title= "eIF3e-only regulated genes")+theme(
    axis.title = element_text(size = 10),   
    axis.text = element_text(size = 8),    
    strip.text = element_text(size = 10)   
  )

ggsave(width = 4, height = 1.7, filename = here("plots/fig1/egenes.pdf"))

```


Visualize e and d specifically
```{r}
# @kate -> i think you really need to look at the normalized counts for ribo AND rna and see if the directionality makes sense. see below for example and figure out if this matches up with what you are seeing.

# visualizing RNA and RIBO counts
allCountsNorm[e_and_d,] %>% 
  as.data.frame() %>%
  rownames_to_column(var = "gene") %>%
  reshape2::melt() %>% 
  separate(variable, into = c("o2","kd","type","rep"), sep = "_") %>% 
  ggplot(aes(x = kd, y = value, color = kd)) +
  geom_jitter() +
  facet_wrap(type~gene, scales = "free_y")

# looking at the reported deltaTE for eif3d in the eif3d kd
res_te_3d %>% filter(gene_id == "ENSG00000100353.17")

# visualizing TE
TE[e_and_d,] %>% 
  as.data.frame() %>%
  rownames_to_column(var = "gene") %>%
  reshape2::melt() %>% 
  separate(variable, into = c("o2","kd","type","rep"), sep = "_") %>% 
  ggplot(aes(x = kd, y = value, color = kd)) +
  geom_jitter() +
  facet_wrap(gene~type, scales = "free_y")

e_and_d <- c("ENSG00000104408.9", "ENSG00000100353.17")

e_d <- TE[e_and_d,] %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_id") %>%
  left_join(., unique(geneInfo[,c(1,4)])) %>%
  reshape2::melt(value.name = "TE", variable.name = "info") %>%
  separate(col = info, into = c("tx","sirna","type","rep"))

e_d$sirna <- relevel(x = factor(e_d$sirna), ref = "sictrl")

e_d <- e_d %>% 
  filter(symbol != "NA")

plot<-
ggstripchart(data = e_d, x = "sirna", y = "TE", color = "sirna", facet.by = "symbol", shape = "sirna", scales="free_y", palette = c("black",si3d_col,si3e_col), position = position_jitterdodge(1.2)) + 
  ylab(expression("log"[2]*" TE (ribo/rna)"))

plot
```

## Venn diagram of 3e/d gene overlaps in normoxia
```{r compare genes}
te_3d_up <- res_te_3d %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

te_3d_down <- res_te_3d %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)

te_3e_up <- res_te_3e %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

te_3e_down <- res_te_3e %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)


norm_te_up <-
  list(
    te_3d_up = res_te_3d %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id),
  te_3e_up = res_te_3e %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)
  )

norm_te_down <-
  list(
    te_3d_down = res_te_3d %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id),
  te_3e_down = res_te_3e %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)
  )


ggvenn(norm_te_up, auto_scale = T, fill_color = c(si3d_col,si3e_col), text_size = 2)

ggsave(width = 2, height = 2, filename = here("plots/fig1/vennup.pdf"))


ggvenn(norm_te_down, auto_scale = T, fill_color = c(si3d_col,si3e_col), text_size = 2)

ggsave(width = 2, height = 2, filename = here("plots/fig1/venndown.pdf"))


# stopifnot(
# identical(res_te_3d$gene_id,res_te_3e$gene_id))
```

### Density scatter plot
```{r}
norm_sig_te <- data.frame(
  gene_id = res_te_3d$gene_id,
  symbol = res_te_3d$symbol,
  te_3d = res_te_3d$log2FoldChange,
  te_3e = res_te_3e$log2FoldChange
) %>%
  filter(gene_id %in% c(te_3e_up,
                        te_3e_down,
                        te_3d_up,
                        te_3d_down)) 

scatter <-
ggscatter(data = norm_sig_te,
          x = "te_3d",
          y = "te_3e",
          xlab= "si-eIF3d TE",
          ylab="si-eIF3e TE",
          alpha = 0.25,
          title = "TE changes with 3d KD vs 3e KD", 
          xlim=c(-2,2), ylim = c(-2,2),
  add = "reg.line")+
   stat_cor(
   aes(label =
         paste(
           after_stat(rr.label),
           ..p.label..,
           sep = "~`,`~")
       ) + theme_prism()
   )+
  stat_regline_equation(
    aes(label =  paste(..eq.label.., after_stat(rr.label), sep = "~~~~")),
    label.x = -1.5, label.y = 1.5
  ) +
  geom_vline(xintercept=0, linetype='dotted') +
  geom_hline(yintercept=0, linetype='dotted') +
  theme_prism() +
  guides(color = FALSE)

scatter

#ggplotly(scat, tooltip="all")

ggsave(width = 5, height = 5, filename = here("plots/fig1/normoxia_quadrants.pdf"))

```


## Pathway analysis setup
```{r prep GSEA analysis}

eif3d_te <- res_te_3d$log2FoldChange
names(eif3d_te) <- res_te_3d$symbol

eif3d_te <- sort(eif3d_te, decreasing = TRUE)

eif3d_te <- eif3d_te[!duplicated(names(eif3d_te))]

eif3e_te <- res_te_3e$log2FoldChange
names(eif3e_te) <- res_te_3e$symbol

eif3e_te <- sort(eif3e_te, decreasing = TRUE)

eif3e_te <- eif3e_te[!duplicated(names(eif3e_te))]


m_df_h <- msigdbr(species = "Homo sapiens") %>%
  filter(gs_cat == "H") %>% 
  dplyr::select(gs_name, gene_symbol)

m_df_c2 <- msigdbr(species = "Homo sapiens") %>%
  filter(gs_cat == "C2" & gs_subcat != "CGP") %>% 
  dplyr::select(gs_name, gene_symbol)

```

### Hallmark pathways
```{r GSEA Hallmarks}

eif3d_te_h <- GSEA(
  geneList = eif3d_te,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )

eif3e_te_h <- GSEA(
  geneList = eif3e_te,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )

norm_te_h <- dplyr::full_join(
  eif3d_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3d_", .), is.numeric),
  eif3e_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3d_pvalue < 0.05 | eif3e_pvalue < 0.05)  %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = eif3e_NES - eif3d_NES)


norm_te_h$ID <- gsub(pattern = "HALLMARK_", replacement = "", x = norm_te_h$ID)

norm_te_h <- norm_te_h %>% 
  mutate(color = ifelse(ID=="HYPOXIA", "red", "black"))

ggscatter(data = norm_te_h,
          x = "eif3d_NES",
          y = "eif3e_NES",
          color = "color", palette = c("black", "red"),
          xlab = "si-eIF3d NES",
          ylab = "si-eIF3e NES") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5, 2.5) +
  xlim(-2.5, 2.5) +
  theme_prism(base_size = 12) +
  guides(color = "none") +
  geom_text_repel(aes(label = ID), size = 3)

ggsave(width = 4, height = 4, filename = here("plots/fig1/normoxia_halmarks.pdf"))

norm_h_table <- norm_te_h %>% 
  mutate(direction = ifelse(eif3d_NES < 0 & eif3e_NES < 0, "down_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES > 0, "up_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES < 0, "up_3d_down_3e",
                            ifelse(eif3d_NES < 0 & eif3e_NES > 0, "up_3e_down_3d", "NA")))))

write_csv(norm_h_table, here("counts/fig1/norm_h_table.csv"))

```



### KEGG pathways

```{r GSEA KEGG}
eif3d_te_kegg <- GSEA(
  geneList = eif3d_te,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^KEGG', gs_name))
  )

eif3e_te_kegg <- GSEA(
  geneList = eif3e_te,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^KEGG', gs_name))
  )

norm_te_kegg <- dplyr::full_join(
  eif3d_te_kegg@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3d_", .), is.numeric),
  eif3e_te_kegg@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3d_qvalue < 0.05 | eif3e_qvalue < 0.05) %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = eif3e_NES - eif3d_NES)

norm_te_kegg$ID <- gsub(pattern = "KEGG_", replacement = "", x = norm_te_kegg$ID)

ggscatter(data = norm_te_kegg,
          x = "eif3d_NES",
          y = "eif3e_NES",
          label = "ID",
          repel = TRUE,
          title = "Kegg - Normoxia",
          font.label = 7) +
  scale_color_continuous(type = "viridis") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5,2.5) +
  xlim(-2.5,2.5) + theme_prism()


DT::datatable(norm_te_kegg) %>%
  formatRound(which(sapply(norm_te_kegg,is.numeric)), digits = 3) 


ggsave(width = 4, height = 4, filename = here("plots/fig1/normoxia_kegg.pdf"))

# @kate -> change this (and any other case of nested ifelse) to using case_when
norm_kegg_table <- norm_te_kegg %>% 
  mutate(direction = ifelse(eif3d_NES < 0 & eif3e_NES < 0, "down_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES > 0, "up_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES < 0, "up_3d_down_3e",
                            ifelse(eif3d_NES < 0 & eif3e_NES > 0, "up_3e_down_3d", "NA")))))

write_csv(norm_kegg_table, here("counts/fig1/norm_kegg_table.csv"))







```


### REACTOME pathways

```{r GSEA REACTOME}
eif3d_te_reactome <- GSEA(
  geneList = eif3d_te,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^REACTOME', gs_name))
  )

eif3e_te_reactome <- GSEA(
  geneList = eif3e_te,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^REACTOME', gs_name))
  )

norm_te_react <- dplyr::full_join(
  eif3d_te_reactome@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3d_", .), is.numeric),
  eif3e_te_reactome@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3d_qvalue < 0.05 | eif3e_qvalue < 0.05)  %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = eif3e_NES - eif3d_NES)

norm_te_react$ID <- gsub(pattern = "REACTOME_", replacement = "", x = norm_te_react$ID)

ggscatter(data = norm_te_react,
          x = "eif3d_NES",
          y = "eif3e_NES",
          label = "ID",
          title = "Reactome - Normoxia",
          repel = TRUE,
          font.label = 6) +
  scale_color_continuous(type = "viridis") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5,2.5) +
  xlim(-2.5,2.5)


DT::datatable(norm_te_react) %>%
  formatRound(which(sapply(norm_te_react,is.numeric)), digits = 3) 


ggsave(width = 4, height = 4, filename = here("plots/fig1/normoxia_reactome.pdf"))


norm_react_table <- norm_te_react %>% 
  mutate(direction = ifelse(eif3d_NES < 0 & eif3e_NES < 0, "down_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES > 0, "up_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES < 0, "up_3d_down_3e",
                            ifelse(eif3d_NES < 0 & eif3e_NES > 0, "up_3e_down_3d", "NA")))))

write_csv(norm_react_table, here("counts/fig1/norm_react_table.csv"))

```
