---
title: "eIF3e and eIF3d are necessary for the acute hypoxic translational response"
author: "Kate"
date: "2024-05-27"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, echo=FALSE}
knitr::opts_chunk$set(echo = FALSE, message=FALSE, warning=FALSE)
```

```{r install packages, echo=FALSE}
library(here)
library(janitor)
library(tximport)
library(tidyverse)
library(EnhancedVolcano)
library(msigdbr)
library(viridis)
library(gridExtra)
library(ggthemes)
library(ggrepel)
library(scales)
library(DESeq2)
library(limma)
library(clusterProfiler)
library(plotly)
library(ggokabeito)
library(ggvenn)
library(ggpubr)
library(DT)
library(edgeR)
library(cowplot)
library(ggExtra)
library(ggprism)
```

colors
```{r}
a209_col = "#009E73"
si3e_col = "#56B4E9"
si3d_col = "#E69F00"
```


## Load and filter metadata
```{r}
metadata <- read_csv(here("metadata.csv"))

metadata_kds <- metadata %>% 
  filter(treatment %in% c("sictrl","si3d","si3e"))
```

## load salmon files
```{r import and filter, message=FALSE, warning=FALSE}
# create the tx2 gene file for tximport
geneInfo <- read_csv(here("accessories","gencode.v26.primary.info.csv.gz"), col_names = F) %>%
  filter(!str_detect(X2, "pre_"))

colnames(geneInfo) <- c("gene_id","transcript_id","biotype","symbol")

#map transcripts to gene
tx2gene <- geneInfo[,c(2,1)]
gene_symbol <- geneInfo %>% dplyr::select(gene_id, symbol) %>% unique()

#load quant files based on metadata names
myquantfiles <- paste(here(), "/salmon/",
                      metadata_kds$sample_id,
                      "/quant.sf.gz",
                      sep = "")

names(myquantfiles) <- paste(metadata_kds$condition,
                             metadata_kds$treatment,
                             metadata_kds$library,
                             metadata_kds$rep,
                             sep = "_")


myTxi <- tximport(myquantfiles, type = "salmon", tx2gene = tx2gene)



#only keeping rows with enough counts - want around 10-15k genes

# hist(log2(rowSums(myTxi$counts)), breaks = 40)
keepGenes <- rownames(myTxi$counts[log2(rowSums(myTxi$counts)) > 7,])

pc <-geneInfo %>%
  filter(biotype =="protein_coding") %>%
  pull(gene_id) %>%
  unique()
pc_keep <- pc[pc %in% keepGenes]

# remove non-polyadenylated mRNAs
nonpolyA <- readxl::read_xlsx(path = here("accessories","polyAminus.xlsx"), sheet = 1) %>% 
   pull(Gene)

nonpolyA_ID <- geneInfo %>%
  filter(symbol %in% nonpolyA) %>% 
  pull(gene_id) %>% 
  unique()

# remove any genes that match these annotation categories # before we do DE analysis
blacklist_smRNA <- c("snoRNA","miRNA","snRNA","rRNA","Mt_tRNA","scaRNA","Mt_tRNA","Mt_rRNA")

# there are 5,355 genes we want to exclude
blacklist_smRNA_ID <-geneInfo %>%
  filter(biotype %in% blacklist_smRNA) %>% 
  pull(gene_id) %>% 
  unique()

genes_to_kill <- unique(c(blacklist_smRNA_ID,nonpolyA_ID))

```

### Set up translational efficiency analysis
- RNA and ribo counts are separated out

```{r deltaTE count setup, message=FALSE, warning=FALSE}
# separate out RNA counts versus ribo count
# from https://currentprotocols.onlinelibrary.wiley.com/doi/full/10.1002/cpmb.108

allCounts <- myTxi$counts %>%
  as.data.frame() %>% 
  rownames_to_column(var = "gene") %>% 
  filter(gene %in% keepGenes) %>%
  filter(!gene %in% genes_to_kill) %>%
  column_to_rownames(var = "gene") %>% 
  as.matrix() %>%
  round(.,0)


ribo_counts <- as.matrix(allCounts[,grep(pattern = "ribo",colnames(allCounts))])
rna_counts <- as.matrix(allCounts[,grep(pattern = "rna",colnames(allCounts))])
```

# Differential gene expression to compare:
## 1. Normoxia -> Hypoxia
## 2. Normoxia KD effect
## 3. Hypoxia KD effect

### Deseq2:
# Acute Hypoxic Response

1. siCTRL norm -> hypox
2. eIF3e norm -> hypox
3. eIF3d norm -> hypox

```{r}
loop_vector <- c("sictrl", "si3e", "si3d")
```

# DeSeq for all TE comparisons in a loop

```{r}

# set shell dataframes ----------------------------------------------------------

all_te <- list()
all_counts <- tribble(~sample, ~TEupreg, ~TEdownreg)

for (i in loop_vector) {

# set up data -------------------------------------------------------------------
  
  #i="sictrl"
  
  metadata_loop <- metadata_kds %>% 
  filter(treatment == i)
  
  # make sample info from metadata
  sample_info <- metadata_loop %>% dplyr::select(condition, treatment, rep, library, sample_id)
  
  sample_info <- sample_info %>% 
    mutate("name" = 
  paste(sample_info$condition, sample_info$treatment, sample_info$library, sample_info$rep, sep = "_"))
  
  # colnames(sample_info) <- c("condition", "treatment","rep", "library", "sample_id", "name")
  sample_info$library <- relevel(x = factor(sample_info$library), ref = "rna")
    sample_info$condition <- relevel(x = factor(sample_info$condition), ref = "normoxia")
    
    levels(sample_info$condition)
    levels(sample_info$library)

  #filter AllCounts for i
    
    names <- as.character(sample_info$name)
    
    allCounts_subset <- allCounts %>% 
      as.data.frame() 
    
    allCounts_subset <- allCounts_subset[colnames(allCounts_subset) %in% names]

# Deseq analysis ----------------------------------------------------------------
  
  ddsMat <- DESeqDataSetFromMatrix(countData = allCounts_subset, 
                                 colData = sample_info, 
                                 design = ~condition+library+condition:library)
  ddsMat <- DESeq(ddsMat)
  
  resultsNames(ddsMat)[4]

  res_te_loop <- results(ddsMat, name=resultsNames(ddsMat)[4], cooksCutoff=FALSE, independentFiltering=FALSE)

  res_te_loop <- lfcShrink(dds = ddsMat, coef = resultsNames(ddsMat)[4], res = res_te_loop) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA)  

  all_te[[i]] <- res_te_loop
  
# visualization -----------------------------------------------------------------
  
  volcano <- EnhancedVolcano(toptable = res_te_loop,
                x = "log2FoldChange",
                y = "padj",
                lab = res_te_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = paste0(i, "_TE_volcano"),
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano)


# count table -------------------------------------------------------------------

upreg <- res_te_loop %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_te_loop %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

all_counts <- rbind(all_counts, data.frame(i,upreg, downreg))


}

# writexl::write_xlsx(x = res_te, path = here("counts","si3e_normoxiatohypoxia_TE.xlsx"), col_names = T)


```

## Supplemental: Differential RNA and RIBO loops

### Differential RNA loop

```{r RNA, fig.height=9, fig.width=10, message=FALSE, warning=FALSE}

# set shell dataframes ----------------------------------------------------------

all_rna <- list()
all_counts_rna <- tribble(~sample, ~RNAupreg, ~RNAdownreg)

for (i in loop_vector) {

# set up data -------------------------------------------------------------------
  
  metadata_loop <- metadata_kds %>% 
  filter(treatment == i)
  
  # make sample info from metadata
  sample_info <- metadata_loop %>% 
    filter(library == "rna")
  
  sample_info <- sample_info %>% 
    mutate("name" = 
  paste(sample_info$condition, sample_info$treatment, sample_info$library, sample_info$rep, sep = "_"))
  
  #filter AllCounts for i
  
  
  names <- as.character(sample_info$name)
    
    rna_subset <- allCounts %>% 
      as.data.frame() 
    
   rna_subset <- rna_subset[colnames(rna_subset) %in% names]
   
sample_info$condition <- relevel(x=factor(sample_info$condition), ref="normoxia")
sample_info$library <- relevel(x=factor(sample_info$library), ref="rna")

# Deseq analysis ----------------------------------------------------------------
  
  ddsMat_rna <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~condition)
  ddsMat_rna <- DESeq(ddsMat_rna)
  

  res_rna_loop <- results(ddsMat_rna, name=resultsNames(ddsMat_rna)[2], cooksCutoff=FALSE, independentFiltering=FALSE)
  
  resultsNames(ddsMat_rna)[2]

  res_rna_loop <- lfcShrink(dds = ddsMat_rna, coef = resultsNames(ddsMat_rna)[2], res = res_rna_loop) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA)  

  all_rna[[i]] <- res_rna_loop
  
  
  # visualization -----------------------------------------------------------------
  
  volcano_rna <- EnhancedVolcano(toptable = res_rna_loop,
                x = "log2FoldChange",
                y = "padj",
                lab = res_rna_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = paste0(i, "_rna_volcano"),
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano_rna)


# count table -------------------------------------------------------------------

upreg_rna <- res_rna_loop %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg_rna <- res_rna_loop %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()


all_counts_rna <- rbind(all_counts_rna, data.frame(i,upreg_rna, downreg_rna))

}

  # writexl::write_xlsx(x = res_te, path = here("counts","si3e_normoxiatohypoxia_TE.xlsx"), col_names = T)


```


### Differential RIBO loop

```{r RNA, fig.height=9, fig.width=10, message=FALSE, warning=FALSE}

# set shell dataframes ----------------------------------------------------------

all_ribo <- list()
all_counts_ribo <- tribble(~sample, ~RIBOupreg, ~RIBOdownreg)

for (i in loop_vector) {

# set up data -------------------------------------------------------------------
  
  metadata_loop <- metadata_kds %>% 
  filter(treatment == i)
  
  # make sample info from metadata
  sample_info <- metadata_loop %>% 
    filter(library == "ribo")
  
  sample_info <- sample_info %>% 
    mutate("name" = 
  paste(sample_info$condition, sample_info$treatment, sample_info$library, sample_info$rep, sep = "_"))
  
  #filter AllCounts for i
  
  
  names <- as.character(sample_info$name)
    
    ribo_subset <- allCounts %>% 
      as.data.frame() 
    
   ribo_subset <- ribo_subset[colnames(ribo_subset) %in% names]

  sample_info$condition <- relevel(x=factor(sample_info$condition), ref="normoxia")

# Deseq analysis ----------------------------------------------------------------
  
  ddsMat_ribo <- DESeqDataSetFromMatrix(countData = ribo_subset, 
                                 colData = sample_info, 
                                 design = ~condition)
  ddsMat_ribo <- DESeq(ddsMat_ribo)
  

  res_ribo_loop <- results(ddsMat_ribo, name=resultsNames(ddsMat_ribo)[2], cooksCutoff=FALSE, independentFiltering=FALSE)

  res_ribo_loop <- lfcShrink(dds = ddsMat_ribo, coef = resultsNames(ddsMat_ribo)[2], res = res_ribo_loop) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA)  

  all_ribo[[i]] <- res_ribo_loop
  
  
  # visualization -----------------------------------------------------------------
  
  volcano_ribo <- EnhancedVolcano(toptable = res_ribo_loop,
                x = "log2FoldChange",
                y = "padj",
                lab = res_ribo_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = paste0(i, "_ribo_volcano"),
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano_ribo)


# count table -------------------------------------------------------------------

upreg_ribo <- res_ribo_loop %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg_ribo <- res_ribo_loop %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()


all_counts_ribo <- rbind(all_counts_ribo, data.frame(i,upreg_ribo, downreg_ribo))

}

  # writexl::write_xlsx(x = res_te, path = here("counts","si3e_normoxiatohypoxia_TE.xlsx"), col_names = T)


```


### Total counts

```{r}
total_counts <- left_join(all_counts, all_counts_rna, by="i")
total_counts <- left_join(total_counts, all_counts_ribo, by="i")
total_counts

# visualize ---------------------------------------------------------------------

long_counts <- total_counts %>% 
  pivot_longer(2:5, names_to = "category", values_to= "counts") %>% 
  filter(i=="sictrl")

long_counts$category <- factor(long_counts$category, levels = c("downreg_rna", "upreg_rna", "downreg", "upreg"))

long_counts <- long_counts %>% 
  mutate(color = ifelse(category== "upreg_rna" | category== "upreg", "red", "blue"))


ggbarplot(long_counts, x="category", y="counts", fill="color")+
  scale_fill_identity()



```

```{r}
sictrl_counts <- total_counts %>% 
  filter(i=="sictrl") %>% 
  pivot_longer(2:7, names_to = "type", values_to = "counts") %>% 
  filter(type!= "upreg") %>% 
  filter(type != "downreg")



ggbarplot(sictrl_counts, x="type", y="counts", label=TRUE)

ggsave(here("plots/fig3/sictrl_counts.pdf"), width=4, height=4)
```


## One table for all genes and their acute changes

```{r}

sictrl_te <- all_te$sictrl[,c(1,2,4,7)]
colnames(sictrl_te)[3] ="LFC_te_sictrl"
colnames(sictrl_te)[4] ="padj_te_sictrl"

sictrl_ribo <- all_ribo$sictrl[,c(1,4,7)]
colnames(sictrl_ribo)[2] ="LFC_ribo_sictrl"
colnames(sictrl_ribo)[3] ="padj_ribo_sictrl"

sictrl_rna <- all_rna$sictrl[,c(1,4,7)]
colnames(sictrl_rna)[2] ="LFC_rna_sictrl"
colnames(sictrl_rna)[3] ="padj_rna_sictrl"

si3e_te <- all_te$si3e[,c(1,4,7)]
colnames(si3e_te)[2] ="LFC_te_si3e"
colnames(si3e_te)[3] ="padj_te_3e"

si3e_ribo <- all_ribo$si3e[,c(1,4,7)]
colnames(si3e_ribo)[2] ="LFC_ribo_si3e"
colnames(si3e_ribo)[3] ="padj_ribo_si3e"

si3e_rna <- all_rna$si3e[,c(1,4,7)]
colnames(si3e_rna)[2] ="LFC_rna_si3e"
colnames(si3e_rna)[3] ="padj_rna_si3e"

si3d_te <- all_te$si3d[,c(1,4,7)]
colnames(si3d_te)[2] ="LFC_te_si3d"
colnames(si3d_te)[3] ="padj_te_si3d"

si3d_ribo <- all_ribo$si3d[,c(1,4,7)]
colnames(si3d_ribo)[2] ="LFC_ribo_si3d"
colnames(si3d_ribo)[3] ="padj_ribo_si3d"

si3d_rna <- all_rna$si3d[,c(1,4,7)]
colnames(si3d_rna)[2] ="LFC_rna_si3d"
colnames(si3d_rna)[3] ="padj_rna_si3d"

#bind cols ----------------------------------------------------------------------

all_changes <- purrr::reduce(list(sictrl_te,sictrl_ribo,sictrl_rna,si3e_te,si3e_ribo,si3e_rna,si3d_te,si3d_ribo,si3d_rna), dplyr::left_join, by = 'gene_id')
```

## Visualize acute hypoxic response with histograms

## Acute hypoxic changes

```{r}

ctrl <- all_changes %>% 
  filter(padj_rna_sictrl < 0.05 | padj_ribo_sictrl < 0.05)

ctrl_acute <- ggplot(data = ctrl, aes(x = LFC_rna_sictrl, y = LFC_ribo_sictrl)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote(Log[2]~"si-CTRL LFC"~"RIBO: hypoxia vs normoxia")) +
  xlab(bquote(Log[2]~"si-CTRL LFC"~"RNA: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 5)+
  theme_prism()

ctrl_acute <- ggMarginal(ctrl_acute, type = "histogram",
           xparams = list(binwidth = .1, fill = "grey"),
           yparams = list(binwidth = .1, fill = "red")
           )

ggsave(filename = here("plots/fig3/rna_ribo_acute.pdf"), ctrl_acute, width = 4, height = 4)

ctrl_acute
```


## TE changes - 3e KD

```{r}

sig_sictrl <- all_changes %>% 
  filter(padj_te_sictrl <0.05)

scatter <- ggplot(data = sig_sictrl, aes(x = LFC_te_sictrl, y = LFC_te_si3e)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote(Log[2]~"si-eIF3e LFC"~"TE: hypoxia vs normoxia")) +
  xlab(bquote(Log[2]~"si-CTRL LFC"~"TE: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 9)+
  theme_prism()

scatter <- ggMarginal(scatter, type = "histogram",
           xparams = list(binwidth = .1, fill = "grey"),
           yparams = list(binwidth = .1, fill = "#56B4E9")
           )

scatter

ggsave(filename= here("plots/fig3/acute_3e.pdf"), scatter, height=4, width=4)

```

## TE changes - 3d KD

```{r}

scatter <- ggplot(data = sig_sictrl, aes(x = LFC_te_sictrl, y = LFC_te_si3d)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote(Log[2]~"si-eIF3d LFC"~"TE: hypoxia vs normoxia")) +
  xlab(bquote(Log[2]~"si-ctrl LFC"~"TE: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 9)+
  theme_prism()

scatter <- ggMarginal(scatter, type = "histogram",
           xparams = list(binwidth = .1, fill = "grey"),
           yparams = list(binwidth = .1, fill = "#E69F00")
           )

scatter

ggsave(here("plots/fig3/acute_3d.pdf"), scatter, height=4, width=4)

```


#################################################################################


# Differential gene expression to compare:
## 1. Normoxia -> Hypoxia
## 2. Normoxia KD effect
## 3. Hypoxia KD effect

## 2 and 3

1. sictrl to eIF3e
2. sictrl to eIF3d

```{r}
# comparisons -------------------------------------------------------------------

control_vector <- "sictrl"
test_vector <- c("si3e", "si3d")
condition_vector <- c("normoxia", "hypoxia")


# set shell dataframes ----------------------------------------------------------

comparisons <- list()
big_comparisons <- list()
counts <- tribble(~condition, ~sample, ~TEupreg, ~TEdownreg)
te_values <-list()

for(i in condition_vector) {
  
    metadata_cond <- metadata_kds %>% 
    filter(condition== i)
  
for (j in test_vector) {
# set up data -------------------------------------------------------------------

  
  # i = "hypoxia"
  # j = "si3e"
  
  metadata_loop <- metadata_cond %>% 
  filter(treatment == "sictrl" | treatment == j)
  
  # make sample info from metadata
  sample_info <- metadata_loop %>% dplyr::select(condition, treatment, rep, library, sample_id)
  
  
  sample_info <- sample_info %>% 
    mutate("name" = 
  paste(sample_info$condition, sample_info$treatment, sample_info$library, sample_info$rep, sep = "_"))
  
    sample_info$library <- relevel(x = factor(sample_info$library), ref = "rna")
    sample_info$treatment <- relevel(x = factor(sample_info$treatment), ref = "sictrl")

  #filter AllCounts for i
  
  names <- as.character(sample_info$name)
    
  allCounts_subset <- allCounts[, colnames(allCounts) %in% names]
  

# Deseq analysis ----------------------------------------------------------------
  
  ddsMat <- DESeqDataSetFromMatrix(countData = allCounts_subset, 
                                 colData = sample_info, 
                                 design = ~treatment+library+treatment:library)
  ddsMat <- DESeq(ddsMat)

  res_te_loop <- results(ddsMat, name=resultsNames(ddsMat)[4], cooksCutoff=FALSE, independentFiltering=FALSE)

  res_te_loop <- lfcShrink(dds = ddsMat, coef = resultsNames(ddsMat)[4], res = res_te_loop) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA)  

  comparisons[[j]] <- res_te_loop

  
# visualization -----------------------------------------------------------------
  
  volcano <- EnhancedVolcano(toptable = res_te_loop,
                x = "log2FoldChange",
                y = "padj",
                lab = res_te_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = paste0(i, j, "_TE_volcano"),
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano)


# count table -------------------------------------------------------------------

up <- res_te_loop %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

down <- res_te_loop %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()



new_df <- data.frame(i, j, up, down) %>% 
  rename("sample" = "j") %>% 
rename("condition" = "i")

counts <- rbind(counts, new_df)

}

  big_comparisons[[i]] <- comparisons
}

```

# pivot counts
```{r}
long_counts <- counts %>% 
  pivot_longer(3:4, names_to = "direction", values_to = "counts")
```

# effect of si3e KD norm and hypoxia
```{r}

eif3e_counts <- long_counts %>% 
  filter(sample=="si3e")

eif3e_bar <- ggbarplot(data = eif3e_counts, x = "condition", y = "counts", fill = "direction", palette = c("#2492a6",si3e_col), position = position_dodge(0.75), label = T, title= "eIF3e")+
  ylab("Transcript counts")+
  xlab("")+
  theme_prism()+
  theme_cowplot(font_size = 12)+
  scale_y_continuous(limits = c(0, 1600))

eif3e_bar

ggsave(here("plots/fig3/eIF3ebars.pdf"), eif3e_bar, width = 3, height = 4)

```


# effect of si3d KD norm and hypoxia
```{r}

eif3d_counts <- long_counts %>% 
  filter(sample=="si3d")


eif3d_bar <- ggbarplot(data = eif3d_counts, x = "condition", y = "counts", fill = "direction", palette = c("#ffd412",si3d_col), position = position_dodge(0.75), label = T, title="eIF3d")+
   ylab("Transcript counts")+
  xlab("")+
  theme_prism()+
    theme_cowplot(font_size = 12)+
  scale_y_continuous(limits = c(0, 1600))

eif3d_bar

ggsave(here("plots/fig3/eIF3dbars.pdf"), eif3d_bar, width = 3, height = 4)

```

## Make gene lists of up and down, norm, hyp, kds
These will be first for venn diagrams, then for pathway analysis

```{r}

gene_type <- geneInfo %>% dplyr::select(biotype, gene_id)
gene_type <- unique(gene_type)

```

Grab TE lists - 6

```{r}

norm_3e <- big_comparisons$normoxia$si3e %>% 
    mutate(condition = "normoxic") %>% 
  inner_join(., gene_type, by = "gene_id") %>% 
  filter(., biotype == "protein_coding") %>% 
  na.omit()


hyp_3e <- big_comparisons$hypoxia$si3e %>% 
    mutate(condition = "hypoxic") %>% 
  inner_join(., gene_type, by = "gene_id") %>% 
  filter(., biotype == "protein_coding") %>% 
  na.omit()


norm_3d <- big_comparisons$normoxia$si3d %>% 
    mutate(condition = "normoxic") %>% 
  inner_join(., gene_type, by = "gene_id") %>% 
  filter(., biotype == "protein_coding") %>% 
  na.omit()

hyp_3d <- big_comparisons$hypoxia$si3d %>% 
      mutate(condition = "hypoxic") %>% 
  inner_join(., gene_type, by = "gene_id") %>% 
  filter(., biotype == "protein_coding") %>% 
  na.omit()



gsea_names <- c("norm_3e", "hyp_3e", "norm_3d", "hyp_3d")
gsea_lists <- list(norm_3e, hyp_3e, norm_3d, hyp_3d)

```


```{r}

write_csv(norm_3d, here("counts/fig3/norm_3d_te.csv"))
write_csv(norm_3e, here("counts/fig3/norm_3e_te.csv"))
write_csv(hyp_3d, here("counts/fig3/hyp_3d_te.csv"))
write_csv(hyp_3e, here("counts/fig3/hyp_3e_te.csv"))
```


Now, separate lists by direction of change - use for venn diagrams and for pathway analyses
```{r}
norm_3d_up <- norm_3d %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

norm_3d_down <- norm_3d %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)

norm_3e_up <- norm_3e %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

norm_3e_down <- norm_3e %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)



hyp_3d_up <- hyp_3d %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

hyp_3d_down <- hyp_3d %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)

hyp_3e_up <- hyp_3e %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

hyp_3e_down <- hyp_3e %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)

norm_te_up <- list(
  norm_3d_up = norm_3d_up,
  norm_3e_up = norm_3e_up
)

norm_te_down <- list(
  norm_3d_down = norm_3d_down,
  norm_3e_down = norm_3e_down
)

hyp_te_up <- list(
  hyp_3d_up = hyp_3d_up,
  hyp_3e_up = hyp_3e_up
)

hyp_te_down <- list(
  hyp_3d_down = hyp_3d_down,
  hyp_3e_down = hyp_3e_down
)


```



## Venn diagram of changes - normoxia

```{r}
ggvenn(norm_te_up, auto_scale = T, fill_color = c("#E69F00","#56B4E9"))

ggsave(here("plots/normteup_venn.pdf"), width = 3, height=2.5)

ggvenn(norm_te_down, auto_scale = T, fill_color = c("#E69F00","#56B4E9"))

ggsave(here("plots/normtedown_venn.pdf"), width = 3, height=2.5)


stopifnot(
identical(norm_3d$gene_id,norm_3e$gene_id))

norm_sig_te <- data.frame(
  gene_id = norm_3d$gene_id,
  symbol = norm_3d$symbol,
  te_3d = norm_3d$log2FoldChange,
  padj_3d = norm_3d$padj,
  te_3e = norm_3e$log2FoldChange,
  padj_3e = norm_3e$padj
) %>%
  filter(gene_id %in% c(norm_3e_up,
                        norm_3e_down,
                       norm_3d_up,
                        norm_3d_down)) 







scatter <- ggscatter(data = norm_sig_te,
          x = "te_3d",
          xlab="si-eIF3d TE",
          y = "te_3e",
          ylab="si-eIF3e TE",
          alpha = 0.25,
          title = "Normoxia TE - sig genes",
xlim=c(-2,2), ylim = c(-2,2))+
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed")+
   stat_cor(
   aes(label =
         paste(
           after_stat(rr.label),
           ..p.label..,
           sep = "~`,`~")
       ) + theme_prism()
   )




scatter <- ggMarginal(scatter, type = "histogram",
           xparams = list(binwidth = .3, fill = si3e_col),
           yparams = list(binwidth = .3, fill = si3d_col)
           )


scatter

ggsave(here("plots/fig3/norm_e_d.pdf"), scatter, height=4, width=4)


```



## Venn diagram of changes - hypoxia

```{r}
ggvenn(hyp_te_up, auto_scale = T, fill_color = c("#E69F00","#56B4E9"))

ggsave(here("plots/hypteup_venn.pdf"), width = 3, height=2.5)


ggvenn(hyp_te_down, auto_scale = T, fill_color = c("#E69F00","#56B4E9"))

ggsave(here("plots/hyptedown_venn.pdf"), width = 3, height=2.5)


setdiff(hyp_3e$gene_id, hyp_3d$gene_id)
# remove from dataset

hyp_3e <- hyp_3e %>% 
  filter(gene_id != "ENSG00000186844.5")

hyp_sig_te <- data.frame(
  gene_id = hyp_3d$gene_id,
  symbol = hyp_3d$symbol,
  te_3d = hyp_3d$log2FoldChange,
  padj_3d = hyp_3d$padj,
  te_3e = hyp_3e$log2FoldChange,
  padj_3e = hyp_3e$padj) %>%
  filter(gene_id %in% c(hyp_3e_up,
                        hyp_3e_down,
                        hyp_3d_up,
                        hyp_3d_down)) 



scatter <- ggscatter(data = hyp_sig_te,
          x = "te_3d",
          xlab="si-eIF3d TE",
          y = "te_3e",
          ylab="si-eIF3e TE",
          alpha = 0.25,
          title = "Hypoxia TE - sig genes",
xlim=c(-2,2), ylim = c(-2,2))+
  geom_smooth(method = "lm", color = "black", se = FALSE) +
  stat_regline_equation(
    aes(label =  paste(..eq.label.., after_stat(rr.label), sep = "~~~~")),
    label.x = -1.5, label.y = 1.5
  )+
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed")+
   stat_cor(
   aes(label =
         paste(
           after_stat(rr.label),
           ..p.label..,
           sep = "~`,`~")
       )
   )




scatter <- ggMarginal(scatter, type = "histogram",
           xparams = list(binwidth = .3, fill = si3e_col),
           yparams = list(binwidth = .3, fill = si3d_col)
           )


scatter

ggsave(here("plots/fig3/hyp_e_d.pdf"), scatter, width=4, height=4)



```


## Pathway analysis setup from below
## Hypoxia only

3d/e

```{r prep GSEA analysis}

#3d lists

eif3d_hyp <- hyp_3d$log2FoldChange
names(eif3d_hyp) <- hyp_3d$symbol

eif3d_hyp <- sort(eif3d_hyp, decreasing = TRUE)

eif3d_hyp <- eif3d_hyp[!duplicated(names(eif3d_hyp))]

#3e lists

eif3e_hyp <- hyp_3e$log2FoldChange

names(eif3e_hyp) <- hyp_3e$symbol

eif3e_hyp <- sort(eif3e_hyp, decreasing = TRUE)

eif3e_hyp <- eif3e_hyp[!duplicated(names(eif3e_hyp))]


```


```{r}
# hallmark and c2 setup

m_df_h <- msigdbr(species = "Homo sapiens") %>%
  filter(gs_cat == "H") %>% 
  dplyr::select(gs_name, gene_symbol)

m_df_c2 <- msigdbr(species = "Homo sapiens") %>%
  filter(gs_cat == "C2" & gs_subcat != "CGP") %>% 
  dplyr::select(gs_name, gene_symbol)
```


### Hallmark pathways enriched in hypoxia

```{r GSEA Hallmarks}

eif3d_te_h <- GSEA(
  geneList = eif3d_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )


eif3e_te_h <- GSEA(
  geneList = eif3e_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )



hyp_te_h <- dplyr::full_join(
  eif3d_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3d_", .), is.numeric),
  eif3e_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3d_pvalue < 0.05 | eif3e_pvalue < 0.05)  %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = eif3e_NES - eif3d_NES)


hyp_te_h$ID <- gsub(pattern = "HALLMARK_", replacement = "", x = hyp_te_h$ID)

hyp_te_h <- hyp_te_h %>% 
  mutate(color= ifelse(ID=="HYPOXIA", "red", "black"))


ggscatter(data = hyp_te_h,
          x = "eif3d_NES",
          y = "eif3e_NES",
          title = "Hallmark pathways - Hypoxia",
          label = "ID",
          repel = TRUE,
          font.label=7,
  color="color",
  palette=c("black", "red"))+
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5,2.5) +
  xlim(-2.5,2.5) + theme_prism()+
  guides(color = FALSE)+
    geom_text_repel(aes(label = ID), size = 2)

DT::datatable(hyp_te_h) %>%
  formatRound(which(sapply(hyp_te_h,is.numeric)), digits = 3)



ggsave(width = 4, height = 4, filename = here("plots/fig3/hypoxia_halmarks.pdf"))


hyp_h_table <- hyp_te_h %>% 
  mutate(direction = ifelse(eif3d_NES < 0 & eif3e_NES < 0, "down_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES > 0, "up_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES < 0, "up_3d_down_3e",
                            ifelse(eif3d_NES < 0 & eif3e_NES > 0, "up_3e_down_3d", "NA")))))

write_csv(hyp_h_table, here("counts/fig3/hyp_hallmarks_table.csv"))





```



### KEGG

```{r GSEA KEGG}
eif3d_te_kegg <- GSEA(
  geneList = eif3d_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^KEGG', gs_name))
  )

eif3e_te_kegg <- GSEA(
  geneList = eif3e_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^KEGG', gs_name))
  )

hyp_te_kegg <- dplyr::full_join(
  eif3d_te_kegg@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3d_", .), is.numeric),
  eif3e_te_kegg@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3d_qvalue < 0.05 | eif3e_qvalue < 0.05) %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = eif3e_NES - eif3d_NES)

hyp_te_kegg$ID <- gsub(pattern = "KEGG_", replacement = "", x = hyp_te_kegg$ID)

ggscatter(data = hyp_te_kegg,
          x = "eif3d_NES",
          y = "eif3e_NES",
          label = "ID",
          title = "KEGG pathways - hypoxia",
          repel = TRUE,
          font.label = 6) +
  scale_color_continuous(type = "viridis") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5,2.5) +
  xlim(-2.5,2.5) +
  theme_prism()


#DT::datatable(hyp_te_kegg) %>%
 # formatRound(which(sapply(hyp_te_kegg,is.numeric)), digits = 3) 




ggsave(width = 4, height = 4, filename = here("plots/fig3/hypoxia_kegg.pdf"))


hyp_kegg_table <- hyp_te_kegg %>% 
  mutate(direction = ifelse(eif3d_NES < 0 & eif3e_NES < 0, "down_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES > 0, "up_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES < 0, "up_3d_down_3e",
                            ifelse(eif3d_NES < 0 & eif3e_NES > 0, "up_3e_down_3d", "NA")))))

write_csv(hyp_kegg_table, here("counts/fig3/hyp_kegg_table.csv"))


```


### REACTOME

```{r GSEA REACTOME}
eif3d_te_reactome <- GSEA(
  geneList = eif3d_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^REACTOME', gs_name))
  )

eif3e_te_reactome <- GSEA(
  geneList = eif3e_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^REACTOME', gs_name))
  )

hyp_te_react <- dplyr::full_join(
  eif3d_te_reactome@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3d_", .), is.numeric),
  eif3e_te_reactome@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3d_qvalue < 0.05 | eif3e_qvalue < 0.05)  %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = eif3e_NES - eif3d_NES)

hyp_te_react$ID <- gsub(pattern = "REACTOME_", replacement = "", x = hyp_te_react$ID)

ggscatter(data = hyp_te_react,
          x = "eif3d_NES",
          y = "eif3e_NES",
          label = "ID",
          repel = TRUE,
          title = "Reactome pathways - hypoxia",
          font.label = 6) +
  scale_color_continuous(type = "viridis") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5,2.5) +
  xlim(-2.5,2.5) +
  theme_prism()


# DT::datatable(hyp_te_react) %>%
#   formatRound(which(sapply(hyp_te_react,is.numeric)), digits = 3) 



ggsave(width = 4, height = 4, filename = here("plots/fig3/hypoxia_reactome.pdf"))


hyp_react_table <- hyp_te_react %>% 
  mutate(direction = ifelse(eif3d_NES < 0 & eif3e_NES < 0, "down_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES > 0, "up_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES < 0, "up_3d_down_3e",
                            ifelse(eif3d_NES < 0 & eif3e_NES > 0, "up_3e_down_3d", "NA")))))

write_csv(hyp_react_table, here("counts/fig3/hyp_react_table.csv"))
```


# Extracting RNA changes in with 3e and 3d knockdown in normoxia and hypoxia

# Differential gene expression to compare:
## 2. Normoxia KD effect
## 3. Hypoxia KD effect

## 2 and 3

1. sictrl to eIF3e
2. sictrl to eIF3d

```{r}
# comparisons -------------------------------------------------------------------

control_vector <- "sictrl"
test_vector <- c("si3e", "si3d")
condition_vector <- c("normoxia", "hypoxia")


# set shell dataframes ----------------------------------------------------------

rna_comparisons <- list()
sig_rna_comparisons <- list()
big_sig_rna_comparisons <- list()
big_rna_comparisons <- list()
counts <- tribble(~condition, ~sample, ~RNAupreg, ~RNAdownreg)
rna_values <-list()

for(i in condition_vector) {
  
    metadata_cond <- metadata_kds %>% 
    filter(condition== i) %>% 
      filter(library=="rna")
    
    metadata_cond$treatment <- as.factor(metadata_cond$treatment)
    
    metadata_cond$treatment <- factor(metadata_cond$treatment, levels=c("sictrl", "si3e", "si3d"))
    
    levels(metadata_cond$treatment)
    

for (j in test_vector) {
# set up data -------------------------------------------------------------------

  
  # i = "hypoxia"
  # j = "si3e"
  
  metadata_loop <- metadata_cond %>% 
  filter(treatment == "sictrl" | treatment == j)
  
  # make sample info from metadata
  sample_info <- metadata_loop %>% dplyr::select(condition, treatment, rep, library, sample_id)
  
  
  sample_info <- sample_info %>% 
    mutate("name" = 
  paste(sample_info$condition, sample_info$treatment, sample_info$library, sample_info$rep, sep = "_"))
  
  
  
    sample_info$treatment <- relevel(x = factor(sample_info$treatment), ref = "sictrl")

  #filter AllCounts for i
  
  names <- as.character(sample_info$name)
    
  allCounts_subset <- allCounts[, colnames(allCounts) %in% names]
  

# Deseq analysis ----------------------------------------------------------------
  
  ddsMat <- DESeqDataSetFromMatrix(countData = allCounts_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  ddsMat <- DESeq(ddsMat)

  res_rna_loop <- results(ddsMat, name=resultsNames(ddsMat)[2], cooksCutoff=FALSE, independentFiltering=FALSE)
  
  print(resultsNames(ddsMat)[2])

  res_rna_loop <- lfcShrink(dds = ddsMat, coef = resultsNames(ddsMat)[2], res = res_rna_loop) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA)  
  
  sig_rna_loop <- res_rna_loop %>% 
    filter(padj < 0.05)
  
  sig_rna_loop_up <- sig_rna_loop %>% 
    filter(log2FoldChange>0)
  
  sig_rna_loop_down <- sig_rna_loop %>% 
    filter(log2FoldChange<0)
  
  sig_rna_comparisons[[j]] <- sig_rna_loop

  rna_comparisons[[j]] <- res_rna_loop

  
# visualization -----------------------------------------------------------------
  
  volcano <- EnhancedVolcano(toptable = res_rna_loop,
                x = "log2FoldChange",
                y = "padj",
                lab = res_rna_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = paste0(i, j, "_RNA_volcano"),
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano)


# count table -------------------------------------------------------------------

up <- res_rna_loop %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

down <- res_rna_loop %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()



new_df <- data.frame(i, j, up, down) %>% 
  rename("sample" = "j") %>% 
rename("condition" = "i")

counts <- rbind(counts, new_df)

}
    big_sig_rna_comparisons[[i]] <- sig_rna_comparisons

  big_rna_comparisons[[i]] <- rna_comparisons
}

```

Export rna lists as excel docs

```{r}
library(openxlsx)

wb <- createWorkbook()

# Loop through each category and dataframe to add them as sheets
for (category in names(big_rna_comparisons)) {
  for (df_name in names(big_rna_comparisons[[category]])) {
    # Create a sheet name based on category and dataframe name
    sheet_name <- paste(category, df_name, sep = "_")
    # Add the dataframe to the workbook
    addWorksheet(wb, sheet_name)
    writeData(wb, sheet = sheet_name, x = big_rna_comparisons[[category]][[df_name]])
  }
}

# Save the workbook to an Excel file
saveWorkbook(wb, file = here("counts/fig3/rna_values_after_KD.xlsx"), overwrite = TRUE)

```

# now export sig only rna changes
```{r}
wb <- createWorkbook()

# Loop through each category and dataframe to add them as sheets
for (category in names(big_sig_rna_comparisons)) {
  for (df_name in names(big_sig_rna_comparisons[[category]])) {
    # Create a sheet name based on category and dataframe name
    sheet_name <- paste(category, df_name, sep = "_")
    # Add the dataframe to the workbook
    addWorksheet(wb, sheet_name)
    writeData(wb, sheet = sheet_name, x = big_sig_rna_comparisons[[category]][[df_name]])
  }
}

# Save the workbook to an Excel file
saveWorkbook(wb, file = here("counts/fig3/rna_values_after_KD.xlsx"), overwrite = TRUE)

```


