---
title: "fig6_3e_a209"
author: "Kate"
date: "2024-06-12"
output: html_document
editor_options: 
  chunk_output_type: console
---

# a209 TE changes in acute hypoxic response
# bar chart of te changes in normoxia and hypoxia, indicating amplification in hypoxia
# odds ratio comparing a209 and 3e in normoxia then in hypoxia
# venn diagram of overlap of 3e and 209 in normoxia, up and down separated, and then in hypoxia
# pathway analysis of overlap


```{r install packages, echo=FALSE}
library(here)
library(janitor)
library(tximport)
library(tidyverse)
library(EnhancedVolcano)
library(msigdbr)
library(viridis)
library(gridExtra)
library(ggthemes)
library(ggrepel)
library(scales)
library(DESeq2)
library(limma)
library(clusterProfiler)
library(plotly)
library(ggokabeito)
library(ggvenn)
library(ggpubr)
library(DT)
library(edgeR)
library(cowplot)
library(ggExtra)
library(ggprism)
library(GeneOverlap)
library(writexl)

```

colors
```{r}

a209_col = "#CC79A7"
si3e_col = "#56B4E9"
si3d_col = "#E69F00"
```



## Load and filter metadata
```{r}
metadata <- read_csv(here("metadata.csv"))

metadata$treatment <- factor(metadata$treatment, levels = c("sictrl", "si3e", "si3d", "DMSO", "a209"))
metadata$condition <- factor(metadata$condition, levels = c("normoxia", "hypoxia"))
metadata$library <- factor(metadata$library, levels= c("rna", "ribo"))


metadata_treat <- metadata


```

## load salmon files
```{r import and filter, message=FALSE, warning=FALSE}
# create the tx2 gene file for tximport
geneInfo <- read_csv(here("accessories","gencode.v26.primary.info.csv.gz"), col_names = F) %>%
  filter(!str_detect(X2, "pre_"))

colnames(geneInfo) <- c("gene_id","transcript_id","biotype","symbol")

#map transcripts to gene
tx2gene <- geneInfo[,c(2,1)]
gene_symbol <- geneInfo %>% dplyr::select(gene_id, symbol) %>% unique()

#load quant files based on metadata names
myquantfiles <- paste(here(), "/salmon/",
                      metadata_treat$sample_id,
                      "/quant.sf.gz",
                      sep = "")

names(myquantfiles) <- paste(metadata_treat$condition,
                             metadata_treat$treatment,
                             metadata_treat$library,
                             metadata_treat$rep,
                             sep = "_")


myTxi <- tximport(myquantfiles, type = "salmon", tx2gene = tx2gene)



#only keeping rows with enough counts - want around 10-15k genes

# hist(log2(rowSums(myTxi$counts)), breaks = 40)
keepGenes <- rownames(myTxi$counts[log2(rowSums(myTxi$counts)) > 7,])

pc <-geneInfo %>%
  filter(biotype =="protein_coding") %>%
  pull(gene_id) %>%
  unique()
pc_keep <- pc[pc %in% keepGenes]

# remove non-polyadenylated mRNAs
nonpolyA <- readxl::read_xlsx(path = here("accessories","polyAminus.xlsx"), sheet = 1) %>% 
   pull(Gene)

nonpolyA_ID <- geneInfo %>%
  filter(symbol %in% nonpolyA) %>% 
  pull(gene_id) %>% 
  unique()

# we want to remove any genes that match these annotation categories # before we do DE analysis
blacklist_smRNA <- c("snoRNA","miRNA","snRNA","rRNA","Mt_tRNA","scaRNA","Mt_tRNA","Mt_rRNA")

# there are 5,355 genes we want to exclude
blacklist_smRNA_ID <-geneInfo %>%
  filter(biotype %in% blacklist_smRNA) %>% 
  pull(gene_id) %>% 
  unique()

genes_to_kill <- unique(c(blacklist_smRNA_ID,nonpolyA_ID))


```

### Set up translational efficiency analysis
- RNA and ribo counts are separated out

```{r deltaTE count setup, message=FALSE, warning=FALSE}
# separate out RNA counts versus ribo count
# from https://currentprotocols.onlinelibrary.wiley.com/doi/full/10.1002/cpmb.108

allCounts <- myTxi$counts %>%
  as.data.frame() %>% 
  rownames_to_column(var = "gene") %>% 
  filter(gene %in% keepGenes) %>%
  filter(!gene %in% genes_to_kill) %>%
  column_to_rownames(var = "gene") %>% 
  as.matrix() %>%
  round(.,0)


ribo_counts <- as.matrix(allCounts[,grep(pattern = "ribo",colnames(allCounts))])
rna_counts <- as.matrix(allCounts[,grep(pattern = "rna",colnames(allCounts))])
```

# Differential gene expression to compare:
## 1. Normoxia -> Hypoxia
## 2. Normoxia KD effect
## 3. Hypoxia KD effect


# acute hypoxic response

```{r}
loop_vector <- c("sictrl", "si3e", "DMSO", "a209")
```

# DeSeq for all TE comparisons in a loop

```{r}
# set shell dataframes ----------------------------------------------------------

all_te <- list()
all_counts <- tribble(~sample, ~TEupreg, ~TEdownreg)

for (i in loop_vector) {

# set up data -------------------------------------------------------------------
  
  #i="si3e"
  
  metadata_loop <- metadata_treat %>% 
  filter(treatment == i)
  
  # make sample info from metadata
  sample_info <- metadata_loop %>% dplyr::select(condition, treatment, rep, library, sample_id)
  
  sample_info <- sample_info %>% 
    mutate("name" = 
  paste(sample_info$condition, sample_info$treatment, sample_info$library, sample_info$rep, sep = "_"))
  
  # colnames(sample_info) <- c("condition", "treatment","rep", "library", "sample_id", "name")
  sample_info$library <- relevel(x = factor(sample_info$library), ref = "rna")
    sample_info$condition <- relevel(x = factor(sample_info$condition), ref = "normoxia")

  #filter AllCounts for i
    
    names <- as.character(sample_info$name)
    
    allCounts_subset <- allCounts %>% 
      as.data.frame() 
    
    allCounts_subset <- allCounts_subset[colnames(allCounts_subset) %in% names]

# Deseq analysis ----------------------------------------------------------------
  
  ddsMat <- DESeqDataSetFromMatrix(countData = allCounts_subset, 
                                 colData = sample_info, 
                                 design = ~condition+library+condition:library)
  ddsMat <- DESeq(ddsMat)

  res_te_loop <- results(ddsMat, name=resultsNames(ddsMat)[4], cooksCutoff=FALSE, independentFiltering=FALSE)

  res_te_loop <- lfcShrink(dds = ddsMat, coef = resultsNames(ddsMat)[4], res = res_te_loop) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA)  

  all_te[[i]] <- res_te_loop
  
# visualization -----------------------------------------------------------------
  
  volcano <- EnhancedVolcano(toptable = res_te_loop,
                x = "log2FoldChange",
                y = "padj",
                lab = res_te_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = paste0(i, "_TE_volcano"),
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano)


# count table -------------------------------------------------------------------

upreg <- res_te_loop %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_te_loop %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

all_counts <- rbind(all_counts, data.frame(i,upreg, downreg))


}

# writexl::write_xlsx(x = res_te, path = here("counts","si3e_normoxiatohypoxia_TE.xlsx"), col_names = T)


```

## Supplemental: Differential RNA and RIBO loops

### Differential RNA loop

```{r RNA, fig.height=9, fig.width=10, message=FALSE, warning=FALSE}

# set shell dataframes ----------------------------------------------------------

all_rna <- list()
all_counts_rna <- tribble(~sample, ~RNAupreg, ~RNAdownreg)

for (i in loop_vector) {

# set up data -------------------------------------------------------------------
  
  metadata_loop <- metadata_treat %>% 
  filter(treatment == i)
  
  # make sample info from metadata
  sample_info <- metadata_loop %>% 
    filter(library == "rna")
  
  sample_info <- sample_info %>% 
    mutate("name" = 
  paste(sample_info$condition, sample_info$treatment, sample_info$library, sample_info$rep, sep = "_"))
     sample_info$condition <- relevel(x = factor(sample_info$condition), ref = "normoxia")
  
  levels(sample_info$condition)
  levels(sample_info$treatment)
  
  #filter AllCounts for i
  
  
  names <- as.character(sample_info$name)
    
    rna_subset <- allCounts %>% 
      as.data.frame() 
    
   rna_subset <- rna_subset[colnames(rna_subset) %in% names]

  
# Deseq analysis ----------------------------------------------------------------
  
  ddsMat_rna <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~condition)
  ddsMat_rna <- DESeq(ddsMat_rna)
  

  res_rna_loop <- results(ddsMat_rna, name=resultsNames(ddsMat_rna)[2], cooksCutoff=FALSE, independentFiltering=FALSE)

  res_rna_loop <- lfcShrink(dds = ddsMat_rna, coef = resultsNames(ddsMat_rna)[2], res = res_rna_loop) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA)  

  all_rna[[i]] <- res_rna_loop
  

  
  
  # visualization -----------------------------------------------------------------
  
  volcano_rna <- EnhancedVolcano(toptable = res_rna_loop,
                x = "log2FoldChange",
                y = "padj",
                lab = res_rna_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = paste0(i, "_rna_volcano"),
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano_rna)


# count table -------------------------------------------------------------------

upreg_rna <- res_rna_loop %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg_rna <- res_rna_loop %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()


all_counts_rna <- rbind(all_counts_rna, data.frame(i,upreg_rna, downreg_rna))

}

  # writexl::write_xlsx(x = res_te, path = here("counts","si3e_normoxiatohypoxia_TE.xlsx"), col_names = T)


#export sictrl


writexl::write_xlsx(x = all_rna$sictrl, path = here("counts/rna_sictrl_normoxia_to_hypoxia.xlsx"), col_names = T)



```


### Differential RIBO loop

```{r diff ribo, fig.height=9, fig.width=10, message=FALSE, warning=FALSE}

# set shell dataframes ----------------------------------------------------------

all_ribo <- list()
all_counts_ribo <- tribble(~sample, ~RIBOupreg, ~RIBOdownreg)

for (i in loop_vector) {

# set up data -------------------------------------------------------------------
  
  metadata_loop <- metadata_treat %>% 
  filter(treatment == i)
  
  # make sample info from metadata
  sample_info <- metadata_loop %>% 
    filter(library == "ribo")
  
  sample_info <- sample_info %>% 
    mutate("name" = 
  paste(sample_info$condition, sample_info$treatment, sample_info$library, sample_info$rep, sep = "_"))
  
     sample_info$condition <- relevel(x = factor(sample_info$condition), ref = "normoxia")
  
  
  #filter AllCounts for i
  
  
  names <- as.character(sample_info$name)
    
    ribo_subset <- allCounts %>% 
      as.data.frame() 
    
   ribo_subset <- ribo_subset[colnames(ribo_subset) %in% names]

  
# Deseq analysis ----------------------------------------------------------------
  
  ddsMat_ribo <- DESeqDataSetFromMatrix(countData = ribo_subset, 
                                 colData = sample_info, 
                                 design = ~condition)
  ddsMat_ribo <- DESeq(ddsMat_ribo)
  

  res_ribo_loop <- results(ddsMat_ribo, name=resultsNames(ddsMat_ribo)[2], cooksCutoff=FALSE, independentFiltering=FALSE)

  res_ribo_loop <- lfcShrink(dds = ddsMat_ribo, coef = resultsNames(ddsMat_ribo)[2], res = res_ribo_loop) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA)  

  all_ribo[[i]] <- res_ribo_loop
  
  
  # visualization -----------------------------------------------------------------
  
  volcano_ribo <- EnhancedVolcano(toptable = res_ribo_loop,
                x = "log2FoldChange",
                y = "padj",
                lab = res_ribo_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = paste0(i, "_ribo_volcano"),
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano_ribo)


# count table -------------------------------------------------------------------

upreg_ribo <- res_ribo_loop %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg_ribo <- res_ribo_loop %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()


all_counts_ribo <- rbind(all_counts_ribo, data.frame(i,upreg_ribo, downreg_ribo))

}




```


### Total counts

```{r}
total_counts <- left_join(all_counts, all_counts_rna, by="i")
total_counts <- left_join(total_counts, all_counts_ribo, by="i")
total_counts

# visualize ---------------------------------------------------------------------

long_counts <- total_counts %>% 
  pivot_longer(2:5, names_to = "category", values_to= "counts") %>% 
  filter(i=="sictrl")

long_counts$category <- factor(long_counts$category, levels = c("downreg_rna", "upreg_rna", "downreg", "upreg"))

long_counts <- long_counts %>% 
  mutate(color = ifelse(category== "upreg_rna" | category== "upreg", "red", "blue"))


ggbarplot(long_counts, x="category", y="counts", fill="color")+
  scale_fill_identity()


```

## One table for all genes and their acute changes
# change names HERE

```{r}

sictrl_te <- all_te$sictrl[,c(1,2,4,7)]
colnames(sictrl_te)[3] ="LFC_te_sictrl"
colnames(sictrl_te)[4] ="padj_te_sictrl"

sictrl_ribo <- all_ribo$sictrl[,c(1,4,7)]
colnames(sictrl_ribo)[2] ="LFC_ribo_sictrl"
colnames(sictrl_ribo)[3] ="padj_ribo_sictrl"

sictrl_rna <- all_rna$sictrl[,c(1,4,7)]
colnames(sictrl_rna)[2] ="LFC_rna_sictrl"
colnames(sictrl_rna)[3] ="padj_rna_sictrl"

si3e_te <- all_te$si3e[,c(1,4,7)]
colnames(si3e_te)[2] ="LFC_te_si3e"
colnames(si3e_te)[3] ="padj_te_3e"

si3e_ribo <- all_ribo$si3e[,c(1,4,7)]
colnames(si3e_ribo)[2] ="LFC_ribo_si3e"
colnames(si3e_ribo)[3] ="padj_ribo_si3e"

si3e_rna <- all_rna$si3e[,c(1,4,7)]
colnames(si3e_rna)[2] ="LFC_rna_si3e"
colnames(si3e_rna)[3] ="padj_rna_si3e"

DMSO_te <- all_te$DMSO[,c(1,4,7)]
colnames(DMSO_te)[2] ="LFC_te_DMSO"
colnames(DMSO_te)[3] ="padj_te_DMSO"

DMSO_ribo <- all_ribo$DMSO[,c(1,4,7)]
colnames(DMSO_ribo)[2] ="LFC_ribo_DMSO"
colnames(DMSO_ribo)[3] ="padj_ribo_DMSO"

DMSO_rna <- all_rna$DMSO[,c(1,4,7)]
colnames(DMSO_rna)[2] ="LFC_rna_DMSO"
colnames(DMSO_rna)[3] ="padj_rna_DMSO"


a209_te <- all_te$a209[,c(1,4,7)]
colnames(a209_te)[2] ="LFC_te_a209"
colnames(a209_te)[3] ="padj_te_a209"

a209_ribo <- all_ribo$a209[,c(1,4,7)]
colnames(a209_ribo)[2] ="LFC_ribo_a209"
colnames(a209_ribo)[3] ="padj_ribo_a209"

a209_rna <- all_rna$a209[,c(1,4,7)]
colnames(a209_rna)[2] ="LFC_rna_a209"
colnames(a209_rna)[3] ="padj_rna_a209"


#bind cols ----------------------------------------------------------------------

all_changes <- purrr::reduce(list(sictrl_te,sictrl_ribo,sictrl_rna,si3e_te,si3e_ribo,si3e_rna,DMSO_te,DMSO_ribo,DMSO_rna, a209_te,a209_ribo,a209_rna), dplyr::left_join, by = 'gene_id')
```

## Visualize acute hypoxic response with histograms

## Acute hypoxic changes - sictrl

```{r}
ctrl <- all_changes %>% 
  filter(padj_rna_sictrl < 0.05 | padj_ribo_sictrl < 0.05)

ctrl_acute <- ggplot(data = ctrl, aes(x = LFC_rna_sictrl, y = LFC_ribo_sictrl)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote("si-CTRL LFC"~"RIBO: hypoxia vs normoxia")) +
  xlab(bquote("si-CTRL LFC"~"RNA: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 16)+
  theme_prism()

ctrl_acute <- ggMarginal(ctrl_acute, type = "histogram",
           xparams = list(binwidth = .1, fill = "grey"),
           yparams = list(binwidth = .1, fill = "red")
           )

ctrl_acute

#ggsave(here("plots/sictrlacute.pdf", width=4,height=4))

```


## TE changes - 3e KD

```{r}

sig_sictrl <- all_changes %>% 
  filter(padj_te_sictrl <0.05)

scatter <- ggplot(data = sig_sictrl, aes(x = LFC_te_sictrl, y = LFC_te_si3e)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote("LFC si-eIF3e"~"TE: hypoxia vs normoxia")) +
  xlab(bquote("LFC si-CTRL"~"TE: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 9)+
  theme_prism()

scatter <- ggMarginal(scatter, type = "histogram",
           xparams = list(binwidth = .1, fill = "grey"),
           yparams = list(binwidth = .1, fill = si3e_col)
           )

scatter

ggsave(here("plots/acute_3e.pdf"), scatter, width=4, height=4)

```



## Acute hypoxic changes - dmso (supplement)
```{r}
dmso <- all_changes %>% 
  filter(padj_rna_DMSO < 0.05 | padj_ribo_DMSO < 0.05)

dmso_acute <- ggplot(data = dmso, aes(x = LFC_rna_DMSO, y = LFC_ribo_DMSO)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote("LFC DMSO"~"RIBO: hypoxia vs normoxia")) +
  xlab(bquote("LFC DMSO"~"RNA: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 16)+
  theme_prism()

ctrl_dmso <- ggMarginal(dmso_acute, type = "histogram",
           xparams = list(binwidth = .1, fill = "grey"),
           yparams = list(binwidth = .1, fill = "red")
           )

ctrl_dmso

# ggsave(here("plots/dmso.pdf"), width=4, height=4)


```


## TE changes - a209
```{r}

sig_DMSO <- all_changes %>% 
  filter(padj_te_DMSO <0.05)

a209_col = "#CC79A7"


scatter <- ggplot(data = sig_DMSO, aes(x = LFC_te_DMSO, y = LFC_te_a209)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote("LFC a209"~"TE: hypoxia vs normoxia")) +
  xlab(bquote("LFC DMSO"~"TE: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 16)+
  theme_prism()


scatter <- ggMarginal(scatter, type = "histogram",
           xparams = list(binwidth = .1, fill = "grey"),
           yparams = list(binwidth = .1, fill = a209_col)
           )

scatter

ggsave(here("plots/fig6/a209.pdf"), width=4, height=4)

```




# TE changes for normoxia first then hypoxia --------------------------------------

condition vector changes to analyze normoxia and hypoxia
# Make a loop to do:
1. sictrl to eIF3e
2. sictrl to eIF3d
3. DMSO to a209

```{r}
# comparisons -------------------------------------------------------------------

control_vector <- c("sictrl", "DMSO")
test_vector <- c("si3e", "si3d")
condition_vector <- c("normoxia", "hypoxia")


# set shell dataframes ----------------------------------------------------------

comparisons <- list()
big_comparisons <- list()
counts <- tribble(~condition, ~sample, ~TEupreg, ~TEdownreg)
te_values <-list()

metadata_treat <- metadata

levels(metadata_treat$treatment)

for (i in condition_vector) {
  
    metadata_cond <- metadata_treat %>% 
    filter(condition== i)
    
    levels(metadata_cond$treatment)
  
for (j in test_vector) {
# set up data -------------------------------------------------------------------

  
  # i = "hypoxia"
  # j = "si3d"
  
  metadata_loop <- metadata_cond %>% 
  filter(treatment == "sictrl" | treatment == j)
  
  levels(metadata_loop$treatment)
  
  # make sample info from metadata
   sample_info <- metadata_loop %>% dplyr::select(condition, treatment, rep, library, sample_id)
   
   levels(sample_info$treatment)
  
  
  sample_info <- sample_info %>% 
    mutate("name" = 
  paste(sample_info$condition, sample_info$treatment, sample_info$library, sample_info$rep, sep = "_"))
  
    sample_info$library <- relevel(x = factor(sample_info$library), ref = "rna")

    levels(sample_info$library)
  #filter AllCounts for i
  
  names <- as.character(sample_info$name)
    
  allCounts_subset <- allCounts[, colnames(allCounts) %in% names]
  

# Deseq analysis ----------------------------------------------------------------
  
  ddsMat <- DESeqDataSetFromMatrix(countData = allCounts_subset, 
                                 colData = sample_info, 
                                 design = ~treatment+library+treatment:library)
  ddsMat <- DESeq(ddsMat)
  
  print(levels(sample_info$condition))
    print(levels(sample_info$treatment))
      print(levels(sample_info$library))


  
  print(resultsNames(ddsMat)[4])

  res_te_loop <- results(ddsMat, name=resultsNames(ddsMat)[4], cooksCutoff=FALSE, independentFiltering=FALSE)

  res_te_loop <- lfcShrink(dds = ddsMat, coef = resultsNames(ddsMat)[4], res = res_te_loop) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA)  

  comparisons[[j]] <- res_te_loop

  
# visualization -----------------------------------------------------------------
  
  volcano <- EnhancedVolcano(toptable = res_te_loop,
                x = "log2FoldChange",
                y = "padj",
                lab = res_te_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = paste0(i, j, "_TE_volcano"),
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano)


# count table -------------------------------------------------------------------

up <- res_te_loop %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

down <- res_te_loop %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()



new_df <- data.frame(i, j, up, down) %>% 
  rename("sample" = "j") %>% 
rename("condition" = "i")

counts <- rbind(counts, new_df)

}

  big_comparisons[[i]] <- comparisons


# DMSO a209 --------------------------------------------------------------------

  metadata_loop <- metadata_cond %>% 
  filter(treatment == "DMSO" | treatment == "a209")
  
  levels(metadata_loop$treatment)
  
  # make sample info from metadata
  sample_info <- metadata_loop %>% dplyr::select(condition, treatment, rep, library, sample_id)
  
  print(levels(sample_info$condition))
    print(levels(sample_info$treatment))
      print(levels(sample_info$library))
      
      
  sample_info <- sample_info %>% 
    mutate("name" = 
  paste(sample_info$condition, sample_info$treatment, sample_info$library, sample_info$rep, sep = "_"))
  
  # colnames(sample_info) <- c("hypoxia", "treatment","rep", "library", "sample_id", "name")
  sample_info$library <- relevel(x = factor(sample_info$library), ref = "rna")
  
  #filter AllCounts for i
  
  names <- as.character(sample_info$name)
    
  allCounts_subset <- allCounts[, colnames(allCounts) %in% names]
  

# Deseq analysis ----------------------------------------------------------------
  
  ddsMat <- DESeqDataSetFromMatrix(countData = allCounts_subset, 
                                 colData = sample_info, 
                                 design = ~treatment+library+treatment:library)
  ddsMat <- DESeq(ddsMat)

  res_te_loop <- results(ddsMat, name=resultsNames(ddsMat)[4], cooksCutoff=FALSE, independentFiltering=FALSE)
  
  print(resultsNames(ddsMat)[4])

  res_te_loop <- lfcShrink(dds = ddsMat, coef = resultsNames(ddsMat)[4], res = res_te_loop) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA)  

  comparisons[["a209"]] <- res_te_loop
  
# export to excel ---------------------------------------------------------------
  
   writexl::write_xlsx(x = res_te_loop, path = here(paste0("counts/", i, "_", j, "_", "tecounts", ".xlsx")), col_names = T)

# visualization -----------------------------------------------------------------
  
  volcano <- EnhancedVolcano(toptable = res_te_loop,
                x = "log2FoldChange",
                y = "padj",
                lab = res_te_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = paste0(i, "a209", "_TE_volcano"),
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano)


# count table -------------------------------------------------------------------

up <- res_te_loop %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

down <- res_te_loop %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

k = "a209"

new_df <- data.frame(i, k, up, down) %>% 
  rename("sample" = "k") %>% 
  rename("condition" = "i")

counts <- rbind(counts, new_df)

  big_comparisons[[i]] <- comparisons

}



```

Export TE
```{r}
write_csv(big_comparisons$normoxia$si3d, here("counts/fig6/norm_3d_all.csv"))
write_csv(big_comparisons$normoxia$si3e, here("counts/fig6/norm_3e_all.csv"))
write_csv(big_comparisons$hypoxia$si3d, here("counts/fig6/hyp_3d_all.csv"))
write_csv(big_comparisons$hypoxia$si3e, here("counts/fig6/hyp_3e_all.csv"))
write_csv(big_comparisons$normoxia$a209, here("counts/fig6/norm_a209_all.csv"))
write_csv(big_comparisons$hypoxia$a209, here("counts/fig6/hyp_a209_all.csv"))
```






# pivot counts
```{r}

long_counts <- counts %>% 
  pivot_longer(3:4, names_to = "direction", values_to = "counts")



```


#RNA loop
```{r}
# comparisons -------------------------------------------------------------------

test_vector <- c("si3e", "si3d")
condition_vector <- c("normoxia", "hypoxia")


# set shell dataframes ----------------------------------------------------------

rna_comparisons <- list()
big_rna_comparisons <- list()
rna_counts <- tribble(~condition, ~sample, ~RNAupreg, ~RNAdownreg)
rna_values <-list()


metadata_treat <- metadata


for (i in condition_vector) {
  
    metadata_cond <- metadata_treat %>% 
    filter(condition== i) %>% 
      filter(library=="rna")
  
for (j in test_vector) {
# set up data -------------------------------------------------------------------

  
  # i = "hypoxia"
  # j = "si3d"
  
  levels(metadata_cond$treatment)
  
  metadata_loop <- metadata_cond %>% 
  filter(treatment == "sictrl" | treatment == j)
  
  # make sample info from metadata
   sample_info <- metadata_loop %>% dplyr::select(condition, treatment, rep, library, sample_id)
  
  
  sample_info <- sample_info %>% 
    mutate("name" = 
  paste(sample_info$condition, sample_info$treatment, sample_info$library, sample_info$rep, sep = "_"))
  
   print(levels(sample_info$condition))
    print(levels(sample_info$treatment))
      print(levels(sample_info$library))
  

  #filter AllCounts for i
  
  names <- as.character(sample_info$name)
  
  rna_subset <- allCounts %>% as.data.frame()
    
  rna_subset <- rna_subset[, colnames(rna_subset) %in% names]
  

# Deseq analysis ----------------------------------------------------------------
  
  ddsMat <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  ddsMat <- DESeq(ddsMat)

  res_rna_loop <- results(ddsMat, name=resultsNames(ddsMat)[2], cooksCutoff=FALSE, independentFiltering=FALSE)
  
  print(resultsNames(ddsMat)[2])

  res_rna_loop <- lfcShrink(dds = ddsMat, coef = resultsNames(ddsMat)[2], res = res_rna_loop) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA)  

  rna_comparisons[[j]] <- res_rna_loop
  
   writexl::write_xlsx(x = res_rna_loop, path = here(paste0("counts/", i, "_", j, "_", "rnacounts", ".xlsx")), col_names = T)

  
# visualization -----------------------------------------------------------------
  
  volcano <- EnhancedVolcano(toptable = res_rna_loop,
                x = "log2FoldChange",
                y = "padj",
                lab = res_rna_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = paste0(i, j, "_RNA_volcano"),
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano)


# count table -------------------------------------------------------------------

up <- res_rna_loop %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

down <- res_rna_loop %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()



new_df <- data.frame(i, j, up, down) %>% 
  rename("sample" = "j") %>% 
rename("condition" = "i")

rna_counts <- rbind(rna_counts, new_df)

}

  big_rna_comparisons[[i]] <- rna_comparisons




# DMSO a209 --------------------------------------------------------------------

  metadata_loop <- metadata_cond %>% 
  filter(treatment == "DMSO" | treatment == "a209")
  
  # make sample info from metadata
  sample_info <- metadata_loop %>% dplyr::select(condition, treatment, rep, library, sample_id)
  
  sample_info <- sample_info %>% 
    mutate("name" = 
  paste(sample_info$condition, sample_info$treatment, sample_info$library, sample_info$rep, sep = "_"))
  
   print(levels(sample_info$condition))
    print(levels(sample_info$treatment))
      print(levels(sample_info$library))
  
  
  #filter AllCounts for i
  
  names <- as.character(sample_info$name)
  
  rna_subset <- allCounts %>% as.data.frame()
    
  rna_subset <- rna_subset[, colnames(rna_subset) %in% names]
  

# Deseq analysis ----------------------------------------------------------------
  
  ddsMat <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  ddsMat <- DESeq(ddsMat)

  res_rna_loop <- results(ddsMat, name=resultsNames(ddsMat)[2], cooksCutoff=FALSE, independentFiltering=FALSE)
  
  print(resultsNames(ddsMat)[2])

  res_rna_loop <- lfcShrink(dds = ddsMat, coef = resultsNames(ddsMat)[2], res = res_rna_loop) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA)  

  rna_comparisons[["a209"]] <- res_rna_loop
  
# export to excel ---------------------------------------------------------------
  
   # writexl::write_xlsx(x = res_rna_loop, path = here(paste0("counts/", i, "_", "a209", "_", "rnacounts", ".xlsx")), col_names = T)

# visualization -----------------------------------------------------------------
  
  volcano <- EnhancedVolcano(toptable = res_rna_loop,
                x = "log2FoldChange",
                y = "padj",
                lab = res_rna_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = paste0(i, "a209", "_RNA_volcano"),
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano)


# count table -------------------------------------------------------------------

up <- res_rna_loop %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

down <- res_rna_loop %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

k = "a209"

new_df <- data.frame(i, k, up, down) %>% 
  rename("sample" = "k") %>% 
  rename("condition" = "i")

rna_counts <- rbind(counts, new_df)

  big_rna_comparisons[[i]] <- rna_comparisons

}

```

# Flipping RNA analyses to create RNA signatures for figure 8. Looking at 3e signature (3e up) ---------------------------------------------------------

DESeq for a single comparison at a time.
Takeaways: 
- need to use coef rather than contrast argument when using apelgm method for LFC shrinkage
- this means factor levels are critical - need to be checked before each analysis

Normoxia-- si3e compared to sictrl
```{r}

metadata$treatment <- factor(metadata$treatment, levels= c("si3e", "si3d", "sictrl", "a209", "DMSO"))

metadata_loop <- metadata %>% 
  filter(library=="rna") %>% 
  filter(condition=="normoxia") %>% 
  filter(treatment=="sictrl"|treatment=="si3e")

sample_info <- metadata_loop %>% 
    mutate("name" = paste(condition, treatment, library, rep, sep = "_")) %>% 
  droplevels()
  
  names <- as.character(sample_info$name)
  names
  rna_subset <- allCounts %>% as.data.frame()
    
  rna_subset <- rna_subset[, colnames(rna_subset) %in% names]
  
  print(levels(sample_info$treatment))

  
# Deseq analysis ----------------------------------------------------------------
  
  dds <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  dds <- DESeq(dds)
  
  #double check factor levels
 resultsNames(dds)

  res_og <- results(dds, name=resultsNames(dds)[2], cooksCutoff=FALSE, independentFiltering=FALSE) #what i did originally

  res_og_df_3e_norm <- lfcShrink(dds = dds, coef = resultsNames(dds)[2], res = res_og) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA) 

# export to excel ---------------------------------------------------------------
  
   writexl::write_xlsx(x = res_og_df_3e_norm, path = here("counts/fig6/rna_signature_3e_normoxia.xlsx"), col_names = T)

# visualization -----------------------------------------------------------------
  
  volcano <- EnhancedVolcano(toptable = res_og_df_3e_norm,
                x = "log2FoldChange",
                y = "padj",
                lab = res_rna_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = "3e normoxia RNA",
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano)


```
# Now eif3e is up -- do the same for hypoxia


# 3e hypoxia rna
```{r}

metadata$treatment <- factor(metadata$treatment, levels= c("si3e", "si3d", "sictrl", "a209", "DMSO"))

metadata_loop <- metadata %>% 
  filter(library=="rna") %>% 
  filter(condition=="hypoxia") %>% 
  filter(treatment=="sictrl"|treatment=="si3e")

sample_info <- metadata_loop %>% 
    mutate("name" = 
  paste(condition, treatment, library, rep, sep = "_"))
  
  names <- as.character(sample_info$name)
  names
  rna_subset <- allCounts %>% as.data.frame()
    
  rna_subset <- rna_subset[, colnames(rna_subset) %in% names]
  
  print(levels(sample_info$treatment))

  
# Deseq analysis ----------------------------------------------------------------
  
  dds <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  dds <- DESeq(dds)
  
    #double check factor levels
  print(resultsNames(dds))

  res_og <- results(dds, name=resultsNames(dds)[2], cooksCutoff=FALSE, independentFiltering=FALSE) #what i did originally

  res_og_df_hypoxia_3e <- lfcShrink(dds = dds, coef = resultsNames(dds)[2], res = res_og) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA) 

# export to excel ---------------------------------------------------------------
  
   writexl::write_xlsx(x = res_og_df_hypoxia_3e, path = here("counts/fig6/rna_signature_3e_hypoxia.xlsx"), col_names = T)

# visualization -----------------------------------------------------------------
  
  volcano <- EnhancedVolcano(toptable = res_og_df_hypoxia_3e,
                x = "log2FoldChange",
                y = "padj",
                lab = res_rna_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = "3e hypoxia RNA",
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano)


```

## 3d analysis -- normoxia

```{r}

metadata$treatment <- factor(metadata$treatment, levels= c("si3e", "si3d", "sictrl", "a209", "DMSO"))

metadata_loop <- metadata %>% 
  filter(library=="rna") %>% 
  filter(condition=="normoxia") %>% 
  filter(treatment=="sictrl"|treatment=="si3d")

sample_info <- metadata_loop %>% 
    mutate("name" = 
  paste(condition, treatment, library, rep, sep = "_"))
  
  names <- as.character(sample_info$name)
  
  rna_subset <- allCounts %>% as.data.frame()
    
  rna_subset <- rna_subset[, colnames(rna_subset) %in% names]
  
  print(levels(sample_info$treatment))

  
# Deseq analysis ----------------------------------------------------------------
  
  dds <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  dds <- DESeq(dds)
  
  #double check factor levels
  print(resultsNames(dds))

  res_og <- results(dds, name=resultsNames(dds)[2], cooksCutoff=FALSE, independentFiltering=FALSE) #what i did originally

  res_og_df_normoxia_3d <- lfcShrink(dds = dds, coef = resultsNames(dds)[2], res = res_og) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA) 

# export to excel ---------------------------------------------------------------
  
   writexl::write_xlsx(x = res_og_df_normoxia_3d, path = here("counts/fig6/rna_signature_3d_normoxia.xlsx"), col_names = T)

# visualization -----------------------------------------------------------------
  
  volcano <- EnhancedVolcano(toptable = res_og_df_normoxia_3d,
                x = "log2FoldChange",
                y = "padj",
                lab = res_rna_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = "Normoxia 3d RNA",
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano)


```



# 3d hypoxia rna
```{r}

metadata$treatment <- factor(metadata$treatment, levels= c("si3e", "si3d", "sictrl", "a209", "DMSO"))

metadata_loop <- metadata %>% 
  filter(library=="rna") %>% 
  filter(condition=="hypoxia") %>% 
  filter(treatment=="sictrl"|treatment=="si3d")

sample_info <- metadata_loop %>% 
    mutate("name" = 
  paste(condition, treatment, library, rep, sep = "_"))
  
  names <- as.character(sample_info$name)
  names
  rna_subset <- allCounts %>% as.data.frame()
    
  rna_subset <- rna_subset[, colnames(rna_subset) %in% names]
  
  print(levels(sample_info$treatment))

  
# Deseq analysis ----------------------------------------------------------------
  
  dds <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  dds <- DESeq(dds)
  
    #double check factor levels
  print(resultsNames(dds))

  res_og <- results(dds, name=resultsNames(dds)[2], cooksCutoff=FALSE, independentFiltering=FALSE) #what i did originally

  res_og_normoxia_3d <- lfcShrink(dds = dds, coef = resultsNames(dds)[2], res = res_og) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA) 

# export to excel ---------------------------------------------------------------
  
   writexl::write_xlsx(x = res_og_normoxia_3d, path = here("counts/rna_signature_3d_hypoxia.xlsx"), col_names = T)

# visualization -----------------------------------------------------------------
  
  volcano <- EnhancedVolcano(toptable = res_og_normoxia_3d,
                x = "log2FoldChange",
                y = "padj",
                lab = res_rna_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = "Normoxia 3d RNA",
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano)


```

# a209 normoxia
```{r}

metadata$treatment <- factor(metadata$treatment, levels= c("si3e", "si3d", "sictrl", "a209", "DMSO"))

metadata_loop <- metadata %>% 
  filter(library=="rna") %>% 
  filter(condition=="normoxia") %>% 
  filter(treatment=="DMSO"|treatment=="a209")

sample_info <- metadata_loop %>% 
    mutate("name" = 
  paste(condition, treatment, library, rep, sep = "_"))

levels(metadata_loop$treatment)
  
  names <- as.character(sample_info$name)
  names
  rna_subset <- allCounts %>% as.data.frame()
    
  rna_subset <- rna_subset[, colnames(rna_subset) %in% names]
  
  print(levels(sample_info$treatment))

  
# Deseq analysis ----------------------------------------------------------------
  
  dds <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  dds <- DESeq(dds)
  
    #double check factor levels
  print(resultsNames(dds))

  res_og <- results(dds, name=resultsNames(dds)[2], cooksCutoff=FALSE, independentFiltering=FALSE) #what i did originally

  res_og_normoxia_a209 <- lfcShrink(dds = dds, coef = resultsNames(dds)[2], res = res_og) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA) 

# export to excel ---------------------------------------------------------------
  
   writexl::write_xlsx(x = res_og_normoxia_a209, path = here("counts/fig6/rna_signature_a209_normoxia.xlsx"), col_names = T)

# visualization -----------------------------------------------------------------
  
  volcano <- EnhancedVolcano(toptable = res_og_normoxia_a209,
                x = "log2FoldChange",
                y = "padj",
                lab = res_rna_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = "a209_normoxia",
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano)


```



# a209 hypoxia rna
```{r}

metadata$treatment <- factor(metadata$treatment, levels= c("si3e", "si3d", "sictrl", "a209", "DMSO"))

metadata_loop <- metadata %>% 
  filter(library=="rna") %>% 
  filter(condition=="hypoxia") %>% 
  filter(treatment=="DMSO"|treatment=="a209")

sample_info <- metadata_loop %>% 
    mutate("name" = 
  paste(condition, treatment, library, rep, sep = "_"))
  
  names <- as.character(sample_info$name)
  names
  rna_subset <- allCounts %>% as.data.frame()
    
  rna_subset <- rna_subset[, colnames(rna_subset) %in% names]
  
  print(levels(sample_info$treatment))

  
# Deseq analysis ----------------------------------------------------------------
  
  dds <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  dds <- DESeq(dds)
  
    #double check factor levels
  print(resultsNames(dds))

  res_og <- results(dds, name=resultsNames(dds)[2], cooksCutoff=FALSE, independentFiltering=FALSE) #what i did originally

  res_og_hypoxia_a209 <- lfcShrink(dds = dds, coef = resultsNames(dds)[2], res = res_og) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .) %>% 
  filter(!symbol %in% nonpolyA) 

# export to excel ---------------------------------------------------------------
  
   writexl::write_xlsx(x = res_og_hypoxia_a209, path = here("counts/fig6/rna_signature_a209_hypoxia.xlsx"), col_names = T)

# visualization -----------------------------------------------------------------
  
  volcano <- EnhancedVolcano(toptable = res_og_hypoxia_a209,
                x = "log2FoldChange",
                y = "padj",
                lab = res_rna_loop$symbol,
                pCutoff = 0.05,
                xlim = c(-5,5),
                title = "a209 hypoxia RNA",
                subtitle = "",
                legendPosition = "right",
                col = c('grey', 'black', 'blue', 'red3'),
                legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'))

print(volcano)


```


universe
```{r}
# 17988 genes in allCounts RNA
```





#############################

## Continuing with TE analysis


### effect of si3e KD norm and hypoxia (have this in previous code)
```{r}

eif3e_counts <- long_counts %>% 
  filter(sample=="si3e")

eif3e_bar <- ggbarplot(data = eif3e_counts, x = "condition", y = "counts", fill = "direction", palette = c("#2492a6",si3e_col), position = position_dodge(0.75), label = T, title= "eIF3e")+
  theme_cowplot(font_size = 16)

eif3e_bar

ggsave(here("plots/eIF3ebars.pdf"), eif3e_bar, width = 2.5, height = 3)

```



# effect of a209 KD norm and hypoxia
```{r}

a209_counts <- long_counts %>% 
  filter(sample=="a209")

a209_bar <- ggbarplot(data = a209_counts, x = "condition", y = "counts", fill = "direction", palette = c("pink",a209_col), position = position_dodge(0.75), label = T, title= "TE Changes - a209")+
  theme_cowplot(font_size = 16)

a209_bar

ggsave(here("plots/fig6/a209bars.pdf"), a209_bar, width = 3.5, height = 3.5)

```




# Is a209 on target? ------------------------------------------------------------

# compare TE changes between the knockdowns and compound

```{r}

norm_3e <- big_comparisons$normoxia$si3e %>% 
    mutate(condition = "normoxia") 
  
hyp_3e <- big_comparisons$hypoxia$si3e %>% 
    mutate(condition = "hypoxia") 


si3e_single <- rbind(norm_3e, hyp_3e) %>% 
  rename("LFC_3e" ="log2FoldChange") %>%
  rename("padj_3e" = "padj")



norm_3d <- big_comparisons$normoxia$si3d %>% 
    mutate(condition = "normoxia") 
  
hyp_3d <- big_comparisons$hypoxia$si3d %>% 
    mutate(condition = "hypoxia") 


si3d_single <- rbind(norm_3d, hyp_3d) %>% 
  rename("LFC_3d" ="log2FoldChange") %>%
  rename("padj_3d" = "padj")

norm_a209 <- big_comparisons$normoxia$a209 %>% 
  mutate(condition = "normoxia")


hyp_a209  <- big_comparisons$hypoxia$a209 %>% 
      mutate(condition = "hypoxia")

a209_single <- rbind(norm_a209, hyp_a209) %>% 
  rename("LFC_a209" ="log2FoldChange") %>%
  rename("padj_a209" = "padj")


a209_short <- a209_single %>% 
  dplyr::select(LFC_a209, padj_a209, gene_id, condition)


single_condition_all <- si3e_single %>% 
  left_join(a209_short, by=c("gene_id", "condition"))
  

 single_condition_all <- single_condition_all %>% 
  left_join(si3d_single, by=c("gene_id", "condition"))
  
  

  
 single_condition_all$condition <- relevel(x = factor(single_condition_all$condition), ref = "normoxia")

#this df compares norm only and hyp only, compared to respective controls
```
  
# Export TE signatures! These are 3e KD signatures or a209 sigantures, (eIF3e low signatures)
```{r}

#3e ---------------------------------------------

#norm sig ----------------------------------------

# no cutoff

norm_3e_no_cutoff <- norm_3e %>% 
  filter(padj < 0.05)

nrow(norm_3e_no_cutoff)


# 0.5 cutoff

norm_3e_0.5cutoff <- norm_3e %>% 
  filter(padj < 0.05) %>% 
  filter(log2FoldChange < -0.5 | log2FoldChange > 0.5)

nrow(norm_3e_0.5cutoff)

# 1 cutoff

norm_3e_1cutoff <- norm_3e %>% 
  filter(padj < 0.05) %>% 
  filter(log2FoldChange < -1 | log2FoldChange > 1)

nrow(norm_3e_1cutoff)


#hyp sig -----------------------------------------

# no cutoff

hyp_3e_no_cutoff <- hyp_3e %>% 
  filter(padj < 0.05)

nrow(hyp_3e_no_cutoff)


# 0.5 cutoff

hyp_3e_0.5cutoff <- hyp_3e %>% 
  filter(padj < 0.05) %>% 
  filter(log2FoldChange < -0.5 | log2FoldChange > 0.5)

nrow(hyp_3e_0.5cutoff)

# 1 cutoff

hyp_3e_1cutoff <- hyp_3e %>% 
  filter(padj < 0.05) %>% 
  filter(log2FoldChange < -1 | log2FoldChange > 1)

nrow(hyp_3e_1cutoff)




#3d ---------------------------------------------

#norm sig ----------------------------------------

# no cutoff

norm_3d_no_cutoff <- norm_3d %>% 
  filter(padj < 0.05)

nrow(norm_3d_no_cutoff)


# 0.5 cutoff

norm_3d_0.5cutoff <- norm_3d %>% 
  filter(padj < 0.05) %>% 
  filter(log2FoldChange < -0.5 | log2FoldChange > 0.5)

nrow(norm_3d_0.5cutoff)

# 1 cutoff

norm_3d_1cutoff <- norm_3d %>% 
  filter(padj < 0.05) %>% 
  filter(log2FoldChange < -1 | log2FoldChange > 1)

nrow(norm_3d_1cutoff)


#hyp sig -----------------------------------------

# no cutoff

hyp_3d_no_cutoff <- hyp_3d %>% 
  filter(padj < 0.05)

nrow(hyp_3d_no_cutoff)


# 0.5 cutoff

hyp_3d_0.5cutoff <- hyp_3d %>% 
  filter(padj < 0.05) %>% 
  filter(log2FoldChange < -0.5 | log2FoldChange > 0.5)

nrow(hyp_3d_0.5cutoff)

# 1 cutoff

hyp_3d_1cutoff <- hyp_3d %>% 
  filter(padj < 0.05) %>% 
  filter(log2FoldChange < -1 | log2FoldChange > 1)

nrow(hyp_3d_1cutoff)






#a209 ---------------------------------------------

#norm sig ----------------------------------------

# no cutoff

norm_a209_no_cutoff <- norm_a209 %>% 
  filter(padj < 0.05)

nrow(norm_a209_no_cutoff)


# 0.5 cutoff

norm_a209_0.5cutoff <- norm_a209 %>% 
  filter(padj < 0.05) %>% 
  filter(log2FoldChange < -0.5 | log2FoldChange > 0.5)

nrow(norm_a209_0.5cutoff)

# 1 cutoff

norm_a209_1cutoff <- norm_a209 %>% 
  filter(padj < 0.05) %>% 
  filter(log2FoldChange < -1 | log2FoldChange > 1)

nrow(norm_a209_1cutoff)


#hyp sig -----------------------------------------

# no cutoff

hyp_a209_no_cutoff <- hyp_a209 %>% 
  filter(padj < 0.05)

nrow(hyp_a209_no_cutoff)


# 0.5 cutoff

hyp_a209_0.5cutoff <- hyp_a209 %>% 
  filter(padj < 0.05) %>% 
  filter(log2FoldChange < -0.5 | log2FoldChange > 0.5)

nrow(hyp_a209_0.5cutoff)

# 1 cutoff

hyp_a209_1cutoff <- hyp_a209 %>% 
  filter(padj < 0.05) %>% 
  filter(log2FoldChange < -1 | log2FoldChange > 1)

nrow(hyp_a209_1cutoff)


```

Save to workbooks
```{r}

# no cutoff
write_xlsx(list(eIF3e_normoxia = norm_3e_no_cutoff,
                eIF3e_hypoxia = hyp_3e_no_cutoff,
                eIF3d_normoxia = norm_3d_no_cutoff,
                eIF3d_hypoxia = hyp_3d_no_cutoff,
                a209_normoxia = norm_a209_no_cutoff,
                a209_hypoxia = hyp_a209_no_cutoff),
           here("counts/fig6/TE_signatures_no_LFC_cutoff.xlsx"))


#.5 cutoff

write_xlsx(list(eIF3e_normoxia = norm_3e_0.5cutoff,
                eIF3e_hypoxia = hyp_3e_0.5cutoff,
                eIF3d_normoxia = norm_3d_0.5cutoff,
                eIF3d_hypoxia = hyp_3d_0.5cutoff,
                a209_normoxia = norm_a209_0.5cutoff,
                a209_hypoxia = hyp_a209_0.5cutoff),
           here("counts/fig6/TE_signatures_0.5_LFC_cutoff.xlsx"))


#1 cutoff

write_xlsx(list(eIF3e_normoxia = norm_3e_1cutoff,
                eIF3e_hypoxia = hyp_3e_1cutoff,
                eIF3d_normoxia = norm_3d_1cutoff,
                eIF3d_hypoxia = hyp_3d_1cutoff,
                a209_normoxia = norm_a209_1cutoff,
                a209_hypoxia = hyp_a209_1cutoff),
           here("counts/fig6/TE_signatures_1_LFC_cutoff.xlsx"))
```

  
  
## Supplemental 3e / a209 comparisons  
3e compared to a209 - supplemental data?

```{r}


si3e_209 <- single_condition_all %>% 
  filter(padj_3e < 0.05 | padj_a209 < 0.05)


ggplot(si3e_209, aes(x = LFC_a209, y=LFC_3e))+
  geom_point(alpha=.5)+
  theme_cowplot()+
  facet_wrap(~condition)+
  xlim(-2.5,2.5)+
  ylim(-2.5,2.5)+
  geom_smooth(method=lm)+
  stat_cor(method="pearson")

```


#3d compared to a209
```{r}

si3d_209 <- single_condition_all %>% 
  filter(padj_3d < 0.05 | padj_a209 < 0.05)

ggplot(si3d_209, aes(x = LFC_a209, y=LFC_3d))+
  geom_point(alpha=.5)+
  theme_cowplot()+
  facet_wrap(~condition)+
  xlim(-2.5,2.5)+
  ylim(-2.5,2.5)+
  geom_smooth(method=lm)+
  stat_cor(method="pearson")



```


Visualizing correlation coefficients
```{r}

corr_coef <- tibble("condition" = rep(c("normoxia","hypoxia"), each = 2),
       "treatment" = rep(c("eIF3e_a209","eIF3d_a209"), times = 2),
       "corr" = c(0.032,-0.022,0.18,0.0051),
       "p_value" = c(0.38,0.5, 7.3e-10, 0.77))


ggbarplot(data = corr_coef, x = "treatment", y = "corr", fill = "condition", palette = c("red","blue"), position = position_dodge(0.75), label = T)+
  theme_cowplot(font_size = 16)

#only the first bar is significant


```


# Odds ratios for 3e and a209 overlap in normoxia

The following analysis determines the number of genes that overlap in each condition and the statistical significance (p-value) of their overlap (odds ratio).

Make gene lists - normoxia
```{r}
#up

si3e_normoxia_up <- big_comparisons$normoxia$si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

si3e_normoxia_up$gs_name <- rep("si3e_normoxia_up")


si3d_normoxia_up <- big_comparisons$normoxia$si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
  dplyr::select(symbol)

si3d_normoxia_up$gs_name <- rep("si3d_normoxia_up")



a209_normoxia_up <- big_comparisons$normoxia$a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
  dplyr::select(symbol)

a209_normoxia_up$gs_name <- rep("a209_normoxia_up")


# down

si3e_normoxia_down <- big_comparisons$normoxia$si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
  dplyr::select(symbol)

si3e_normoxia_down$gs_name <- rep("si3e_normoxia_down")


si3d_normoxia_down <- big_comparisons$normoxia$si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
  dplyr::select(symbol)

si3d_normoxia_down$gs_name <- rep("si3d_normoxia_down")


a209_normoxia_down <- big_comparisons$normoxia$a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
  dplyr::select(symbol)

a209_normoxia_down$gs_name <- rep("a209_normoxia_down")


```

### Odds ratios in hypoxia

Make gene lists - Hypoxia
```{r}

#up

si3e_hypoxia_up <- big_comparisons$hypoxia$si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
  dplyr::select(symbol)

si3e_hypoxia_up$gs_name <- rep("si3e_hypoxia_up")


si3d_hypoxia_up <- big_comparisons$hypoxia$si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
  dplyr::select(symbol)

si3d_hypoxia_up$gs_name <- rep("si3d_hypoxia_up")



a209_hypoxia_up <- big_comparisons$hypoxia$a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
  dplyr::select(symbol)

a209_hypoxia_up$gs_name <- rep("a209_hypoxia_up")



# down

si3e_hypoxia_down <- big_comparisons$hypoxia$si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
  dplyr::select(symbol)

si3e_hypoxia_down$gs_name <- rep("si3e_hypoxia_down")


si3d_hypoxia_down <- big_comparisons$hypoxia$si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
  dplyr::select(symbol)

si3d_hypoxia_down$gs_name <- rep("si3d_hypoxia_down")



a209_hypoxia_down <- big_comparisons$hypoxia$a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
  dplyr::select(symbol)

a209_hypoxia_down$gs_name <- rep("a209_hypoxia_down")





```


Create a matrix that compares each condition for gene overlap analysis.
```{r}
all_lists <- bind_rows(si3e_normoxia_up, si3d_normoxia_up, a209_normoxia_up, si3e_normoxia_down, si3d_normoxia_down, a209_normoxia_down,
                       si3e_hypoxia_up, si3d_hypoxia_up, a209_hypoxia_up, si3e_hypoxia_down, si3d_hypoxia_down, a209_hypoxia_down) %>% 
  separate(gs_name, c("sample", "condition", "direction"))
```



### Normoxia analysis
```{r}

norm_matrix <- all_lists %>%
  filter(condition=="normoxia") %>% 
  dplyr::select(-condition)


norm_matrix_comb <- norm_matrix %>% 
  tidyr::unite("description", 2:3, sep=', ')

gom.allsamples <- split(norm_matrix_comb$symbol, norm_matrix_comb$description)

gom.all <- newGOM(gom.allsamples, gom.allsamples, genome.size = 9608)

norm_all.odds.matrix <- getMatrix(gom.all, "odds")
norm_all.pval.matrix <- getMatrix(gom.all, "pval")


```

### Odds values
```{r}
#knitr::kable(all.odds.matrix, valign = 't')
```

### P values
```{r}
#knitr::kable(all.pval.matrix, valign = 't')
```

# plot odds ratios as bars with p values
```{r}

#merge odds and pval matricies

odds <- norm_all.odds.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:7, names_to = "comparison", values_to="odds_value")

pval <- norm_all.pval.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:7, names_to = "comparison", values_to="pval_value")


odds_pval <- left_join(odds, pval, by=c("sample", "comparison")) %>% 
  tidyr::unite("comparison", 1:2, sep="-") %>% 
  filter(comparison == "a209, up-si3e, up"| 
         comparison =="a209, down-si3e, down"|
           comparison == "a209, up-si3d, up"|
           comparison == "a209, down-si3d, down") %>% 
  mutate(comparison_with_a209 = sub(".*-", "", comparison)) %>% 
  mutate(pvalue_plot = signif(pval_value, digits=3)) %>% 
  mutate(label = paste0("p = ", pvalue_plot, sep=" "))

odds_pval$comparison_with_a209 <- factor(odds_pval$comparison_with_a209, levels = c("si3e, up", "si3e, down", "si3d, up", "si3d, down"))

odds_pval <- odds_pval %>% 
  mutate(group = ifelse(comparison== "a209, down-si3d, down" | comparison== "a209, up-si3d, up", "si3d", "si3e"))


ggpubr::ggbarplot(odds_pval, x="comparison_with_a209", y="odds_value", fill="group", label = odds_pval$label, title="a209 overlaps in normoxia", palette = c(si3d_col, si3e_col), ylim=(c(0,6)))+
  geom_hline(aes(yintercept=1))


ggsave(here("plots/fig6/normoverlap.pdf"), width=4, height=3)

```

### Hypoxia
```{r}

hyp_matrix <- all_lists %>%
  filter(condition=="hypoxia") %>% 
  dplyr::select(-condition)


hyp_matrix_comb <- hyp_matrix %>% 
  tidyr::unite("description", 2:3, sep=', ')

gom.allsamples <- split(hyp_matrix_comb$symbol, hyp_matrix_comb$description)

gom.all <- newGOM(gom.allsamples, gom.allsamples, genome.size = 9608)

hyp_all.odds.matrix <- getMatrix(gom.all, "odds")
hyp_all.pval.matrix <- getMatrix(gom.all, "pval")


```

# plot odds ratios as bars with p values

```{r}

#merge odds and pval matricies

odds <- hyp_all.odds.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:7, names_to = "comparison", values_to="odds_value")

pval <- hyp_all.pval.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:7, names_to = "comparison", values_to="pval_value")


odds_pval <- left_join(odds, pval, by=c("sample", "comparison")) %>% 
  tidyr::unite("comparison", 1:2, sep="-") %>% 
  filter(comparison == "a209, up-si3e, up"| 
         comparison =="a209, down-si3e, down"|
           comparison == "a209, up-si3d, up"|
           comparison == "a209, down-si3d, down") %>% 
  mutate(comparison_with_a209 = sub(".*-", "", comparison)) %>% 
  mutate(pvalue_plot = signif(pval_value, digits=3)) %>% 
  mutate(label = paste0("p = ", pvalue_plot, sep=" "))

odds_pval$comparison_with_a209 <- factor(odds_pval$comparison_with_a209, levels = c("si3e, up", "si3e, down", "si3d, up", "si3d, down"))

odds_pval <- odds_pval %>% 
  mutate(group = ifelse(comparison== "a209, down-si3d, down" | comparison== "a209, up-si3d, up", "si3d", "si3e"))

ggpubr::ggbarplot(odds_pval, x="comparison_with_a209", y="odds_value", fill="group", label = odds_pval$label, title="a209 overlaps in hypoxia", palette = c(si3d_col, si3e_col), ylim=(c(0,6)))+
  geom_hline(aes(yintercept=1))


ggsave(filename=here("plots/fig6/hypoverlap.pdf"), width=4, height=3)



```




### Venn diagram of overlaps of a209 and eIF3e in normoxia and hypoxia (unused in paper)

Now, separate lists by direction of change - use for venn diagrams and for pathway analyses
```{r}

#norm
norm_3d_up <- norm_3d %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

norm_3d_down <- norm_3d %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)

norm_3e_up <- norm_3e %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

norm_3e_down <- norm_3e %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)

norm_a209_up <- norm_a209 %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

norm_a209_down <- norm_a209 %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)





hyp_3d_up <- hyp_3e %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

hyp_3d_down <- hyp_3d %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)

hyp_3e_up <- hyp_3e %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

hyp_3e_down <- hyp_3e %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)

hyp_a209_up <- hyp_a209 %>% 
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

hyp_a209_down <- hyp_a209 %>% 
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)



norm_te_up <- list(
  norm_a209_up = norm_a209_up,
  norm_3e_up = norm_3e_up
)


norm_te_down <- list(
  norm_a209_down = norm_a209_down,
  norm_3e_down = norm_3e_down
)

hyp_te_up <- list(
  hyp_a209_up = hyp_a209_up,
  hyp_3e_up = hyp_3e_up
)


hyp_te_down <- list(
  hyp_a209_down = hyp_a209_down,
  hyp_3e_down = hyp_3e_down
)



```


# Visualize up and down genes combined

## Normoxia

```{r}
ggvenn(norm_te_up, auto_scale = T, fill_color = c(a209_col,si3e_col))

ggvenn(norm_te_down, auto_scale = T, fill_color = c(a209_col,si3e_col))

stopifnot(
identical(norm_3e$gene_id,norm_a209$gene_id))

norm_sig_te <- data.frame(
  gene_id = norm_3e$gene_id,
  symbol = norm_3e$symbol,
  te_a209 = norm_a209$log2FoldChange,
  te_3e = norm_3e$log2FoldChange
) %>%
  filter(gene_id %in% c(norm_3e_up,
                        norm_3e_down,
                        norm_a209_up,
                        norm_a209_down)) 



scatter <- ggscatter(data = norm_sig_te,
          x = "te_a209",
          y = "te_3e",
          xlab="a209 TE",
          ylab="si-eIF3e TE",
          alpha = 0.25,
          title = "Compare TE of sig genes - Normoxia",
xlim=c(-2,2), ylim = c(-2,2))+
   stat_cor(
   aes(label =
         paste(
           after_stat(rr.label),
           ..p.label..,
           sep = "~`,`~")
       )
   )


scatter <- ggMarginal(scatter, type = "histogram",
           xparams = list(binwidth = .3, fill = a209_col),
           yparams = list(binwidth = .3, fill = si3e_col)
           )

scatter


ggsave(here("plots/fig6/norm_3e_a209_te_scatter.pdf"), scatter, width=4, height=4)




```

## Hypoxia

```{r}
ggvenn(hyp_te_up, auto_scale = T, fill_color = c(a209_col,si3e_col))

ggvenn(hyp_te_down, auto_scale = T, fill_color = c(a209_col,si3e_col))

stopifnot(
identical(hyp_3e$gene_id,hyp_a209$gene_id))

hyp_sig_te <- data.frame(
  gene_id = hyp_3e$gene_id,
  symbol = hyp_3e$symbol,
  te_a209 = hyp_a209$log2FoldChange,
  te_3e = hyp_3e$log2FoldChange
) %>%
  filter(gene_id %in% c(hyp_3e_up,
                        hyp_3e_down,
                        hyp_a209_up,
                        hyp_a209_down)) 



scatter <- ggscatter(data = hyp_sig_te,
          x = "te_a209",
          y = "te_3e",
          xlab="a209 TE",
          ylab="si-eIF3e TE",
          alpha = 0.25,
          title = "Compare TE of sig genes - Hypoxia", xlim=c(-2,2), ylim = c(-2,2)) +
   stat_cor(
   aes(label =
         paste(
           after_stat(rr.label),
           ..p.label..,
           sep = "~`,`~")
       )
   )



scatter <- ggMarginal(scatter, type = "histogram",
           xparams = list(binwidth = .3, fill = a209_col),
           yparams = list(binwidth = .3, fill = si3e_col)
           )

scatter

ggsave(filename=here("plots/fig6/hyp_3e_a209_te_scatter.pdf"),scatter,width=4, height=4)



```



#Pathway analysis:

In hypoxia, what pathways are overrepresented in the overlap of a209 and 3e? What pathways are 3e hitting? Which are a209 hitting? how are these distinct from normoxia?


3e and a209
```{r prep GSEA analysis}

#3e lists

#norm

eif3e_norm <- norm_3e$log2FoldChange

names(eif3e_norm) <- norm_3e$symbol

eif3e_norm <- sort(eif3e_norm, decreasing = TRUE)

eif3e_norm <- eif3e_norm[!duplicated(names(eif3e_norm))]



#hyp

eif3e_hyp <- hyp_3e$log2FoldChange

names(eif3e_hyp) <- hyp_3e$symbol

eif3e_hyp <- sort(eif3e_hyp, decreasing = TRUE)

eif3e_hyp <- eif3e_hyp[!duplicated(names(eif3e_hyp))]




#a209 lists

#norm

a209_norm <- norm_a209$log2FoldChange

names(a209_norm) <- norm_a209$symbol

a209_norm <- sort(a209_norm, decreasing = TRUE)

a209_norm <- a209_norm[!duplicated(names(a209_norm))]



#hyp

a209_hyp <- hyp_a209$log2FoldChange

names(a209_hyp) <- hyp_a209$symbol

a209_hyp <- sort(a209_hyp, decreasing = TRUE)

a209_hyp <- a209_hyp[!duplicated(names(a209_hyp))]



```


```{r}
# hallmark and c2 setup

m_df_h <- msigdbr(species = "Homo sapiens") %>%
  filter(gs_cat == "H") %>% 
  dplyr::select(gs_name, gene_symbol)

m_df_c2 <- msigdbr(species = "Homo sapiens") %>%
  filter(gs_cat == "C2" & gs_subcat != "CGP") %>% 
  dplyr::select(gs_name, gene_symbol)
```


## Hallmark Pathways

Which hallmark pathways overlap for a209 and 3e in hypoxia? which are different?

```{r GSEA Hallmarks}

#normoxia

norm_a209_te_h <- GSEA(
  geneList = a209_norm,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )


norm_eif3e_te_h <- GSEA(
  geneList = eif3e_norm,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )



norm_te_h <- dplyr::full_join(
  norm_eif3e_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  norm_a209_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("a209_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3e_pvalue < 0.05 | a209_pvalue < 0.05)  %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = a209_NES - eif3e_NES)


norm_te_h$ID <- gsub(pattern = "HALLMARK_", replacement = "", x = norm_te_h$ID)




hallmarks <- ggscatter(data = norm_te_h,
          x = "eif3e_NES",
          y = "a209_NES",
          label = "ID",
          repel = TRUE,
          title = "Hallmarks - normoxia",
          font.label = 8) +
 scale_color_continuous(type = "viridis") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5,2.5) +
  xlim(-2.5,2.5) + theme_prism()

hallmarks

DT::datatable(norm_te_h) %>%
  formatRound(which(sapply(norm_te_h,is.numeric)), digits = 3)

ggsave(here("plots/fig6/normoxia_hallmarks.pdf"), width=4, height=4)


```


Which pathways overlap for 3e and a209 in hypoxia? Which are different?
```{r}

#hypoxia

hyp_a209_te_h <- GSEA(
  geneList = a209_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )


hyp_eif3e_te_h <- GSEA(
  geneList = eif3e_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )


hyp_te_h <- dplyr::full_join(
  hyp_eif3e_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  hyp_a209_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("a209_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3e_pvalue < 0.05 | a209_pvalue < 0.05)  %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = a209_NES - eif3e_NES)


hyp_te_h$ID <- gsub(pattern = "HALLMARK_", replacement = "", x = hyp_te_h$ID)




scatter <- ggscatter(data = hyp_te_h,
          x = "eif3e_NES",
          y = "a209_NES",
          label = "ID",
          repel = TRUE,
          title="Hallmarks - Hypoxia",
          font.label = 10) +
 scale_color_continuous(type = "viridis") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5,2.5) +
  xlim(-2.5,2.5) + theme_prism()

scatter

DT::datatable(hyp_te_h) %>%
  formatRound(which(sapply(hyp_te_h,is.numeric)), digits = 3)

ggsave(here("plots/fig6/hypoxia_hallmarks.pdf"), width=4, height = 4)



```


## C2 pathways

C2 reactome - pathways that are similar and different in normoxia

```{r c2 Hallmarks}

#normoxia

norm_a209_te_react <- GSEA(
  geneList = a209_norm,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^REACTOME', gs_name)))
  


norm_eif3e_te_react <- GSEA(
  geneList = eif3e_norm,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^REACTOME', gs_name)))

  



norm_te_react <- dplyr::full_join(
  norm_eif3e_te_react@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  norm_a209_te_react@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("a209_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3e_pvalue < 0.05 | a209_pvalue < 0.05)  %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = a209_NES - eif3e_NES)


norm_te_react$ID <- gsub(pattern = "REACTOME_", replacement = "", x = norm_te_react$ID)



reactome <- ggscatter(data = norm_te_react,
          x = "eif3e_NES",
          y = "a209_NES",
          label = "ID",
          title = "Reactome - normoxia",
          repel = TRUE,
          font.label = 6) +
 scale_color_continuous(type = "viridis") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5,2.5) +
  xlim(-2.5,2.5)

reactome

DT::datatable(norm_te_react) %>%
  formatRound(which(sapply(norm_te_react,is.numeric)), digits = 3)

ggsave(here("plots/fig6/normoxia_reactome.pdf"), reactome)




```


C2 - Which pathways overlap for 3e and a209 in hypoxia? Which are different?
```{r}

#hypoxia

hyp_a209_te_react <- GSEA(
  geneList = a209_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^REACTOME', gs_name)))
  


hyp_eif3e_te_react <- GSEA(
  geneList = eif3e_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^REACTOME', gs_name)))

  



hyp_te_react <- dplyr::full_join(
  hyp_eif3e_te_react@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  hyp_a209_te_react@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("a209_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3e_pvalue < 0.05 | a209_pvalue < 0.05)  %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = a209_NES - eif3e_NES)


hyp_te_react$ID <- gsub(pattern = "REACTOME_", replacement = "", x = hyp_te_react$ID)



reactome <- ggscatter(data = hyp_te_react,
          x = "eif3e_NES",
          y = "a209_NES",
          label = "ID",
          title = "Reactome - hypoxia",
          repel = TRUE,
          font.label = 6) +
 scale_color_continuous(type = "viridis") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5,2.5) +
  xlim(-2.5,2.5)

reactome

DT::datatable(hyp_te_react) %>%
  formatRound(which(sapply(hyp_te_react,is.numeric)), digits = 3)

ggsave(here("plots/fig6/hypoxia_reactome.pdf"), reactome)



```
Can include kegg as well