---
title: "Figure 8 - 3e/d RNA signatures in patient data"
author: "Kate"
date: "2024-08-27"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(here)
library(tidyverse)
library(janitor)
library(readxl)
library(DESeq2)
library(ggprism)
library(survminer)
library(cowplot)
library(ggpubr)
library(survival)



```

```{r}
theme_set(theme_prism())

a209_col = "#CC79A7"
si3e_col = "#56B4E9"
si3d_col = "#E69F00"

```


# Generate RNA signatures

Pull in raw RNA data and do differential gene expression

## load metadata
```{r}
metadata <- read_csv(here("counts/calculate_te/metadata.csv"))
```

```{r}
geneInfo <- read_csv(here("accessories","gencode.v26.primary.info.csv.gz"), col_names = F) %>%
  filter(!str_detect(X2, "pre_"))

colnames(geneInfo) <- c("gene_id","transcript_id","biotype","symbol")

#map transcripts to gene
tx2gene <- geneInfo[,c(2,1)]
gene_symbol <- geneInfo %>% dplyr::select(gene_id, symbol) %>% unique()

```

## Separate ribo counts, rna counts, and te counts
```{r}
raw_counts <- read_csv(here("counts/calculate_te/raw_filtered_allCounts.csv")) %>% 
  column_to_rownames("gene_id")

ribo_counts <- as.matrix(raw_counts[,grep(pattern = "ribo",colnames(raw_counts))])
rna_counts <- as.matrix(raw_counts[,grep(pattern = "rna",colnames(raw_counts))])

rna_ribo_counts <- as.matrix(raw_counts)

```

# 3e hypoxia rna @Kate if we keep generate signatures code we can skip the deseq2 and import from there
```{r}

metadata_loop <- metadata %>% 
  filter(library=="rna") %>% 
  filter(condition=="hypoxia") %>% 
  filter(treatment=="sictrl"|treatment=="si3e")

metadata_loop$treatment <- factor(metadata_loop$treatment, levels= c("si3e", "sictrl"))
levels(metadata_loop$treatment)

sample_info <- metadata_loop %>% 
    mutate("name" = 
  paste(condition, treatment, rep, library, sep = "_"))
  
  names <- as.character(sample_info$name)
  names
  
  rna_subset <- rna_ribo_counts[, colnames(rna_ribo_counts) %in% names]
  
  print(levels(sample_info$treatment))

  
# Deseq analysis ----------------------------------------------------------------
  
  dds <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  dds <- DESeq(dds)
  
  print(resultsNames(dds))

  res <- results(dds, name=resultsNames(dds)[2], cooksCutoff=FALSE, independentFiltering=FALSE)

  res_hyp_3e <- lfcShrink(dds = dds, coef = resultsNames(dds)[2], res = res) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)

# export to excel ---------------------------------------------------------------
  
   writexl::write_xlsx(x = res_hyp_3e, path = here("counts/fig8/rna_signature_3e_hypoxia.xlsx"), col_names = T)

#confirm direction
  res_hyp_3e %>% 
    filter(symbol=="EIF3E")

```

# 3d hypoxia rna
```{r}

metadata_loop <- metadata %>% 
  filter(library=="rna") %>% 
  filter(condition=="hypoxia") %>% 
  filter(treatment=="sictrl"|treatment=="si3d")

metadata_loop$treatment <- factor(metadata_loop$treatment, levels= c("si3d", "sictrl"))
levels(metadata_loop$treatment)

sample_info <- metadata_loop %>% 
    mutate("name" = 
  paste(condition, treatment, rep, library, sep = "_"))
  
  names <- as.character(sample_info$name)
  names
  
  rna_subset <- rna_ribo_counts[, colnames(rna_ribo_counts) %in% names]
  
  print(levels(sample_info$treatment))

  
# Deseq analysis ----------------------------------------------------------------
  
  dds <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  dds <- DESeq(dds)
  
  print(resultsNames(dds))

  res <- results(dds, name=resultsNames(dds)[2], cooksCutoff=FALSE, independentFiltering=FALSE)

  res_hyp_3d <- lfcShrink(dds = dds, coef = resultsNames(dds)[2], res = res) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)

# export to excel ---------------------------------------------------------------
  
   writexl::write_xlsx(x = res_hyp_3d, path = here("counts/fig8/rna_signature_3d_hypoxia.xlsx"), col_names = T)

#confirm direction
  res_hyp_3d %>% 
    filter(symbol=="EIF3D")

```
#@Kate RIBO and TE sig-- in another file not for pub?

#@Neel the major thing here is I'm doing this analysis using a cutoff of 1 LFC for the rna signature (rather than 0.5 LFC cutoff etc). We may want 0.5-- the interpretation is very similar either way

# Normoxia analysis for supplement
# 3e normoxia rna @Kate if we keep generate signatures code we can skip the deseq2 and import from there
```{r}

metadata_loop <- metadata %>% 
  filter(library=="rna") %>% 
  filter(condition=="normoxia") %>% 
  filter(treatment=="sictrl"|treatment=="si3e")

metadata_loop$treatment <- factor(metadata_loop$treatment, levels= c("si3e", "sictrl"))
levels(metadata_loop$treatment)

sample_info <- metadata_loop %>% 
    mutate("name" = 
  paste(condition, treatment, rep, library, sep = "_"))
  
  names <- as.character(sample_info$name)
  names
  
  rna_subset <- rna_ribo_counts[, colnames(rna_ribo_counts) %in% names]
  
  print(levels(sample_info$treatment))

  
# Deseq analysis ----------------------------------------------------------------
  
  dds <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  dds <- DESeq(dds)
  
  print(resultsNames(dds))

  res <- results(dds, name=resultsNames(dds)[2], cooksCutoff=FALSE, independentFiltering=FALSE)

  res_norm_3e <- lfcShrink(dds = dds, coef = resultsNames(dds)[2], res = res) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)

# export to excel ---------------------------------------------------------------
  
   writexl::write_xlsx(x = res_norm_3e, path = here("counts/fig8/rna_signature_3e_normoxia.xlsx"), col_names = T)

#confirm direction
  res_norm_3e %>% 
    filter(symbol=="EIF3E")

```

# 3d normoxia rna
```{r}

metadata_loop <- metadata %>% 
  filter(library=="rna") %>% 
  filter(condition=="normoxia") %>% 
  filter(treatment=="sictrl"|treatment=="si3d")

metadata_loop$treatment <- factor(metadata_loop$treatment, levels= c("si3d", "sictrl"))
levels(metadata_loop$treatment)

sample_info <- metadata_loop %>% 
    mutate("name" = 
  paste(condition, treatment, rep, library, sep = "_"))
  
  names <- as.character(sample_info$name)
  names
  
  rna_subset <- rna_ribo_counts[, colnames(rna_ribo_counts) %in% names]
  
  print(levels(sample_info$treatment))

  
# Deseq analysis ----------------------------------------------------------------
  
  dds <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  dds <- DESeq(dds)
  
  print(resultsNames(dds))

  res <- results(dds, name=resultsNames(dds)[2], cooksCutoff=FALSE, independentFiltering=FALSE)

  res_norm_3d <- lfcShrink(dds = dds, coef = resultsNames(dds)[2], res = res) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)

# export to excel ---------------------------------------------------------------
  
   writexl::write_xlsx(x = res_norm_3d, path = here("counts/fig8/rna_signature_3d_normoxia.xlsx"), col_names = T)

#confirm direction
  res_norm_3d %>% 
    filter(symbol=="EIF3D")

```

# Prepare survival analysis
### Separate RNA signatures by direction

```{r}
# 3e hypoxia

sig_rna_3e_hyp <- res_hyp_3e %>% 
  filter(padj < 0.05)

up_3e_hyp <- sig_rna_3e_hyp %>% 
  filter(log2FoldChange > 1)%>% 
  pull(symbol)

down_3e_hyp <- sig_rna_3e_hyp %>% 
  filter(log2FoldChange < -1)%>% 
  pull(symbol)
  
```

```{r}
# 3d hypoxia

sig_rna_3d_hyp <- res_hyp_3d %>% 
  filter(padj < 0.05)

up_3d_hyp <- sig_rna_3d_hyp %>% 
  filter(log2FoldChange > 1) %>% 
  pull(symbol)
  

down_3d_hyp <- sig_rna_3d_hyp %>% 
  filter(log2FoldChange < -1)%>% 
  pull(symbol)
  

```


```{r}
# 3e normoxia

sig_rna_3e_norm <- res_norm_3e %>% 
  filter(padj < 0.05)

up_3e_norm <- sig_rna_3e_norm %>% 
  filter(log2FoldChange > 1)%>% 
  pull(symbol)

down_3e_norm <- sig_rna_3e_norm %>% 
  filter(log2FoldChange < -1)%>% 
  pull(symbol)
  
```

```{r}
# 3d normoxia

sig_rna_3d_norm <- res_norm_3d %>% 
  filter(padj < 0.05)

up_3d_norm <- sig_rna_3d_norm %>% 
  filter(log2FoldChange > 1) %>% 
  pull(symbol)
  

down_3d_norm <- sig_rna_3d_norm %>% 
  filter(log2FoldChange < -1)%>% 
  pull(symbol)
  

```

# Import METABRIC microarray data from cbioportal - read patient z scores and matching clinical data
Nature 2012 and Nat Commun 2016

```{r cache = TRUE}
zscores <- read.table(here("cbioportal_data/brca_metabric/data_mrna_illumina_microarray_zscores_ref_diploid_samples.txt"), sep="\t", header=T)
row.names(zscores) <- gsub("_", ".",row.names(zscores))

zscores_all_symbol <- zscores %>% 
  dplyr::select(-Entrez_Gene_Id)

### Average scores for duplicate patients

dup_genes <- zscores$Hugo_Symbol[duplicated(zscores$Hugo_Symbol)]

dedup <- zscores %>%
  filter(Hugo_Symbol %in% dup_genes) %>%
  group_by(Hugo_Symbol) %>%
  summarise(across(everything(), mean))


merged_zscores <- zscores %>%
  filter(!Hugo_Symbol %in% dup_genes) %>% 
  bind_rows(., dedup) %>% 
  as.tibble() %>%
  column_to_rownames("Hugo_Symbol") %>% 
  as.data.frame()


```

# eIF3e hypoxia signature and survival prediction
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3e <- merged_zscores %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% sig_rna_3e_hyp$symbol) %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  dplyr::select(-Entrez_Gene_Id)

# empty df
sig_zscores_3e <- data.frame(patient=colnames(exp_3e),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3e)) {
  sig_zscores_3e[i,2] <- mean(exp_3e[up_3e_hyp,i],  na.rm=TRUE)
  sig_zscores_3e[i,3] <- mean(exp_3e[down_3e_hyp,i],  na.rm=TRUE)
  }

sig_zscores_3e <- sig_zscores_3e %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3e$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3e$quantile <- cut(sig_zscores_3e$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_3e$patient <- sub("-[^-]*$", "", sig_zscores_3e$patient)

sig_zscores_3e$patient <- gsub("\\.", "-", sig_zscores_3e$patient)


```


# eIF3d hypoxia signature and survival prediction
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3d <- merged_zscores %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% sig_rna_3d_hyp$symbol) %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  dplyr::select(-Entrez_Gene_Id)

# empty df
sig_zscores_3d <- data.frame(patient=colnames(exp_3d),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3d)) {
  sig_zscores_3d[i,2] <- mean(exp_3d[up_3d_hyp,i],  na.rm=TRUE)
  sig_zscores_3d[i,3] <- mean(exp_3d[down_3d_hyp,i],  na.rm=TRUE)
  }

sig_zscores_3d <- sig_zscores_3d %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3d$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3d$quantile <- cut(sig_zscores_3d$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)


sig_zscores_3d$patient <- gsub("\\.", "-", sig_zscores_3d$patient)

```

# Normoxia
# eIF3e normoxia signature and survival prediction
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3e_norm <- merged_zscores %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% sig_rna_3e_norm$symbol) %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  dplyr::select(-Entrez_Gene_Id)

# empty df
sig_zscores_3e_norm <- data.frame(patient=colnames(exp_3e_norm),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3e_norm)) {
  sig_zscores_3e_norm[i,2] <- mean(exp_3e_norm[up_3e_norm,i],  na.rm=TRUE)
  sig_zscores_3e_norm[i,3] <- mean(exp_3e_norm[down_3e_norm,i],  na.rm=TRUE)
  }

sig_zscores_3e_norm <- sig_zscores_3e_norm %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3e$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3e_norm$quantile <- cut(sig_zscores_3e_norm$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_3e_norm$patient <- sub("-[^-]*$", "", sig_zscores_3e_norm$patient)

sig_zscores_3e_norm$patient <- gsub("\\.", "-", sig_zscores_3e_norm$patient)


```


# eIF3d normoxia signature and survival prediction
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3d_norm <- merged_zscores %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% sig_rna_3d_norm$symbol) %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  dplyr::select(-Entrez_Gene_Id)

# empty df
sig_zscores_3d_norm <- data.frame(patient=colnames(exp_3d_norm),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3d_norm)) {
  sig_zscores_3d_norm[i,2] <- mean(exp_3d_norm[up_3d_norm,i],  na.rm=TRUE)
  sig_zscores_3d_norm[i,3] <- mean(exp_3d_norm[down_3d_norm,i],  na.rm=TRUE)
  }

sig_zscores_3d_norm <- sig_zscores_3d_norm %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3d_norm$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3d_norm$quantile <- cut(sig_zscores_3d_norm$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)


sig_zscores_3d_norm$patient <- gsub("\\.", "-", sig_zscores_3d_norm$patient)

```


# Import clinical data
```{r}
clin <- read_excel(here("cbioportal_data/brca_metabric/data_clinical_patient_edit.xlsx"))
clin$OS_MONTHS <- as.numeric(clin$OS_MONTHS)
clin$OS_STATUS[which(clin$OS_STATUS == "1:DECEASED" & clin$OS_MONTHS > 120)] <- "0:LIVING"
clin$OS_STATUS <- clin$OS_STATUS == "1:DECEASED"
clin <- cbind(clin, OS_YEARS = clin$OS_MONTHS/12)
clin$OS_YEARS[which(clin$OS_YEARS >= 10)] <- 10

```

### visualizing high and low quartile in each survival plot for the paper.
@Kate delete violin plots below. 
## 3e survival
```{r}
#bind clin data with z score data

all_data_3e_hyp <- left_join(clin, sig_zscores_3e, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA")

all_data_3e_hyp <- all_data_3e_hyp %>% 
  filter(quantile == 1 | quantile == 4)

fit_all_3e_hyp <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_3e_hyp)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_3e_hyp)
summary_coxfit_3e_norm <- summary(coxfit)


hr <- round(summary_coxfit_3e_norm$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_3e_norm$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

#3e_colors
colors <- grDevices::colorRampPalette(c("grey", si3e_col))(2)


surv_plot_3e_hyp<-
ggsurvplot(fit_all_3e_hyp,
           data = all_data_3e_hyp,
           size = 1,
           palette =colors, 
          legend.labs=c("Low 3e", "High 3e"),
         #  legend.labs=c("Low 3e", "Med-Low 3e", "Med-High 3e", "High 3e"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3e_hyp$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("3e_hyp_surv_plot.pdf", path=here("plots/fig8"),width = 11.5, height=8)




#boxplot of quartile expression

ggplot(all_data_3e_hyp, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3e hypoxia signature")


#@Kate-- decide if needed. Connor wanted to see this but not used in paper. Same with charts below in this chunk

ggviolin(data = all_data_3e_hyp %>% filter(ER_IHC != ""), y = "total_z_score", x = "ER_IHC", fill = "ER_IHC", palette = ggokabeito::palette_okabe_ito(), add = "median_iqr" ) +
  rotate_x_text(45)

ggviolin(data = all_data_3e_hyp %>% filter(THREEGENE != ""), y = "total_z_score", x = "THREEGENE", fill = "THREEGENE", palette = ggokabeito::palette_okabe_ito(), add = "boxplot", add.params = list(fill = "white")) +
  rotate_x_text(45)


#claudin subtype
ggviolin(data = all_data_3e_hyp , y = "total_z_score", x = "CLAUDIN_SUBTYPE", fill = "CLAUDIN_SUBTYPE", palette = ggokabeito::palette_okabe_ito(), add = "median_iqr" ) +
  rotate_x_text(45)


```

## 3d survival
@Kate delete violin plots below
```{r}
#bind clin data with z score data

all_data_3d_hyp <- left_join(clin, sig_zscores_3d, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA")

all_data_3d_hyp <- all_data_3d_hyp %>% 
  filter(quantile == 1 | quantile == 4)

fit_all_3d_hyp <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_3d_hyp)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_3d_hyp)
summary_coxfit_3d_norm <- summary(coxfit)


hr <- round(summary_coxfit_3d_norm$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_3d_norm$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", si3d_col))(2)

surv_plot_3d_hyp<-
ggsurvplot(fit_all_3d_hyp,
           data = all_data_3d_hyp,
           size = 1,
           palette = colors, 
          legend.labs=c("Low 3d", "High 3d"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3d_hyp$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("3d_hyp_surv_plot.pdf", path=here("plots/fig8"),width = 11.5, height=8)




#boxplot of quartile expression

ggplot(all_data_3d_hyp, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3d hypoxia signature")

#@Kate decide if this is wanted-- probably is not
# maybe something like this could be relevant
ggviolin(data = all_data_3d_hyp %>% filter(ER_IHC != ""), y = "total_z_score", x = "ER_IHC", fill = "ER_IHC", palette = ggokabeito::palette_okabe_ito(), add = "median_iqr" ) +
  rotate_x_text(45)

ggviolin(data = all_data_3d_hyp %>% filter(THREEGENE != ""), y = "total_z_score", x = "THREEGENE", fill = "THREEGENE", palette = ggokabeito::palette_okabe_ito(), add = "boxplot", add.params = list(fill = "white")) +
  rotate_x_text(45)

```


## 3e survival - normoxia
```{r}
#bind clin data with z score data

all_data_3e_norm <- left_join(clin, sig_zscores_3e_norm, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA")

fit_all_3e_norm <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_3e_norm)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_3e_norm)
summary_coxfit_3e_norm <- summary(coxfit)


hr <- round(summary_coxfit_3e_norm$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_3e_norm$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

#3e_colors
colors <- grDevices::colorRampPalette(c("grey", si3e_col))(4)

surv_plot_3e_norm<-
ggsurvplot(fit_all_3e_norm,
           data = all_data_3e_norm,
           size = 1,
           palette =colors, 
           legend.labs=c("Low 3e", "Med-Low 3e", "Med-High 3e", "High 3e"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3e_norm$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("3e_norm_surv_plot.pdf", path=here("plots/fig8"),width = 11.5, height=8)




#boxplot of quartile expression

ggplot(all_data_3e_norm, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3e normoxia signature")

```

## 3d survival - normoxia
```{r}
#bind clin data with z score data

all_data_3d_norm <- left_join(clin, sig_zscores_3d_norm, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA")

fit_all_3d_norm <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_3d_norm)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_3d_norm)
summary_coxfit_3d_norm <- summary(coxfit)


hr <- round(summary_coxfit_3d_norm$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_3d_norm$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

#3d_colors
colors <- grDevices::colorRampPalette(c("grey", si3d_col))(4)

surv_plot_3d_norm<-
ggsurvplot(fit_all_3d_norm,
           data = all_data_3d_norm,
           size = 1,
           palette =colors, 
           legend.labs=c("Low 3d", "Med-Low 3d", "Med-High 3d", "High 3d"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3d_norm$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("3d_norm_surv_plot.pdf", path=here("plots/fig8"),width = 11.5, height=8)




#boxplot of quartile expression

ggplot(all_data_3d_norm, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3d normoxia signature")

```




## Import hypoxia signature from ye et al
Hypoxia signature in breast cancer:
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6279594/#:~:text=The%2042%2Dgene%20expression%20signature,mediate%20the%20conserved%20hypoxic%20response


```{r}
ye_hyp_sig <- read_excel(here("other_data/hypoxic_sig_ye.xlsx"))
#up values only

#calculate z scores for enrichment of hypoxia signature

# filter merged z score data based on genes in my signature
exp_hyp_sig <- merged_zscores %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% ye_hyp_sig$up_in_hypoxia) %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  dplyr::select(-Entrez_Gene_Id)

# empty df
sig_zscores_hyp_signature <- data.frame(patient=colnames(exp_hyp_sig),
                         z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_hyp_signature)) {
  sig_zscores_hyp_signature[i,2] <- mean(exp_hyp_sig[,i],  na.rm=TRUE)
}


quantile_groups <- quantile(sig_zscores_hyp_signature$z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_hyp_signature$quantile <- cut(sig_zscores_hyp_signature$z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_hyp_signature$patient <- sub("-[^-]*$", "", sig_zscores_hyp_signature$patient)

sig_zscores_hyp_signature$patient <- gsub("\\.", "-", sig_zscores_hyp_signature$patient)


#Bind z score hypoxia signature with clin data


all_data_hif_ye <- left_join(clin, sig_zscores_hyp_signature, by=c("PATIENT_ID"="patient"))

all_data_hif_ye <- all_data_hif_ye %>% 
  filter(quantile == 1 | quantile == 4)

#plot

fit_all_data_hif_ye <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_hif_ye)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_hif_ye)
summary_coxfit_ye <- summary(coxfit)

exp(summary_coxfit_ye$coefficients[1])

hr <- round(summary_coxfit_ye$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_ye$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", "red"))(2)

surv_plot_ye_hyp<-
ggsurvplot(fit_all_data_hif_ye,
           data = all_data_hif_ye,
           size = 1,
           palette = colors, 
           break.time.by = 1,
          legend.labs=c("Low Hypoxia", "High Hypoxia"),
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_ye_hyp$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("surv_plot_ye_hyp.pdf", path=here("plots/fig8"),width = 11.5, height=8)


#boxplot of quartile expression

ggplot(all_data_hif_ye, aes(x=quantile,y=z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3d hypoxia signature")

```


## correlation of 3e and hypoxia score

```{r}

patient_ye_and_3e <- left_join(sig_zscores_3e, sig_zscores_hyp_signature, by="patient") %>% 
  dplyr::rename("hypoxia_z_score"="z_score") %>% 
   dplyr::rename("eIF3e_z_score"="total_z_score")


```

## Divide patients into 3e sig high/low and hypoxia signature high/low

Make four groups based on quandrants -- 3e levels in hypoxia
```{r}

patient_ye_and_3e <- patient_ye_and_3e %>% 
  mutate(quadrant = ifelse(hypoxia_z_score < 0 & eIF3e_z_score < 0, "low hypoxia and low 3e",
                           ifelse(hypoxia_z_score > 0 & eIF3e_z_score < 0, "high hypoxia and low 3e",
                                  ifelse(hypoxia_z_score > 0 & eIF3e_z_score > 0, "high hypoxia and high 3e",
                                         ifelse(hypoxia_z_score < 0 & eIF3e_z_score > 0, "low hypoxia and high 3e", "other")))))


patient_ye_and_3e$quadrant <- factor(patient_ye_and_3e$quadrant, levels=c("low hypoxia and low 3e","high hypoxia and low 3e","low hypoxia and high 3e", "high hypoxia and high 3e"))


```

Determine correlation and color by group
```{r}
ggplot(patient_ye_and_3e, aes(x=hypoxia_z_score, y= eIF3e_z_score, color=quadrant))+
  geom_point()+
  xlab("Ye hypoxia signature z score")+
  ylab("eIF3e-hyp z score")+
  theme_prism()+ 
  theme(text = element_text(size = 14))+
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0)+
  scale_color_manual(values=c("high hypoxia and high 3e"="#9B4D96", 
                       "high hypoxia and low 3e"="red", 
                       "low hypoxia and low 3e"="grey", 
                       "low hypoxia and high 3e"= "#56B4E9"))


ggsave("3e_hyp_and_ye.pdf", path=here("plots/fig8"),width = 5.5, height=3)

cor.test(patient_ye_and_3e$hypoxia_z_score, patient_ye_and_3e$eIF3e_z_score, method="pearson")
```

Plot survival curve
```{r}
#bind with clin data
patient_ye_and_3e_hyp_clin <- left_join(clin, patient_ye_and_3e, by=c("PATIENT_ID"="patient"))

#plot

fit_all_hypoxia_3e <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = patient_ye_and_3e_hyp_clin)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, patient_ye_and_3e_hyp_clin)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)


levels(patient_ye_and_3e_hyp_clin$quadrant)

surv_plot_3e_and_hypoxia_sig<-
ggsurvplot(fit_all_hypoxia_3e,
           data = patient_ye_and_3e_hyp_clin,
           size = 1,
           palette = c("grey", "red", si3e_col, "#9B4D96"), 
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 7,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3e_and_hypoxia_sig$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("surv_plot_3e_and_hypoxia_sig.pdf", path=here("plots/fig8"),width = 11.5, height=8)

```

## Metabric data controls:
1. exclusing points in the middle of the correlation chart as negative controls.
2. Neg control/random quantiles
3. Pos control - checking death with cancer status

Control: excluding points in the middle as a negative control for the survival curve analysis
Same as earlier with a filter
@Kate and @Neel do we need this in here?? -- I think I can remove this whole chunk
```{r}

ctrl_patient_ye_and_3e <- patient_ye_and_3e %>% 
  mutate(change = case_when(
    (hypoxia_z_score < -0.1 | hypoxia_z_score > 0.1) &
      (eIF3e_z_score < -0.1 | eIF3e_z_score > 0.1) ~ "big_change",
    .default="small_change"
  ))

ctrl_patient_ye_and_3e <- ctrl_patient_ye_and_3e %>% 
  filter(change == "big_change")


ggplot(ctrl_patient_ye_and_3e, aes(x=hypoxia_z_score, y= eIF3e_z_score))+
  geom_point()+
  xlab("Ye hypoxia signature z score")+
  ylab("eIF3e-hyp z score")+
  theme_cowplot()+ 
  theme(text = element_text(size = 14))+
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0)


ggsave("3e_hyp_and_ye.pdf", path=here("plots/fig8"),width = 4, height=3)

cor.test(ctrl_patient_ye_and_3e$hypoxia_z_score, ctrl_patient_ye_and_3e$eIF3e_z_score, method="pearson")

#p value still signficiant

#. test survival curve

ctrl_patient_ye_and_3e <- ctrl_patient_ye_and_3e %>% 
  mutate(quadrant = ifelse(hypoxia_z_score < 0 & eIF3e_z_score < 0, "low hypoxia and low eIF3e (hypoxia)",
                           ifelse(hypoxia_z_score > 0 & eIF3e_z_score < 0, "high hypoxia and low eIF3e (hypoxia)",
                                  ifelse(hypoxia_z_score > 0 & eIF3e_z_score > 0, "high hypoxia and high eIF3e (hypoxia)",
                                         ifelse(hypoxia_z_score < 0 & eIF3e_z_score > 0, "low hypoxia sig and high eIF3e (hypoxia)", "other")))))

#bind with clin data
ctrl_patient_ye_and_3e_hyp_clin <- left_join(clin, ctrl_patient_ye_and_3e, by=c("PATIENT_ID"="patient"))

#plot

fit_all_hypoxia_3e <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = ctrl_patient_ye_and_3e_hyp_clin)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, patient_ye_and_3e_hyp_clin)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

ctrl_patient_ye_and_3e_hyp_clin$quadrant <- factor(ctrl_patient_ye_and_3e_hyp_clin$quadrant, levels=c("low hypoxia and low eIF3e (hypoxia)", "high hypoxia and low eIF3e (hypoxia)", "low hypoxia sig and high eIF3e (hypoxia)", "high hypoxia and high eIF3e (hypoxia)"))

quadrant_levels <- levels(ctrl_patient_ye_and_3e_hyp_clin$quadrant)

# ctrl_surv_plot_3e_and_hypoxia_sig<-
# ggsurvplot(fit_all_hypoxia_3e,
#            data = ctrl_patient_ye_and_3e_hyp_clin,
#            size = 1,
#            palette = c("#E7B800", "pink", "purple", "#2E9FDF"), 
#            break.time.by = 1,
#            ylab="Overall Survival",
#            xlab = "Time in years",
#            xlim = c(0,10),
#            font.x = 30,      
#            font.y = 30,      
#            font.tickslab = 25,
#            font.legend = 7,
#            conf.int = FALSE,
#            pval = F,
#            risk.table = FALSE,
#            risk.table.col = "strata",
#            risk.table.height = 0.25,
#            ggtheme = theme_prism())
# 
# ctrl_surv_plot_3e_and_hypoxia_sig$plot +
#   annotate(
#     "text", 
#     x = 3, y = 0.2, 
#     label = paste("HR =", hr, "\np =", pval), 
#     size = 12, 
#     color = "black"
#   )
# 

#trend remains, even when taking out values with low levels of changes


```

## Controls

## Positive control

Visualize OS_STATUS

```{r}
fit_pos <- survfit(Surv(OS_YEARS, OS_STATUS) ~ OS_STATUS, data = all_data_3e_hyp)


ggsurvplot(fit_pos, data = all_data_3e_hyp, size = 1, palette = c("#E7B800", "#2E9FDF"),legend.labs = c("alive", "died_from_cancer"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = TRUE, risk.table = TRUE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())

```

## Negative Control for eIF3e signature

Now neg control -- random genes comprising signatures-- same number of genes just random

In my 3e hyp signature, there are:
78 up genes
77 down genes

Create random gene signatures and evaluate how they impact survival
```{r}

all_genes <- rownames(merged_zscores)

fake_pos_list <-all_genes[sample(1:length(all_genes), 78)]
fake_neg_list <-all_genes[sample(1:length(all_genes), 77)]


# empty df
neg_ctrl_zscores <- data.frame(patient=colnames(exp_3e),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(neg_ctrl_zscores)) {
  neg_ctrl_zscores[i,2] <- mean(exp_3e[fake_pos_list,i],  na.rm=TRUE)
  neg_ctrl_zscores[i,3] <- mean(exp_3e[fake_neg_list,i],  na.rm=TRUE)
  }

neg_ctrl_zscores <- neg_ctrl_zscores %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(neg_ctrl_zscores$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

neg_ctrl_zscores$quantile <- cut(neg_ctrl_zscores$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

neg_ctrl_zscores$patient <- gsub("\\.", "-", neg_ctrl_zscores$patient)

# Survival plot

#Bind z score hypoxia signature with clin data


all_data_neg_ctrl <- left_join(clin, neg_ctrl_zscores, by=c("PATIENT_ID"="patient"))


#plot

neg_ctrl_fit <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_neg_ctrl)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_neg_ctrl)
summary_coxfit_ye <- summary(coxfit)


hr <- round(summary_coxfit_ye$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_ye$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)


surv_plot_neg_ctrl<-
ggsurvplot(neg_ctrl_fit,
           data = all_data_neg_ctrl,
           size = 1,
           palette = c("#E7B800", "pink", "purple", "#2E9FDF"), 
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_neg_ctrl$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )


#boxplot of quartile expression

ggplot(all_data_neg_ctrl, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("neg ctrl z scores")


```

Repeat this 1000 times and determine how many times after 1000 we get a significant p value at random
Keeping this commented out because it takes forever @Kate probably remove, or if needed do this faster

```{r}
# Vector to store p-values
# pval_results <- numeric(1000)
# 
# for (i in 1:1000) {
#   # Select random samples for positive and negative gene lists
#   fake_pos_list <- all_genes[sample(1:length(all_genes), 78)]
#   fake_neg_list <- all_genes[sample(1:length(all_genes), 77)]
# 
#   # Reset neg_ctrl_zscores each iteration
#   neg_ctrl_zscores <- data.frame(patient = colnames(exp_3e),
#                                  up_z_score = NA,
#                                  down_z_score = NA)
# 
#   # Calculate z-scores
#   for (j in 1:nrow(neg_ctrl_zscores)) {
#     neg_ctrl_zscores[j, 2] <- mean(exp_3e[fake_pos_list, j], na.rm = TRUE)
#     neg_ctrl_zscores[j, 3] <- mean(exp_3e[fake_neg_list, j], na.rm = TRUE)
#   }
# 
#   neg_ctrl_zscores <- neg_ctrl_zscores %>%
#     mutate(total_z_score = up_z_score - down_z_score)
# 
#   # Determine quantiles and add to dataframe
#   quantile_groups <- quantile(neg_ctrl_zscores$total_z_score, probs = seq(0, 1, 0.25), na.rm = TRUE)
#   neg_ctrl_zscores$quantile <- cut(neg_ctrl_zscores$total_z_score,
#                                    breaks = quantile_groups,
#                                    include.lowest = TRUE,
#                                    labels = FALSE)
# 
#   neg_ctrl_zscores$patient <- gsub("\\.", "-", neg_ctrl_zscores$patient)
# 
# 
#   # Merge with clinical data
#   all_data_neg_ctrl <- left_join(clin, neg_ctrl_zscores, by = c("PATIENT_ID" = "patient"))
# 
#   # Perform Cox regression
#   coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_neg_ctrl)
#   summary_coxfit <- summary(coxfit)
# 
#   # Extract p-value and store
#   pval_results[i] <- summary_coxfit$coefficients[1, "Pr(>|z|)"]
# }
# 
# # Calculate the proportion of p-values below 0.05
# num_significant <- sum(pval_results < 0.05)
# proportion_significant <- num_significant / 1000
# 
# cat("Number of significant p-values:", num_significant, "\n")
# cat("Proportion of significant p-values:", proportion_significant, "\n")

```


# Repeat analysis using TCGA dataset --- REMOVE for this publication
TCGA Firehouse legacy
Downloaded from cbioportal

1,108 samples
```{r cache = TRUE}
zscores_tcga <- read.table(here("cbioportal_data/brca_tcga/tcga_data_mrna_seq_v2_rsem_zscores_ref_diploid_samples.txt"), sep="\t", header=T)
row.names(zscores) <- gsub("_", ".",row.names(zscores))

zscores_tcga_all_symbol <- zscores_tcga %>% 
  dplyr::select(-Entrez_Gene_Id)

### Average scores for duplicates

dup_genes_tcga <- zscores_tcga_all_symbol$Hugo_Symbol[duplicated(zscores_tcga_all_symbol$Hugo_Symbol)]

dedup <- zscores_tcga_all_symbol %>%
  filter(Hugo_Symbol %in% dup_genes_tcga) %>%
  group_by(Hugo_Symbol) %>%
  summarise(across(everything(), mean))

merged_zscores_tcga <- zscores_tcga_all_symbol %>%
  filter(!Hugo_Symbol %in% dup_genes_tcga) %>% 
  bind_rows(., dedup) %>% 
  as.tibble() %>%
  column_to_rownames("Hugo_Symbol") %>% 
  as.data.frame()

```

# eIF3e hypoxia signature and survival prediction
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3e_tcga <- merged_zscores_tcga %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% sig_rna_3e_hyp$symbol) %>% 
  column_to_rownames("Hugo_Symbol")

# empty df
sig_zscores_3e_tcga <- data.frame(patient=colnames(exp_3e_tcga),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3e_tcga)) {
  sig_zscores_3e_tcga[i,2] <- mean(exp_3e_tcga[up_3e_hyp,i],  na.rm=TRUE)
  sig_zscores_3e_tcga[i,3] <- mean(exp_3e_tcga[down_3e_hyp,i],  na.rm=TRUE)
  }

sig_zscores_3e_tcga <- sig_zscores_3e_tcga %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3e_tcga$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3e_tcga$quantile <- cut(sig_zscores_3e_tcga$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_3e_tcga$patient <- gsub("\\.", "-", as.character(sig_zscores_3e_tcga$patient))
sig_zscores_3e_tcga$patient <- sub("-[^-]*$", "", sig_zscores_3e_tcga$patient)

```

## Import TCGA patient data
```{r}
clin_tcga <- read_excel(here("cbioportal_data/new_tcga_clinical.xlsx"))
colnames(clin_tcga)

#do not need all these columns

clin_small <- clin_tcga %>% 
  dplyr::select(OS_STATUS, OS_MONTHS, PATIENT_ID)

# years and months in this are not reading correctly
#get rid of not available
clin_small <- clin_small %>%
  filter(OS_MONTHS != "[Not Available]") %>% 
  filter(OS_MONTHS > 0)

clin_small$OS_MONTHS <- as.numeric(clin_small$OS_MONTHS)

clin_small$OS_STATUS[which(clin_small$OS_STATUS == "1:DECEASED" & clin_small$OS_MONTHS > 120)] <- "0:LIVING"
 
 clin_small$OS_STATUS <- clin_small$OS_STATUS == "1:DECEASED"

clin$OS_MONTHS <- as.numeric(clin$OS_MONTHS)

clin_small <- clin_small %>% 
  mutate(OS_YEARS= clin_small$OS_MONTHS/12)

clin_small$OS_YEARS[which(clin_small$OS_YEARS >= 10)] <- 10
```


## 3e survival
```{r}
#bind clin data with z score data

all_tcga_3e <- left_join(clin_small, sig_zscores_3e_tcga, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA") %>% 
  filter(quantile==1 | quantile==4)

fit_tcga_3e <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_tcga_3e)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_tcga_3e)
summary_coxfit<- summary(coxfit)

hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", si3e_col))(2)


surv_plot_3e_tcga<-
ggsurvplot(fit_tcga_3e,
           data = all_tcga_3e,
           size = 1,
           palette = colors, 
          legend.labs=c("Low 3e", "High 3e"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = TRUE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3e_tcga$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("tcga_3e.pdf", path=here("plots/fig8"),width = 8, height=6)

#boxplot of quartile expression

ggplot(all_tcga_3e, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3e hypoxia signature")

```

## 3d TCGA
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3d_tcga <- merged_zscores_tcga %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% sig_rna_3d_hyp$symbol) %>% 
  column_to_rownames("Hugo_Symbol")

# empty df
sig_zscores_3d_tcga <- data.frame(patient=colnames(exp_3d_tcga),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3d_tcga)) {
  sig_zscores_3d_tcga[i,2] <- mean(exp_3d_tcga[up_3d_hyp,i],  na.rm=TRUE)
  sig_zscores_3d_tcga[i,3] <- mean(exp_3d_tcga[down_3d_hyp,i],  na.rm=TRUE)
  }

sig_zscores_3d_tcga <- sig_zscores_3d_tcga %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3d_tcga$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3d_tcga$quantile <- cut(sig_zscores_3d_tcga$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_3d_tcga$patient <- gsub("\\.", "-", as.character(sig_zscores_3d_tcga$patient))
sig_zscores_3d_tcga$patient <- sub("-[^-]*$", "", sig_zscores_3d_tcga$patient)

```

```{r}
#bind clin data with z score data

all_tcga_3d <- left_join(clin_small, sig_zscores_3d_tcga, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA") %>% 
    filter(quantile==1 | quantile==4)


fit_tcga_3d <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_tcga_3d)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_tcga_3d)
summary_coxfit<- summary(coxfit)

hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", si3d_col))(2)


surv_plot_3d_tcga<-
ggsurvplot(fit_tcga_3d,
           data = all_tcga_3d,
           size = 1,
           palette = colors,
                     legend.labs=c("Low 3d", "High 3d"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = TRUE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3d_tcga$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("tcga_3d.pdf", path=here("plots/fig8"),width = 8, height=6)




#boxplot of quartile expression

ggplot(all_tcga_3d, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3d hypoxia signature")

```

## Hypoxia signature TCGA

```{r}

#calculate z scores for enrichment of hypoxia signature

# filter merged z score data based on genes in my signature
exp_hyp_sig_tcga <- merged_zscores_tcga %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% ye_hyp_sig$up_in_hypoxia) %>% 
  column_to_rownames("Hugo_Symbol")

# empty df
sig_zscores_hyp_signature_tcga <- data.frame(patient=colnames(exp_hyp_sig_tcga),z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_hyp_signature_tcga)) {
  sig_zscores_hyp_signature_tcga[i,2] <- mean(exp_hyp_sig_tcga[,i],  na.rm=TRUE)
}


quantile_groups <- quantile(sig_zscores_hyp_signature_tcga$z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_hyp_signature_tcga$quantile <- cut(sig_zscores_hyp_signature_tcga$z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_hyp_signature_tcga$patient <- sub("-[^-]*$", "", sig_zscores_hyp_signature_tcga$patient)

sig_zscores_hyp_signature_tcga$patient <- gsub("\\.", "-", sig_zscores_hyp_signature_tcga$patient)

sig_zscores_hyp_signature_tcga$patient <- gsub("-01$", "", sig_zscores_hyp_signature_tcga$patient)




#Bind z score hypoxia signature with clin data


all_data_hif_ye_tcga <- left_join(clin_small, sig_zscores_hyp_signature_tcga, by=c("PATIENT_ID"="patient")) %>% 
      filter(quantile==1 | quantile==4)



#plot

all_data_hif_ye_tcga_surv <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_hif_ye_tcga)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_hif_ye_tcga)
summary_coxfit_ye <- summary(coxfit)


hr <- round(summary_coxfit_ye$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_ye$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", "red"))(2)

surv_plot_ye_hyp_tcga<-
ggsurvplot(all_data_hif_ye_tcga_surv,
           data = all_data_hif_ye_tcga,
           size = 1,
           palette = colors, 
           break.time.by = 1,
          legend.labs=c("Low Hypoxia", "High Hypoxia"),
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_ye_hyp_tcga$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("surv_plot_ye_hyp_tcga.pdf", path=here("plots/fig8"),width = 8, height=6)


#boxplot of quartile expression

ggplot(all_data_hif_ye_tcga, aes(x=quantile,y=z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3d hypoxia signature")

```




## Positive control

Visualize OS_STATUS

```{r}
fit_pos <- survfit(Surv(OS_YEARS, OS_STATUS) ~ OS_STATUS, data = all_tcga_3e)


ggsurvplot(fit_pos, data = all_data_3e_hyp, size = 1, palette = c("#E7B800", "#2E9FDF"),legend.labs = c("alive", "died_from_cancer"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = TRUE, risk.table = TRUE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())



```

## Negative Control for eIF3e signature @Kate need to find a way to do this better (like done in MPRA) with P values

Now neg control -- random genes comprising signatures-- same number of genes just random

In my 3e hyp signature, there are:
78 up genes
77 down genes

Create random gene signatures and evaluate how they impact survival
```{r}

all_genes <- rownames(merged_zscores)

fake_pos_list <-all_genes[sample(1:length(all_genes), 78)]
fake_neg_list <-all_genes[sample(1:length(all_genes), 77)]


# empty df
neg_ctrl_zscores <- data.frame(patient=colnames(exp_3e),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(neg_ctrl_zscores)) {
  neg_ctrl_zscores[i,2] <- mean(exp_3e[fake_pos_list,i],  na.rm=TRUE)
  neg_ctrl_zscores[i,3] <- mean(exp_3e[fake_neg_list,i],  na.rm=TRUE)
  }

neg_ctrl_zscores <- neg_ctrl_zscores %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(neg_ctrl_zscores$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

neg_ctrl_zscores$quantile <- cut(neg_ctrl_zscores$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

neg_ctrl_zscores$patient <- gsub("\\.", "-", neg_ctrl_zscores$patient)

# Survival plot

#Bind z score hypoxia signature with clin data


all_data_neg_ctrl <- left_join(clin, neg_ctrl_zscores, by=c("PATIENT_ID"="patient"))


#plot

neg_ctrl_fit <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_neg_ctrl)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_neg_ctrl)
summary_coxfit_ye <- summary(coxfit)


hr <- round(summary_coxfit_ye$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_ye$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)


surv_plot_neg_ctrl<-
ggsurvplot(neg_ctrl_fit,
           data = all_data_neg_ctrl,
           size = 1,
           palette = c("#E7B800", "pink", "purple", "#2E9FDF"), 
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_neg_ctrl$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )

```

Repeat this 1000 times
Keeping this commented out because it takes forever @Kate fix-- and decide if we want TCGA data in the paper
```{r}
# #Vector to store p-values
# pval_results <- numeric(1000)
# 
# for (i in 1:1000) {
#   # Select random samples for positive and negative gene lists
#   fake_pos_list <- all_genes[sample(1:length(all_genes), 78)]
#   fake_neg_list <- all_genes[sample(1:length(all_genes), 77)]
# 
#   # Reset neg_ctrl_zscores each iteration
#   neg_ctrl_zscores <- data.frame(patient = colnames(exp_3e),
#                                  up_z_score = NA,
#                                  down_z_score = NA)
# 
#   # Calculate z-scores
#   for (j in 1:nrow(neg_ctrl_zscores)) {
#     neg_ctrl_zscores[j, 2] <- mean(exp_3e[fake_pos_list, j], na.rm = TRUE)
#     neg_ctrl_zscores[j, 3] <- mean(exp_3e[fake_neg_list, j], na.rm = TRUE)
#   }
# 
#   neg_ctrl_zscores <- neg_ctrl_zscores %>%
#     mutate(total_z_score = up_z_score - down_z_score)
# 
#   # Determine quantiles and add to dataframe
#   quantile_groups <- quantile(neg_ctrl_zscores$total_z_score, probs = seq(0, 1, 0.25), na.rm = TRUE)
#   neg_ctrl_zscores$quantile <- cut(neg_ctrl_zscores$total_z_score,
#                                    breaks = quantile_groups,
#                                    include.lowest = TRUE,
#                                    labels = FALSE)
# 
#   neg_ctrl_zscores$patient <- gsub("\\.", "-", neg_ctrl_zscores$patient)
# 
# 
#   # Merge with clinical data
#   all_data_neg_ctrl <- left_join(clin, neg_ctrl_zscores, by = c("PATIENT_ID" = "patient"))
# 
#   # Perform Cox regression
#   coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_neg_ctrl)
#   summary_coxfit <- summary(coxfit)
# 
#   # Extract p-value and store
#   pval_results[i] <- summary_coxfit$coefficients[1, "Pr(>|z|)"]
# }
# 
# # Calculate the proportion of p-values below 0.05
# num_significant <- sum(pval_results < 0.05)
# proportion_significant <- num_significant / 1000
# 
# cat("Number of significant p-values:", num_significant, "\n")
# cat("Proportion of significant p-values:", proportion_significant, "\n")

```


## Looking at copy number alterations in data

```{r}
cna <- read.table(here("cbioportal_data/data_cna.txt"), fill=TRUE, header=TRUE) %>% 
  dplyr::select(-Entrez_Gene_Id)

eIF3e_cna <- cna %>% 
  filter(Hugo_Symbol=="EIF3E") 

# eIF3e_cna[ , 2:2174] <- lapply(eIF3e_cna[ , 2:2174], as.character)


eIF3e_cna<- eIF3e_cna%>%
  pivot_longer(cols= 2:2174, names_to = "patient", values_to = "cna")


#take just MB

eIF3e_cna$patient <- gsub("^MB.", "MB-", eIF3e_cna$patient)

eIF3e_cna_cancer <- eIF3e_cna %>% 
  filter(str_detect(patient, "MB-")) %>% 
  dplyr::select(-Hugo_Symbol)

eIF3e_cna_cancer$cna <- as.numeric(eIF3e_cna_cancer$cna)


#bind with patient data

# ggplot(eIF3e_cna_cancer, aes(x=cna))+
#   geom_histogram()+
#   theme_cowplot()
 
eIF3e_cna_cancer_clin <- left_join(clin, eIF3e_cna_cancer, by=c("PATIENT_ID"="patient"))

eIF3e_cna_cancer_clin <- eIF3e_cna_cancer_clin %>% 
  mutate(copy_number_alt = case_when(
    cna >0 ~ "gain",
    cna == 0 ~ "no_change",
    cna < 0 ~ "loss",
    .default="other"
  ))


eIF3e_cna_cancer_clin %>% 
  filter(cna !="NA") %>% 
  group_by(cna) %>% 
  summarise(count = n()) %>%
  mutate(percentage = count / sum(count) * 100)

#get rid of -1

eIF3e_cna_cancer_clin <- eIF3e_cna_cancer_clin %>% 
  filter(cna > -1)

eIF3e_cna_cancer_clin$cna <- as.factor(eIF3e_cna_cancer_clin$cna)
cna_3e_fit <- survfit(Surv(OS_YEARS, OS_STATUS) ~ cna, data = eIF3e_cna_cancer_clin)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ cna, eIF3e_cna_cancer_clin)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", si3e_col))(3)

surv_plot_cna<-
ggsurvplot(cna_3e_fit,
           data = eIF3e_cna_cancer_clin,
           size = 1,
           palette = c("grey", si3e_col, "#0b6ca3"), 
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_cna$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )

ggsave("cna_3e.pdf", path=here("plots/fig8"),width = 8, height=6)

 #percentage of copy number gains for each



```

## copy numbers of eIF3d
```{r}
eIF3d_cna <- cna %>% 
  filter(Hugo_Symbol=="EIF3D") 

# eIF3d_cna[ , 2:2174] <- lapply(eIF3d_cna[ , 2:2174], as.character)


eIF3d_cna<- eIF3d_cna%>%
  pivot_longer(cols= 2:2174, names_to = "patient", values_to = "cna")


#take just MB

eIF3d_cna$patient <- gsub("^MB.", "MB-", eIF3d_cna$patient)

eIF3d_cna_cancer <- eIF3d_cna %>% 
  filter(str_detect(patient, "MB-")) %>% 
  dplyr::select(-Hugo_Symbol)

eIF3d_cna_cancer$cna <- as.numeric(eIF3d_cna_cancer$cna)


#bind with patient data

# ggplot(eIF3d_cna_cancer, aes(x=cna))+
#   geom_histogram()+
#   theme_cowplot()
 
eIF3d_cna_cancer_clin <- left_join(clin, eIF3d_cna_cancer, by=c("PATIENT_ID"="patient"))

eIF3d_cna_cancer_clin <- eIF3d_cna_cancer_clin %>% 
  mutate(copy_number_alt = case_when(
    cna >0 ~ "gain",
    cna == 0 ~ "no_change",
    cna < 0 ~ "loss",
    .default="other"
  ))


eIF3d_cna_cancer_clin %>% 
  filter(cna !="NA") %>% 
  group_by(cna) %>% 
  summarise(count = n()) %>%
  mutate(percentage = count / sum(count) * 100)

#get rid of -1

eIF3d_cna_cancer_clin <- eIF3d_cna_cancer_clin %>% 
  filter(cna > -2 & cna < 2)


cna_3d_fit <- survfit(Surv(OS_YEARS, OS_STATUS) ~ cna, data = eIF3d_cna_cancer_clin)

eIF3d_cna_cancer_clin$cna <- relevel(as.factor(eIF3d_cna_cancer_clin$cna), ref = "0")

coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ cna, eIF3d_cna_cancer_clin)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)


colors <- grDevices::colorRampPalette(c("grey", si3d_col))(3)


surv_plot_cna<-
ggsurvplot(cna_3d_fit,
           data = eIF3d_cna_cancer_clin,
           size = 1,
           palette = c("#8a8884", "grey", si3d_col), 
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_cna$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )

ggsave("cna_3d.pdf", path=here("plots/fig8"),width = 8, height=6)

#percentage of copy number gains for each


```

barchart of copy numbers
```{r}
cna_bar <-
cna %>% 
  filter(Hugo_Symbol=="EIF3D"|Hugo_Symbol=="EIF3E") %>% 
  pivot_longer(cols= 2:2174, names_to = "patient", values_to = "cna")

cna_bar$patient <- gsub("^MB.", "MB-", cna_bar$patient)

cna_bar_cancer <- cna_bar %>% 
  filter(str_detect(patient, "MB-"))

cna_bar_cancer$cna <- as.numeric(cna_bar_cancer$cna)


cna_bar_cancer %>% 
 group_by(cna) %>% 
  summarise(count = n()) %>%
  mutate(percentage = count / sum(count) * 100)


cna_percent <-
cna_bar_cancer %>%
  group_by(Hugo_Symbol, cna) %>%  # Group by gene symbol and cna
  summarise(count = n()) %>%      # Count occurrences of each cna per gene symbol
  group_by(Hugo_Symbol) %>%       # Group by gene symbol to calculate percentage within each gene
  mutate(percentage = count / sum(count) * 100)


ggplot(cna_percent, aes(x=cna, y=percentage, fill=Hugo_Symbol))+
  geom_bar(stat="identity")+
  facet_wrap(~Hugo_Symbol)+
  scale_fill_manual(values = c(si3d_col, si3e_col))+
  theme_prism()


ggsave("cna_percentages.pdf", path=here("plots/fig8"),width = 4, height=2)








```




myc
```{r}
myc_cna <- cna %>% 
  filter(Hugo_Symbol=="MYC") 

# myc_cna[ , 2:2174] <- lapply(myc_cna[ , 2:2174], as.character)


myc_cna<- myc_cna%>%
  pivot_longer(cols= 2:2174, names_to = "patient", values_to = "cna")


#take just MB

myc_cna$patient <- gsub("^MB.", "MB-", myc_cna$patient)

myc_cna_cancer <- myc_cna %>% 
  filter(str_detect(patient, "MB-")) %>% 
  dplyr::select(-Hugo_Symbol)

myc_cna_cancer$cna <- as.numeric(myc_cna_cancer$cna)


#bind with patient data

# ggplot(myc_cna_cancer, aes(x=cna))+
#   geom_histogram()+
#   theme_cowplot()
 
myc_cna_cancer_clin <- left_join(clin, myc_cna_cancer, by=c("PATIENT_ID"="patient"))

myc_cna_cancer_clin <- myc_cna_cancer_clin %>% 
  mutate(copy_number_alt = case_when(
    cna >0 ~ "gain",
    cna == 0 ~ "no_change",
    cna < 0 ~ "loss",
    .default="other"
  ))


myc_cna_cancer_clin %>% 
  filter(cna !="NA") %>% 
  group_by(cna) %>% 
  summarise(count = n()) %>%
  mutate(percentage = count / sum(count) * 100)

#get rid of -1

myc_cna_cancer_clin <- myc_cna_cancer_clin %>% 
  filter(cna > -1 & cna < 3)


cna_myc_fit <- survfit(Surv(OS_YEARS, OS_STATUS) ~ cna, data = myc_cna_cancer_clin)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ cna, myc_cna_cancer_clin)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)


surv_plot_cna<-
ggsurvplot(cna_myc_fit,
           data = myc_cna_cancer_clin,
           size = 1,
           palette = c("#E7B800", "pink", "purple", "#2E9FDF"), 
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_cna$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )

ggsave("cna_myc.pdf", path=here("plots/fig8"),width = 11.5, height=8)

#percentage of copy number gains for each


```

