---
title: "Figure 8 - 3e/d RNA signatures in patient data"
author: "Kate"
date: "2024-08-27"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(here)
library(tidyverse)
library(janitor)
library(readxl)
library(cBioPortalData)
library(MetaIntegrator)
library(cbioportalR)
library(pheatmap)
library(cowplot)
library(ggpubr)
library(survival)
library(survminer)
library(GSVA)
library(GSEABase)
library(matrixStats)
library(clusterProfiler)
library(msigdbr)
library(GeneOverlap)
library(UpSetR)
library(ComplexHeatmap)
library(reshape2)
```


# Generate RNA signatures

Pull in raw RNA data and do differential gene expression

## load metadata
```{r}
metadata <- read_csv(here("calculate_te/metadata.csv"))

```

```{r}
geneInfo <- read_csv(here("accessories","gencode.v26.primary.info.csv.gz"), col_names = F) %>%
  filter(!str_detect(X2, "pre_"))

colnames(geneInfo) <- c("gene_id","transcript_id","biotype","symbol")

#map transcripts to gene
tx2gene <- geneInfo[,c(2,1)]
gene_symbol <- geneInfo %>% dplyr::select(gene_id, symbol) %>% unique()

```

## Separate ribo counts, rna counts, and te counts
```{r}
raw_counts <- read_csv(here("calculate_te/raw_filtered_allCounts.csv")) %>% 
  column_to_rownames("gene_id")

ribo_counts <- as.matrix(raw_counts[,grep(pattern = "ribo",colnames(raw_counts))])
rna_counts <- as.matrix(raw_counts[,grep(pattern = "rna",colnames(raw_counts))])

rna_ribo_counts <- as.matrix(raw_counts)

```

# 3e hypoxia rna
```{r}

metadata_loop <- metadata %>% 
  filter(library=="rna") %>% 
  filter(condition=="hypoxia") %>% 
  filter(treatment=="sictrl"|treatment=="si3e")

metadata_loop$treatment <- factor(metadata_loop$treatment, levels= c("si3e", "sictrl"))
levels(metadata_loop$treatment)

sample_info <- metadata_loop %>% 
    mutate("name" = 
  paste(condition, treatment, rep, library, sep = "_"))
  
  names <- as.character(sample_info$name)
  names
  
  rna_subset <- rna_ribo_counts[, colnames(rna_ribo_counts) %in% names]
  
  print(levels(sample_info$treatment))

  
# Deseq analysis ----------------------------------------------------------------
  
  dds <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  dds <- DESeq(dds)
  
  print(resultsNames(dds))

  res <- results(dds, name=resultsNames(dds)[2], cooksCutoff=FALSE, independentFiltering=FALSE)

  res_hyp_3e <- lfcShrink(dds = dds, coef = resultsNames(dds)[2], res = res) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)

# export to excel ---------------------------------------------------------------
  
   writexl::write_xlsx(x = res_hyp_3e, path = here("counts/fig8/rna_signature_3e_hypoxia.xlsx"), col_names = T)

#confirm direction
  res_hyp_3e %>% 
    filter(symbol=="EIF3E")

```

# 3d hypoxia rna
```{r}

metadata_loop <- metadata %>% 
  filter(library=="rna") %>% 
  filter(condition=="hypoxia") %>% 
  filter(treatment=="sictrl"|treatment=="si3d")

metadata_loop$treatment <- factor(metadata_loop$treatment, levels= c("si3d", "sictrl"))
levels(metadata_loop$treatment)

sample_info <- metadata_loop %>% 
    mutate("name" = 
  paste(condition, treatment, rep, library, sep = "_"))
  
  names <- as.character(sample_info$name)
  names
  
  rna_subset <- rna_ribo_counts[, colnames(rna_ribo_counts) %in% names]
  
  print(levels(sample_info$treatment))

  
# Deseq analysis ----------------------------------------------------------------
  
  dds <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  dds <- DESeq(dds)
  
  print(resultsNames(dds))

  res <- results(dds, name=resultsNames(dds)[2], cooksCutoff=FALSE, independentFiltering=FALSE)

  res_hyp_3d <- lfcShrink(dds = dds, coef = resultsNames(dds)[2], res = res) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)

# export to excel ---------------------------------------------------------------
  
   writexl::write_xlsx(x = res_hyp_3d, path = here("counts/fig8/rna_signature_3d_hypoxia.xlsx"), col_names = T)

#confirm direction
  res_hyp_3d %>% 
    filter(symbol=="EIF3D")

```
# RIBO and TE sig-- in another file not for pub?


# Prepare survival analysis
### Separate RNA signatures by direction

```{r}
# 3e hypoxia

sig_rna_3e_hyp <- res_hyp_3e %>% 
  filter(padj < 0.05)

up_3e_hyp <- sig_rna_3e_hyp %>% 
  filter(log2FoldChange > 1)%>% 
  pull(symbol)

down_3e_hyp <- sig_rna_3e_hyp %>% 
  filter(log2FoldChange < -1)%>% 
  pull(symbol)
  
```

```{r}
# 3d hypoxia

sig_rna_3d_hyp <- res_hyp_3d %>% 
  filter(padj < 0.05)

up_3d_hyp <- sig_rna_3d_hyp %>% 
  filter(log2FoldChange > 1) %>% 
  pull(symbol)
  

down_3d_hyp <- sig_rna_3d_hyp %>% 
  filter(log2FoldChange < -1)%>% 
  pull(symbol)
  

```


# Import METABRIC microarray data from cbioportal - read patient z scores and matching clinical data
Nature 2012 and Nat Commun 2016

```{r cache = TRUE}
zscores <- read.table(here("cbioportal_data/brca_metabric/data_mrna_illumina_microarray_zscores_ref_diploid_samples.txt"), sep="\t", header=T)
row.names(zscores) <- gsub("_", ".",row.names(zscores))

zscores_all_symbol <- zscores %>% 
  dplyr::select(-Entrez_Gene_Id)

### Average scores for duplicate patients

dup_genes <- zscores$Hugo_Symbol[duplicated(zscores$Hugo_Symbol)]

dedup <- zscores %>%
  filter(Hugo_Symbol %in% dup_genes) %>%
  group_by(Hugo_Symbol) %>%
  summarise(across(everything(), mean))


z_clean <- zscores %>%
  filter(!Hugo_Symbol %in% dup_genes) %>% 
  bind_rows(., dedup)

merged_zscores <- zscores %>%
  filter(!Hugo_Symbol %in% dup_genes) %>% 
  bind_rows(., dedup) %>% 
  as.tibble() %>%
  column_to_rownames("Hugo_Symbol") %>% 
  as.data.frame()

# exp <- merged_zscores[raw$symbol,] don't want to subset this yet

```

# eIF3e hypoxia signature and survival prediction
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3e <- merged_zscores %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% sig_rna_3e_hyp$symbol) %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  dplyr::select(-Entrez_Gene_Id)

# empty df
sig_zscores_3e <- data.frame(patient=colnames(exp_3e),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3e)) {
  sig_zscores_3e[i,2] <- mean(exp_3e[up_3e_hyp,i],  na.rm=TRUE)
  sig_zscores_3e[i,3] <- mean(exp_3e[down_3e_hyp,i],  na.rm=TRUE)
  }

sig_zscores_3e <- sig_zscores_3e %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3e$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3e$quantile <- cut(sig_zscores_3e$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_3e$patient <- sub("-[^-]*$", "", sig_zscores_3e$patient)

sig_zscores_3e$patient <- gsub("\\.", "-", sig_zscores_3e$patient)


```


# eIF3d hypoxia signature and survival prediction
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3d <- merged_zscores %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% sig_rna_3d_hyp$symbol) %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  dplyr::select(-Entrez_Gene_Id)

# empty df
sig_zscores_3d <- data.frame(patient=colnames(exp_3d),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3d)) {
  sig_zscores_3d[i,2] <- mean(exp_3d[up_3d_hyp,i],  na.rm=TRUE)
  sig_zscores_3d[i,3] <- mean(exp_3d[down_3d_hyp,i],  na.rm=TRUE)
  }

sig_zscores_3d <- sig_zscores_3d %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3d$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3d$quantile <- cut(sig_zscores_3d$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)


sig_zscores_3d$patient <- gsub("\\.", "-", sig_zscores_3d$patient)

```

# import clinical data
```{r}
clin <- read_excel(here("cbioportal_data/brca_metabric/data_clinical_patient_edit.xlsx"))
clin$OS_MONTHS <- as.numeric(clin$OS_MONTHS)
clin$OS_STATUS[which(clin$OS_STATUS == "1:DECEASED" & clin$OS_MONTHS > 120)] <- "0:LIVING"
clin$OS_STATUS <- clin$OS_STATUS == "1:DECEASED"
clin <- cbind(clin, OS_YEARS = clin$OS_MONTHS/12)
clin$OS_YEARS[which(clin$OS_YEARS >= 10)] <- 10

```

## 3e survival
```{r}
#bind clin data with z score data

all_data_3e_hyp <- left_join(clin, sig_zscores_3e, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA")

fit_all_3e_hyp <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_3e_hyp)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_3e_hyp)
summary_coxfit_3e_norm <- summary(coxfit)


hr <- round(summary_coxfit_3e_norm$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_3e_norm$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

# surv_plot_3e_hyp <- ggsurvplot(fit_all_3e_hyp, data = all_data_3e_hyp, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),legend.labs = c("depleted for eIF3e signature (hypoxia): low (Q1)","depleted: medium-low (Q2)", "enriched: medium-high (Q3)", "enriched for eIF3e signature (hypoxia): high (Q4)"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = FALSE,  risk.table.col = NA, risk.table.height = 0.25, ggtheme = theme_bw())
# 
# surv_plot_3e_hyp <- ggsurvplot(fit_all_3e_hyp, data = all_data_3e_hyp, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = FALSE,  risk.table.col = NA, risk.table.height = 0.25, ggtheme = theme_bw())
# 
# surv_plot_3e_hyp$plot <- surv_plot_3e_hyp$plot + 
#   annotate(
#     "text", 
#     x = 3, y = 0.2, 
#     label = paste("HR =", hr, "\np =", pval), 
#     size = 5, 
#     color = "black"
#   )+
#   theme(text = element_text(size = 14))+
#   ggtitle("Patient quartiles by eIF3e hypoxia signature")
# 
# surv_plot_3e_hyp
# 
# ggsave("3e_hyp_surv_plot.pdf", path=here("plots"),width = 11.5, height=8)
# 
# 

#new

surv_plot_3e_hyp<-
ggsurvplot(fit_all_3e_hyp,
           data = all_data_3e_hyp,
           size = 1,
           palette = c("#E7B800", "pink", "purple", "#2E9FDF"), 
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3e_hyp$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("3e_hyp_surv_plot.pdf", path=here("plots/fig8"),width = 11.5, height=8)




#boxplot of quartile expression

ggplot(all_data_3e_hyp, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3e hypoxia signature")


# maybe something like this could be relevant
ggviolin(data = all_data_3e_hyp %>% filter(ER_IHC != ""), y = "total_z_score", x = "ER_IHC", fill = "ER_IHC", palette = ggokabeito::palette_okabe_ito(), add = "median_iqr" ) +
  rotate_x_text(45)

ggviolin(data = all_data_3e_hyp %>% filter(THREEGENE != ""), y = "total_z_score", x = "THREEGENE", fill = "THREEGENE", palette = ggokabeito::palette_okabe_ito(), add = "boxplot", add.params = list(fill = "white")) +
  rotate_x_text(45)

```

## 3d survival
```{r}
#bind clin data with z score data

all_data_3d_hyp <- left_join(clin, sig_zscores_3d, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA")

fit_all_3d_hyp <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_3d_hyp)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_3d_hyp)
summary_coxfit_3d_norm <- summary(coxfit)


hr <- round(summary_coxfit_3d_norm$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_3d_norm$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

# surv_plot_3d_hyp <- ggsurvplot(fit_all_3d_hyp, data = all_data_3d_hyp, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),legend.labs = c("depleted for eIF3d signature (hypoxia): low (Q1)","depleted: medium-low (Q2)", "enriched: medium-high (Q3)", "enriched for eIF3d signature (hypoxia): high (Q4)"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = FALSE,  risk.table.col = NA, risk.table.height = 0.25, ggtheme = theme_bw())
# 
# surv_plot_3d_hyp <- ggsurvplot(fit_all_3d_hyp, data = all_data_3d_hyp, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = FALSE,  risk.table.col = NA, risk.table.height = 0.25, ggtheme = theme_bw())
# 
# surv_plot_3d_hyp$plot <- surv_plot_3d_hyp$plot + 
#   annotate(
#     "text", 
#     x = 3, y = 0.2, 
#     label = paste("HR =", hr, "\np =", pval), 
#     size = 5, 
#     color = "black"
#   )+
#   theme(text = element_text(size = 14))+
#   ggtitle("Patient quartiles by eIF3d hypoxia signature")
# 
# surv_plot_3d_hyp
# 
# ggsave("3d_hyp_surv_plot.pdf", path=here("plots"),width = 11.5, height=8)
# 
# 

#new

surv_plot_3d_hyp<-
ggsurvplot(fit_all_3d_hyp,
           data = all_data_3d_hyp,
           size = 1,
           palette = c("#E7B800", "pink", "purple", "#2E9FDF"), 
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3d_hyp$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("3d_hyp_surv_plot.pdf", path=here("plots/fig8"),width = 11.5, height=8)




#boxplot of quartile expression

ggplot(all_data_3d_hyp, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3d hypoxia signature")


# maybe something like this could be relevant
ggviolin(data = all_data_3d_hyp %>% filter(ER_IHC != ""), y = "total_z_score", x = "ER_IHC", fill = "ER_IHC", palette = ggokabeito::palette_okabe_ito(), add = "median_iqr" ) +
  rotate_x_text(45)

ggviolin(data = all_data_3d_hyp %>% filter(THREEGENE != ""), y = "total_z_score", x = "THREEGENE", fill = "THREEGENE", palette = ggokabeito::palette_okabe_ito(), add = "boxplot", add.params = list(fill = "white")) +
  rotate_x_text(45)

```

# stopped update here

## Import hypoxia signature from ye et al
Hypoxia signature in breast cancer:
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6279594/#:~:text=The%2042%2Dgene%20expression%20signature,mediate%20the%20conserved%20hypoxic%20response

```{r}
ye_hyp_sig <- read_excel(here("other_data/hypoxic_sig_ye.xlsx"))

#these are only up values-- so give z score values here


exp <- merged_zscores_matrix %>%
  as.data.frame() %>% 
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% ye_hyp_sig$up_in_hypoxia) %>% 
  column_to_rownames("Hugo_Symbol")

# up loop

patients <- colnames(merged_zscores_all)
patients <- patients[patients !="Hugo_Symbol"]



ye_hyp_sig$up_in_hypoxia %in% rownames(exp)



ye_hif_dataset <- data.frame(patient=character(), avg_z_score=numeric(), stringsAsFactors = FALSE)

for (i in patients) {
  
  #filter dataframe
  patient_level <- exp %>% 
    dplyr::select(i) %>%
    rownames_to_column("Hugo_Symbol") %>% 
    filter(Hugo_Symbol %in% ye_hyp_sig$up_in_hypoxia)
  
  all_values <- as.vector(patient_level[,2])
  
  
  #geometric mean
  
  up_genes_z_score <- mean(all_values, na.rm=TRUE)
  
  loop_dataset <- data.frame(patient=i, ye_hyp_sig_up = up_genes_z_score)
  
  ye_hif_dataset <- rbind(ye_hif_dataset, loop_dataset)
}

#plot distribution
ggplot(ye_hif_dataset, aes(x=ye_hyp_sig_up))+
  geom_histogram(bins = 400)


ye_hif_dataset$patient <- gsub("\\.", "-", as.character(ye_hif_dataset$patient))


### Stratify based on z-score

quantile_groups <- quantile(ye_hif_dataset$ye_hyp_sig_up, probs=seq(0,1,.25), na.rm=TRUE)

ye_hif_dataset$quantile <- cut(ye_hif_dataset$ye_hyp_sig_up, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

ye_hif_dataset$patient <- gsub("\\.", "-", as.character(ye_hif_dataset$patient))


#bind clin data with z score

all_data_hif_ye <- left_join(clin, ye_hif_dataset, by=c("PATIENT_ID"="patient"))


#plot

fit_all_data_hif_ye <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_hif_ye)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_hif_ye)
summary_coxfit_ye <- summary(coxfit)


hr <- round(summary_coxfit_ye$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_ye$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

surv_plot_ye <- ggsurvplot(fit_all_data_hif_ye, data = all_data_hif_ye, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"), break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = FALSE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())



surv_plot_ye$plot <- surv_plot_ye$plot + 
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 5, 
    color = "black"
  ) + theme(text = element_text(size = 14))

surv_plot_ye

ggsave("surv_plot_ye.pdf", path=here("plots"),width = 11.5, height=8)





ggplot(all_data_hif_ye, aes(x=quantile,y=ye_hyp_sig_up, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for Ye et al hypoxia signature")

ggsave("ye hyp boxplot.pdf",  path=here("plots"), width=4.5, height=4)




```


See if correlated with our sigs


#plot 3e score against ye hypoxia score

```{r}

patient_ye_and_3e <- left_join(patient_scores, ye_hif_dataset, by="patient")

ggplot(patient_ye_and_3e, aes(x=ye_hyp_sig_up, y= total_z_score))+
  geom_point()+
  xlab("Ye Hypoxia z score")+
  ylab("eIF3e (normoxia) z score")+
  theme_cowplot()+ 
  theme(text = element_text(size = 14))


ggsave("3e_hyp_and_ye.pdf", path=here("plots"),width = 4, height=3)




cor.test(patient_ye_and_3e$ye_hyp_sig_up, patient_ye_and_3e$total_z_score)
```


# now try for hypoxia 3e

```{r}

patient_ye_and_3e_hyp <- left_join(hyp_3e_patient_scores, ye_hif_dataset, by="patient")

ggplot(patient_ye_and_3e_hyp, aes(x=ye_hyp_sig_up, y= total_z_score))+
  geom_point()+
  xlab("Ye Hypoxia z score")+
  ylab("eIF3e (hypoxia) z score")+
  theme_cowplot() +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0)+
  theme(text = element_text(size = 14))


ggsave("3e_hyp_and_ye.pdf", path=here("plots"),width = 5, height=4)

# remove z scores less than 0.2, 0.2

patient_ye_and_3e_hyp <- patient_ye_and_3e_hyp %>% 
  mutate(change = ifelse(abs(total_z_score) < 0.1 & abs(ye_hyp_sig_up) < 0.1, "small_change", "big_change"))




cor.test(patient_ye_and_3e_hyp$ye_hyp_sig_up, patient_ye_and_3e_hyp$total_z_score)
```

Make four groups based on quandrants
```{r}

patient_ye_and_3e_hyp <- patient_ye_and_3e_hyp %>% 
  mutate(quadrant = ifelse(ye_hyp_sig_up < 0 & total_z_score < 0, "low hypoxia and low eIF3e (hypoxia)",
                           ifelse(ye_hyp_sig_up > 0 & total_z_score < 0, "high hypoxia and low eIF3e (hypoxia)",
                                  ifelse(ye_hyp_sig_up > 0 & total_z_score > 0, "high hypoxia and high eIF3e (hypoxia)",
                                         ifelse(ye_hyp_sig_up < 0 & total_z_score > 0, "low hypoxia sig and high eIF3e (hypoxia)", "other")))))



```

Plot survival curve
```{r}
#bind with clin data
patient_ye_and_3e_hyp_clin <- left_join(clin, patient_ye_and_3e_hyp, by=c("PATIENT_ID"="patient"))


#plot

fit_all_hypoxia <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = patient_ye_and_3e_hyp_clin)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, patient_ye_and_3e_hyp_clin)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

patient_ye_and_3e_hyp_clin$quadrant <- as.factor(patient_ye_and_3e_hyp_clin$quadrant)

quadrant_levels <- levels(patient_ye_and_3e_hyp_clin$quadrant)


surv_plot <- ggsurvplot(fit_all_hypoxia, data = patient_ye_and_3e_hyp_clin, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),legend.labs = quadrant_levels,  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = FALSE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())



surv_plot$plot <- surv_plot$plot + 
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 5, 
    color = "black"
  )+ theme(text = element_text(size = 14))+
  ggtitle("Ye et al hypoxia signature and eIF3e hypoxia signature vs survival")

surv_plot

ggsave("surv_plot_3e_hyp_ye.pdf", path=here("plots"),width = 11.5, height=8)







```


Control: excluding points in the middle as a negative control for the survival curve analysis
```{r}

big_changes_hyp <- patient_ye_and_3e_hyp %>% 
  filter(change=="big_change")

ggplot(big_changes_hyp, aes(x=ye_hyp_sig_up, y= total_z_score))+
  geom_point()+
  xlab("Ye Hypoxia z score")+
  ylab("eIF3e (hypoxia) z score")+
  theme_cowplot() +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0)+
  theme(text = element_text(size = 14))

# ggsave("3e_hyp_and_ye_big_changes.pdf", path=here("plots"),width = 5, height=4)

cor.test(big_changes_hyp$ye_hyp_sig_up, big_changes_hyp$total_z_score)


# make four groups

big_changes_hyp <- big_changes_hyp %>% 
  mutate(quadrant = ifelse(ye_hyp_sig_up < 0 & total_z_score < 0, "depleted hypoxia sig and depleted eIF3e (hypoxia) sig",
                           ifelse(ye_hyp_sig_up > 0 & total_z_score < 0, "enriched hypoxia sig and depleted eIF3e (hypoxia) sig",
                                  ifelse(ye_hyp_sig_up > 0 & total_z_score > 0, "enriched hypoxia sig and enriched eIF3e (hypoxia) sig",
                                         ifelse(ye_hyp_sig_up < 0 & total_z_score > 0, "depleted hypoxia sig and enriched eIF3e (hypoxia) sig", "other")))))


# plot surival curve with those groups

#bind with clin data
patient_ye_and_3e_hyp_clin_big_changes <- left_join(clin, big_changes_hyp, by=c("PATIENT_ID"="patient"))


#plot

fit_all_hypoxia <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = patient_ye_and_3e_hyp_clin_big_changes)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, patient_ye_and_3e_hyp_clin_big_changes)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

patient_ye_and_3e_hyp_clin$quadrant <- as.factor(patient_ye_and_3e_hyp_clin_big_changes$quadrant)

quadrant_levels <- levels(patient_ye_and_3e_hyp_clin_big_changes$quadrant)


surv_plot <- ggsurvplot(fit_all_hypoxia, data = patient_ye_and_3e_hyp_clin, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),legend.labs = quadrant_levels,  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = FALSE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())



surv_plot$plot <- surv_plot$plot + 
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 5, 
    color = "black"
  )+ theme(text = element_text(size = 14))

surv_plot

ggsave("surv_plot_3e_hyp_ye_big changes.pdf", path=here("plots"),width = 11.5, height=8)










```


Control: Limit to changes over 0.1 as a control
```{r}
big_changes_hyp <- patient_ye_and_a209_hyp %>% 
  filter(change=="big_change")

ggplot(big_changes_hyp, aes(x=ye_hyp_sig_up, y= total_z_score))+
  geom_point()+
  xlab("Ye Hypoxia z score")+
  ylab("a209 (hypoxia) z score")+
  theme_cowplot() +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0)+
  theme(text = element_text(size = 14))

ggsave("a209_hyp_and_ye_big_changes.pdf", path=here("plots"),width = 5, height=4)


cor.test(big_changes_hyp$ye_hyp_sig_up, big_changes_hyp$total_z_score)


# make four groups
big_changes_hyp <- big_changes_hyp %>% 
  mutate(quadrant = ifelse(ye_hyp_sig_up < 0 & total_z_score < 0, "depleted hypoxia sig and depleted a209 (hypoxia) sig",
                           ifelse(ye_hyp_sig_up > 0 & total_z_score < 0, "enriched hypoxia sig and depleted a209 (hypoxia) sig",
                                  ifelse(ye_hyp_sig_up > 0 & total_z_score > 0, "enriched hypoxia sig and enriched a209 (hypoxia) sig",
                                         ifelse(ye_hyp_sig_up < 0 & total_z_score > 0, "depleted hypoxia sig and enriched a209 (hypoxia) sig", "other")))))




# plot surival curve with those groups
#bind with clin data
patient_ye_and_a209_hyp_clin <- left_join(clin, big_changes_hyp, by=c("PATIENT_ID"="patient"))


#plot

fit_all_hypoxia <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = patient_ye_and_a209_hyp_clin)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, patient_ye_and_a209_hyp_clin)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

patient_ye_and_a209_hyp_clin$quadrant <- as.factor(patient_ye_and_a209_hyp_clin$quadrant)

quadrant_levels <- levels(patient_ye_and_a209_hyp_clin$quadrant)


surv_plot <- ggsurvplot(fit_all_hypoxia, data = patient_ye_and_a209_hyp_clin, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),legend.labs = quadrant_levels,  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = FALSE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())



surv_plot$plot <- surv_plot$plot + 
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 5, 
    color = "black"
  )+theme(text = element_text(size = 14))

surv_plot

ggsave("big_changes_surv_plot_a209_hyp_ye.pdf", path=here("plots"),width = 11.5, height=8)









```



# eIF3e and eIF3d expression in patients

# eIF3e
```{r}
#expression matrix

eif3e_exp <- merged_zscores_matrix %>%
  as.data.frame() %>% 
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol =="EIF3E") %>% 
  dplyr::select(-Hugo_Symbol) %>% 
  melt(value.name = "eIF3e_level") %>% 
  rename(patient="variable")

 

ggplot(eif3e_exp, aes(x=eIF3e_level))+
  geom_histogram(bins = 400)


### Stratify based on z-score

quantile_groups <- quantile(eif3e_exp$eIF3e_level, probs=seq(0,1,.25), na.rm=TRUE)

eif3e_exp$quantile <- cut(eif3e_exp$eIF3e_level, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

eif3e_exp$patient <- gsub("\\.", "-", as.character(eif3e_exp$patient))


#bind clin data with z score

eif3e_all_data <- left_join(clin, eif3e_exp, by=c("PATIENT_ID"="patient"))


#plot

fit_all_eIF3e <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = eif3e_all_data)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, eif3e_all_data)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

surv_plot_eIF3e_only <- ggsurvplot(fit_all_eIF3e, data = eif3e_all_data, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = TRUE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())



surv_plot_eIF3e_only$plot <- surv_plot_eIF3e_only$plot + 
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 5, 
    color = "black"
  )+
  theme(text = element_text(size = 14))+
  ggtitle("Patient quartiles by eIF3e expression")

surv_plot_eIF3e_only

# ggsave("3d_norm_surv_plot.pdf", path=here("plots"),width = 11.5, height=8)


ggplot(eif3e_all_data, aes(x=quantile,y=eIF3e_level, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3e")

# ggsave("3e boxplot.pdf",  path=here("plots"), width=4.5, height=4)




```


# eIF3d
```{r}

eif3d_exp <- merged_zscores_matrix %>%
  as.data.frame() %>% 
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol =="EIF3D") %>% 
  dplyr::select(-Hugo_Symbol) %>% 
  melt(value.name = "eIF3d_level") %>% 
  rename(patient="variable")

 

ggplot(eif3d_exp, aes(x=eIF3d_level))+
  geom_histogram(bins = 400)


### Stratify based on z-score

quantile_groups <- quantile(eif3d_exp$eIF3d_level, probs=seq(0,1,.25), na.rm=TRUE)

eif3d_exp$quantile <- cut(eif3d_exp$eIF3d_level, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

eif3d_exp$patient <- gsub("\\.", "-", as.character(eif3d_exp$patient))


#bind clin data with z score

eif3d_all_data <- left_join(clin, eif3d_exp, by=c("PATIENT_ID"="patient"))


#plot

fit_all_eIF3d <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = eif3d_all_data)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, eif3d_all_data)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

surv_plot_eIF3d_only <- ggsurvplot(fit_all_eIF3d, data = eif3d_all_data, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = FALSE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())



surv_plot_eIF3d_only$plot <- surv_plot_eIF3d_only$plot + 
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 5, 
    color = "black"
  )+
  theme(text = element_text(size = 14))+
  ggtitle("Patient quartiles by eIF3d expression")

surv_plot_eIF3d_only

# ggsave("3d_norm_surv_plot.pdf", path=here("plots"),width = 11.5, height=8)


ggplot(eif3d_all_data, aes(x=quantile,y=eIF3d_level, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3d")

# ggsave("3d boxplot.pdf",  path=here("plots"), width=4.5, height=4)




```

## Looking at copy number alterations in data

```{r}
cna <- read_excel(here("cbioportal_data/cna_metabric_matrix.xlsx"))

eIF3e_cna <- cna %>% 
  filter(Hugo_Symbol=="EIF3E") 

eIF3e_cna[ , 2:2174] <- lapply(eIF3e_cna[ , 2:2174], as.character)


eIF3e_cna<- eIF3e_cna%>%
  pivot_longer(cols= 2:2174, names_to = "patient", values_to = "cna")


#take just MB

eIF3e_cna_cancer <- eIF3e_cna %>% 
  filter(str_detect(patient, "MB-"))

eIF3e_cna_cancer$cna <- as.numeric(eIF3e_cna_cancer$cna)


ggplot(eIF3e_cna_cancer, aes(x=cna))+
  geom_histogram()+
  theme_cowplot()


```

3e or 3d

```{r}
eIF3_cna <- cna %>% 
  filter(Hugo_Symbol=="EIF3E" | Hugo_Symbol=="EIF3D") 

eIF3_cna[ , 2:2174] <- lapply(eIF3_cna[ , 2:2174], as.character)


eIF3_cna<- eIF3_cna%>%
  pivot_longer(cols= 2:2174, names_to = "patient", values_to = "cna")


#take just MB

eIF3_cna_cancer <- eIF3_cna %>% 
  filter(str_detect(patient, "MB-"))

eIF3_cna_cancer$cna <- as.numeric(eIF3_cna_cancer$cna)


ggplot(eIF3_cna_cancer, aes(x=cna))+
  geom_histogram()+
  theme_cowplot()+
  facet_wrap(~Hugo_Symbol)

library(scales)

ggplot(eIF3_cna_cancer, aes(x = cna)) +
  geom_bar()+
  theme_cowplot() +
  facet_wrap(~Hugo_Symbol)+
  scale_y_continuous(breaks=seq(0, 2000, by=200),
                       labels=seq(0, 2000, by =200))


```



# Controls


## Sanity check: positive-ish controls

VITAL_STATUS = living or died of disease


```{r}

# clin2 <- all_data[c(which(all_data$low_signature), which(all_data$high_signature)),]

alive_data <- all_data %>% 
  mutate(alive = ifelse(VITAL_STATUS=="Living", TRUE, FALSE))

alive_data <- alive_data %>% 
  mutate(died_from_cancer = ifelse(VITAL_STATUS=="Died of Disease", TRUE, FALSE))

fit_pos <- survfit(Surv(OS_YEARS, OS_STATUS) ~ died_from_cancer, data = alive_data)


ggsurvplot(fit_pos, data = alive_data, size = 1, palette = c("#E7B800", "#2E9FDF"),legend.labs = c("alive", "died_from_cancer"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = TRUE, risk.table = TRUE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())



```
Good sanity check! just need to clean up some of the alive vs died data??? maybe?

## Negative Control

Now neg control -- random genes comprising signatures-- same number of genes just random

pos genes = 1747 genes
neg genes = 1638 genes

pick random from all genes expressed in clinical data?? or in my data?


for loop generating z score for each patient



1. Create expression data
```{r}

exp_full <- merged_zscores_matrix %>%
  as.data.frame()

```

2. Pick random up and down values
```{r}

all_genes <- rownames(exp_full)

fake_pos_list <-all_genes[sample(1:length(all_genes), 1747)]
fake_neg_list <-all_genes[sample(1:length(all_genes), 1638)]
```

# fake up signature values - neg ctrl


Loop through each patient
```{r}

patients <- colnames(merged_zscores_all)
patients <- patients[patients !="Hugo_Symbol"]


fake_up_dataset <- data.frame(patient=character(), avg_z_score=numeric(), stringsAsFactors = FALSE)

for (i in patients) {

  #filter dataframe
  patient_level <- exp %>% 
    dplyr::select(i) %>%
    rownames_to_column("Hugo_Symbol") %>% 
    filter(Hugo_Symbol %in% fake_pos_list)

    all_values <- as.vector(patient_level[,2])
  
  
  #geometric mean
  
  up_genes_z_score <- mean(all_values)

  loop_dataset <- data.frame(patient=i, norm_3d_z_score_up = up_genes_z_score)
  
  fake_up_dataset <- rbind(fake_up_dataset, loop_dataset)
}

```





#fake down sig values - neg ctrl

```{r}

patients <- colnames(merged_zscores_all)
patients <- patients[patients !="Hugo_Symbol"]


fake_down_dataset <- data.frame(patient=character(), avg_z_score=numeric(), stringsAsFactors = FALSE)

for (i in patients) {

  #filter dataframe
  patient_level <- exp %>% 
    dplyr::select(i) %>%
    rownames_to_column("Hugo_Symbol") %>% 
    filter(Hugo_Symbol %in% fake_neg_list)

    all_values <- as.vector(patient_level[,2])
  
  
  #geometric mean
  
  down_genes_z_score <- mean(all_values)

  loop_dataset <- data.frame(patient=i, norm_3e_z_score_down = down_genes_z_score)
  
  fake_down_dataset <- rbind(fake_down_dataset, loop_dataset)
}



```


Merge two datasets
```{r}
neg_ctrl_fake_patient_scores <- left_join(fake_up_dataset, fake_down_dataset, by="patient")

neg_ctrl_fake_patient_scores <- neg_ctrl_fake_patient_scores %>% 
  mutate(total_z_score = norm_3e_z_score_up - norm_3e_z_score_down)

```

NEG CONTROL - visualize scores among patient cohort
```{r}
ggplot(neg_ctrl_fake_patient_scores, aes(x=total_z_score))+
  geom_histogram(bins = 400)
```


Stratify into groups based on z score

```{r}

neg_ctrl_quantile_groups <- quantile(neg_ctrl_fake_patient_scores$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

neg_ctrl_fake_patient_scores$quantile <- cut(neg_ctrl_fake_patient_scores$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

neg_ctrl_fake_patient_scores$patient <- gsub("\\.", "-", as.character(neg_ctrl_fake_patient_scores$patient))


colnames(neg_ctrl_fake_patient_scores) <- paste(colnames(neg_ctrl_fake_patient_scores),"neg_ctrl",sep="_") 

names(neg_ctrl_fake_patient_scores)[names(neg_ctrl_fake_patient_scores) == "patient_neg_ctrl"] <- "patient"




```


Now plot neg control!


```{r}
all_data <- left_join(clin, neg_ctrl_fake_patient_scores, by=c("PATIENT_ID"="patient"))


all_data <- all_data %>% 
  mutate(neg_ctrl_high_signature = ifelse(quantile_neg_ctrl==4, TRUE, FALSE))

all_data <- all_data %>% 
  mutate(neg_ctrl_low_signature = ifelse(quantile_neg_ctrl==1, TRUE, FALSE))
```


```{r}
clin2 <- all_data[c(which(all_data$neg_ctrl_low_signature), which(all_data$neg_ctrl_high_signature)),]
fit <- survfit(Surv(OS_YEARS, OS_STATUS) ~ neg_ctrl_low_signature, data = clin2)


ggsurvplot(fit, data = clin2, size = 1, palette = c("#E7B800", "#2E9FDF"),legend.labs = c("NEG CTRL eIF3e KD signature high", "NEG CTRL eIF3e KD signature low"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = TRUE, risk.table = TRUE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())

```



# Repeat analysis with TCGA dataset

1,108 samples

```{r cache = TRUE}
zscores_tcga <- read.table(here("cbioportal_data/brca_tcga/tcga_data_mrna_seq_v2_rsem_zscores_ref_diploid_samples.txt"), sep="\t", header=T)
row.names(zscores) <- gsub("_", ".",row.names(zscores))

# there are duplicates in this list. What happens when i subset for the genes we are interested in?

zscores_tcga_all_symbol <- zscores_tcga %>% 
  dplyr::select(-Entrez_Gene_Id)

### Average scores for duplicates

merged_tcga_zscores_all <- zscores_tcga_all_symbol %>%
  group_by(Hugo_Symbol) %>%
  summarise(across(everything(), mean))


merged_tcga_zscores_matrix <- merged_tcga_zscores_all %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  as.matrix()


```

Define genes of interest
```{r}
#genes <- sig_3e_hyp$symbol
```



for loop generating z score for each patient

#3e hypoxia

1. Create expression data
```{r}

exp <- merged_tcga_zscores_matrix %>%
  as.data.frame() %>% 
  rownames_to_column("Hugo_Symbol") %>% 
    filter(Hugo_Symbol %in% sig_3e_norm$symbol) %>% 
 column_to_rownames("Hugo_Symbol")


```




#up values


Loop through each patient
```{r}

patients <- colnames(merged_tcga_zscores_all)
patients <- patients[patients !="Hugo_Symbol"]


tcga_up_3e_norm_dataset <- data.frame(patient=character(), avg_z_score=numeric(), stringsAsFactors = FALSE)

for (i in patients) {

  #filter dataframe
  patient_level <- exp %>% 
    dplyr::select(i) %>%
    rownames_to_column("Hugo_Symbol") %>% 
    filter(Hugo_Symbol %in% up_3e_norm$symbol)

    all_values <- as.vector(patient_level[,2])
  
  
  #geometric mean
  
  up_genes_z_score <- mean(all_values, na.rm=TRUE)

  loop_dataset <- data.frame(patient=i, norm_3e_z_score_up = up_genes_z_score)
  
  tcga_up_3e_norm_dataset <- rbind(tcga_up_3e_norm_dataset, loop_dataset)
}

```





#down values

```{r}


tcga_down_3e_norm_dataset <- data.frame(patient=character(), avg_z_score=numeric(), stringsAsFactors = FALSE)

for (i in patients) {

  #filter dataframe
  patient_level <- exp %>% 
    dplyr::select(i) %>%
    rownames_to_column("Hugo_Symbol") %>% 
    filter(Hugo_Symbol %in% down_3e_norm$symbol)

    all_values <- as.vector(patient_level[,2])
  
  
  #geometric mean
  
  down_genes_z_score <- mean(all_values, na.rm=TRUE)

  loop_dataset <- data.frame(patient=i, norm_3e_z_score_down = down_genes_z_score)
  
  tcga_down_3e_norm_dataset <- rbind(tcga_down_3e_norm_dataset, loop_dataset)
}



```


Merge two datasets
```{r}
patient_scores_tcga <- left_join(tcga_up_3e_norm_dataset, tcga_down_3e_norm_dataset, by="patient")

patient_scores_tcga <- patient_scores_tcga %>% 
  mutate(total_z_score = norm_3e_z_score_up - norm_3e_z_score_down)

```

visualize scores among patient cohort
```{r}
ggplot(patient_scores_tcga, aes(x=total_z_score))+
  geom_histogram(bins = 400)
```


Stratify into groups based on z score

```{r}

quantile_groups <- quantile(patient_scores_tcga$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

patient_scores_tcga$quantile <- cut(patient_scores_tcga$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

patient_scores_tcga$patient <- gsub("\\.", "-", as.character(patient_scores_tcga$patient))



```

## Switch to using survival data
Read in and clean up clin data

```{r}
clin_tcga <- read_excel(here("cbioportal_data/new_tcga_clinical.xlsx"))
colnames(clin_tcga)


#modify clin


clin_small <- clin_tcga %>% 
  dplyr::select(OS_STATUS, OS_MONTHS, PATIENT_ID)

# years and months in this are not reading correctly

#get rid of not available

clin_small <- clin_small %>%
  filter(OS_MONTHS != "[Not Available]") %>% 
  filter(OS_MONTHS > 0)

clin_small$OS_MONTHS <- as.numeric(clin_small$OS_MONTHS)

clin_small <- clin_small %>% 
  arrange(OS_MONTHS)


clin_small$OS_STATUS[which(clin_small$OS_STATUS == "1:DECEASED" & clin_small$OS_MONTHS > 120)] <- "0:LIVING"
 
 clin_small$OS_STATUS <- clin_small$OS_STATUS == "1:DECEASED"


clin$OS_MONTHS <- as.numeric(clin$OS_MONTHS)

clin_small <- clin_small %>% 
  mutate(OS_YEARS= clin_small$OS_MONTHS/12)


clin_small$OS_YEARS[which(clin_small$OS_YEARS >= 10)] <- 10



```


Bind clin data with patient z score data
```{r}

patient_scores_tcga$patient <- sub("-01$", "", patient_scores_tcga$patient)
all_data_tcga <- left_join(clin_small, patient_scores_tcga, by=c("PATIENT_ID"="patient"))

```

## eif3e signature vs survival
Visualize survival data by 3e normoxia signature expression

```{r}
fit_all <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_tcga)

coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_tcga)
summary(coxfit)

str(survfit(coxfit))

#report hazards ratio on plot

summary_coxfit <- summary(coxfit)

hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

```

```{r}

surv_plot <- ggsurvplot(fit_all, data = all_data_tcga, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = TRUE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())



surv_plot$plot <- surv_plot$plot + 
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 5, 
    color = "black"
  )+
  theme(text = element_text(size = 14))+
  ggtitle("Patient quatiles by eIF3e normoxia signature")

surv_plot

ggsave("3e_normoxia_surv_plot.pdf", path=here("plots"),width = 11.5, height=8)





ggplot(all_data, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3e normoxia signature")

ggsave("3e norm boxplot.pdf",  path=here("plots"), width=4.5, height=4)



```



Repeat in hypoxia 3e


```{r}
# For loop generating z score for each patient

# expression matrix changes based on signature

exp <- merged_tcga_zscores_matrix %>%
  as.data.frame() %>% 
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% sig_3e_hyp$symbol) %>% 
  column_to_rownames("Hugo_Symbol")

# up loop

up_3e_hyp_dataset <- data.frame(patient=character(), avg_z_score=numeric(), stringsAsFactors = FALSE)

for (i in patients) {
  
  #filter dataframe
  patient_level <- exp %>% 
    dplyr::select(i) %>%
    rownames_to_column("Hugo_Symbol") %>% 
    filter(Hugo_Symbol %in% up_3e_hyp$symbol)
  
  all_values <- as.vector(patient_level[,2])
  
  
  #geometric mean
  
  up_genes_z_score <- mean(all_values, na.rm=TRUE)
  
  loop_dataset <- data.frame(patient=i, hyp_3e_z_score_up = up_genes_z_score)
  
  up_3e_hyp_dataset <- rbind(up_3e_hyp_dataset, loop_dataset)
}

ggplot(up_3e_hyp_dataset, aes(x=hyp_3e_z_score_up))+
  geom_histogram(bins = 400)


# down loop

down_3e_hyp_dataset <- data.frame(patient=character(), avg_z_score=numeric(), stringsAsFactors = FALSE)

for (i in patients) {
  
  #filter dataframe
  patient_level <- exp %>% 
    dplyr::select(i) %>%
    rownames_to_column("Hugo_Symbol") %>% 
    filter(Hugo_Symbol %in% down_3e_hyp$symbol)
  
  all_values <- as.vector(patient_level[,2])
  
  
  #geometric mean
  
  down_genes_z_score <- mean(all_values, na.rm=TRUE)
  
  loop_dataset <- data.frame(patient=i, hyp_3e_z_score_down = down_genes_z_score)
  
  down_3e_hyp_dataset <- rbind(down_3e_hyp_dataset, loop_dataset)
}

ggplot(down_3e_hyp_dataset, aes(x=hyp_3e_z_score_down))+
  geom_histogram(bins = 400)


# merge datasets

hyp_3e_patient_scores <- left_join(up_3e_hyp_dataset, down_3e_hyp_dataset, by="patient")

hyp_3e_patient_scores <- hyp_3e_patient_scores %>% 
  mutate(total_z_score = hyp_3e_z_score_up - hyp_3e_z_score_down)

#visualize distribution

ggplot(hyp_3e_patient_scores, aes(x=total_z_score))+
  geom_histogram(bins = 400)



### Stratify based on z-score

quantile_groups <- quantile(hyp_3e_patient_scores$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

hyp_3e_patient_scores$quantile <- cut(hyp_3e_patient_scores$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

class(hyp_3e_patient_scores$patient)
class(clin$PATIENT_ID)

hyp_3e_patient_scores$patient %in% clin_small$PATIENT_ID



hyp_3e_patient_scores$patient <- hyp_3e_patient_scores$patient %>%
  sub("\\.01$", "", .) %>%
  gsub("\\.", "-", .)
#bind clin data with z score

hyp_3e_patient_scores$patient %in% clin_small$PATIENT_ID


all_data_3e_hyp <- left_join(clin_small, hyp_3e_patient_scores, by=c("PATIENT_ID"="patient"))


#plot

fit_all_3e_hyp <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_3e_hyp)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_3e_hyp)
summary_coxfit_3e_norm <- summary(coxfit)


hr <- round(summary_coxfit_3e_norm$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_3e_norm$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

surv_plot_3e_hyp <- ggsurvplot(fit_all_3e_hyp, data = all_data_3e_hyp, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),legend.labs = c("depleted for eIF3e signature (hypoxia): low (Q1)","depleted: medium-low (Q2)", "enriched: medium-high (Q3)", "enriched for eIF3e signature (hypoxia): high (Q4)"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = TRUE,  risk.table.col = NA, risk.table.height = 0.25, ggtheme = theme_bw())

surv_plot_3e_hyp <- ggsurvplot(fit_all_3e_hyp, data = all_data_3e_hyp, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = TRUE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())

surv_plot_3e_hyp

surv_plot_3e_hyp$plot <- surv_plot_3e_hyp$plot + 
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 5, 
    color = "black"
  )+
  theme(text = element_text(size = 14))+
  ggtitle("Patient quartiles by eIF3e hypoxia signature")

surv_plot_3e_hyp

ggsave("3e_hyp_surv_plot_tcga.pdf", path=here("plots"),width = 11.5, height=8)



#boxplot of quartile expression

ggplot(all_data_3e_hyp, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3e hypoxia signature")

```


## 3d hypoxia

```{r}
# For loop generating z score for each patient

# expression matrix changes based on signature

exp <- merged_tcga_zscores_matrix %>%
  as.data.frame() %>% 
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% sig_3d_hyp$symbol) %>% 
  column_to_rownames("Hugo_Symbol")

# up loop

up_3d_hyp_dataset <- data.frame(patient=character(), avg_z_score=numeric(), stringsAsFactors = FALSE)

for (i in patients) {
  
  #filter dataframe
  patient_level <- exp %>% 
    dplyr::select(i) %>%
    rownames_to_column("Hugo_Symbol") %>% 
    filter(Hugo_Symbol %in% up_3d_hyp$symbol)
  
  all_values <- as.vector(patient_level[,2])
  
  
  #geometric mean
  
  up_genes_z_score <- mean(all_values, na.rm=TRUE)
  
  loop_dataset <- data.frame(patient=i, hyp_3d_z_score_up = up_genes_z_score)
  
  up_3d_hyp_dataset <- rbind(up_3d_hyp_dataset, loop_dataset)
}

ggplot(up_3d_hyp_dataset, aes(x=hyp_3d_z_score_up))+
  geom_histogram(bins = 400)


# down loop

down_3d_hyp_dataset <- data.frame(patient=character(), avg_z_score=numeric(), stringsAsFactors = FALSE)

for (i in patients) {
  
  #filter dataframe
  patient_level <- exp %>% 
    dplyr::select(i) %>%
    rownames_to_column("Hugo_Symbol") %>% 
    filter(Hugo_Symbol %in% down_3d_hyp$symbol)
  
  all_values <- as.vector(patient_level[,2])
  
  
  #geometric mean
  
  down_genes_z_score <- mean(all_values, na.rm=TRUE)
  
  loop_dataset <- data.frame(patient=i, hyp_3d_z_score_down = down_genes_z_score)
  
  down_3d_hyp_dataset <- rbind(down_3d_hyp_dataset, loop_dataset)
}

ggplot(down_3d_hyp_dataset, aes(x=hyp_3d_z_score_down))+
  geom_histogram(bins = 400)


# merge datasets

hyp_3d_patient_scores <- left_join(up_3d_hyp_dataset, down_3d_hyp_dataset, by="patient")

hyp_3d_patient_scores <- hyp_3d_patient_scores %>% 
  mutate(total_z_score = hyp_3d_z_score_up - hyp_3d_z_score_down)

#visualize distribution

ggplot(hyp_3d_patient_scores, aes(x=total_z_score))+
  geom_histogram(bins = 400)



### Stratify based on z-score

quantile_groups <- quantile(hyp_3d_patient_scores$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

hyp_3d_patient_scores$quantile <- cut(hyp_3d_patient_scores$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

class(hyp_3d_patient_scores$patient)
class(clin$PATIENT_ID)

hyp_3d_patient_scores$patient %in% clin_small$PATIENT_ID



hyp_3d_patient_scores$patient <- hyp_3d_patient_scores$patient %>%
  sub("\\.01$", "", .) %>%
  gsub("\\.", "-", .)
#bind clin data with z score

hyp_3d_patient_scores$patient %in% clin_small$PATIENT_ID


all_data_3d_hyp <- left_join(clin_small, hyp_3d_patient_scores, by=c("PATIENT_ID"="patient"))


#plot

fit_all_3d_hyp <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_3d_hyp)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_3d_hyp)
summary_coxfit_3d_norm <- summary(coxfit)


hr <- round(summary_coxfit_3d_norm$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_3d_norm$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

surv_plot_3d_hyp <- ggsurvplot(fit_all_3d_hyp, data = all_data_3d_hyp, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),legend.labs = c("depleted for eIF3d signature (hypoxia): low (Q1)","depleted: medium-low (Q2)", "enriched: medium-high (Q3)", "enriched for eIF3d signature (hypoxia): high (Q4)"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = FALSE,  risk.table.col = NA, risk.table.height = 0.25, ggtheme = theme_bw())

surv_plot_3d_hyp <- ggsurvplot(fit_all_3d_hyp, data = all_data_3d_hyp, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = TRUE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())



surv_plot_3d_hyp$plot <- surv_plot_3d_hyp$plot + 
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 5, 
    color = "black"
  )+
  theme(text = element_text(size = 14))+
  ggtitle("Patient quartiles by eIF3d hypoxia signature")

surv_plot_3d_hyp

ggsave("3d_hyp_surv_plot_tcga.pdf", path=here("plots"),width = 4.5, height=5)



#boxplot of quartile expression

ggplot(all_data_3d_hyp, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3d hypoxia signature")

```


## 3d normoxia

```{r}
# For loop generating z score for each patient

# expression matrix changes based on signature

exp <- merged_tcga_zscores_matrix %>%
  as.data.frame() %>% 
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% sig_3d_norm$symbol) %>% 
  column_to_rownames("Hugo_Symbol")

# up loop

up_3d_norm_dataset <- data.frame(patient=character(), avg_z_score=numeric(), stringsAsFactors = FALSE)

for (i in patients) {
  
  #filter dataframe
  patient_level <- exp %>% 
    dplyr::select(i) %>%
    rownames_to_column("Hugo_Symbol") %>% 
    filter(Hugo_Symbol %in% up_3d_norm$symbol)
  
  all_values <- as.vector(patient_level[,2])
  
  
  #geometric mean
  
  up_genes_z_score <- mean(all_values, na.rm=TRUE)
  
  loop_dataset <- data.frame(patient=i, norm_3d_z_score_up = up_genes_z_score)
  
  up_3d_norm_dataset <- rbind(up_3d_norm_dataset, loop_dataset)
}

ggplot(up_3d_norm_dataset, aes(x=norm_3d_z_score_up))+
  geom_histogram(bins = 400)


# down loop

down_3d_norm_dataset <- data.frame(patient=character(), avg_z_score=numeric(), stringsAsFactors = FALSE)

for (i in patients) {
  
  #filter dataframe
  patient_level <- exp %>% 
    dplyr::select(i) %>%
    rownames_to_column("Hugo_Symbol") %>% 
    filter(Hugo_Symbol %in% down_3d_norm$symbol)
  
  all_values <- as.vector(patient_level[,2])
  
  
  #geometric mean
  
  down_genes_z_score <- mean(all_values, na.rm=TRUE)
  
  loop_dataset <- data.frame(patient=i, norm_3d_z_score_down = down_genes_z_score)
  
  down_3d_norm_dataset <- rbind(down_3d_norm_dataset, loop_dataset)
}

ggplot(down_3d_norm_dataset, aes(x=norm_3d_z_score_down))+
  geom_histogram(bins = 400)


# merge datasets

norm_3d_patient_scores <- left_join(up_3d_norm_dataset, down_3d_norm_dataset, by="patient")

norm_3d_patient_scores <- norm_3d_patient_scores %>% 
  mutate(total_z_score = norm_3d_z_score_up - norm_3d_z_score_down)

#visualize distribution

ggplot(norm_3d_patient_scores, aes(x=total_z_score))+
  geom_histogram(bins = 400)



### Stratify based on z-score

quantile_groups <- quantile(norm_3d_patient_scores$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

norm_3d_patient_scores$quantile <- cut(norm_3d_patient_scores$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

class(norm_3d_patient_scores$patient)
class(clin$PATIENT_ID)

norm_3d_patient_scores$patient %in% clin_small$PATIENT_ID



norm_3d_patient_scores$patient <- norm_3d_patient_scores$patient %>%
  sub("\\.01$", "", .) %>%
  gsub("\\.", "-", .)
#bind clin data with z score

norm_3d_patient_scores$patient %in% clin_small$PATIENT_ID


all_data_3d_norm <- left_join(clin_small, norm_3d_patient_scores, by=c("PATIENT_ID"="patient"))


#plot

fit_all_3d_norm <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_3d_norm)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_3d_norm)
summary_coxfit_3d_norm <- summary(coxfit)


hr <- round(summary_coxfit_3d_norm$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_3d_norm$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

surv_plot_3d_norm <- ggsurvplot(fit_all_3d_norm, data = all_data_3d_norm, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),legend.labs = c("depleted for eIF3d signature (normoxia): low (Q1)","depleted: medium-low (Q2)", "enriched: medium-high (Q3)", "enriched for eIF3d signature (normoxia): high (Q4)"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = FALSE,  risk.table.col = NA, risk.table.height = 0.25, ggtheme = theme_bw())

surv_plot_3d_norm <- ggsurvplot(fit_all_3d_norm, data = all_data_3d_norm, size = 1, palette = c("#E7B800", "pink", "purple", "#2E9FDF"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = FALSE, risk.table = FALSE,  risk.table.col = NA, risk.table.height = 0.25, ggtheme = theme_bw())



surv_plot_3d_norm$plot <- surv_plot_3d_norm$plot + 
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 5, 
    color = "black"
  )+
  theme(text = element_text(size = 14))+
  ggtitle("Patient quartiles by eIF3d normoxia signature")

surv_plot_3d_norm

ggsave("3d_norm_surv_plot_tcga.pdf", path=here("plots"),width = 11.5, height=8)



#boxplot of quartile expression

ggplot(all_data_3d_norm, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3d normoxia signature")

```








## Sanity check: positive-ish controls

VITAL_STATUS = living or died of disease


```{r}

clin2 <- all_data[c(which(all_data$low_signature), which(all_data$high_signature)),]

all_data <- all_data %>% 
  mutate(alive = ifelse(VITAL_STATUS=="Living", TRUE, FALSE))

all_data <- all_data %>% 
  mutate(died_from_cancer = ifelse(VITAL_STATUS=="Died of Disease", TRUE, FALSE))

fit_pos <- survfit(Surv(OS_YEARS, OS_STATUS) ~ died_from_cancer, data = clin2)


ggsurvplot(fit_pos, data = clin2, size = 1, palette = c("#E7B800", "#2E9FDF"),legend.labs = c("alive", "died_from_cancer"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = TRUE, risk.table = TRUE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())



```
Good sanity check! just need to clean up some of the alive vs died data??? maybe?

## Negative Control

Now neg control -- random genes comprising signatures-- same number of genes just random

pos genes = 1747 genes
neg genes = 1638 genes

pick random from all genes expressed in clinical data?? or in my data?


for loop generating z score for each patient



1. Create expression data
```{r}

exp_full <- merged_zscores_matrix %>%
  as.data.frame()

```

2. Pick random up and down values
```{r}

all_genes <- rownames(exp_full)

fake_pos_list <-all_genes[sample(1:length(all_genes), 1747)]
fake_neg_list <-all_genes[sample(1:length(all_genes), 1638)]
```

# fake up signature values - neg ctrl


Loop through each patient
```{r}

patients <- colnames(merged_zscores_all)
patients <- patients[patients !="Hugo_Symbol"]


fake_up_dataset <- data.frame(patient=character(), avg_z_score=numeric(), stringsAsFactors = FALSE)

for (i in patients) {

  #filter dataframe
  patient_level <- exp %>% 
    dplyr::select(i) %>%
    rownames_to_column("Hugo_Symbol") %>% 
    filter(Hugo_Symbol %in% fake_pos_list)

    all_values <- as.vector(patient_level[,2])
  
  
  #geometric mean
  
  up_genes_z_score <- mean(all_values)

  loop_dataset <- data.frame(patient=i, norm_3e_z_score_up = up_genes_z_score)
  
  fake_up_dataset <- rbind(fake_up_dataset, loop_dataset)
}

```





#fake down sig values - neg ctrl

```{r}

patients <- colnames(merged_zscores_all)
patients <- patients[patients !="Hugo_Symbol"]


fake_down_dataset <- data.frame(patient=character(), avg_z_score=numeric(), stringsAsFactors = FALSE)

for (i in patients) {

  #filter dataframe
  patient_level <- exp %>% 
    dplyr::select(i) %>%
    rownames_to_column("Hugo_Symbol") %>% 
    filter(Hugo_Symbol %in% fake_neg_list)

    all_values <- as.vector(patient_level[,2])
  
  
  #geometric mean
  
  down_genes_z_score <- mean(all_values)

  loop_dataset <- data.frame(patient=i, norm_3e_z_score_down = down_genes_z_score)
  
  fake_down_dataset <- rbind(fake_down_dataset, loop_dataset)
}



```


Merge two datasets
```{r}
neg_ctrl_fake_patient_scores <- left_join(fake_up_dataset, fake_down_dataset, by="patient")

neg_ctrl_fake_patient_scores <- neg_ctrl_fake_patient_scores %>% 
  mutate(total_z_score = norm_3e_z_score_up - norm_3e_z_score_down)

```

NEG CONTROL - visualize scores among patient cohort
```{r}
ggplot(neg_ctrl_fake_patient_scores, aes(x=total_z_score))+
  geom_histogram(bins = 400)
```


Stratify into groups based on z score

```{r}

neg_ctrl_quantile_groups <- quantile(neg_ctrl_fake_patient_scores$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

neg_ctrl_fake_patient_scores$quantile <- cut(neg_ctrl_fake_patient_scores$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

neg_ctrl_fake_patient_scores$patient <- gsub("\\.", "-", as.character(neg_ctrl_fake_patient_scores$patient))


colnames(neg_ctrl_fake_patient_scores) <- paste(colnames(neg_ctrl_fake_patient_scores),"neg_ctrl",sep="_") 

names(neg_ctrl_fake_patient_scores)[names(neg_ctrl_fake_patient_scores) == "patient_neg_ctrl"] <- "patient"




```


Now plot neg control!


```{r}
all_data <- left_join(clin, neg_ctrl_fake_patient_scores, by=c("PATIENT_ID"="patient"))


all_data <- all_data %>% 
  mutate(neg_ctrl_high_signature = ifelse(quantile_neg_ctrl==4, TRUE, FALSE))

all_data <- all_data %>% 
  mutate(neg_ctrl_low_signature = ifelse(quantile_neg_ctrl==1, TRUE, FALSE))
```


```{r}
clin2 <- all_data[c(which(all_data$neg_ctrl_low_signature), which(all_data$neg_ctrl_high_signature)),]
fit <- survfit(Surv(OS_YEARS, OS_STATUS) ~ neg_ctrl_low_signature, data = clin2)


ggsurvplot(fit, data = clin2, size = 1, palette = c("#E7B800", "#2E9FDF"),legend.labs = c("NEG CTRL eIF3e KD signature high", "NEG CTRL eIF3e KD signature low"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = TRUE, risk.table = TRUE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())

```



