---
title: "231 riboseq"
author: "Kate"
date: "2025-08-07"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}

library(here)
library(janitor)
library(tximport)
library(tidyverse)
library(EnhancedVolcano)
library(msigdbr)
library(viridis)
library(gridExtra)
library(ggthemes)
library(ggrepel)
library(scales)
library(survival)
library(DESeq2)
library(limma)
library(clusterProfiler)
library(plotly)
library(ggokabeito)
library(ggvenn)
library(ggpubr)
library(DT)
library(edgeR)
library(cowplot)
library(ggExtra)
library(ggprism)
library(GeneOverlap)
library(writexl)
library(survminer)
library(readxl)
library(reshape2)
```

```{r}
a209_col = "#CC79A7"
si3e_col = "#56B4E9"
si3d_col = "#E69F00"
```

## Load transcript-level data

Importing metadata and all salmon counts
```{r meta data}
metadata <- read_csv(here("accessories","metadata_NM_231.csv"), skip = 0) %>% 
  clean_names()

metadata <- metadata %>% 
  mutate(library = ifelse(type=="Qiagen_miRNA", "ribo", "rna"))

#ribosome lysate = ribosome seq
#rRNA depletion = RNA seq
#rename variables remove outlier

#edit factor levels
metadata$library <- relevel(x = factor(metadata$library), ref = "rna")

metadata$treatment <- relevel(x = factor(metadata$treatment), ref = "siCTRL")
metadata$treatment <- recode(metadata$treatment, "siCTRL" = "sictrl")
metadata$condition <- relevel(x = factor(metadata$condition), ref = "normoxia")
metadata$rep <- recode(metadata$rep, RepA = "a", RepB = "b", RepC = "c")

metadata <- metadata %>%
dplyr::select(id, library, treatment, condition, rep) 

```

```{r import gene info}

#import key on RNA type, remove premRNA
geneInfo <- read_csv(here("accessories","gencode.v26.primary.info.csv.gz"), col_names = F) %>%
  filter(!str_detect(X2, "pre_"))

colnames(geneInfo) <- c("gene_id","transcript_id","biotype","symbol")

#import transcripts for gene to transcript mapping
tx2gene <- geneInfo[,c(2,1)]
gene_symbol <- geneInfo %>% dplyr::select(gene_id, symbol) %>% unique()

```

```{r txi import, include=FALSE}
#name files to import
myquantfiles <- paste(here(), "/salmon/",
                      metadata$id,
                      "/quant.sf.gz",
                      sep = "")

names(myquantfiles) <- paste(metadata$condition, metadata$treatment, metadata$rep, metadata$library, sep = "_")

#import salmon files
myTxi <- tximport(myquantfiles, type = "salmon", tx2gene = tx2gene)
```


No spike ins because i aligned only to the human

```{r}
# visualize distribution of counts
hist_count <- hist(log2(rowSums(myTxi$counts)), breaks = 50)
#determine threshold

#distribution of abundance histogram
#hist_abundance <- hist(log2(rowSums(myTxi$abundance)), breaks = 50)

#only keeping rows with counts  log2> 10 across all samples - want around 10-15k genes - want genes that are highly expressed
keepGenes <- rownames(myTxi$counts[log2(rowSums(myTxi$counts)) > 10 ,])

#removes noise and keeps only highly expressed genes
# results show bimodal distribution
#1 RNA per cell = 1 TPM
```




```{r standard deviation filtering}
# log transforming with pseudocount to avoid undefined numbers
qcinput <- log2(myTxi$abundance[keepGenes,] + 1) 

qcinput <- qcinput[rowSds(qcinput) > 1,] 

dim(qcinput)
```



```{r}

mycor <- cor(qcinput, method = "pearson")

heatmap <- pheatmap::pheatmap(
  mat = mycor,
  border_color = "black",  # Black borders around cells
   #color = colorRampPalette(c("blue", "white", "red"))(100),  # Custom color gradient
   scale = "none",  
  cluster_rows = TRUE,  
  cluster_cols = TRUE,  
  fontsize = 8 
)

heatmap

ggsave(here("plots/plots_231/qc.pdf"), heatmap, width=6.5, height = 6)
```


Split normoxia and hypoxia
```{r}

qc_input_normoxia <- qcinput[, grep("normoxia", colnames(qcinput))]

qc_input_hypoxia <- qcinput[, grep("hypoxia", colnames(qcinput))]

```


# Visualizations

# Figure 1: Normoxia

```{r}
#KDs
mycor_norm <- cor(qc_input_normoxia, method = "pearson")

colnames(mycor_norm) <- gsub("^normoxia_", "", colnames(mycor_norm))  # Remove "normoxia_" prefix
rownames(mycor_norm) <- gsub("^normoxia_", "", rownames(mycor_norm))  # Same for row names

# Replace underscores with spaces
colnames(mycor_norm) <- gsub("_", " ", colnames(mycor_norm))
rownames(mycor_norm) <- gsub("_", " ", rownames(mycor_norm))

# Create the heatmap with the cleaned-up names
heatmap_normoxia <- pheatmap::pheatmap(
  mat = mycor_norm,
  border_color = "black",  # Black borders around cells
   #color = colorRampPalette(c("blue", "white", "red"))(100),  # Custom color gradient
   scale = "none",  
  cluster_rows = TRUE,  
  cluster_cols = TRUE,  
  fontsize = 12 
)


heatmap_normoxia
ggsave(heatmap_normoxia, filename=here("plots/plots_231/heatmap_normoxia.pdf"), width=4.5, height=4)

# RIBO average correlation coefficient

qc_input_normoxia_ribo <- qc_input_normoxia[, grep("ribo", colnames(qc_input_normoxia))]

mycor_ribo_norm <- cor(qc_input_normoxia_ribo, method = "pearson")

# Extract the lower triangular part of the matrix without the diagonal
lower_tri <- mycor_ribo_norm[lower.tri(mycor_ribo_norm, diag=FALSE)]

# Calculate the average correlation coefficient (excluding diagonal)
average_correlation <- mean(lower_tri)

# Print the result
average_correlation


# rna average correlation coefficient

qc_input_normoxia_rna <- qc_input_normoxia[, grep("rna", colnames(qc_input_normoxia))]

mycor_rna_norm <- cor(qc_input_normoxia_rna, method = "pearson")


# Extract the lower triangular part of the matrix without the diagonal
lower_tri <- mycor_rna_norm[lower.tri(mycor_rna_norm, diag=FALSE)]

# Calculate the average correlation coefficient (excluding diagonal)
average_correlation <- mean(lower_tri)

# Print the result
average_correlation

```


# Figure 2: Hypoxia, knockdowns


Heatmap

```{r}
mycor_hyp_kds <- cor(qc_input_hypoxia, method = "pearson")

colnames(mycor_hyp_kds) <- gsub("^hypoxia_", "", colnames(mycor_hyp_kds))  # Remove "hypoxia_" prefix
rownames(mycor_hyp_kds) <- gsub("^hypoxia_", "", rownames(mycor_hyp_kds))  # Same for row names

# Replace underscores with spaces
colnames(mycor_hyp_kds) <- gsub("_", " ", colnames(mycor_hyp_kds))
rownames(mycor_hyp_kds) <- gsub("_", " ", rownames(mycor_hyp_kds))

# Create the heatmap with the cleaned-up names
heatmap_hypoxia <- pheatmap::pheatmap(
  mat = mycor_hyp_kds,
  border_color = "black",  # Black borders around cells
   #color = colorRampPalette(c("blue", "white", "red"))(100),  # Custom color gradient
   scale = "none",  
  cluster_rows = TRUE,  
  cluster_cols = TRUE,  
  fontsize = 12 
)





heatmap_hypoxia
ggsave(heatmap_hypoxia, filename=here("plots/plots_231/heatmap_hyp.pdf"), width=4.5, height=4)

```

```{r}

# RIBO average correlation coefficient

qc_input_hypoxia_ribo <- qc_input_hypoxia[, grep("ribo", colnames(qc_input_hypoxia))]

mycor_ribo_hyp <- cor(qc_input_hypoxia_ribo, method = "pearson")


# Extract the lower triangular part of the matrix without the diagonal
lower_tri <- mycor_ribo_hyp[lower.tri(mycor_ribo_hyp, diag=FALSE)]

# Calculate the average correlation coefficient (excluding diagonal)
average_correlation <- mean(lower_tri)

# Print the result
average_correlation



# rna average correlation coefficient

qc_input_hypoxia_rna <- qc_input_hypoxia[, grep("rna", colnames(qc_input_hypoxia))]

mycor_rna_hyp <- cor(qc_input_hypoxia_rna, method = "pearson")


# Extract the lower triangular part of the matrix without the diagonal
lower_tri <- mycor_rna_hyp[lower.tri(mycor_rna_hyp, diag=FALSE)]

# Calculate the average correlation coefficient (excluding diagonal)
average_correlation <- mean(lower_tri)

# Print the result
average_correlation


```

qc looks great.




# Calculate TE


### Filter
```{r keep only protein coding genes}

pc <- geneInfo %>%
  filter(biotype =="protein_coding") %>%
  pull(gene_id) %>%
  unique()

pc_keep <- pc[pc %in% keepGenes]

# remove non-polyadenylated mRNAs
nonpolyA <- readxl::read_xlsx(path = here("accessories","polyAminus.xlsx"), sheet = 1) %>% 
   pull(Gene)

nonpolyA_ID <- geneInfo %>%
  filter(symbol %in% nonpolyA) %>% 
  pull(gene_id) %>% 
  unique()

# we want to remove any genes that match these annotation categories # before we do DE analysis
blacklist_smRNA <- c("snoRNA","miRNA","snRNA","rRNA","Mt_tRNA","scaRNA","Mt_tRNA","Mt_rRNA")

blacklist_smRNA_ID <- geneInfo %>%
  filter(biotype %in% blacklist_smRNA) %>% 
  pull(gene_id) %>% 
  unique()

genes_to_kill <- unique(c(blacklist_smRNA_ID,nonpolyA_ID))
```


```{r filter and extract counts}

allCounts <- myTxi$counts %>%
  as.data.frame() %>% 
  rownames_to_column(var = "gene") %>% 
  filter(gene %in% pc_keep) %>%
  filter(!gene %in% genes_to_kill) %>%
  column_to_rownames(var = "gene") %>% 
  as.matrix() %>%
  round(.,0)

all_raw_filtered_counts <- allCounts %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id")

#export allCounts for DeSeq analysis
write_csv(all_raw_filtered_counts, here("counts/calculate_te/raw_filtered_allCounts_231.csv"))


allCountsNorm <- edgeR::cpm(allCounts[,order(colnames(allCounts))])

riboN <- log2(1+ as.matrix(allCountsNorm[,grep(pattern = "ribo",colnames(allCountsNorm))]))
rnaN <- log2(1 + as.matrix(allCountsNorm[,grep(pattern = "rna",colnames(allCountsNorm))])) 


TE_original <- riboN - rnaN

hist(TE_original)

median(as.data.frame(TE_original) %>% melt() %>% pull(value))

#center data --

TE <- (riboN - rnaN) + 0.5749442
# this is why im using 0.5749442! -- all changes are relative


hist(TE)

colnames(TE) <- gsub(pattern = "ribo", replacement = "TE", x = colnames(TE))


gghistogram(as.data.frame(TE) %>% melt(), x="value")
ribo_counts <- as.matrix(allCounts[,grep(pattern = "ribo",colnames(allCounts))])
rna_counts <- as.matrix(allCounts[,grep(pattern = "rna",colnames(allCounts))])

```


### calculation
```{r}

full_te_list <- TE %>% 
  as.data.frame() %>%
  rownames_to_column("gene_id") %>% 
  left_join(gene_symbol, by="gene_id") %>% 
  dplyr::select(symbol, everything())


full_rna_list <- rnaN %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id")


full_ribo_list <- riboN %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id")

full_te_list <- full_te_list %>% 
  left_join(full_rna_list, by="gene_id")

complete_list <- full_te_list %>% 
  left_join(full_ribo_list, by= "gene_id")


write_csv(complete_list, here("counts/counts_231/complete_te_rna_ribo_list_231.csv"))


```



```{r reformat data}
all_long <- complete_list %>% 
  melt() %>% 
    separate(variable, into=(c("condition", "treatment", "rep", "type")))

```

Sanity check:

Look at RNA for eIF3e and eIF3d KD compared to control


```{r eIF3e RNA}

rna_3e_formatted <- all_long %>%
  filter(symbol=="EIF3E") %>%
  filter(condition=="normoxia") %>%
  filter(type=="rna")


ggplot(rna_3e_formatted, aes(x=treatment, y=value))+
  geom_point()+
  theme_cowplot()

```


```{r eIF3d RNA}

rna_3d_formatted <- all_long %>%
  filter(symbol=="EIF3D") %>%
  filter(condition=="normoxia") %>%
  filter(type=="rna")


ggplot(rna_3d_formatted, aes(x=treatment, y=value))+
  geom_point()+
  theme_cowplot()

```


Example of TE: Analyze ATF4 rna ribo and TE in control compared to KD

Example- ATF4
```{r}
atf4_data <- all_long %>% 
  filter(symbol=="ATF4") %>% 
  filter(condition=="normoxia")


ggplot(atf4_data, aes(x=treatment, y=value))+
  geom_point()+
  facet_wrap(~type)
```

```{r}
eif3e_data <- all_long %>% 
  filter(symbol=="EIF3E") %>% 
  filter(condition=="hypoxia")


ggplot(eif3e_data, aes(x=treatment, y=value))+
  geom_point()+
  facet_wrap(~type)
```

# ###############################################################################
# just doing TE here

## HIF1alpha levels after compound treatment

```{r}

all_gene_te <- complete_list

all_gene_te_long <- all_gene_te %>% 
  melt() %>% 
tidyr::extract(variable, into = c("condition", "treatment", "rep", "type"), regex = "([^_]+)_([^_]+)_([^_]+)_([^_]+)") %>% 
  filter(condition=="hypoxia")

all_gene_te_long$treatment <- relevel(factor(all_gene_te_long$treatment), ref = "sictrl")


hif <-all_gene_te_long %>% 
  filter(symbol=="HIF1A") %>% 
  filter(type != "TE")

hif_summary <- hif %>%
  group_by(symbol, type, treatment) %>%
  summarise(
    mean_value = mean(value),
    sd = sd(value)  # Standard deviation
  )

hif_summary$type <- factor(hif_summary$type, levels=c("rna","ribo"))

ggplot(hif_summary, aes(x = treatment, y = mean_value, fill = treatment)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  geom_errorbar(aes(ymin = mean_value - sd, ymax = mean_value + sd), 
                width = 0.2, position = position_dodge(0.9)) +
  facet_wrap(~ symbol + type) +
  scale_fill_manual(values = c("grey", si3d_col, si3e_col)) +
  theme_cowplot()+
    theme(legend.position = "none")

ggsave(here("plots/plots_231/hif_231_hypoxia.pdf"), width = 4, height = 3)

```

## Acute hypoxic response: DMSO normoxia to hypoxia.

### TE changes
```{r}

#organize metadata
metadata_acute_sictrl <- metadata %>% 
  filter(treatment== "sictrl") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_acute_sictrl$library <- relevel(factor(metadata_acute_sictrl$library), ref = "rna")
metadata_acute_sictrl$condition <- relevel(factor(metadata_acute_sictrl$condition), ref = "normoxia")

levels(metadata_acute_sictrl$library)
levels(metadata_acute_sictrl$condition)

names <- as.character(metadata_acute_sictrl$name)

all_raw <- all_raw_filtered_counts %>% 
  column_to_rownames("gene_id")

acute_sictrl <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_sictrl <- DESeqDataSetFromMatrix(
  countData = acute_sictrl,
  colData = metadata_acute_sictrl,
  design = ~condition + library + condition:library
  )

ddsMat_acute_sictrl <- DESeq(ddsMat_acute_sictrl)
resultsNames(ddsMat_acute_sictrl)
res_acute_sictrl <- results(ddsMat_acute_sictrl, name=resultsNames(ddsMat_acute_sictrl)[4], cooksCutoff=F, independentFiltering=F)


res_acute_sictrl <- lfcShrink(dds = ddsMat_acute_sictrl, coef = resultsNames(ddsMat_acute_sictrl)[4], res = res_acute_sictrl) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_sictrl %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_sictrl %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

all_counts_acute_sictrl <- data.frame(upreg, downreg)

```


### RNA changes
```{r}

#organize metadata
metadata_acute_sictrl_rna <- metadata %>% 
  filter(treatment== "sictrl") %>% 
  filter(library=="rna") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_sictrl_rna$condition <- relevel(factor(metadata_acute_sictrl_rna$condition), ref = "normoxia")

levels(metadata_acute_sictrl_rna$condition)

names <- as.character(metadata_acute_sictrl_rna$name)

acute_sictrl_rna <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_sictrl_rna <- DESeqDataSetFromMatrix(
  countData = acute_sictrl_rna,
  colData = metadata_acute_sictrl_rna,
  design = ~condition
  )

ddsMat_acute_sictrl_rna <- DESeq(ddsMat_acute_sictrl_rna)
resultsNames(ddsMat_acute_sictrl_rna)
res_acute_sictrl_rna <- results(ddsMat_acute_sictrl_rna, name=resultsNames(ddsMat_acute_sictrl_rna)[2], cooksCutoff=F, independentFiltering=F)


res_acute_sictrl_rna <- lfcShrink(dds = ddsMat_acute_sictrl_rna, coef = resultsNames(ddsMat_acute_sictrl_rna)[2], res = res_acute_sictrl_rna) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_sictrl_rna %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_sictrl_rna %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="rna"

all_counts_acute_sictrl_rna <- data.frame(dataset, upreg, downreg)

```


### RIBO changes

```{r}

#organize metadata
metadata_acute_sictrl_ribo <- metadata %>% 
  filter(treatment== "sictrl") %>% 
  filter(library=="ribo") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_sictrl_ribo$condition <- relevel(factor(metadata_acute_sictrl_ribo$condition), ref = "normoxia")

levels(metadata_acute_sictrl_ribo$condition)

names <- as.character(metadata_acute_sictrl_ribo$name)

acute_sictrl_ribo <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_sictrl_ribo <- DESeqDataSetFromMatrix(
  countData = acute_sictrl_ribo,
  colData = metadata_acute_sictrl_ribo,
  design = ~condition
  )

ddsMat_acute_sictrl_ribo <- DESeq(ddsMat_acute_sictrl_ribo)
resultsNames(ddsMat_acute_sictrl_ribo)
res_acute_sictrl_ribo <- results(ddsMat_acute_sictrl_ribo, name=resultsNames(ddsMat_acute_sictrl_ribo)[2], cooksCutoff=F, independentFiltering=F)


res_acute_sictrl_ribo <- lfcShrink(dds = ddsMat_acute_sictrl_ribo, coef = resultsNames(ddsMat_acute_sictrl_ribo)[2], res = res_acute_sictrl_ribo) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_sictrl_ribo %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_sictrl_ribo %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="ribo"

all_counts_acute_sictrl_ribo <- data.frame(dataset, upreg, downreg)

```

## Visualize sictrl acute hypoxic changes
```{r}

acute_ctrl_counts <- rbind(all_counts_acute_sictrl_rna, all_counts_acute_sictrl_ribo) %>% melt()

acute_ctrl_counts$dataset <- as.factor(acute_ctrl_counts$dataset)
acute_ctrl_counts$dataset <- relevel(acute_ctrl_counts$dataset, ref = "rna")

ggbarplot(acute_ctrl_counts, x="variable", y="value")+
  facet_wrap(~dataset)+
    geom_text(aes(label = value), vjust = -0.3)+
  theme_cowplot()

ggsave(here("plots/plots_231/acute_sictrl_counts.pdf"), width=4, height=4)


```


## Acute hypoxic response: 231 normoxia to hypoxia.

### TE changes
```{r}

#organize metadata
metadata_acute_si3e <- metadata %>% 
  filter(treatment== "si3e") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_acute_si3e$library <- relevel(factor(metadata_acute_si3e$library), ref = "rna")
metadata_acute_si3e$condition <- relevel(factor(metadata_acute_si3e$condition), ref = "normoxia")

levels(metadata_acute_si3e$library)
levels(metadata_acute_si3e$condition)

names <- as.character(metadata_acute_si3e$name)

acute_si3e <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_si3e <- DESeqDataSetFromMatrix(
  countData = acute_si3e,
  colData = metadata_acute_si3e,
  design = ~condition + library + condition:library
  )

ddsMat_acute_si3e <- DESeq(ddsMat_acute_si3e)
resultsNames(ddsMat_acute_si3e)
res_acute_si3e <- results(ddsMat_acute_si3e, name=resultsNames(ddsMat_acute_si3e)[4], cooksCutoff=F, independentFiltering=F)


res_acute_si3e <- lfcShrink(dds = ddsMat_acute_si3e, coef = resultsNames(ddsMat_acute_si3e)[4], res = res_acute_si3e) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_si3e %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_si3e %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

all_counts_acute_si3e <- data.frame(upreg, downreg)
all_counts_acute_si3e
```

### RNA changes
```{r}

#organize metadata
metadata_acute_si3e_rna <- metadata %>% 
  filter(treatment== "si3e") %>% 
  filter(library=="rna") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_si3e_rna$condition <- relevel(factor(metadata_acute_si3e_rna$condition), ref = "normoxia")

levels(metadata_acute_si3e_rna$condition)

names <- as.character(metadata_acute_si3e_rna$name)

acute_si3e_rna <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_si3e_rna <- DESeqDataSetFromMatrix(
  countData = acute_si3e_rna,
  colData = metadata_acute_si3e_rna,
  design = ~condition
  )

ddsMat_acute_si3e_rna <- DESeq(ddsMat_acute_si3e_rna)
resultsNames(ddsMat_acute_si3e_rna)
res_acute_si3e_rna <- results(ddsMat_acute_si3e_rna, name=resultsNames(ddsMat_acute_si3e_rna)[2], cooksCutoff=F, independentFiltering=F)


res_acute_si3e_rna <- lfcShrink(dds = ddsMat_acute_si3e_rna, coef = resultsNames(ddsMat_acute_si3e_rna)[2], res = res_acute_si3e_rna) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_si3e_rna %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_si3e_rna %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="rna"

all_counts_acute_si3e_rna <- data.frame(dataset, upreg, downreg)

```


### RIBO changes

```{r}

#organize metadata
metadata_acute_si3e_ribo <- metadata%>% 
  filter(treatment== "si3e") %>% 
  filter(library=="ribo") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_si3e_ribo$condition <- relevel(factor(metadata_acute_si3e_ribo$condition), ref = "normoxia")

levels(metadata_acute_si3e_ribo$condition)

names <- as.character(metadata_acute_si3e_ribo$name)

acute_si3e_ribo <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_si3e_ribo <- DESeqDataSetFromMatrix(
  countData = acute_si3e_ribo,
  colData = metadata_acute_si3e_ribo,
  design = ~condition
  )

ddsMat_acute_si3e_ribo <- DESeq(ddsMat_acute_si3e_ribo)
resultsNames(ddsMat_acute_si3e_ribo)
res_acute_si3e_ribo <- results(ddsMat_acute_si3e_ribo, name=resultsNames(ddsMat_acute_si3e_ribo)[2], cooksCutoff=F, independentFiltering=F)


res_acute_si3e_ribo <- lfcShrink(dds = ddsMat_acute_si3e_ribo, coef = resultsNames(ddsMat_acute_si3e_ribo)[2], res = res_acute_si3e_ribo) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_si3e_ribo %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_si3e_ribo %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="ribo"

all_counts_acute_si3e_ribo <- data.frame(dataset, upreg, downreg)

```

# 3d

### TE changes
```{r}

#organize metadata
metadata_acute_si3d <- metadata %>% 
  filter(treatment== "si3d") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_acute_si3d$library <- relevel(factor(metadata_acute_si3d$library), ref = "rna")
metadata_acute_si3d$condition <- relevel(factor(metadata_acute_si3d$condition), ref = "normoxia")

levels(metadata_acute_si3d$library)
levels(metadata_acute_si3d$condition)

names <- as.character(metadata_acute_si3d$name)

acute_si3d <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_si3d <- DESeqDataSetFromMatrix(
  countData = acute_si3d,
  colData = metadata_acute_si3d,
  design = ~condition + library + condition:library
  )

ddsMat_acute_si3d <- DESeq(ddsMat_acute_si3d)
resultsNames(ddsMat_acute_si3d)
res_acute_si3d <- results(ddsMat_acute_si3d, name=resultsNames(ddsMat_acute_si3d)[4], cooksCutoff=F, independentFiltering=F)


res_acute_si3d <- lfcShrink(dds = ddsMat_acute_si3d, coef = resultsNames(ddsMat_acute_si3d)[4], res = res_acute_si3d) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_si3d %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_si3d %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

all_counts_acute_si3d <- data.frame(upreg, downreg)
all_counts_acute_si3d
```

### RNA changes
```{r}

#organize metadata
metadata_acute_si3d_rna <- metadata %>% 
  filter(treatment== "si3d") %>% 
  filter(library=="rna") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_si3d_rna$condition <- relevel(factor(metadata_acute_si3d_rna$condition), ref = "normoxia")

levels(metadata_acute_si3d_rna$condition)

names <- as.character(metadata_acute_si3d_rna$name)

acute_si3d_rna <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_si3d_rna <- DESeqDataSetFromMatrix(
  countData = acute_si3d_rna,
  colData = metadata_acute_si3d_rna,
  design = ~condition
  )

ddsMat_acute_si3d_rna <- DESeq(ddsMat_acute_si3d_rna)
resultsNames(ddsMat_acute_si3d_rna)
res_acute_si3d_rna <- results(ddsMat_acute_si3d_rna, name=resultsNames(ddsMat_acute_si3d_rna)[2], cooksCutoff=F, independentFiltering=F)


res_acute_si3d_rna <- lfcShrink(dds = ddsMat_acute_si3d_rna, coef = resultsNames(ddsMat_acute_si3d_rna)[2], res = res_acute_si3d_rna) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_si3d_rna %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_si3d_rna %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="rna"

all_counts_acute_si3d_rna <- data.frame(dataset, upreg, downreg)

```


### RIBO changes

```{r}

#organize metadata
metadata_acute_si3d_ribo <- metadata%>% 
  filter(treatment== "si3d") %>% 
  filter(library=="ribo") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_si3d_ribo$condition <- relevel(factor(metadata_acute_si3d_ribo$condition), ref = "normoxia")

levels(metadata_acute_si3d_ribo$condition)

names <- as.character(metadata_acute_si3d_ribo$name)

acute_si3d_ribo <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_si3d_ribo <- DESeqDataSetFromMatrix(
  countData = acute_si3d_ribo,
  colData = metadata_acute_si3d_ribo,
  design = ~condition
  )

ddsMat_acute_si3d_ribo <- DESeq(ddsMat_acute_si3d_ribo)
resultsNames(ddsMat_acute_si3d_ribo)
res_acute_si3d_ribo <- results(ddsMat_acute_si3d_ribo, name=resultsNames(ddsMat_acute_si3d_ribo)[2], cooksCutoff=F, independentFiltering=F)


res_acute_si3d_ribo <- lfcShrink(dds = ddsMat_acute_si3d_ribo, coef = resultsNames(ddsMat_acute_si3d_ribo)[2], res = res_acute_si3d_ribo) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_si3d_ribo %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_si3d_ribo %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="ribo"

all_counts_acute_si3d_ribo <- data.frame(dataset, upreg, downreg)

```






## Visualize acute hypoxic response 
Export all acute changes as a single table

```{r}


sictrl_te <- res_acute_sictrl[,c(1,2,4,7)]
colnames(sictrl_te)[3] ="LFC_te_sictrl"
colnames(sictrl_te)[4] ="padj_te_sictrl"

sictrl_ribo <- res_acute_sictrl_ribo[,c(1,4,7)]
colnames(sictrl_ribo)[2] ="LFC_ribo_sictrl"
colnames(sictrl_ribo)[3] ="padj_ribo_sictrl"

sictrl_rna <- res_acute_sictrl_rna[,c(1,4,7)]
colnames(sictrl_rna)[2] ="LFC_rna_sictrl"
colnames(sictrl_rna)[3] ="padj_rna_sictrl"

si3e_te <- res_acute_si3e[,c(1,4,7)]
colnames(si3e_te)[2] ="LFC_te_si3e"
colnames(si3e_te)[3] ="padj_te_3e"

si3e_ribo <- res_acute_si3e_ribo[,c(1,4,7)]
colnames(si3e_ribo)[2] ="LFC_ribo_si3e"
colnames(si3e_ribo)[3] ="padj_ribo_si3e"

si3e_rna <- res_acute_si3e_rna[,c(1,4,7)]
colnames(si3e_rna)[2] ="LFC_rna_si3e"
colnames(si3e_rna)[3] ="padj_rna_si3e"

si3d_te <- res_acute_si3d[,c(1,4,7)]
colnames(si3d_te)[2] ="LFC_te_si3d"
colnames(si3d_te)[3] ="padj_te_si3d"

si3d_ribo <- res_acute_si3d_ribo[,c(1,4,7)]
colnames(si3d_ribo)[2] ="LFC_ribo_si3d"
colnames(si3d_ribo)[3] ="padj_ribo_si3d"

si3d_rna <- res_acute_si3d_rna[,c(1,4,7)]
colnames(si3d_rna)[2] ="LFC_rna_si3d"
colnames(si3d_rna)[3] ="padj_rna_si3d"

#bind cols ----------------------------------------------------------------------

all_changes <- purrr::reduce(list(sictrl_te,sictrl_ribo,sictrl_rna,si3e_te,si3e_ribo,si3e_rna,si3d_te,si3d_ribo,si3d_rna), dplyr::left_join, by = 'gene_id') %>% 
  as.data.frame()

write_csv(all_changes, here("counts/all_norm_to_hyp_changes_231.csv"))
```



## Visualize acute hypoxic response with histograms

## Acute hypoxic changes

```{r}

sictrl <- all_changes %>% 
  filter(padj_rna_sictrl < 0.05 | padj_ribo_sictrl < 0.05)

sictrl_acute <- ggplot(data = sictrl, aes(x = LFC_rna_sictrl, y = LFC_ribo_sictrl)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote(Log[2]~"sictrl LFC"~"RIBO: hypoxia vs normoxia")) +
  xlab(bquote(Log[2]~"sictrl LFC"~"RNA: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 5)+
  theme_prism()

sictrl_acute <- ggMarginal(sictrl_acute, type = "histogram",
           xparams = list(binwidth = .1, fill = "grey"),
           yparams = list(binwidth = .1, fill = "red")
           )

ggsave(filename = here("plots/plots_231/rna_ribo_sictrl_acute.pdf"), sictrl_acute, width = 4, height = 4)

sictrl_acute
```

# acute hypoxic response with 3e KD


```{r}

sig_sictrl <- all_changes %>% 
  filter(padj_te_sictrl <0.05)

#counts sictrl
ctrl_down<-
sig_sictrl %>% 
  filter(LFC_te_sictrl < 0) %>% 
  filter(padj_te_sictrl < 0.05) %>% 
  nrow()

ctrl_up<-
sig_sictrl %>% 
  filter(LFC_te_sictrl > 0) %>% 
  filter(padj_te_sictrl < 0.05) %>% 
  nrow()

#counts 3eKD
down_3e<-
sig_sictrl %>% 
  filter(LFC_te_si3e < 0) %>% 
  filter(padj_te_3e < 0.05) %>% 
  nrow()

up_3e <-
sig_sictrl %>% 
  filter(LFC_te_si3e > 0) %>% 
  filter(padj_te_3e < 0.05) %>% 
  nrow()

1-(up_3e/ctrl_up)
1-(down_3e/ctrl_down)

scatter_acute_sictrl_3e <- ggplot(data = sig_sictrl, aes(x = LFC_te_sictrl, y = LFC_te_si3e)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote(Log[2]~"si-eIF3e LFC"~"TE: hypoxia vs normoxia")) +
  xlab(bquote(Log[2]~"sictrl LFC"~"TE: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 9)+
  theme_prism()

scatter_acute_sictrl_3e <- ggMarginal(scatter_acute_sictrl_3e, type = "histogram",
           xparams = list(binwidth = .1, fill = "grey"),
           yparams = list(binwidth = .1, fill = si3e_col)
           )

scatter_acute_sictrl_3e

ggsave(filename= here("plots/plots_231/acute_sictrl_si3e.pdf"), scatter_acute_sictrl_3e, height=4, width=4)

sig_sictrl %>% 
  filter(padj_te_3e < 0.05 & LFC_te_si3e > 0) %>% 
  nrow()

sig_sictrl %>% 
  filter(padj_te_3e < 0.05 & LFC_te_si3e < 0) %>% 
  nrow()

```



si3d

```{r}

sig_sictrl <- all_changes %>% 
  filter(padj_te_sictrl <0.05)

#counts sictrl
sig_sictrl %>% 
  filter(LFC_te_sictrl < 0) %>% 
  filter(padj_te_sictrl < 0.05) %>% 
  nrow()

sig_sictrl %>% 
  filter(LFC_te_sictrl > 0) %>% 
  filter(padj_te_sictrl < 0.05) %>% 
  nrow()

#counts 3dKD
down_3d <-
sig_sictrl %>% 
  filter(LFC_te_si3d < 0) %>% 
  filter(padj_te_si3d < 0.05) %>% 
  nrow()

up_3d <-
sig_sictrl %>% 
  filter(LFC_te_si3d > 0) %>% 
  filter(padj_te_si3d < 0.05) %>% 
  nrow()

1-(up_3d/ctrl_up)
1-(down_3d/ctrl_down)



scatter_acute_sictrl_3d <- ggplot(data = sig_sictrl, aes(x = LFC_te_sictrl, y = LFC_te_si3d)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote(Log[2]~"si-eIF3d LFC"~"TE: hypoxia vs normoxia")) +
  xlab(bquote(Log[2]~"sictrl LFC"~"TE: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 9)+
  theme_prism()

scatter_acute_sictrl_3d <- ggMarginal(scatter_acute_sictrl_3d, type = "histogram",
           xparams = list(binwidth = .1, fill = "grey"),
           yparams = list(binwidth = .1, fill = si3d_col)
           )

scatter_acute_sictrl_3d

ggsave(filename= here("plots/plots_231/acute_sictrl_si3d.pdf"), scatter_acute_sictrl_3d, height=4, width=4)


nrow(sig_sictrl)

```



# Normoxia
##3e
```{r}

#organize metadata
metadata_normox_si3e <- metadata %>% 
  filter(treatment== "si3e"| treatment=="sictrl") %>% 
  filter(condition=="normoxia") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_normox_si3e$library <- relevel(factor(metadata_normox_si3e$library), ref = "rna")
metadata_normox_si3e$treatment <- relevel(factor(metadata_normox_si3e$treatment), ref = "sictrl")

levels(metadata_normox_si3e$library)
levels(metadata_normox_si3e$treatment)

names <- as.character(metadata_normox_si3e$name)

normox_si3e <- all_raw[, colnames(all_raw) %in% names]


ddsMat_normox_si3e <- DESeqDataSetFromMatrix(
  countData = normox_si3e,
  colData = metadata_normox_si3e,
  design = ~treatment + library + treatment:library
  )

ddsMat_normox_si3e <- DESeq(ddsMat_normox_si3e)
resultsNames(ddsMat_normox_si3e)
res_normox_si3e <- results(ddsMat_normox_si3e, name=resultsNames(ddsMat_normox_si3e)[4], cooksCutoff=F, independentFiltering=F)


res_normox_si3e <- lfcShrink(dds = ddsMat_normox_si3e, coef = resultsNames(ddsMat_normox_si3e)[4], res = res_normox_si3e) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)



# count table -------------------------------------------------------------------

upreg <- res_normox_si3e %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_normox_si3e %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset= "normoxia_3e"

all_counts_normox_si3e <- data.frame(dataset, upreg, downreg)
all_counts_normox_si3e

write_csv(res_normox_si3e, here("counts/counts_231/normoxia_si_ctrl_3e.csv"))

```

##3d

```{r}

#organize metadata
metadata_normox_si3d <- metadata %>% 
  filter(treatment== "si3d"| treatment=="sictrl") %>% 
  filter(condition=="normoxia") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_normox_si3d$library <- relevel(factor(metadata_normox_si3d$library), ref = "rna")
metadata_normox_si3d$treatment <- relevel(factor(metadata_normox_si3d$treatment), ref = "sictrl")

levels(metadata_normox_si3d$library)
levels(metadata_normox_si3d$treatment)

names <- as.character(metadata_normox_si3d$name)

normox_si3d <- all_raw[, colnames(all_raw) %in% names]


ddsMat_normox_si3d <- DESeqDataSetFromMatrix(
  countData = normox_si3d,
  colData = metadata_normox_si3d,
  design = ~treatment + library + treatment:library
  )

ddsMat_normox_si3d <- DESeq(ddsMat_normox_si3d)
resultsNames(ddsMat_normox_si3d)
res_normox_si3d <- results(ddsMat_normox_si3d, name=resultsNames(ddsMat_normox_si3d)[4], cooksCutoff=F, independentFiltering=F)


res_normox_si3d <- lfcShrink(dds = ddsMat_normox_si3d, coef = resultsNames(ddsMat_normox_si3d)[4], res = res_normox_si3d) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)



# count table -------------------------------------------------------------------

upreg <- res_normox_si3d %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_normox_si3d %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset= "normoxia_3d"

all_counts_normox_si3d <- data.frame(dataset, upreg, downreg)
all_counts_normox_si3d

write_csv(res_normox_si3d, here("counts/counts_231/normoxia_si_ctrl_3d.csv"))

```


# Hypoxia
##3e
```{r}

#organize metadata
metadata_hypox_si3e <- metadata %>% 
  filter(treatment== "si3e"| treatment=="sictrl") %>% 
  filter(condition=="hypoxia") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_hypox_si3e$library <- relevel(factor(metadata_hypox_si3e$library), ref = "rna")
metadata_hypox_si3e$treatment <- relevel(factor(metadata_hypox_si3e$treatment), ref = "sictrl")

levels(metadata_hypox_si3e$library)
levels(metadata_hypox_si3e$treatment)

names <- as.character(metadata_hypox_si3e$name)

hypox_si3e <- all_raw[, colnames(all_raw) %in% names]


ddsMat_hypox_si3e <- DESeqDataSetFromMatrix(
  countData = hypox_si3e,
  colData = metadata_hypox_si3e,
  design = ~treatment + library + treatment:library
  )

ddsMat_hypox_si3e <- DESeq(ddsMat_hypox_si3e)
resultsNames(ddsMat_hypox_si3e)
res_hypox_si3e <- results(ddsMat_hypox_si3e, name=resultsNames(ddsMat_hypox_si3e)[4], cooksCutoff=F, independentFiltering=F)


res_hypox_si3e <- lfcShrink(dds = ddsMat_hypox_si3e, coef = resultsNames(ddsMat_hypox_si3e)[4], res = res_hypox_si3e) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)



# count table -------------------------------------------------------------------

upreg <- res_hypox_si3e %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_hypox_si3e %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset= "hypoxia_3e"

all_counts_hypox_si3e <- data.frame(dataset, upreg, downreg)
all_counts_hypox_si3e

write_csv(res_hypox_si3e, here("counts/counts_231/hypoxia_si_ctrl_3e.csv"))

```

##3d

```{r}

#organize metadata
metadata_hypox_si3d <- metadata %>% 
  filter(treatment== "si3d"| treatment=="sictrl") %>% 
  filter(condition=="hypoxia") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_hypox_si3d$library <- relevel(factor(metadata_hypox_si3d$library), ref = "rna")
metadata_hypox_si3d$treatment <- relevel(factor(metadata_hypox_si3d$treatment), ref = "sictrl")

levels(metadata_hypox_si3d$library)
levels(metadata_hypox_si3d$treatment)

names <- as.character(metadata_hypox_si3d$name)

hypox_si3d <- all_raw[, colnames(all_raw) %in% names]


ddsMat_hypox_si3d <- DESeqDataSetFromMatrix(
  countData = hypox_si3d,
  colData = metadata_hypox_si3d,
  design = ~treatment + library + treatment:library
  )

ddsMat_hypox_si3d <- DESeq(ddsMat_hypox_si3d)
resultsNames(ddsMat_hypox_si3d)
res_hypox_si3d <- results(ddsMat_hypox_si3d, name=resultsNames(ddsMat_hypox_si3d)[4], cooksCutoff=F, independentFiltering=F)


res_hypox_si3d <- lfcShrink(dds = ddsMat_hypox_si3d, coef = resultsNames(ddsMat_hypox_si3d)[4], res = res_hypox_si3d) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)



# count table -------------------------------------------------------------------

upreg <- res_hypox_si3d %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_hypox_si3d %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset= "hypoxia_3d"

all_counts_hypox_si3d <- data.frame(dataset, upreg, downreg)
all_counts_hypox_si3d

write_csv(res_hypox_si3d, here("counts/counts_231/hypoxia_si_ctrl_3d.csv"))

```


### Visualize normoxic vs hyoxic response
```{r}
hyp_vs_norm_kds <- rbind(all_counts_normox_si3e, all_counts_normox_si3d,
                         all_counts_hypox_si3e, all_counts_hypox_si3d) %>%
  melt() %>%
  mutate(
    condition = str_extract(dataset, "^(normox|hypox)"),
    sample = str_extract(dataset, "3[ed]")
  )
  
```

# effect of si3e KD norm and hypoxia
```{r}

eif3e_counts <- hyp_vs_norm_kds %>% 
  filter(sample=="3e")

eif3e_bar <- ggbarplot(data = eif3e_counts, x = "condition", y = "value", fill = "variable", palette = c("#2492a6",si3e_col), position = position_dodge(0.75), label = T, title= "eIF3e")+
  ylab("Transcript counts")+
  xlab("")+
  theme_prism()+
  # theme_cowplot(font_size = 12)+
  scale_y_continuous(limits = c(0, 1600))

eif3e_bar

ggsave(here("plots/plots_231/eIF3ebars.pdf"), eif3e_bar, width = 4, height = 4)

```


# effect of si3d KD norm and hypoxia

```{r}

eif3d_counts <- hyp_vs_norm_kds %>% 
  filter(sample=="3d")

eif3d_bar <- ggbarplot(data = eif3d_counts, x = "condition", y = "value", fill = "variable", palette = c("#ffd412",si3d_col), position = position_dodge(0.75), label = T, title= "eIF3d")+
  ylab("Transcript counts")+
  xlab("")+
  theme_prism()+
  # theme_cowplot(font_size = 12)+
  scale_y_continuous(limits = c(0, 1600))

eif3d_bar

ggsave(here("plots/plots_231/eIF3dbars.pdf"), eif3d_bar, width = 4, height = 4)

```



# Do targets overlap with data from MCF7-SIX1s?

pull in mcf7 six 1 data and label everything by cell line


## Create gene lists for downstream analysis
231s

normoxia
```{r}

#3d

#up

mb231_si3d_normoxia_up <- res_normox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mb231_si3d_normoxia_up$gs_name <- rep("mb231_si3d_normoxia_up")


#down

mb231_si3d_normoxia_down <- res_normox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mb231_si3d_normoxia_down$gs_name <- rep("mb231_si3d_normoxia_down")


#3e
#up

mb231_si3e_normoxia_up <- res_normox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mb231_si3e_normoxia_up$gs_name <- rep("mb231_si3e_normoxia_up")

#down

mb231_si3e_normoxia_down <- res_normox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mb231_si3e_normoxia_down$gs_name <- rep("mb231_si3e_normoxia_down")


```


hypoxia
```{r}

#3d

#up

mb231_si3d_hypoxia_up <- res_hypox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mb231_si3d_hypoxia_up$gs_name <- rep("mb231_si3d_hypoxia_up")


#down

mb231_si3d_hypoxia_down <- res_hypox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mb231_si3d_hypoxia_down$gs_name <- rep("mb231_si3d_hypoxia_down")


#3e
#up

mb231_si3e_hypoxia_up <- res_hypox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mb231_si3e_hypoxia_up$gs_name <- rep("mb231_si3e_hypoxia_up")

#down

mb231_si3e_hypoxia_down <- res_hypox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mb231_si3e_hypoxia_down$gs_name <- rep("mb231_si3e_hypoxia_down")


```


Pull in data from mcf7-six1

```{r}

#3d

#up
mcf7six1_res_normox_si3d <-read_csv(here("counts/fig1/normoxia_sictrl_3d.csv"))

mcf7six1_si3d_normoxia_up <- mcf7six1_res_normox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mcf7six1_si3d_normoxia_up$gs_name <- rep("mcf7six1_si3d_normoxia_up")


#down

mcf7six1_si3d_normoxia_down <- mcf7six1_res_normox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mcf7six1_si3d_normoxia_down$gs_name <- rep("mcf7six1_si3d_normoxia_down")


#3e
#up
mcf7six1_res_normox_si3e <-read_csv(here("counts/fig1/normoxia_sictrl_3e.csv"))


mcf7six1_si3e_normoxia_up <- mcf7six1_res_normox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mcf7six1_si3e_normoxia_up$gs_name <- rep("mcf7six1_si3e_normoxia_up")

#down

mcf7six1_si3e_normoxia_down <- mcf7six1_res_normox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mcf7six1_si3e_normoxia_down$gs_name <- rep("mcf7six1_si3e_normoxia_down")


```

Import hypoxia data from figure 3
```{r}

#3d
mcf7six1_res_hypox_si3d <-read_csv(here("counts/fig3/hypoxia_si_ctrl_3d.csv"))


mcf7six1_si3d_hypoxia_up <- mcf7six1_res_hypox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mcf7six1_si3d_hypoxia_up$gs_name <- rep("mcf7six1_si3d_hypoxia_up")


#down

mcf7six1_si3d_hypoxia_down <- mcf7six1_res_hypox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mcf7six1_si3d_hypoxia_down$gs_name <- rep("mcf7six1_si3d_hypoxia_down")



#3e
mcf7six1_res_hypox_si3e <-read_csv(here("counts/fig3/hypoxia_si_ctrl_3e.csv"))


mcf7six1_si3e_hypoxia_up <- mcf7six1_res_hypox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mcf7six1_si3e_hypoxia_up$gs_name <- rep("mcf7six1_si3e_hypoxia_up")

#down

mcf7six1_si3e_hypoxia_down <- mcf7six1_res_hypox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mcf7six1_si3e_hypoxia_down$gs_name <- rep("mcf7six1_si3e_hypoxia_down")


```



## Gene lists for a209 in normoxia and hypoxia

```{r}

#normoxia a209


res_normox_a209 <-read_csv(here("counts/fig6/normoxia_dmso_a209.csv"))


#up
a209_normoxia_up <- res_normox_a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

a209_normoxia_up$gs_name <- rep("mcf7six1_a209_normoxia_up")

#down

a209_normoxia_down <- res_normox_a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

a209_normoxia_down$gs_name <- rep("mcf7six1_a209_normoxia_down")


#hypoxia a209
res_hypox_a209 <-read_csv(here("counts/fig6/hypoxia_dmso_a209.csv"))

#up

a209_hypoxia_up <- res_hypox_a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

a209_hypoxia_up$gs_name <- rep("mcf7six1_a209_hypoxia_up")

#down

a209_hypoxia_down <- res_hypox_a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

a209_hypoxia_down$gs_name <- rep("mcf7six1_a209_hypoxia_down")


```

# Overlap of 3e targets in MCF7SIX1 and MD-MBA-231


# Odds ratios for a209 3e and 3d overlap in normoxia

The following analysis determines the number of genes that overlap in each condition and the statistical significance (p-value) of their overlap (odds ratio).


Create a matrix that compares each condition for gene overlap analysis.
```{r}
all_lists <- bind_rows(mcf7six1_si3e_normoxia_up, mcf7six1_si3d_normoxia_up, a209_normoxia_up, 
                       mcf7six1_si3e_normoxia_down, mcf7six1_si3d_normoxia_down, a209_normoxia_down, 
                       mcf7six1_si3e_hypoxia_up, mcf7six1_si3d_hypoxia_up, a209_hypoxia_up, 
                       mcf7six1_si3e_hypoxia_down, mcf7six1_si3d_hypoxia_down, a209_hypoxia_down,
                       mb231_si3e_normoxia_up, mb231_si3d_normoxia_up, 
                       mb231_si3e_normoxia_down, mb231_si3d_normoxia_down, 
                       mb231_si3e_hypoxia_up, mb231_si3d_hypoxia_up, 
                       mb231_si3e_hypoxia_down, mb231_si3d_hypoxia_down) %>% 
extract(gs_name,
        into = c("cell_line", "treatment", "condition", "direction"),
        regex = "([^_]+)_([^_]+)_([^_]+)_([^_]+)")
```


### get number of expressed genes for universe value for mcf7six1 and md-mba-231
```{r}
all_3e_symbols_expressed <- rbind(
res_normox_si3e,
res_hypox_si3e,
mcf7six1_res_hypox_si3e,
mcf7six1_res_hypox_si3e)

universe <- unique(all_3e_symbols_expressed$symbol)

```






### Normoxia analysis
```{r}

norm_matrix <- all_lists %>%
  filter(condition=="normoxia") %>% 
  dplyr::select(-condition)


norm_matrix_comb <- norm_matrix %>% 
  tidyr::unite("description", 2:4, sep=', ')

gom.allsamples <- split(norm_matrix_comb$symbol, norm_matrix_comb$description)

#double check genome size here

gom.all <- newGOM(gom.allsamples, gom.allsamples, genome.size = 12495)

norm_all.odds.matrix <- getMatrix(gom.all, "odds")
norm_all.pval.matrix <- getMatrix(gom.all, "pval")


```


# plot odds ratios as bars with p values
```{r}

#merge odds and pval matricies

odds <- norm_all.odds.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:11, names_to = "comparison", values_to="odds_value")

pval <- norm_all.pval.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:11, names_to = "comparison", values_to="pval_value")


odds_pval_norm <- left_join(odds, pval, by=c("sample", "comparison")) %>% 
  tidyr::unite("comparison", 1:2, sep="-") 
```




```{r}
odds_pval_3e<-odds_pval_norm %>% 
  filter(comparison == "mcf7six1, si3e, down-mb231, si3e, down"| 
         comparison =="mcf7six1, si3e, up-mb231, si3e, up"|
           comparison == "mcf7six1, si3d, down-mb231, si3d, down"|
           comparison == "mcf7six1, si3d, up-mb231, si3d, up")


ggpubr::ggbarplot(odds_pval_3e, x="comparison", y="odds_value", label = formatC(odds_pval_3e$pval_value, format = "e", digits = 2), title="Overlap of 3e/3d targets in normoxia", palette = c(si3d_col, si3e_col), ylim=c(0,16))+
  geom_hline(aes(yintercept=1))+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))



ggsave(here("plots/plots_231/normoverlap.pdf"), width=4, height=8)

```

### Hypoxia

```{r}

hyp_matrix <- all_lists %>%
  filter(condition=="hypoxia") %>% 
  dplyr::select(-condition)


hyp_matrix_comb <- hyp_matrix %>% 
  tidyr::unite("description", 2:4, sep=', ')

gom.allsamples <- split(hyp_matrix_comb$symbol, hyp_matrix_comb$description)

#double check genome size here

gom.all <- newGOM(gom.allsamples, gom.allsamples, genome.size = 12495)

hyp_all.odds.matrix <- getMatrix(gom.all, "odds")
hyp_all.pval.matrix <- getMatrix(gom.all, "pval")


```



# plot odds ratios as bars with p values
```{r}

#merge odds and pval matricies

odds <- hyp_all.odds.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:11, names_to = "comparison", values_to="odds_value")

pval <- hyp_all.pval.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:11, names_to = "comparison", values_to="pval_value")


odds_pval_hyp <- left_join(odds, pval, by=c("sample", "comparison")) %>% 
  tidyr::unite("comparison", 1:2, sep="-") 
```


```{r}
odds_pval_3e<-odds_pval_hyp %>% 
  filter(comparison == "mcf7six1, si3e, down-mb231, si3e, down"| 
         comparison =="mcf7six1, si3e, up-mb231, si3e, up"|
           comparison == "mcf7six1, si3d, down-mb231, si3d, down"|
           comparison == "mcf7six1, si3d, up-mb231, si3d, up")


ggpubr::ggbarplot(odds_pval_3e, x="comparison", y="odds_value", label = formatC(odds_pval_3e$pval_value, format = "e", digits = 2), title="Overlap of 3e/3d targets in hypoxia", palette = c(si3d_col, si3e_col), ylim=c(0,16))+
  geom_hline(aes(yintercept=1))+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))



ggsave(here("plots/plots_231/hypoverlap.pdf"), width=4, height=8)

```



# Venn diagrams

```{r}

all_lists <- bind_rows(mcf7six1_si3e_normoxia_up, mcf7six1_si3d_normoxia_up, a209_normoxia_up, 
                       mcf7six1_si3e_normoxia_down, mcf7six1_si3d_normoxia_down, a209_normoxia_down, 
                       mcf7six1_si3e_hypoxia_up, mcf7six1_si3d_hypoxia_up, a209_hypoxia_up, 
                       mcf7six1_si3e_hypoxia_down, mcf7six1_si3d_hypoxia_down, a209_hypoxia_down,
                       mb231_si3e_normoxia_up, mb231_si3d_normoxia_up, 
                       mb231_si3e_normoxia_down, mb231_si3d_normoxia_down, 
                       mb231_si3e_hypoxia_up, mb231_si3d_hypoxia_up, 
                       mb231_si3e_hypoxia_down, mb231_si3d_hypoxia_down) %>% 
extract(gs_name,
        into = c("cell_line", "treatment", "condition", "direction"),
        regex = "([^_]+)_([^_]+)_([^_]+)_([^_]+)")

```

Normoxia
## Venn diagram of 3e/d gene overlaps in normoxia
```{r compare genes}
mcf7six1_si3e_normoxia_up <- mcf7six1_si3e_normoxia_up %>% pull(symbol)
mcf7six1_si3d_normoxia_up <- mcf7six1_si3d_normoxia_up %>% pull(symbol)

mcf7six1_si3e_normoxia_down <- mcf7six1_si3e_normoxia_down %>% pull(symbol)
mcf7six1_si3d_normoxia_down <- mcf7six1_si3d_normoxia_down %>% pull(symbol)

mb231_si3e_normoxia_up <- mb231_si3e_normoxia_up %>% pull(symbol)
mb231_si3d_normoxia_up <- mb231_si3d_normoxia_up %>% pull(symbol)

mb231_si3e_normoxia_down <- mb231_si3e_normoxia_down %>% pull(symbol)
mb231_si3d_normoxia_down <- mb231_si3d_normoxia_down %>% pull(symbol)

#3e

#up

norm_te_up_3e <- list(
  MCF7SIX1 = mcf7six1_si3e_normoxia_up, 
  MDA_MB_231 = mb231_si3e_normoxia_up
)

ggvenn(
  norm_te_up_3e,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3e_col, "blue")
) +
  ggtitle("Up with 3e KD - Normoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3e_norm_vennup.pdf"))


#down

norm_te_down_3e <- list(
  MCF7SIX1 = mcf7six1_si3e_normoxia_down, 
  MDA_MB_231 = mb231_si3e_normoxia_down
)

ggvenn(
  norm_te_down_3e,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3e_col, "blue")
) +
  ggtitle("down with 3e KD - Normoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3e_norm_venndown.pdf"))






#3d

#up

norm_te_up_3d <- list(
  MCF7SIX1 = mcf7six1_si3d_normoxia_up, 
  MDA_MB_231 = mb231_si3d_normoxia_up
)

ggvenn(
  norm_te_up_3d,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3d_col, "red"))+
  ggtitle("Up with 3d KD - Normoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3d_norm_vennup.pdf"))


#down

norm_te_down_3d <- list(
  MCF7SIX1 = mcf7six1_si3d_normoxia_down, 
  MDA_MB_231 = mb231_si3d_normoxia_down
)

ggvenn(
  norm_te_down_3d,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3d_col, "red")
) +
  ggtitle("down with 3d KD - Normoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3d_norm_venndown.pdf"))


```




hypoxia
## Venn diagram of 3e/d gene overlaps in hypoxia
```{r compare genes}
mcf7six1_si3e_hypoxia_up <- mcf7six1_si3e_hypoxia_up %>% pull(symbol)
mcf7six1_si3d_hypoxia_up <- mcf7six1_si3d_hypoxia_up %>% pull(symbol)

mcf7six1_si3e_hypoxia_down <- mcf7six1_si3e_hypoxia_down %>% pull(symbol)
mcf7six1_si3d_hypoxia_down <- mcf7six1_si3d_hypoxia_down %>% pull(symbol)

mb231_si3e_hypoxia_up <- mb231_si3e_hypoxia_up %>% pull(symbol)
mb231_si3d_hypoxia_up <- mb231_si3d_hypoxia_up %>% pull(symbol)

mb231_si3e_hypoxia_down <- mb231_si3e_hypoxia_down %>% pull(symbol)
mb231_si3d_hypoxia_down <- mb231_si3d_hypoxia_down %>% pull(symbol)

#3e

#up

hyp_te_up_3e <- list(
  MCF7SIX1 = mcf7six1_si3e_hypoxia_up, 
  MDA_MB_231 = mb231_si3e_hypoxia_up
)

ggvenn(
  hyp_te_up_3e,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3e_col, "blue")
) +
  ggtitle("Up with 3e KD - hypoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3e_hyp_vennup.pdf"))


#down

hyp_te_down_3e <- list(
  MCF7SIX1 = mcf7six1_si3e_hypoxia_down, 
  MDA_MB_231 = mb231_si3e_hypoxia_down
)

ggvenn(
  hyp_te_down_3e,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3e_col, "blue")
) +
  ggtitle("down with 3e KD - hypoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3e_hyp_venndown.pdf"))






#3d

#up

hyp_te_up_3d <- list(
  MCF7SIX1 = mcf7six1_si3d_hypoxia_up, 
  MDA_MB_231 = mb231_si3d_hypoxia_up
)

ggvenn(
  hyp_te_up_3d,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3d_col, "red"))+
  ggtitle("Up with 3d KD - hypoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3d_hyp_vennup.pdf"))


#down

hyp_te_down_3d <- list(
  MCF7SIX1 = mcf7six1_si3d_hypoxia_down, 
  MDA_MB_231 = mb231_si3d_hypoxia_down
)

ggvenn(
  hyp_te_down_3d,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3d_col, "red")
) +
  ggtitle("down with 3d KD - hypoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3d_hyp_venndown.pdf"))


```





# Acute hypoxic response

Genes collapsed by 3e KD in acute hypoxic response:
```{r}
mb231_acute_3e <- all_changes %>%
  filter(padj_te_sictrl < 0.05 & padj_te_3e > 0.05)

nrow(mb231_acute_3e)

mb231_acute_3d <- all_changes %>%
  filter(padj_te_sictrl < 0.05 & padj_te_si3d > 0.05)

nrow(mb231_acute_3d)
```


Gather same lists from MCF7SIX1 data
```{r}

mcf7six1_acute <- read_csv(here("counts/fig3/all_norm_to_hyp_changes.csv"))

mcf7six1_acute_3e <- mcf7six1_acute %>%
  filter(padj_te_sictrl < 0.05 & padj_te_3e > 0.05)

nrow(mcf7six1_acute_3e)



mcf7six1_acute_3d <- mcf7six1_acute %>%
  filter(padj_te_sictrl < 0.05 & padj_te_si3d > 0.05)

nrow(mcf7six1_acute_3d)


```

3e overlap between cell lines
```{r}
acute_3e_sig_down_MCF7SIX1 <- mcf7six1_acute_3e %>% 
  filter(LFC_te_sictrl < 0) %>% 
  pull(symbol)

acute_3e_sig_up_MCF7SIX1 <- mcf7six1_acute_3e %>% 
  filter(LFC_te_sictrl > 0) %>% 
  pull(symbol)


acute_3e_sig_down_MB231 <- mb231_acute_3e %>% 
  filter(LFC_te_sictrl < 0) %>% 
  pull(symbol)

acute_3e_sig_up_MB231 <- mb231_acute_3e %>% 
  filter(LFC_te_sictrl > 0) %>% 
  pull(symbol)



acute_down <- list(
  "MCF7-SIX1" = acute_3e_sig_down_MCF7SIX1,
  "MDA-MB-231" = acute_3e_sig_down_MB231
)


ggvenn(acute_down)

view(intersect(acute_3e_sig_down_MCF7SIX1, acute_3e_sig_down_MB231))

ggsave(here("plots/plots_231/acute_down.pdf"))






acute_up <- list(
  "MCF7-SIX1" = acute_3e_sig_up_MCF7SIX1,
  "MDA-MB-231" = acute_3e_sig_up_MB231
)


ggvenn(acute_up)

view(intersect(acute_3e_sig_up_MCF7SIX1, acute_3e_sig_up_MB231))

ggsave(here("plots/plots_231/acute_up.pdf"))


```


3d overlap between cell lines
```{r}
acute_3d_sig_down_MCF7SIX1 <- mcf7six1_acute_3d %>% 
  filter(LFC_te_sictrl > 0) %>% 
  pull(symbol)

acute_3d_sig_up_MCF7SIX1 <- mcf7six1_acute_3d %>% 
  filter(LFC_te_sictrl < 0) %>% 
  pull(symbol)


acute_3d_sig_down_MB231 <- mb231_acute_3d %>% 
  filter(LFC_te_sictrl > 0) %>% 
  pull(symbol)

acute_3d_sig_up_MB231 <- mb231_acute_3d %>% 
  filter(LFC_te_sictrl < 0) %>% 
  pull(symbol)



acute_down <- list(
  "MCF7-SIX1" = acute_3d_sig_down_MCF7SIX1,
  "MDA-MB-231" = acute_3d_sig_down_MB231
)


ggvenn(acute_down)

view(intersect(acute_3d_sig_down_MCF7SIX1, acute_3d_sig_down_MB231))

ggsave(here("plots/plots_231/acute_down_3d.pdf"))






acute_up <- list(
  "MCF7-SIX1" = acute_3d_sig_up_MCF7SIX1,
  "MDA-MB-231" = acute_3d_sig_up_MB231
)


ggvenn(acute_up)

view(intersect(acute_3d_sig_up_MCF7SIX1, acute_3d_sig_up_MB231))

ggsave(here("plots/plots_231/acute_up_d.pdf"))


```


# Compare the acute hypoxic response in both cell lines


Correlation between TE changes with control in both cell lines
```{r}

colnames(mcf7six1_acute)

mcf7six1_acute_ctrl <- mcf7six1_acute %>% 
  dplyr::select(gene_id, LFC_te_sictrl, padj_te_sictrl) %>% 
  rename("LFC_te_sictrl_mcf7six1" = "LFC_te_sictrl") %>% 
    rename("padj_te_sictrl_mcf7six1" = "padj_te_sictrl")




mb231_acute_ctrl <- all_changes %>% 
  dplyr::select(gene_id, LFC_te_sictrl, padj_te_sictrl) %>% 
  rename("LFC_te_sictrl_mb231" = "LFC_te_sictrl") %>% 
    rename("padj_te_sictrl_mb231" = "padj_te_sictrl")

both_lines_ctrl <- left_join(mcf7six1_acute_ctrl, mb231_acute_ctrl)




```


```{r}

scatter_acute_both_cells <- ggplot(data = both_lines_ctrl, aes(x = LFC_te_sictrl_mcf7six1, y = LFC_te_sictrl_mb231)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote(Log[2]~"MB231 si-ctrl LFC"~"TE: hypoxia vs normoxia")) +
  xlab(bquote(Log[2]~"MCF7SIX1 si-ctrl LFC"~"TE: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 9)+
  theme_prism()+
  stat_cor(
    method = "pearson", 
    label.x = -1.8,   # adjust placement
    label.y = 1.8,
    size = 5)

scatter_acute_both_cells <- ggMarginal(scatter_acute_both_cells, type = "histogram",
           xparams = list(binwidth = .1, fill = "orange"),
           yparams = list(binwidth = .1, fill = "pink")
           )

scatter_acute_both_cells

ggsave(filename= here("plots/plots_231/acute_sictrl.pdf"), scatter_acute_both_cells, height=4, width=4)

```

# Overlap of a209 targets in mcf7six1 with e and d in 231s
# NORMOXIA
```{r}
odds_pval_a209<-odds_pval_norm %>% 
  filter(comparison == "mcf7six1, a209, down-mb231, si3e, down"| 
         comparison =="mcf7six1, a209, up-mb231, si3e, up"|
           comparison == "mcf7six1, a209, down-mb231, si3d, down"|
           comparison == "mcf7six1, a209, up-mb231, si3d, up")


ggpubr::ggbarplot(odds_pval_a209, x="comparison", y="odds_value", label = formatC(odds_pval_3e$pval_value, format = "e", digits = 2), title="Overlap of 3e/3d targets in MB-231s in normoxia with a209 in MCF7SIX1s", palette = c(si3d_col, si3e_col), ylim=c(0,6))+
  geom_hline(aes(yintercept=1))+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))



ggsave(here("plots/plots_231/normoverlap209.pdf"), width=7, height=7)

```


#HYPOXIA
```{r}
odds_pval_a209_hyp<-odds_pval_hyp %>% 
 filter(comparison == "mcf7six1, a209, down-mb231, si3e, down"| 
         comparison =="mcf7six1, a209, up-mb231, si3e, up"|
           comparison == "mcf7six1, a209, down-mb231, si3d, down"|
           comparison == "mcf7six1, a209, up-mb231, si3d, up")


ggpubr::ggbarplot(odds_pval_a209_hyp, x="comparison", y="odds_value", label = formatC(odds_pval_3e$pval_value, format = "e", digits = 2), title="Overlap of 3e/3d targets in MB-231s in normoxia with a209 in MCF7SIX1s", palette = c(si3d_col, si3e_col), ylim=c(0,6))+
  geom_hline(aes(yintercept=1))+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))



ggsave(here("plots/plots_231/hypoverlap.pdf"), width=7, height=7)

```



# Venn diagrams

```{r}

all_lists <- bind_rows(mcf7six1_si3e_normoxia_up, mcf7six1_si3d_normoxia_up, a209_normoxia_up, 
                       mcf7six1_si3e_normoxia_down, mcf7six1_si3d_normoxia_down, a209_normoxia_down, 
                       mcf7six1_si3e_hypoxia_up, mcf7six1_si3d_hypoxia_up, a209_hypoxia_up, 
                       mcf7six1_si3e_hypoxia_down, mcf7six1_si3d_hypoxia_down, a209_hypoxia_down,
                       mb231_si3e_normoxia_up, mb231_si3d_normoxia_up, 
                       mb231_si3e_normoxia_down, mb231_si3d_normoxia_down, 
                       mb231_si3e_hypoxia_up, mb231_si3d_hypoxia_up, 
                       mb231_si3e_hypoxia_down, mb231_si3d_hypoxia_down) %>% 
extract(gs_name,
        into = c("cell_line", "treatment", "condition", "direction"),
        regex = "([^_]+)_([^_]+)_([^_]+)_([^_]+)")

```

Normoxia
## Venn diagram of 3e/d gene overlaps in normoxia
```{r compare genes}
mcf7six1_si3e_normoxia_up <- mcf7six1_si3e_normoxia_up %>% pull(symbol)
mcf7six1_si3d_normoxia_up <- mcf7six1_si3d_normoxia_up %>% pull(symbol)

mcf7six1_si3e_normoxia_down <- mcf7six1_si3e_normoxia_down %>% pull(symbol)
mcf7six1_si3d_normoxia_down <- mcf7six1_si3d_normoxia_down %>% pull(symbol)

mb231_si3e_normoxia_up <- mb231_si3e_normoxia_up %>% pull(symbol)
mb231_si3d_normoxia_up <- mb231_si3d_normoxia_up %>% pull(symbol)

mb231_si3e_normoxia_down <- mb231_si3e_normoxia_down %>% pull(symbol)
mb231_si3d_normoxia_down <- mb231_si3d_normoxia_down %>% pull(symbol)

#3e

#up

norm_te_up_3e <- list(
  MCF7SIX1 = mcf7six1_si3e_normoxia_up, 
  MDA_MB_231 = mb231_si3e_normoxia_up
)

ggvenn(
  norm_te_up_3e,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3e_col, "blue")
) +
  ggtitle("Up with 3e KD - Normoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3e_norm_vennup.pdf"))


#down

norm_te_down_3e <- list(
  MCF7SIX1 = mcf7six1_si3e_normoxia_down, 
  MDA_MB_231 = mb231_si3e_normoxia_down
)

ggvenn(
  norm_te_down_3e,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3e_col, "blue")
) +
  ggtitle("down with 3e KD - Normoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3e_norm_venndown.pdf"))






#3d

#up

norm_te_up_3d <- list(
  MCF7SIX1 = mcf7six1_si3d_normoxia_up, 
  MDA_MB_231 = mb231_si3d_normoxia_up
)

ggvenn(
  norm_te_up_3d,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3d_col, "red"))+
  ggtitle("Up with 3d KD - Normoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3d_norm_vennup.pdf"))


#down

norm_te_down_3d <- list(
  MCF7SIX1 = mcf7six1_si3d_normoxia_down, 
  MDA_MB_231 = mb231_si3d_normoxia_down
)

ggvenn(
  norm_te_down_3d,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3d_col, "red")
) +
  ggtitle("down with 3d KD - Normoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3d_norm_venndown.pdf"))


```

## Venn diagram of 3e/d gene overlaps in hypoxia
```{r compare genes}
mcf7six1_si3e_hypoxia_up <- mcf7six1_si3e_hypoxia_up %>% pull(symbol)
mcf7six1_si3d_hypoxia_up <- mcf7six1_si3d_hypoxia_up %>% pull(symbol)

mcf7six1_si3e_hypoxia_down <- mcf7six1_si3e_hypoxia_down %>% pull(symbol)
mcf7six1_si3d_hypoxia_down <- mcf7six1_si3d_hypoxia_down %>% pull(symbol)

mb231_si3e_hypoxia_up <- mb231_si3e_hypoxia_up %>% pull(symbol)
mb231_si3d_hypoxia_up <- mb231_si3d_hypoxia_up %>% pull(symbol)

mb231_si3e_hypoxia_down <- mb231_si3e_hypoxia_down %>% pull(symbol)
mb231_si3d_hypoxia_down <- mb231_si3d_hypoxia_down %>% pull(symbol)

#3e

#up

hyp_te_up_3e <- list(
  MCF7SIX1 = mcf7six1_si3e_hypoxia_up, 
  MDA_MB_231 = mb231_si3e_hypoxia_up
)

intersect_up_3e <- intersect(mcf7six1_si3e_hypoxia_up,mb231_si3e_hypoxia_up)

ggvenn(
  hyp_te_up_3e,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3e_col, "blue")
) +
  ggtitle("Up with 3e KD - hypoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )


ggsave(width = 6, height = 6, filename = here("plots/plots_231/3e_hyp_vennup.pdf"))


#down

hyp_te_down_3e <- list(
  MCF7SIX1 = mcf7six1_si3e_hypoxia_down, 
  MDA_MB_231 = mb231_si3e_hypoxia_down
)

ggvenn(
  hyp_te_down_3e,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3e_col, "blue")
) +
  ggtitle("down with 3e KD - hypoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

intersect_down_3e <- intersect(mcf7six1_si3e_hypoxia_down,mb231_si3e_hypoxia_down)


ggsave(width = 6, height = 6, filename = here("plots/plots_231/3e_hyp_venndown.pdf"))






#3d

#up

hyp_te_up_3d <- list(
  MCF7SIX1 = mcf7six1_si3d_hypoxia_up, 
  MDA_MB_231 = mb231_si3d_hypoxia_up
)

ggvenn(
  hyp_te_up_3d,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3d_col, "red"))+
  ggtitle("Up with 3d KD - hypoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3d_hyp_vennup.pdf"))

intersect_up_3d <- intersect(mcf7six1_si3d_hypoxia_up,mb231_si3d_hypoxia_up)


#down

hyp_te_down_3d <- list(
  MCF7SIX1 = mcf7six1_si3d_hypoxia_down, 
  MDA_MB_231 = mb231_si3d_hypoxia_down
)

ggvenn(
  hyp_te_down_3d,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3d_col, "red")
) +
  ggtitle("down with 3d KD - hypoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3d_hyp_venndown.pdf"))
intersect_down_3d <- intersect(mcf7six1_si3d_hypoxia_down,mb231_si3d_hypoxia_down)


```


# Export major lists for hypoxia
```{r}
res_231_3d<-
res_hypox_si3d %>%
  dplyr::select(symbol, log2FoldChange, padj) %>% 
  rename("LFC_231_3d" = "log2FoldChange") %>% 
    rename("padj_231_3d" = "padj")

res_231_3e<-
res_hypox_si3e %>%
  dplyr::select(symbol, log2FoldChange, padj) %>% 
  rename("LFC_231_3e" = "log2FoldChange") %>% 
    rename("padj_231_3e" = "padj")


res_mcf7six1_3d<-
mcf7six1_res_hypox_si3d %>%
  dplyr::select(symbol, log2FoldChange, padj) %>% 
  rename("LFC_mcf7six1_3d" = "log2FoldChange") %>% 
    rename("padj_mcf7six1_3d" = "padj")

res_mcf7six1_3e<-
mcf7six1_res_hypox_si3d %>%
  dplyr::select(symbol, log2FoldChange, padj) %>% 
  rename("LFC_mcf7six1_3e" = "log2FoldChange") %>% 
    rename("padj_mcf7six1_3e" = "padj")



all_hypox_KD <- 
  res_231_3d %>% 
  left_join(res_231_3e) %>% 
    left_join(res_mcf7six1_3d) %>% 
  left_join(res_mcf7six1_3e) %>% 
  mutate(intersect_up_3eKD = case_when(
    symbol %in% intersect_up_3e ~ "intersect",
    .default = "not_intersect"
  )) %>% 
  mutate(intersect_down_3eKD = case_when(
    symbol %in% intersect_down_3e ~ "intersect",
    .default = "not_intersect"
  )) %>% 
    mutate(intersect_up_3dKD = case_when(
    symbol %in% intersect_up_3d ~ "intersect",
    .default = "not_intersect"
  )) %>% 
  mutate(intersect_down_3dKD = case_when(
    symbol %in% intersect_down_3d ~ "intersect",
    .default = "not_intersect"
  ))


write_xlsx(all_hypox_KD, here("counts/counts_231/all_hypoxia_intersect.xlsx"))


```





# Pathway analysis


## Lists for pathway analysis
Separate lists by direction of change -- delete? -- add back depending on whether we want that scatter plot in the supplement
```{r}

hyp_3d_up <- res_hypox_si3d %>%
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

hyp_3d_down <- res_hypox_si3d %>%
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)

hyp_3e_up <- res_hypox_si3e %>%
  filter(padj < 0.05 & log2FoldChange > 0) %>% pull(gene_id)

hyp_3e_down <- res_hypox_si3e %>%
  filter(padj < 0.05 & log2FoldChange < 0) %>% pull(gene_id)


```


## Pathway analysis gene list setup
```{r prep GSEA analysis}

#3d lists

eif3d_hyp <- res_hypox_si3d$log2FoldChange
names(eif3d_hyp) <- res_hypox_si3d$symbol

eif3d_hyp <- sort(eif3d_hyp, decreasing = TRUE)

eif3d_hyp <- eif3d_hyp[!duplicated(names(eif3d_hyp))]

#3e lists

eif3e_hyp <- res_hypox_si3e$log2FoldChange
names(eif3e_hyp) <- res_hypox_si3e$symbol

eif3e_hyp <- sort(eif3e_hyp, decreasing = TRUE)

eif3e_hyp <- eif3e_hyp[!duplicated(names(eif3e_hyp))]


```

### Hallmark and C2 pathway setup
```{r}
m_df_h <- msigdbr(species = "Homo sapiens") %>%
  filter(gs_cat == "H") %>% 
  dplyr::select(gs_name, gene_symbol)

m_df_c2 <- msigdbr(species = "Homo sapiens") %>%
  filter(gs_cat == "C2" & gs_subcat != "CGP") %>% 
  dplyr::select(gs_name, gene_symbol)



m_df_c2_curated <- msigdbr(species = "Homo sapiens") %>%
  filter(gs_cat == "C2") %>% 
  dplyr::select(gs_name, gene_symbol)
```


Hypoxia
```{r}
eif3d_te_h <- GSEA(
  geneList = eif3d_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )

eif3e_te_h <- GSEA(
  geneList = eif3e_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )



hyp_te_h_3d <-
  eif3d_te_h@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) %>% 
    mutate(sample="eIF3d")


hyp_te_h_3e <- 
  eif3e_te_h@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) %>% 
  mutate(sample="eIF3e")



hyp_hallmark <- rbind(hyp_te_h_3d, hyp_te_h_3e) %>% 
  mutate(significant = case_when(
    p.adjust < 0.05 ~"*",
    .default=""
  ))

pathways_to_include <- hyp_hallmark %>% 
  filter(significant=="*")

pathways_to_include <- pathways_to_include$ID

hyp_hallmark %>%
  filter(ID %in% pathways_to_include) %>%
  ggplot(aes(x = ID, y = NES, fill=sample)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.9)) +
  geom_text(aes(label = significant), 
            position = position_dodge(width = 0.9), 
            hjust = -0.2) +  # Adjust hjust for better alignment
  # theme_cowplot() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
  scale_fill_manual(values=c(si3d_col, si3e_col))+
  coord_flip()+
  theme_cowplot()+
  ggtitle("Pathways enriched with KD - Hypoxia")

ggsave(here("plots/plots_231/pathway_bars_hypoxia.pdf"))






```

Hypoxia C2
```{r}
eif3d_te_c2 <- GSEA(
  geneList = eif3d_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2
  )

eif3e_te_c2 <- GSEA(
  geneList = eif3e_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2
  )



hyp_te_c2_3d <-
  eif3d_te_c2@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) %>% 
    mutate(sample="eIF3d")


hyp_te_c2_3e <- 
  eif3e_te_c2@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) %>% 
  mutate(sample="eIF3e")



hyp_c2 <- rbind(hyp_te_c2_3d, hyp_te_c2_3e) %>% 
  mutate(significant = case_when(
    p.adjust < 0.05 ~"*",
    .default=""
  ))

pathways_to_include <- hyp_c2 %>%
  filter(significant=="*")

pathways_to_include <- pathways_to_include$ID

hyp_c2 %>%
  filter(ID %in% pathways_to_include) %>%
  ggplot(aes(x = ID, y = NES, fill=sample)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.9)) +
  geom_text(aes(label = significant), 
            position = position_dodge(width = 0.9), 
            hjust = -0.2) +  # Adjust hjust for better alignment
  # theme_cowplot() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
  scale_fill_manual(values=c(si3d_col, si3e_col))+
  coord_flip()+
  theme_cowplot()+
  ggtitle("Pathways enriched with KD - Hypoxia")

ggsave(here("plots/plots_231/pathway_bars_hypoxia_c2.pdf"))


hyp_c2_sig <- hyp_c2 %>% 
  filter(significant=="*")




```

make table
```{r}
hyp_te_c2_3d_indiv <-
  eif3d_te_c2@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) %>% 
  rename("NES_3d" = "NES", "p.adjust_3d"="p.adjust", "core_enrichment_3d" = "core_enrichment")


hyp_te_c2_3e_indiv <- 
  eif3e_te_c2@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) %>% 
    rename("NES_3e" = "NES", "p.adjust_3e"="p.adjust", "core_enrichment_3e" = "core_enrichment")

colnames(hyp_te_c2_3e_indiv)


hyp_c2 <- left_join(hyp_te_c2_3d_indiv, hyp_te_c2_3e_indiv)


hyp_c2 <- rbind(hyp_te_c2_3d, hyp_te_c2_3e) %>% 
  mutate(significant = case_when(
    p.adjust < 0.05 ~"*",
    .default=""
  ))

pathways_to_include <- hyp_c2 %>%
  filter(significant=="*")
```


Hypoxia C2

```{r}
eif3d_te_c2 <- GSEA(
  geneList = eif3d_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2_curated
  )

eif3e_te_c2 <- GSEA(
  geneList = eif3e_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2_curated
  )



hyp_te_c2_3d <-
  eif3d_te_c2@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) %>% 
    mutate(sample="eIF3d")


hyp_te_c2_3e <- 
  eif3e_te_c2@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) %>% 
  mutate(sample="eIF3e")



hyp_c2 <- rbind(hyp_te_c2_3d, hyp_te_c2_3e) %>% 
  mutate(significant = case_when(
    p.adjust < 0.05 ~"*",
    .default=""
  ))

pathways_to_include <- hyp_c2 %>%
  filter(significant=="*")

pathways_to_include <- pathways_to_include$ID

hyp_c2 %>%
  filter(ID %in% pathways_to_include) %>%
  ggplot(aes(x = ID, y = NES, fill=sample)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.9)) +
  geom_text(aes(label = significant), 
            position = position_dodge(width = 0.9), 
            hjust = -0.2) +  # Adjust hjust for better alignment
  # theme_cowplot() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
  scale_fill_manual(values=c(si3d_col, si3e_col))+
  coord_flip()+
  theme_cowplot()+
  ggtitle("Pathways enriched with KD - Hypoxia")

ggsave(here("plots/plots_231/pathway_bars_hypoxia_c2.pdf"))


hyp_c2_sig <- hyp_c2 %>% 
  filter(significant=="*")




```

make table
```{r}
hyp_te_c2_3d_indiv <-
  eif3d_te_c2@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) %>% 
  rename("NES_3d" = "NES", "p.adjust_3d"="p.adjust", "core_enrichment_3d" = "core_enrichment")


hyp_te_c2_3e_indiv <- 
  eif3e_te_c2@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) %>% 
    rename("NES_3e" = "NES", "p.adjust_3e"="p.adjust", "core_enrichment_3e" = "core_enrichment")

colnames(hyp_te_c2_3e_indiv)


hyp_c2 <- left_join(hyp_te_c2_3d_indiv, hyp_te_c2_3e_indiv)


hyp_c2 <- rbind(hyp_te_c2_3d, hyp_te_c2_3e) %>% 
  mutate(significant = case_when(
    p.adjust < 0.05 ~"*",
    .default=""
  ))

pathways_to_include <- hyp_c2 %>%
  filter(significant=="*")


write_xlsx(hyp_c2, here("counts/counts_231/MB231_hypoxia_c2_all.xlsx"))


#invasion pathways
hyp_c2 <- rbind(hyp_te_c2_3d, hyp_te_c2_3e) %>% 
  mutate(significant = case_when(
    p.adjust < 0.05 ~"*",
    .default=""
  )) %>% 
  filter(str_detect(ID, "INVA"))


write_xlsx(hyp_c2, here("counts/counts_231/MB231_hypoxia_c2curated_invasion.xlsx"))
```





Hypoxia kegg
```{r GSEA KEGG}
eif3d_te_kegg <- GSEA(
  geneList = eif3d_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^KEGG', gs_name))
  )

eif3e_te_kegg <- GSEA(
  geneList = eif3e_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^KEGG', gs_name))
  )

hyp_te_kegg_3d <-
  eif3d_te_kegg@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) %>% 
    mutate(sample="eIF3d")

hyp_te_kegg_3e <- 
  eif3e_te_kegg@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) %>% 
      mutate(sample="eIF3e")






hyp_kegg <- rbind(hyp_te_kegg_3d, hyp_te_kegg_3e) %>% 
  mutate(significant = case_when(
    p.adjust < 0.05 ~"*",
    .default=""
  ))

pathways_to_include <- hyp_kegg %>% 
  filter(significant=="*")

pathways_to_include <- pathways_to_include$ID

hyp_kegg %>%
  filter(ID %in% pathways_to_include) %>%
  ggplot(aes(x = ID, y = NES, fill=sample)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.9)) +
  geom_text(aes(label = significant), 
            position = position_dodge(width = 0.9), 
            hjust = -0.2) +  # Adjust hjust for better alignment
  # theme_cowplot() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
  scale_fill_manual(values=c(si3d_col, si3e_col))+
  coord_flip()+
  theme_cowplot()+
  ggtitle("Pathways enriched with KD - hypoxia")

ggsave(here("plots/plots_231/pathway_bars_hypoxia_kegg.pdf"))





```



### Hallmark pathway enrichment analysis

```{r GSEA Hallmarks}

eif3d_te_h <- GSEA(
  geneList = eif3d_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )


eif3e_te_h <- GSEA(
  geneList = eif3e_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )


hyp_te_h <- dplyr::full_join(
  eif3d_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3d_", .), is.numeric),
  eif3e_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3d_pvalue < 0.05 | eif3e_pvalue < 0.05)  %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = eif3e_NES - eif3d_NES)


hyp_te_h$ID <- gsub(pattern = "HALLMARK_", replacement = "", x = hyp_te_h$ID)

hyp_te_h <- hyp_te_h %>% 
  mutate(color= ifelse(ID=="HYPOXIA", "red", "black"))


ggscatter(data = hyp_te_h,
          x = "eif3d_NES",
          y = "eif3e_NES",
          title = "Hallmark pathways - Hypoxia",
          label = "ID",
          repel = TRUE,
          font.label=7,
  color="color",
  palette=c("black", "red"))+
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5,2.5) +
  xlim(-2.5,2.5) + theme_prism()+
  guides(color = FALSE)

DT::datatable(hyp_te_h) %>%
  formatRound(which(sapply(hyp_te_h,is.numeric)), digits = 3)



ggsave(width = 4, height = 4, filename = here("plots/fig3/hypoxia_halmarks.pdf"))


hyp_h_table <- hyp_te_h %>% 
  mutate(direction = ifelse(eif3d_NES < 0 & eif3e_NES < 0, "down_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES > 0, "up_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES < 0, "up_3d_down_3e",
                            ifelse(eif3d_NES < 0 & eif3e_NES > 0, "up_3e_down_3d", "NA")))))

write_csv(hyp_h_table, here("counts/fig3/hyp_hallmarks_table.csv"))





```

Visualize a different way -- with bars

```{r}

hyp_te_h_3d <-
  eif3d_te_h@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>% 
  mutate(sample="eIF3d")

hyp_te_h_3e <-
  eif3e_te_h@result %>% 
  dplyr::select(ID,NES,pvalue,qvalue) %>% 
  mutate(sample="eIF3e")

hyp_hallmark <- rbind(hyp_te_h_3d, hyp_te_h_3e) %>% 
  mutate(significant = case_when(
    qvalue < 0.05 ~"*",
    .default=""
  ))

pathways_to_include <- hyp_hallmark %>% 
  filter(significant=="*")

pathways_to_include <- pathways_to_include$ID

hyp_hallmark %>%
  filter(ID %in% pathways_to_include) %>%
  ggplot(aes(x = ID, y = NES, fill = sample)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.9)) +
  geom_text(aes(label = significant), 
            position = position_dodge(width = 0.9), 
            hjust = -0.2) +  # Adjust hjust for better alignment
  # theme_cowplot() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
  coord_flip()


ggsave(here("plots/fig3/hallmark_bars.pdf"), width=8, height = 6)
```


### KEGG

```{r GSEA KEGG}
eif3d_te_kegg <- GSEA(
  geneList = eif3d_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^KEGG', gs_name))
  )

eif3e_te_kegg <- GSEA(
  geneList = eif3e_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^KEGG', gs_name))
  )

hyp_te_kegg <- dplyr::full_join(
  eif3d_te_kegg@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3d_", .), is.numeric),
  eif3e_te_kegg@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3d_qvalue < 0.05 | eif3e_qvalue < 0.05) %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = eif3e_NES - eif3d_NES)

hyp_te_kegg$ID <- gsub(pattern = "KEGG_", replacement = "", x = hyp_te_kegg$ID)

ggscatter(data = hyp_te_kegg,
          x = "eif3d_NES",
          y = "eif3e_NES",
          label = "ID",
          title = "KEGG pathways - hypoxia",
          repel = TRUE,
          font.label = 6) +
  scale_color_continuous(type = "viridis") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5,2.5) +
  xlim(-2.5,2.5) +
  theme_prism()


#DT::datatable(hyp_te_kegg) %>%
 # formatRound(which(sapply(hyp_te_kegg,is.numeric)), digits = 3) 




ggsave(width = 4, height = 4, filename = here("plots/fig3/hypoxia_kegg.pdf"))


hyp_kegg_table <- hyp_te_kegg %>% 
  mutate(direction = ifelse(eif3d_NES < 0 & eif3e_NES < 0, "down_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES > 0, "up_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES < 0, "up_3d_down_3e",
                            ifelse(eif3d_NES < 0 & eif3e_NES > 0, "up_3e_down_3d", "NA")))))

write_csv(hyp_kegg_table, here("counts/fig3/hyp_kegg_table.csv"))


```


### REACTOME

```{r GSEA REACTOME}
eif3d_te_reactome <- GSEA(
  geneList = eif3d_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^REACTOME', gs_name))
  )

eif3e_te_reactome <- GSEA(
  geneList = eif3e_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^REACTOME', gs_name))
  )

hyp_te_react <- dplyr::full_join(
  eif3d_te_reactome@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3d_", .), is.numeric),
  eif3e_te_reactome@result %>%
  dplyr::select(ID,NES,pvalue,qvalue) %>%
  rename_with(~ paste0("eif3e_", .), is.numeric),
  by = "ID") %>% 
  filter(eif3d_qvalue < 0.05 | eif3e_qvalue < 0.05)  %>%
  dplyr::arrange(-eif3e_NES) %>%
  mutate(delta = eif3e_NES - eif3d_NES)

hyp_te_react$ID <- gsub(pattern = "REACTOME_", replacement = "", x = hyp_te_react$ID)

ggscatter(data = hyp_te_react,
          x = "eif3d_NES",
          y = "eif3e_NES",
          label = "ID",
          repel = TRUE,
          title = "Reactome pathways - hypoxia",
          font.label = 6) +
  scale_color_continuous(type = "viridis") +
  geom_vline(xintercept = 0, linetype = "dashed") +
  geom_hline(yintercept = 0, linetype = "dashed") +
  ylim(-2.5,2.5) +
  xlim(-2.5,2.5) +
  theme_prism()


# DT::datatable(hyp_te_react) %>%
#   formatRound(which(sapply(hyp_te_react,is.numeric)), digits = 3) 



ggsave(width = 4, height = 4, filename = here("plots/fig3/hypoxia_reactome.pdf"))


hyp_react_table <- hyp_te_react %>% 
  mutate(direction = ifelse(eif3d_NES < 0 & eif3e_NES < 0, "down_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES > 0, "up_both",
                            ifelse(eif3d_NES > 0 & eif3e_NES < 0, "up_3d_down_3e",
                            ifelse(eif3d_NES < 0 & eif3e_NES > 0, "up_3e_down_3d", "NA")))))

write_csv(hyp_react_table, here("counts/fig3/hyp_react_table.csv"))
```


# Patient survival data ---------------------------------------------------------

Create RNA signatures
Confirm direction -- opposite of previous

# 3e hypoxia rna
```{r}

metadata_loop <- metadata %>% 
  filter(library=="rna") %>% 
  filter(condition=="hypoxia") %>% 
  filter(treatment=="sictrl"|treatment=="si3e")

metadata_loop$treatment <- factor(metadata_loop$treatment, levels= c("si3e", "sictrl"))
levels(metadata_loop$treatment)

sample_info <- metadata_loop %>% 
    mutate("name" = 
  paste(condition, treatment, rep, library, sep = "_"))
  
  names <- as.character(sample_info$name)
  names
  
  rna_subset <- all_raw[, colnames(all_raw) %in% names]
  
  print(levels(sample_info$treatment))

  
# Deseq analysis ----------------------------------------------------------------
  
  dds <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  dds <- DESeq(dds)
  
  print(resultsNames(dds))

  res <- results(dds, name=resultsNames(dds)[2], cooksCutoff=FALSE, independentFiltering=FALSE)

  res_hyp_3e <- lfcShrink(dds = dds, coef = resultsNames(dds)[2], res = res) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)

# export to excel ---------------------------------------------------------------
  
   writexl::write_xlsx(x = res_hyp_3e, path = here("counts/counts_231/rna_signature_3e_hypoxia_231.xlsx"), col_names = T)

#confirm direction
  res_hyp_3e %>% 
    filter(symbol=="EIF3E")

```

# 3d hypoxia rna
```{r}

metadata_loop <- metadata %>% 
  filter(library=="rna") %>% 
  filter(condition=="hypoxia") %>% 
  filter(treatment=="sictrl"|treatment=="si3d")

metadata_loop$treatment <- factor(metadata_loop$treatment, levels= c("si3d", "sictrl"))
levels(metadata_loop$treatment)

sample_info <- metadata_loop %>% 
    mutate("name" = 
  paste(condition, treatment, rep, library, sep = "_"))
  
  names <- as.character(sample_info$name)
  names
  
  rna_subset <- all_raw[, colnames(all_raw) %in% names]
  
  print(levels(sample_info$treatment))

  
# Deseq analysis ----------------------------------------------------------------
  
  dds <- DESeqDataSetFromMatrix(countData = rna_subset, 
                                 colData = sample_info, 
                                 design = ~treatment)
  dds <- DESeq(dds)
  
  print(resultsNames(dds))

  res <- results(dds, name=resultsNames(dds)[2], cooksCutoff=FALSE, independentFiltering=FALSE)

  res_hyp_3d <- lfcShrink(dds = dds, coef = resultsNames(dds)[2], res = res) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)

# export to excel ---------------------------------------------------------------
  
   writexl::write_xlsx(x = res_hyp_3d, path = here("counts/counts_231/rna_signature_3d_hypoxia_231.xlsx"), col_names = T)

#confirm direction
  res_hyp_3d %>% 
    filter(symbol=="EIF3D")

```


# Prepare survival analysis
### Separate RNA signatures by direction

```{r}
# 3e hypoxia

sig_rna_3e_hyp <- res_hyp_3e %>% 
  filter(padj < 0.05)

up_3e_hyp <- sig_rna_3e_hyp %>% 
  filter(log2FoldChange > 1)%>% 
  pull(symbol)

down_3e_hyp <- sig_rna_3e_hyp %>% 
  filter(log2FoldChange < -1)%>% 
  pull(symbol)
  
```

```{r}
# 3d hypoxia

sig_rna_3d_hyp <- res_hyp_3d %>% 
  filter(padj < 0.05)

up_3d_hyp <- sig_rna_3d_hyp %>% 
  filter(log2FoldChange > 1) %>% 
  pull(symbol)
  

down_3d_hyp <- sig_rna_3d_hyp %>% 
  filter(log2FoldChange < -1)%>% 
  pull(symbol)
  

```


# Import METABRIC microarray data from cbioportal - read patient z scores and matching clinical data
Nature 2012 and Nat Commun 2016

```{r cache = TRUE}
zscores <- read.table(here("cbioportal_data/brca_metabric/data_mrna_illumina_microarray_zscores_ref_diploid_samples.txt"), sep="\t", header=T)
row.names(zscores) <- gsub("_", ".",row.names(zscores))

zscores_all_symbol <- zscores %>% 
  dplyr::select(-Entrez_Gene_Id)

### Average scores for duplicate patients

dup_genes <- zscores$Hugo_Symbol[duplicated(zscores$Hugo_Symbol)]

dedup <- zscores %>%
  filter(Hugo_Symbol %in% dup_genes) %>%
  group_by(Hugo_Symbol) %>%
  summarise(across(everything(), mean))


merged_zscores <- zscores %>%
  filter(!Hugo_Symbol %in% dup_genes) %>% 
  bind_rows(., dedup) %>% 
  as.tibble() %>%
  column_to_rownames("Hugo_Symbol") %>% 
  as.data.frame()


```

# eIF3e hypoxia signature and survival prediction
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3e <- merged_zscores %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% sig_rna_3e_hyp$symbol) %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  dplyr::select(-Entrez_Gene_Id)

# empty df
sig_zscores_3e <- data.frame(patient=colnames(exp_3e),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3e)) {
  sig_zscores_3e[i,2] <- mean(exp_3e[up_3e_hyp,i],  na.rm=TRUE)
  sig_zscores_3e[i,3] <- mean(exp_3e[down_3e_hyp,i],  na.rm=TRUE)
  }

sig_zscores_3e <- sig_zscores_3e %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3e$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3e$quantile <- cut(sig_zscores_3e$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_3e$patient <- sub("-[^-]*$", "", sig_zscores_3e$patient)

sig_zscores_3e$patient <- gsub("\\.", "-", sig_zscores_3e$patient)


```


# eIF3d hypoxia signature and survival prediction
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3d <- merged_zscores %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% sig_rna_3d_hyp$symbol) %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  dplyr::select(-Entrez_Gene_Id)

# empty df
sig_zscores_3d <- data.frame(patient=colnames(exp_3d),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3d)) {
  sig_zscores_3d[i,2] <- mean(exp_3d[up_3d_hyp,i],  na.rm=TRUE)
  sig_zscores_3d[i,3] <- mean(exp_3d[down_3d_hyp,i],  na.rm=TRUE)
  }

sig_zscores_3d <- sig_zscores_3d %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3d$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3d$quantile <- cut(sig_zscores_3d$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)


sig_zscores_3d$patient <- gsub("\\.", "-", sig_zscores_3d$patient)

```


# Import clinical data
```{r}
clin <- read_excel(here("cbioportal_data/brca_metabric/data_clinical_patient_edit.xlsx"))
clin$OS_MONTHS <- as.numeric(clin$OS_MONTHS)
clin$OS_STATUS[which(clin$OS_STATUS == "1:DECEASED" & clin$OS_MONTHS > 120)] <- "0:LIVING"
clin$OS_STATUS <- clin$OS_STATUS == "1:DECEASED"
clin <- cbind(clin, OS_YEARS = clin$OS_MONTHS/12)
clin$OS_YEARS[which(clin$OS_YEARS >= 10)] <- 10

```

### visualizing high and low quartile in each survival plot for the paper.
## 3e survival
```{r}
#bind clin data with z score data

all_data_3e_hyp <- left_join(clin, sig_zscores_3e, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA")

all_data_3e_hyp <- all_data_3e_hyp %>% 
  filter(quantile == 1 | quantile == 4)

fit_all_3e_hyp <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_3e_hyp)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_3e_hyp)
summary_coxfit_3e_norm <- summary(coxfit)


hr <- round(summary_coxfit_3e_norm$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_3e_norm$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

#3e_colors
colors <- grDevices::colorRampPalette(c("grey", si3e_col))(2)


surv_plot_3e_hyp<-
ggsurvplot(fit_all_3e_hyp,
           data = all_data_3e_hyp,
           size = 1,
           palette =colors, 
          legend.labs=c("Low 3e", "High 3e"),
         #  legend.labs=c("Low 3e", "Med-Low 3e", "Med-High 3e", "High 3e"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3e_hyp$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("3e_hyp_surv_plot_231.pdf", path=here("plots/plots_231"),width = 11.5, height=8)


```

## 3d survival
```{r}
#bind clin data with z score data

all_data_3d_hyp <- left_join(clin, sig_zscores_3d, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA")

all_data_3d_hyp <- all_data_3d_hyp %>% 
  filter(quantile == 1 | quantile == 4)

fit_all_3d_hyp <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_3d_hyp)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_3d_hyp)
summary_coxfit_3d_norm <- summary(coxfit)


hr <- round(summary_coxfit_3d_norm$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_3d_norm$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", si3d_col))(2)

surv_plot_3d_hyp<-
ggsurvplot(fit_all_3d_hyp,
           data = all_data_3d_hyp,
           size = 1,
           palette = colors, 
          legend.labs=c("Low 3d", "High 3d"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3d_hyp$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("3d_hyp_surv_plot_231.pdf", path=here("plots/plots_231"),width = 11.5, height=8)


```


# TCGA

## Import hypoxia signature from ye et al
Hypoxia signature in breast cancer:
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6279594/#:~:text=The%2042%2Dgene%20expression%20signature,mediate%20the%20conserved%20hypoxic%20response


```{r}
ye_hyp_sig <- read_excel(here("other_data/hypoxic_sig_ye.xlsx"))
#up values only

#calculate z scores for enrichment of hypoxia signature

# filter merged z score data based on genes in my signature
exp_hyp_sig <- merged_zscores %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% ye_hyp_sig$up_in_hypoxia) %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  dplyr::select(-Entrez_Gene_Id)

# empty df
sig_zscores_hyp_signature <- data.frame(patient=colnames(exp_hyp_sig),
                         z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_hyp_signature)) {
  sig_zscores_hyp_signature[i,2] <- mean(exp_hyp_sig[,i],  na.rm=TRUE)
}


quantile_groups <- quantile(sig_zscores_hyp_signature$z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_hyp_signature$quantile <- cut(sig_zscores_hyp_signature$z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_hyp_signature$patient <- sub("-[^-]*$", "", sig_zscores_hyp_signature$patient)

sig_zscores_hyp_signature$patient <- gsub("\\.", "-", sig_zscores_hyp_signature$patient)


#Bind z score hypoxia signature with clin data


all_data_hif_ye <- left_join(clin, sig_zscores_hyp_signature, by=c("PATIENT_ID"="patient"))

all_data_hif_ye <- all_data_hif_ye %>% 
  filter(quantile == 1 | quantile == 4)

#plot

fit_all_data_hif_ye <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_hif_ye)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_hif_ye)
summary_coxfit_ye <- summary(coxfit)

exp(summary_coxfit_ye$coefficients[1])

hr <- round(summary_coxfit_ye$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_ye$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", "red"))(2)

surv_plot_ye_hyp<-
ggsurvplot(fit_all_data_hif_ye,
           data = all_data_hif_ye,
           size = 1,
           palette = colors, 
           break.time.by = 1,
          legend.labs=c("Low Hypoxia", "High Hypoxia"),
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_ye_hyp$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("surv_plot_ye_hyp.pdf", path=here("plots/fig8"),width = 11.5, height=8)


#boxplot of quartile expression

ggplot(all_data_hif_ye, aes(x=quantile,y=z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3d hypoxia signature")

```


## correlation of 3e and hypoxia score

```{r}

patient_ye_and_3e <- left_join(sig_zscores_3e, sig_zscores_hyp_signature, by="patient") %>% 
  dplyr::rename("hypoxia_z_score"="z_score") %>% 
   dplyr::rename("eIF3e_z_score"="total_z_score")


```

## Divide patients into 3e sig high/low and hypoxia signature high/low

Make four groups based on quandrants -- 3e levels in hypoxia
```{r}

patient_ye_and_3e <- patient_ye_and_3e %>% 
  mutate(quadrant = ifelse(hypoxia_z_score < 0 & eIF3e_z_score < 0, "low hypoxia and low 3e",
                           ifelse(hypoxia_z_score > 0 & eIF3e_z_score < 0, "high hypoxia and low 3e",
                                  ifelse(hypoxia_z_score > 0 & eIF3e_z_score > 0, "high hypoxia and high 3e",
                                         ifelse(hypoxia_z_score < 0 & eIF3e_z_score > 0, "low hypoxia and high 3e", "other")))))


patient_ye_and_3e$quadrant <- factor(patient_ye_and_3e$quadrant, levels=c("low hypoxia and low 3e","high hypoxia and low 3e","low hypoxia and high 3e", "high hypoxia and high 3e"))


```

Determine correlation and color by group
```{r}
ggplot(patient_ye_and_3e, aes(x=hypoxia_z_score, y= eIF3e_z_score, color=quadrant))+
  geom_point()+
  xlab("Ye hypoxia signature z score")+
  ylab("eIF3e-hyp z score")+
  theme_prism()+ 
  theme(text = element_text(size = 14))+
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0)+
  scale_color_manual(values=c("high hypoxia and high 3e"="#9B4D96", 
                       "high hypoxia and low 3e"="red", 
                       "low hypoxia and low 3e"="grey", 
                       "low hypoxia and high 3e"= "#56B4E9"))


ggsave("3e_hyp_and_ye.pdf", path=here("plots/fig8"),width = 5.5, height=3)

cor.test(patient_ye_and_3e$hypoxia_z_score, patient_ye_and_3e$eIF3e_z_score, method="pearson")
```

Plot survival curve
```{r}
#bind with clin data
patient_ye_and_3e_hyp_clin <- left_join(clin, patient_ye_and_3e, by=c("PATIENT_ID"="patient"))

#plot

fit_all_hypoxia_3e <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = patient_ye_and_3e_hyp_clin)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, patient_ye_and_3e_hyp_clin)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)


levels(patient_ye_and_3e_hyp_clin$quadrant)

surv_plot_3e_and_hypoxia_sig<-
ggsurvplot(fit_all_hypoxia_3e,
           data = patient_ye_and_3e_hyp_clin,
           size = 1,
           palette = c("grey", "red", si3e_col, "#9B4D96"), 
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 7,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3e_and_hypoxia_sig$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("surv_plot_3e_and_hypoxia_sig.pdf", path=here("plots/plots_231"),width = 11.5, height=8)

#compare middle two quadrants

levels(patient_ye_and_3e_hyp_clin$quadrant)

# Relevel the 'quadrant' factor to set the desired reference group
patient_ye_and_3e_hyp_clin$quadrant <- relevel(patient_ye_and_3e_hyp_clin$quadrant, ref = "high hypoxia and low 3e")

# Fit the Cox model
coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = patient_ye_and_3e_hyp_clin)
summary_coxfit <- summary(coxfit)
print(summary_coxfit)



```

## Metabric data controls:
1. exclusing points in the middle of the correlation chart as negative controls.
2. Neg control/random quantiles
3. Pos control - checking death with cancer status

Control: excluding points in the middle as a negative control for the survival curve analysis
```{r}

ctrl_patient_ye_and_3e <- patient_ye_and_3e %>% 
  mutate(change = case_when(
    (hypoxia_z_score < -0.1 | hypoxia_z_score > 0.1) &
      (eIF3e_z_score < -0.1 | eIF3e_z_score > 0.1) ~ "big_change",
    .default="small_change"
  ))

ctrl_patient_ye_and_3e <- ctrl_patient_ye_and_3e %>% 
  filter(change == "big_change")


ggplot(ctrl_patient_ye_and_3e, aes(x=hypoxia_z_score, y= eIF3e_z_score))+
  geom_point()+
  xlab("Ye hypoxia signature z score")+
  ylab("eIF3e-hyp z score")+
  theme_cowplot()+ 
  theme(text = element_text(size = 14))+
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0)


ggsave("3e_hyp_and_ye.pdf", path=here("plots/fig8"),width = 4, height=3)

cor.test(ctrl_patient_ye_and_3e$hypoxia_z_score, ctrl_patient_ye_and_3e$eIF3e_z_score, method="pearson")

#p value still signficiant

#. test survival curve

ctrl_patient_ye_and_3e <- ctrl_patient_ye_and_3e %>% 
  mutate(quadrant = ifelse(hypoxia_z_score < 0 & eIF3e_z_score < 0, "low hypoxia and low eIF3e (hypoxia)",
                           ifelse(hypoxia_z_score > 0 & eIF3e_z_score < 0, "high hypoxia and low eIF3e (hypoxia)",
                                  ifelse(hypoxia_z_score > 0 & eIF3e_z_score > 0, "high hypoxia and high eIF3e (hypoxia)",
                                         ifelse(hypoxia_z_score < 0 & eIF3e_z_score > 0, "low hypoxia sig and high eIF3e (hypoxia)", "other")))))

#bind with clin data
ctrl_patient_ye_and_3e_hyp_clin <- left_join(clin, ctrl_patient_ye_and_3e, by=c("PATIENT_ID"="patient"))

#plot

fit_all_hypoxia_3e <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = ctrl_patient_ye_and_3e_hyp_clin)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, patient_ye_and_3e_hyp_clin)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

ctrl_patient_ye_and_3e_hyp_clin$quadrant <- factor(ctrl_patient_ye_and_3e_hyp_clin$quadrant, levels=c("low hypoxia and low eIF3e (hypoxia)", "high hypoxia and low eIF3e (hypoxia)", "low hypoxia sig and high eIF3e (hypoxia)", "high hypoxia and high eIF3e (hypoxia)"))

quadrant_levels <- levels(ctrl_patient_ye_and_3e_hyp_clin$quadrant)

# ctrl_surv_plot_3e_and_hypoxia_sig<-
# ggsurvplot(fit_all_hypoxia_3e,
#            data = ctrl_patient_ye_and_3e_hyp_clin,
#            size = 1,
#            palette = c("#E7B800", "pink", "purple", "#2E9FDF"), 
#            break.time.by = 1,
#            ylab="Overall Survival",
#            xlab = "Time in years",
#            xlim = c(0,10),
#            font.x = 30,      
#            font.y = 30,      
#            font.tickslab = 25,
#            font.legend = 7,
#            conf.int = FALSE,
#            pval = F,
#            risk.table = FALSE,
#            risk.table.col = "strata",
#            risk.table.height = 0.25,
#            ggtheme = theme_prism())
# 
# ctrl_surv_plot_3e_and_hypoxia_sig$plot +
#   annotate(
#     "text", 
#     x = 3, y = 0.2, 
#     label = paste("HR =", hr, "\np =", pval), 
#     size = 12, 
#     color = "black"
#   )
# 

#trend remains, even when taking out values with low levels of changes


```

## Controls

## Positive control

Visualize OS_STATUS

```{r}
fit_pos <- survfit(Surv(OS_YEARS, OS_STATUS) ~ OS_STATUS, data = all_data_3e_hyp)


ggsurvplot(fit_pos, data = all_data_3e_hyp, size = 1, palette = c("#E7B800", "#2E9FDF"),legend.labs = c("alive", "died_from_cancer"),  break.time.by = 1, ylab="Overall Survival", xlab = "Time in years", xlim = c(0,10),conf.int = TRUE, pval = TRUE, risk.table = TRUE,  risk.table.col = "strata", risk.table.height = 0.25, ggtheme = theme_bw())

```

## Negative Control for eIF3e signature

For a negative control -- random genes comprising signatures-- same number of genes just random

In my 3e hyp signature, there are:
78 up genes
77 down genes

Create random gene signatures and evaluate how they impact survival
```{r}

all_genes <- rownames(merged_zscores)

fake_pos_list <-all_genes[sample(1:length(all_genes), 78)]
fake_neg_list <-all_genes[sample(1:length(all_genes), 77)]


# empty df
neg_ctrl_zscores <- data.frame(patient=colnames(exp_3e),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(neg_ctrl_zscores)) {
  neg_ctrl_zscores[i,2] <- mean(exp_3e[fake_pos_list,i],  na.rm=TRUE)
  neg_ctrl_zscores[i,3] <- mean(exp_3e[fake_neg_list,i],  na.rm=TRUE)
  }

neg_ctrl_zscores <- neg_ctrl_zscores %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(neg_ctrl_zscores$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

neg_ctrl_zscores$quantile <- cut(neg_ctrl_zscores$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

neg_ctrl_zscores$patient <- gsub("\\.", "-", neg_ctrl_zscores$patient)

# Survival plot

#Bind z score hypoxia signature with clin data


all_data_neg_ctrl <- left_join(clin, neg_ctrl_zscores, by=c("PATIENT_ID"="patient"))


#plot

neg_ctrl_fit <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_neg_ctrl)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_neg_ctrl)
summary_coxfit_ye <- summary(coxfit)


hr <- round(summary_coxfit_ye$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_ye$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)


surv_plot_neg_ctrl<-
ggsurvplot(neg_ctrl_fit,
           data = all_data_neg_ctrl,
           size = 1,
           palette = c("#E7B800", "pink", "purple", "#2E9FDF"), 
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_neg_ctrl$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )


#boxplot of quartile expression

ggplot(all_data_neg_ctrl, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("neg ctrl z scores")


```




# Repeat analysis using TCGA dataset
TCGA Firehouse legacy
Downloaded from cbioportal

1,108 samples
```{r cache = TRUE}
zscores_tcga <- read.table(here("cbioportal_data/brca_tcga/tcga_data_mrna_seq_v2_rsem_zscores_ref_diploid_samples.txt"), sep="\t", header=T)
row.names(zscores) <- gsub("_", ".",row.names(zscores))

zscores_tcga_all_symbol <- zscores_tcga %>% 
  dplyr::select(-Entrez_Gene_Id)

### Average scores for duplicates

dup_genes_tcga <- zscores_tcga_all_symbol$Hugo_Symbol[duplicated(zscores_tcga_all_symbol$Hugo_Symbol)]

dedup <- zscores_tcga_all_symbol %>%
  filter(Hugo_Symbol %in% dup_genes_tcga) %>%
  group_by(Hugo_Symbol) %>%
  summarise(across(everything(), mean))

merged_zscores_tcga <- zscores_tcga_all_symbol %>%
  filter(!Hugo_Symbol %in% dup_genes_tcga) %>% 
  bind_rows(., dedup) %>% 
  as.tibble() %>%
  column_to_rownames("Hugo_Symbol") %>% 
  as.data.frame()

```

# eIF3e hypoxia signature and survival prediction
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3e_tcga <- merged_zscores_tcga %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% sig_rna_3e_hyp$symbol) %>% 
  column_to_rownames("Hugo_Symbol")

# empty df
sig_zscores_3e_tcga <- data.frame(patient=colnames(exp_3e_tcga),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3e_tcga)) {
  sig_zscores_3e_tcga[i,2] <- mean(exp_3e_tcga[up_3e_hyp,i],  na.rm=TRUE)
  sig_zscores_3e_tcga[i,3] <- mean(exp_3e_tcga[down_3e_hyp,i],  na.rm=TRUE)
  }

sig_zscores_3e_tcga <- sig_zscores_3e_tcga %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3e_tcga$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3e_tcga$quantile <- cut(sig_zscores_3e_tcga$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_3e_tcga$patient <- gsub("\\.", "-", as.character(sig_zscores_3e_tcga$patient))
sig_zscores_3e_tcga$patient <- sub("-[^-]*$", "", sig_zscores_3e_tcga$patient)

```

## Import TCGA patient data
```{r}
clin_tcga <- read_excel(here("cbioportal_data/new_tcga_clinical.xlsx"))
colnames(clin_tcga)

#do not need all these columns

clin_small <- clin_tcga %>% 
  dplyr::select(OS_STATUS, OS_MONTHS, PATIENT_ID)

# years and months in this are not reading correctly
#get rid of not available
clin_small <- clin_small %>%
  filter(OS_MONTHS != "[Not Available]") %>% 
  filter(OS_MONTHS > 0)

clin_small$OS_MONTHS <- as.numeric(clin_small$OS_MONTHS)

clin_small$OS_STATUS[which(clin_small$OS_STATUS == "1:DECEASED" & clin_small$OS_MONTHS > 120)] <- "0:LIVING"
 
 clin_small$OS_STATUS <- clin_small$OS_STATUS == "1:DECEASED"

clin$OS_MONTHS <- as.numeric(clin$OS_MONTHS)

clin_small <- clin_small %>% 
  mutate(OS_YEARS= clin_small$OS_MONTHS/12)

clin_small$OS_YEARS[which(clin_small$OS_YEARS >= 10)] <- 10
```


## 3e survival
```{r}
#bind clin data with z score data

all_tcga_3e <- left_join(clin_small, sig_zscores_3e_tcga, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA") %>% 
  filter(quantile==1 | quantile==4)

fit_tcga_3e <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_tcga_3e)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_tcga_3e)
summary_coxfit<- summary(coxfit)

hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", si3e_col))(2)


surv_plot_3e_tcga<-
ggsurvplot(fit_tcga_3e,
           data = all_tcga_3e,
           size = 1,
           palette = colors, 
          legend.labs=c("Low 3e", "High 3e"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = TRUE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3e_tcga$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("tcga_3e_231.pdf", path=here("plots/plots_231"),width = 8, height=6)

#boxplot of quartile expression

ggplot(all_tcga_3e, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3e hypoxia signature")

```

## 3d TCGA
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3d_tcga <- merged_zscores_tcga %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% sig_rna_3d_hyp$symbol) %>% 
  column_to_rownames("Hugo_Symbol")

# empty df
sig_zscores_3d_tcga <- data.frame(patient=colnames(exp_3d_tcga),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3d_tcga)) {
  sig_zscores_3d_tcga[i,2] <- mean(exp_3d_tcga[up_3d_hyp,i],  na.rm=TRUE)
  sig_zscores_3d_tcga[i,3] <- mean(exp_3d_tcga[down_3d_hyp,i],  na.rm=TRUE)
  }

sig_zscores_3d_tcga <- sig_zscores_3d_tcga %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3d_tcga$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3d_tcga$quantile <- cut(sig_zscores_3d_tcga$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_3d_tcga$patient <- gsub("\\.", "-", as.character(sig_zscores_3d_tcga$patient))
sig_zscores_3d_tcga$patient <- sub("-[^-]*$", "", sig_zscores_3d_tcga$patient)

```

```{r}
#bind clin data with z score data

all_tcga_3d <- left_join(clin_small, sig_zscores_3d_tcga, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA") %>% 
    filter(quantile==1 | quantile==4)


fit_tcga_3d <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_tcga_3d)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_tcga_3d)
summary_coxfit<- summary(coxfit)

hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", si3d_col))(2)


surv_plot_3d_tcga<-
ggsurvplot(fit_tcga_3d,
           data = all_tcga_3d,
           size = 1,
           palette = colors,
                     legend.labs=c("Low 3d", "High 3d"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = TRUE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3d_tcga$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("tcga_3d_231.pdf", path=here("plots/plots_231"),width = 8, height=6)




#boxplot of quartile expression

ggplot(all_tcga_3d, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3d hypoxia signature")

```


# Find the intersection of 3e and 3d RNA signature, do surival analysis with that

## Import MCF7-SIX1 data

```{r}

six1_res_hyp_3d <- read_xlsx(path = here("counts/fig8/rna_signature_3d_hypoxia.xlsx"), col_names = T)

#confirm direction
  six1_res_hyp_3d %>% 
    filter(symbol=="EIF3D")
  
six1_res_hyp_3e <- read_xlsx(path = here("counts/fig8/rna_signature_3e_hypoxia.xlsx"), col_names = T)

#confirm direction
  six1_res_hyp_3e %>% 
    filter(symbol=="EIF3E")

```


Doing a cutoff of LFC of 1 leads to very few genes in the intersect 

```{r}

six1_sig_rna_3d_hyp <- six1_res_hyp_3d %>% 
  filter(padj < 0.05)

six1_up_3d_hyp <- six1_sig_rna_3d_hyp %>% 
  filter(log2FoldChange > 1) %>% 
  pull(symbol)
  

six1_down_3d_hyp <- six1_sig_rna_3d_hyp %>% 
  filter(log2FoldChange < -1)%>% 
  pull(symbol)
  
```

```{r}
intersect_hyp_up_3d <- intersect(six1_up_3d_hyp, up_3d_hyp) %>% view()

intersect_hyp_down_3d <- intersect(six1_down_3d_hyp, down_3d_hyp) %>% view()

```

```{r}

six1_sig_rna_3e_hyp <- six1_res_hyp_3e %>% 
  filter(padj < 0.05)

six1_up_3e_hyp <- six1_sig_rna_3e_hyp %>% 
  filter(log2FoldChange > 1) %>% 
  pull(symbol)
  

six1_down_3e_hyp <- six1_sig_rna_3e_hyp %>% 
  filter(log2FoldChange < -1)%>% 
  pull(symbol)
  
```

```{r}
intersect_hyp_up_3e <- intersect(six1_up_3e_hyp, up_3e_hyp) %>% view()

intersect_hyp_down_3e <- intersect(six1_down_3e_hyp, down_3e_hyp) %>% view()

```

# eIF3e hypoxia signature and survival prediction
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3e <- merged_zscores %>%
  rownames_to_column("Hugo_Symbol") %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  dplyr::select(-Entrez_Gene_Id)

# empty df
sig_zscores_3e <- data.frame(patient=colnames(exp_3e),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3e)) {
  sig_zscores_3e[i,2] <- mean(exp_3e[intersect_hyp_up_3e,i],  na.rm=TRUE)
  sig_zscores_3e[i,3] <- mean(exp_3e[intersect_hyp_down_3e,i],  na.rm=TRUE)
  }

sig_zscores_3e <- sig_zscores_3e %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3e$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3e$quantile <- cut(sig_zscores_3e$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_3e$patient <- sub("-[^-]*$", "", sig_zscores_3e$patient)

sig_zscores_3e$patient <- gsub("\\.", "-", sig_zscores_3e$patient)


```


# eIF3d hypoxia signature and survival prediction
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3d <- merged_zscores %>%
  rownames_to_column("Hugo_Symbol") %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  dplyr::select(-Entrez_Gene_Id)

# empty df
sig_zscores_3d <- data.frame(patient=colnames(exp_3d),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3d)) {
  sig_zscores_3d[i,2] <- mean(exp_3d[intersect_hyp_up_3d,i],  na.rm=TRUE)
  sig_zscores_3d[i,3] <- mean(exp_3d[intersect_hyp_down_3d,i],  na.rm=TRUE)
  }

sig_zscores_3d <- sig_zscores_3d %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3d$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3d$quantile <- cut(sig_zscores_3d$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)


sig_zscores_3d$patient <- gsub("\\.", "-", sig_zscores_3d$patient)

```


# Import clinical data
```{r}
clin <- read_excel(here("cbioportal_data/brca_metabric/data_clinical_patient_edit.xlsx"))
clin$OS_MONTHS <- as.numeric(clin$OS_MONTHS)
clin$OS_STATUS[which(clin$OS_STATUS == "1:DECEASED" & clin$OS_MONTHS > 120)] <- "0:LIVING"
clin$OS_STATUS <- clin$OS_STATUS == "1:DECEASED"
clin <- cbind(clin, OS_YEARS = clin$OS_MONTHS/12)
clin$OS_YEARS[which(clin$OS_YEARS >= 10)] <- 10

```

### visualizing high and low quartile in each survival plot for the paper.
## 3e survival
```{r}
#bind clin data with z score data

all_data_3e_hyp <- left_join(clin, sig_zscores_3e, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA")

all_data_3e_hyp <- all_data_3e_hyp %>% 
  filter(quantile == 1 | quantile == 4)

fit_all_3e_hyp <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_3e_hyp)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_3e_hyp)
summary_coxfit_3e_norm <- summary(coxfit)


hr <- round(summary_coxfit_3e_norm$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_3e_norm$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

#3e_colors
colors <- grDevices::colorRampPalette(c("grey", si3e_col))(2)


surv_plot_3e_hyp<-
ggsurvplot(fit_all_3e_hyp,
           data = all_data_3e_hyp,
           size = 1,
           palette =colors, 
          legend.labs=c("Low 3e", "High 3e"),
         #  legend.labs=c("Low 3e", "Med-Low 3e", "Med-High 3e", "High 3e"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3e_hyp$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("3e_hyp_surv_plot_intersect.pdf", path=here("plots/plots_231"),width = 11.5, height=8)




#boxplot of quartile expression

ggplot(all_data_3e_hyp, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3e hypoxia signature")


#examples of more analyses, not used in paper
ggviolin(data = all_data_3e_hyp %>% filter(ER_IHC != ""), y = "total_z_score", x = "ER_IHC", fill = "ER_IHC", palette = ggokabeito::palette_okabe_ito(), add = "median_iqr" ) +
  rotate_x_text(45)

ggviolin(data = all_data_3e_hyp %>% filter(THREEGENE != ""), y = "total_z_score", x = "THREEGENE", fill = "THREEGENE", palette = ggokabeito::palette_okabe_ito(), add = "boxplot", add.params = list(fill = "white")) +
  rotate_x_text(45)


#claudin subtype
ggviolin(data = all_data_3e_hyp , y = "total_z_score", x = "CLAUDIN_SUBTYPE", fill = "CLAUDIN_SUBTYPE", palette = ggokabeito::palette_okabe_ito(), add = "median_iqr" ) +
  rotate_x_text(45)


```

## 3d survival
```{r}
#bind clin data with z score data

all_data_3d_hyp <- left_join(clin, sig_zscores_3d, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA")

all_data_3d_hyp <- all_data_3d_hyp %>% 
  filter(quantile == 1 | quantile == 4)

fit_all_3d_hyp <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_3d_hyp)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_3d_hyp)
summary_coxfit_3d_norm <- summary(coxfit)


hr <- round(summary_coxfit_3d_norm$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_3d_norm$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", si3d_col))(2)

surv_plot_3d_hyp<-
ggsurvplot(fit_all_3d_hyp,
           data = all_data_3d_hyp,
           size = 1,
           palette = colors, 
          legend.labs=c("Low 3d", "High 3d"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3d_hyp$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("3d_hyp_surv_plot_intersect.pdf", path=here("plots/plots_231"),width = 11.5, height=8)




#boxplot of quartile expression

ggplot(all_data_3d_hyp, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3d hypoxia signature")

count(all_data_3d_hyp$quantile==1)

ggviolin(data = all_data_3d_hyp %>% filter(ER_IHC != ""), y = "total_z_score", x = "ER_IHC", fill = "ER_IHC", palette = ggokabeito::palette_okabe_ito(), add = "median_iqr" ) +
  rotate_x_text(45)

ggviolin(data = all_data_3d_hyp %>% filter(THREEGENE != ""), y = "total_z_score", x = "THREEGENE", fill = "THREEGENE", palette = ggokabeito::palette_okabe_ito(), add = "boxplot", add.params = list(fill = "white")) +
  rotate_x_text(45)

```
## Sanity check:
REpeat analyses using random subsets of the group adn make sure results look the same

3e
```{r}
all_data_3e_hyp <- left_join(clin, sig_zscores_3e, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA")

all_data_3e_hyp <- all_data_3e_hyp %>% 
  filter(quantile == 1 | quantile == 4) %>% 
    slice_sample(n = 100)


fit_all_3e_hyp <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_3e_hyp)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_3e_hyp)
summary_coxfit_3e_norm <- summary(coxfit)


hr <- round(summary_coxfit_3e_norm$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_3e_norm$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

#3e_colors
colors <- grDevices::colorRampPalette(c("grey", si3e_col))(2)


surv_plot_3e_hyp<-
ggsurvplot(fit_all_3e_hyp,
           data = all_data_3e_hyp,
           size = 1,
           palette =colors, 
          legend.labs=c("Low 3e", "High 3e"),
         #  legend.labs=c("Low 3e", "Med-Low 3e", "Med-High 3e", "High 3e"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3e_hyp$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )

```

3d -- random subset

```{r}
all_data_3d_hyp <- all_data_3d_hyp %>% 
  filter(quantile == 1 | quantile == 4) %>% 
  slice_sample(n = 400)

fit_all_3d_hyp <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_3d_hyp)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_3d_hyp)
summary_coxfit_3d_norm <- summary(coxfit)


hr <- round(summary_coxfit_3d_norm$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_3d_norm$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", si3d_col))(2)

surv_plot_3d_hyp<-
ggsurvplot(fit_all_3d_hyp,
           data = all_data_3d_hyp,
           size = 1,
           palette = colors, 
          legend.labs=c("Low 3d", "High 3d"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3d_hyp$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
```





# Intersect TCGA


# eIF3e hypoxia signature and survival prediction
### Calculate z scores

# eIF3e hypoxia signature and survival prediction
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3e_tcga <- merged_zscores_tcga %>%
  rownames_to_column("Hugo_Symbol") %>% 
  column_to_rownames("Hugo_Symbol")

# empty df
sig_zscores_3e_tcga <- data.frame(patient=colnames(exp_3e_tcga),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3e_tcga)) {
  sig_zscores_3e_tcga[i,2] <- mean(exp_3e_tcga[intersect_hyp_up_3e,i],  na.rm=TRUE)
  sig_zscores_3e_tcga[i,3] <- mean(exp_3e_tcga[intersect_hyp_down_3e,i],  na.rm=TRUE)
  }

sig_zscores_3e_tcga <- sig_zscores_3e_tcga %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3e_tcga$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3e_tcga$quantile <- cut(sig_zscores_3e_tcga$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_3e_tcga$patient <- gsub("\\.", "-", as.character(sig_zscores_3e_tcga$patient))
sig_zscores_3e_tcga$patient <- sub("-[^-]*$", "", sig_zscores_3e_tcga$patient)

```


## 3e survival
```{r}
#bind clin data with z score data

all_tcga_3e <- left_join(clin_small, sig_zscores_3e_tcga, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA") %>% 
  filter(quantile==1 | quantile==4)

fit_tcga_3e <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_tcga_3e)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_tcga_3e)
summary_coxfit<- summary(coxfit)

hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", si3e_col))(2)


surv_plot_3e_tcga<-
ggsurvplot(fit_tcga_3e,
           data = all_tcga_3e,
           size = 1,
           palette = colors, 
          legend.labs=c("Low 3e", "High 3e"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = TRUE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3e_tcga$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("tcga_3e_intersect.pdf", path=here("plots/plots_231"),width = 8, height=6)

#boxplot of quartile expression

ggplot(all_tcga_3e, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3e hypoxia signature")

```

## 3d TCGA
### Calculate z scores
```{r}
# filter merged z score data based on genes in my signature
exp_3d_tcga <- merged_zscores_tcga %>%
  rownames_to_column("Hugo_Symbol") %>% 
  column_to_rownames("Hugo_Symbol")

# empty df
sig_zscores_3d_tcga <- data.frame(patient=colnames(exp_3d_tcga),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3d_tcga)) {
  sig_zscores_3d_tcga[i,2] <- mean(exp_3d_tcga[intersect_hyp_up_3d,i],  na.rm=TRUE)
  sig_zscores_3d_tcga[i,3] <- mean(exp_3d_tcga[intersect_hyp_down_3d,i],  na.rm=TRUE)
  }

sig_zscores_3d_tcga <- sig_zscores_3d_tcga %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3d_tcga$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3d_tcga$quantile <- cut(sig_zscores_3d_tcga$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_3d_tcga$patient <- gsub("\\.", "-", as.character(sig_zscores_3d_tcga$patient))
sig_zscores_3d_tcga$patient <- sub("-[^-]*$", "", sig_zscores_3d_tcga$patient)

```

```{r}
#bind clin data with z score data

all_tcga_3d <- left_join(clin_small, sig_zscores_3d_tcga, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA") %>% 
    filter(quantile==1 | quantile==4)


fit_tcga_3d <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_tcga_3d)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_tcga_3d)
summary_coxfit<- summary(coxfit)

hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", si3d_col))(2)


surv_plot_3d_tcga<-
ggsurvplot(fit_tcga_3d,
           data = all_tcga_3d,
           size = 1,
           palette = colors,
                     legend.labs=c("Low 3d", "High 3d"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = TRUE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3d_tcga$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("tcga_3d_intersect.pdf", path=here("plots/plots_231"),width = 8, height=6)




#boxplot of quartile expression

ggplot(all_tcga_3d, aes(x=quantile,y=total_z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3d hypoxia signature")

```


3d subsample sanity check
```{r}
all_tcga_3d <- left_join(clin_small, sig_zscores_3d_tcga, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA") %>% 
    filter(quantile==1 | quantile==4) %>% 
    slice_sample(n = 100)



fit_tcga_3d <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_tcga_3d)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_tcga_3d)
summary_coxfit<- summary(coxfit)

hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", si3d_col))(2)


surv_plot_3d_tcga<-
ggsurvplot(fit_tcga_3d,
           data = all_tcga_3d,
           size = 1,
           palette = colors,
                     legend.labs=c("Low 3d", "High 3d"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = TRUE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3d_tcga$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
```

3e sanity check
```{r}
all_tcga_3e <- left_join(clin_small, sig_zscores_3e_tcga, by=c("PATIENT_ID"="patient")) %>% 
  filter(quantile != "NA") %>% 
  filter(quantile==1 | quantile==4) %>% 
  slice_sample(n=100)

fit_tcga_3e <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_tcga_3e)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_tcga_3e)
summary_coxfit<- summary(coxfit)

hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", si3e_col))(2)


surv_plot_3e_tcga<-
ggsurvplot(fit_tcga_3e,
           data = all_tcga_3e,
           size = 1,
           palette = colors, 
          legend.labs=c("Low 3e", "High 3e"),
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = TRUE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3e_tcga$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
```





# Hypoxia signature

On the intersect signature for both cell lines


## Import hypoxia signature from ye et al
Hypoxia signature in breast cancer:
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6279594/#:~:text=The%2042%2Dgene%20expression%20signature,mediate%20the%20conserved%20hypoxic%20response


```{r}
ye_hyp_sig <- read_excel(here("other_data/hypoxic_sig_ye.xlsx"))
#up values only

#calculate z scores for enrichment of hypoxia signature

# filter merged z score data based on genes in my signature
exp_hyp_sig <- merged_zscores %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% ye_hyp_sig$up_in_hypoxia) %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  dplyr::select(-Entrez_Gene_Id)

# empty df
sig_zscores_hyp_signature <- data.frame(patient=colnames(exp_hyp_sig),
                         z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_hyp_signature)) {
  sig_zscores_hyp_signature[i,2] <- mean(exp_hyp_sig[,i],  na.rm=TRUE)
}


quantile_groups <- quantile(sig_zscores_hyp_signature$z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_hyp_signature$quantile <- cut(sig_zscores_hyp_signature$z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_hyp_signature$patient <- sub("-[^-]*$", "", sig_zscores_hyp_signature$patient)

sig_zscores_hyp_signature$patient <- gsub("\\.", "-", sig_zscores_hyp_signature$patient)


#Bind z score hypoxia signature with clin data


all_data_hif_ye <- left_join(clin, sig_zscores_hyp_signature, by=c("PATIENT_ID"="patient"))

all_data_hif_ye <- all_data_hif_ye %>% 
  filter(quantile == 1 | quantile == 4)

#plot

fit_all_data_hif_ye <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_hif_ye)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_hif_ye)
summary_coxfit_ye <- summary(coxfit)

exp(summary_coxfit_ye$coefficients[1])

hr <- round(summary_coxfit_ye$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_ye$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", "red"))(2)

surv_plot_ye_hyp<-
ggsurvplot(fit_all_data_hif_ye,
           data = all_data_hif_ye,
           size = 1,
           palette = colors, 
           break.time.by = 1,
          legend.labs=c("Low Hypoxia", "High Hypoxia"),
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_ye_hyp$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("surv_plot_ye_hyp.pdf", path=here("plots/fig8"),width = 11.5, height=8)


#boxplot of quartile expression

ggplot(all_data_hif_ye, aes(x=quantile,y=z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3d hypoxia signature")

```


## correlation of 3e and hypoxia score

```{r}

exp_3e <- merged_zscores %>%
  rownames_to_column("Hugo_Symbol") %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  dplyr::select(-Entrez_Gene_Id)

# empty df
sig_zscores_3e <- data.frame(patient=colnames(exp_3e),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3e)) {
  sig_zscores_3e[i,2] <- mean(exp_3e[intersect_hyp_up_3e,i],  na.rm=TRUE)
  sig_zscores_3e[i,3] <- mean(exp_3e[intersect_hyp_down_3e,i],  na.rm=TRUE)
  }

sig_zscores_3e <- sig_zscores_3e %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3e$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3e$quantile <- cut(sig_zscores_3e$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_3e$patient <- sub("-[^-]*$", "", sig_zscores_3e$patient)

sig_zscores_3e$patient <- gsub("\\.", "-", sig_zscores_3e$patient)



patient_ye_and_3e <- left_join(sig_zscores_3e, sig_zscores_hyp_signature, by="patient") %>% 
  dplyr::rename("hypoxia_z_score"="z_score") %>% 
   dplyr::rename("eIF3e_z_score"="total_z_score")


```

## Divide patients into 3e sig high/low and hypoxia signature high/low

Make four groups based on quandrants -- 3e levels in hypoxia
```{r}

patient_ye_and_3e <- patient_ye_and_3e %>% 
  mutate(quadrant = ifelse(hypoxia_z_score < 0 & eIF3e_z_score < 0, "low hypoxia and low 3e",
                           ifelse(hypoxia_z_score > 0 & eIF3e_z_score < 0, "high hypoxia and low 3e",
                                  ifelse(hypoxia_z_score > 0 & eIF3e_z_score > 0, "high hypoxia and high 3e",
                                         ifelse(hypoxia_z_score < 0 & eIF3e_z_score > 0, "low hypoxia and high 3e", "other")))))


patient_ye_and_3e$quadrant <- factor(patient_ye_and_3e$quadrant, levels=c("low hypoxia and low 3e","high hypoxia and low 3e","low hypoxia and high 3e", "high hypoxia and high 3e"))


```

Determine correlation and color by group
```{r}
ggplot(patient_ye_and_3e, aes(x=hypoxia_z_score, y= eIF3e_z_score, color=quadrant))+
  geom_point()+
  xlab("Ye hypoxia signature z score")+
  ylab("eIF3e-hyp z score")+
  theme_prism()+ 
  theme(text = element_text(size = 14))+
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0)+
  scale_color_manual(values=c("high hypoxia and high 3e"="#9B4D96", 
                       "high hypoxia and low 3e"="red", 
                       "low hypoxia and low 3e"="grey", 
                       "low hypoxia and high 3e"= "#56B4E9"))


ggsave("intersect_3e_hyp_and_ye.pdf", path=here("plots/plots_231"),width = 5.5, height=3)

cor.test(patient_ye_and_3e$hypoxia_z_score, patient_ye_and_3e$eIF3e_z_score, method="pearson")
```

Plot survival curve
```{r}
#bind with clin data
patient_ye_and_3e_hyp_clin <- left_join(clin, patient_ye_and_3e, by=c("PATIENT_ID"="patient"))

#plot

fit_all_hypoxia_3e <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = patient_ye_and_3e_hyp_clin)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, patient_ye_and_3e_hyp_clin)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)


levels(patient_ye_and_3e_hyp_clin$quadrant)

surv_plot_3e_and_hypoxia_sig<-
ggsurvplot(fit_all_hypoxia_3e,
           data = patient_ye_and_3e_hyp_clin,
           size = 1,
           palette = c("grey", "red", si3e_col, "#9B4D96"), 
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 7,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3e_and_hypoxia_sig$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("intersect_surv_plot_3e_and_hypoxia_sig.pdf", path=here("plots/plots_231"),width = 11.5, height=8)

#compare middle two quadrants

levels(patient_ye_and_3e_hyp_clin$quadrant)

# Relevel the 'quadrant' factor to set the desired reference group
patient_ye_and_3e_hyp_clin$quadrant <- relevel(patient_ye_and_3e_hyp_clin$quadrant, ref = "low hypoxia and low 3e")

# Fit the Cox model
coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = patient_ye_and_3e_hyp_clin)
summary_coxfit <- summary(coxfit)
print(summary_coxfit)

summary(patient_ye_and_3e_hyp_clin$quadrant)

```


#3d


## correlation of 3d and hypoxia score

```{r}

exp_3d <- merged_zscores %>%
  rownames_to_column("Hugo_Symbol") %>% 
  column_to_rownames("Hugo_Symbol") %>% 
  dplyr::select(-Entrez_Gene_Id)

# empty df
sig_zscores_3d <- data.frame(patient=colnames(exp_3d),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3d)) {
  sig_zscores_3d[i,2] <- mean(exp_3d[intersect_hyp_up_3d,i],  na.rm=TRUE)
  sig_zscores_3d[i,3] <- mean(exp_3d[intersect_hyp_down_3d,i],  na.rm=TRUE)
  }

sig_zscores_3d <- sig_zscores_3d %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3d$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3d$quantile <- cut(sig_zscores_3d$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_3d$patient <- sub("-[^-]*$", "", sig_zscores_3d$patient)

sig_zscores_3d$patient <- gsub("\\.", "-", sig_zscores_3d$patient)



patient_ye_and_3d <- left_join(sig_zscores_3d, sig_zscores_hyp_signature, by="patient") %>% 
  dplyr::rename("hypoxia_z_score"="z_score") %>% 
   dplyr::rename("eIF3d_z_score"="total_z_score")


```

## Divide patients into 3d sig high/low and hypoxia signature high/low

Make four groups based on quandrants -- 3d levels in hypoxia
```{r}

patient_ye_and_3d <- patient_ye_and_3d %>% 
  mutate(quadrant = ifelse(hypoxia_z_score < 0 & eIF3d_z_score < 0, "low hypoxia and low 3d",
                           ifelse(hypoxia_z_score > 0 & eIF3d_z_score < 0, "high hypoxia and low 3d",
                                  ifelse(hypoxia_z_score > 0 & eIF3d_z_score > 0, "high hypoxia and high 3d",
                                         ifelse(hypoxia_z_score < 0 & eIF3d_z_score > 0, "low hypoxia and high 3d", "other")))))


patient_ye_and_3d$quadrant <- factor(patient_ye_and_3d$quadrant, levels=c("low hypoxia and low 3d","high hypoxia and low 3d","low hypoxia and high 3d", "high hypoxia and high 3d"))


```

Determine correlation and color by group
```{r}
ggplot(patient_ye_and_3d, aes(x=hypoxia_z_score, y= eIF3d_z_score, color=quadrant))+
  geom_point()+
  xlab("Ye hypoxia signature z score")+
  ylab("eIF3d-hyp z score")+
  theme_prism()+ 
  theme(text = element_text(size = 14))+
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0)+
  scale_color_manual(values=c("high hypoxia and high 3d"="#9B4D96", 
                       "high hypoxia and low 3d"="red", 
                       "low hypoxia and low 3d"="grey", 
                       "low hypoxia and high 3d"= "#56B4E9"))


ggsave("intersect_3d_hyp_and_ye.pdf", path=here("plots/plots_231"),width = 5.5, height=3)

cor.test(patient_ye_and_3d$hypoxia_z_score, patient_ye_and_3d$eIF3d_z_score, method="pearson")
```

Plot survival curve
```{r}
#bind with clin data
patient_ye_and_3d_hyp_clin <- left_join(clin, patient_ye_and_3d, by=c("PATIENT_ID"="patient"))

#plot

fit_all_hypoxia_3d <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = patient_ye_and_3d_hyp_clin)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, patient_ye_and_3d_hyp_clin)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)


levels(patient_ye_and_3d_hyp_clin$quadrant)

surv_plot_3d_and_hypoxia_sig<-
ggsurvplot(fit_all_hypoxia_3d,
           data = patient_ye_and_3d_hyp_clin,
           size = 1,
           palette = c("grey", "red", "#56B4E9", "#9B4D96"), 
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 7,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3d_and_hypoxia_sig$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("intersect_surv_plot_3d_and_hypoxia_sig.pdf", path=here("plots/plots_231"),width = 11.5, height=8)

#compare middle two quadrants

levels(patient_ye_and_3d_hyp_clin$quadrant)

# Relevel the 'quadrant' factor to set the desired reference group
patient_ye_and_3d_hyp_clin$quadrant <- relevel(patient_ye_and_3d_hyp_clin$quadrant, ref = "low hypoxia and low 3d")

# Fit the Cox model
coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = patient_ye_and_3d_hyp_clin)
summary_coxfit <- summary(coxfit)
print(summary_coxfit)

summary(patient_ye_and_3d_hyp_clin$quadrant)

```





# TCGA version of this

TCGA hypoxia signature.

```{r}
#calculate z scores for enrichment of hypoxia signature

# filter merged z score data based on genes in my signature
exp_hyp_sig <- merged_zscores_tcga %>%
  rownames_to_column("Hugo_Symbol") %>% 
  filter(Hugo_Symbol %in% ye_hyp_sig$up_in_hypoxia) %>% 
  column_to_rownames("Hugo_Symbol")


# empty df
sig_zscores_hyp_signature <- data.frame(patient=colnames(exp_hyp_sig),
                         z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_hyp_signature)) {
  sig_zscores_hyp_signature[i,2] <- mean(exp_hyp_sig[,i],  na.rm=TRUE)
}


quantile_groups <- quantile(sig_zscores_hyp_signature$z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_hyp_signature$quantile <- cut(sig_zscores_hyp_signature$z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)


sig_zscores_hyp_signature$patient <- gsub("\\.", "-", as.character(sig_zscores_hyp_signature$patient))
sig_zscores_hyp_signature$patient <- sub("-[^-]*$", "", sig_zscores_hyp_signature$patient)


#Bind z score hypoxia signature with clin data


all_data_hif_ye <- left_join(clin_small, sig_zscores_hyp_signature, by=c("PATIENT_ID"="patient"))

all_data_hif_ye <- all_data_hif_ye %>% 
  filter(quantile == 1 | quantile == 4)

#plot

fit_all_data_hif_ye <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quantile, data = all_data_hif_ye)





coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quantile, all_data_hif_ye)
summary_coxfit_ye <- summary(coxfit)

exp(summary_coxfit_ye$coefficients[1])

hr <- round(summary_coxfit_ye$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit_ye$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)

colors <- grDevices::colorRampPalette(c("grey", "red"))(2)

surv_plot_ye_hyp<-
ggsurvplot(fit_all_data_hif_ye,
           data = all_data_hif_ye,
           size = 1,
           palette = colors, 
           break.time.by = 1,
          legend.labs=c("Low Hypoxia", "High Hypoxia"),
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 25,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_ye_hyp$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("surv_plot_ye_hyp.pdf", path=here("plots/fig8"),width = 11.5, height=8)


#boxplot of quartile expression

ggplot(all_data_hif_ye, aes(x=quantile,y=z_score, group=quantile)) +
  geom_boxplot()+
  theme_cowplot()+
  ggtitle("Patient scores for eIF3d hypoxia signature")

```




## correlation of 3e and hypoxia score

```{r}

exp_3e_tcga_intersect <- merged_zscores_tcga %>%
  rownames_to_column("Hugo_Symbol") %>% 
  column_to_rownames("Hugo_Symbol")



# empty df
sig_zscores_3e <- data.frame(patient=colnames(exp_3e_tcga_intersect),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3e)) {
  sig_zscores_3e[i,2] <- mean(exp_3e[intersect_hyp_up_3e$x,i],  na.rm=TRUE)
  sig_zscores_3e[i,3] <- mean(exp_3e[intersect_hyp_down_3e$x,i],  na.rm=TRUE)
  }

sig_zscores_3e <- sig_zscores_3e %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3e$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3e$quantile <- cut(sig_zscores_3e$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_3e$patient <- gsub("\\.", "-", as.character(sig_zscores_3e$patient))
sig_zscores_3e$patient <- sub("-[^-]*$", "", sig_zscores_3e$patient)


```








```{r}


patient_ye_and_3e <- left_join(sig_zscores_3e, sig_zscores_hyp_signature, by="patient") %>% 
  dplyr::rename("hypoxia_z_score"="z_score") %>% 
   dplyr::rename("eIF3e_z_score"="total_z_score")

```


## Divide patients into 3e sig high/low and hypoxia signature high/low

Make four groups based on quandrants -- 3e levels in hypoxia
```{r}

patient_ye_and_3e <- patient_ye_and_3e %>% 
  mutate(quadrant = ifelse(hypoxia_z_score < 0 & eIF3e_z_score < 0, "low hypoxia and low 3e",
                           ifelse(hypoxia_z_score > 0 & eIF3e_z_score < 0, "high hypoxia and low 3e",
                                  ifelse(hypoxia_z_score > 0 & eIF3e_z_score > 0, "high hypoxia and high 3e",
                                         ifelse(hypoxia_z_score < 0 & eIF3e_z_score > 0, "low hypoxia and high 3e", "other")))))


patient_ye_and_3e$quadrant <- factor(patient_ye_and_3e$quadrant, levels=c("low hypoxia and low 3e","high hypoxia and low 3e","low hypoxia and high 3e", "high hypoxia and high 3e"))


```

Determine correlation and color by group
```{r}
ggplot(patient_ye_and_3e, aes(x=hypoxia_z_score, y= eIF3e_z_score, color=quadrant))+
  geom_point()+
  xlab("Ye hypoxia signature z score")+
  ylab("eIF3e-hyp z score")+
  theme_prism()+ 
  theme(text = element_text(size = 14))+
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0)+
  scale_color_manual(values=c("high hypoxia and high 3e"="#9B4D96", 
                       "high hypoxia and low 3e"="red", 
                       "low hypoxia and low 3e"="grey", 
                       "low hypoxia and high 3e"= "#56B4E9"))


ggsave("intersect_3e_hyp_and_ye_tcga.pdf", path=here("plots/plots_231"),width = 5.5, height=3)

cor.test(patient_ye_and_3e$hypoxia_z_score, patient_ye_and_3e$eIF3e_z_score, method="pearson")
```

Plot survival curve
```{r}
#bind with clin data
patient_ye_and_3e_hyp_clin <- left_join(clin_small, patient_ye_and_3e, by=c("PATIENT_ID"="patient"))

#plot

fit_all_hypoxia_3e <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = patient_ye_and_3e_hyp_clin)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, patient_ye_and_3e_hyp_clin)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)


levels(patient_ye_and_3e_hyp_clin$quadrant)

surv_plot_3e_and_hypoxia_sig<-
ggsurvplot(fit_all_hypoxia_3e,
           data = patient_ye_and_3e_hyp_clin,
           size = 1,
           palette = c("grey", "red", si3e_col, "#9B4D96"), 
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 7,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3e_and_hypoxia_sig$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("intersect_surv_plot_3e_and_hypoxia_sig_tcga.pdf", path=here("plots/plots_231"),width = 11.5, height=8)

#compare middle two quadrants

levels(patient_ye_and_3e_hyp_clin$quadrant)

# Relevel the 'quadrant' factor to set the desired reference group
patient_ye_and_3e_hyp_clin$quadrant <- relevel(patient_ye_and_3e_hyp_clin$quadrant, ref = "low hypoxia and low 3e")

# Fit the Cox model
coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = patient_ye_and_3e_hyp_clin)
summary_coxfit <- summary(coxfit)
print(summary_coxfit)

summary(patient_ye_and_3e_hyp_clin$quadrant)

```


#3d


## correlation of 3d and hypoxia score

```{r}

exp_3d_tcga_intersect <- merged_zscores_tcga %>%
  rownames_to_column("Hugo_Symbol") %>% 
  column_to_rownames("Hugo_Symbol")



# empty df
sig_zscores_3d <- data.frame(patient=colnames(exp_3d_tcga_intersect),
                         up_z_score=NA,
                         down_z_score=NA)
#calculate z score

for (i in 1:nrow(sig_zscores_3d)) {
  sig_zscores_3d[i,2] <- mean(exp_3d[intersect_hyp_up_3d$x,i],  na.rm=TRUE)
  sig_zscores_3d[i,3] <- mean(exp_3d[intersect_hyp_down_3d$x,i],  na.rm=TRUE)
  }

sig_zscores_3d <- sig_zscores_3d %>%
  mutate(total_z_score = up_z_score - down_z_score)

quantile_groups <- quantile(sig_zscores_3d$total_z_score, probs=seq(0,1,.25), na.rm=TRUE)

sig_zscores_3d$quantile <- cut(sig_zscores_3d$total_z_score, breaks = quantile_groups, include.lowest = TRUE, labels = FALSE)

sig_zscores_3d$patient <- gsub("\\.", "-", as.character(sig_zscores_3d$patient))
sig_zscores_3d$patient <- sub("-[^-]*$", "", sig_zscores_3d$patient)


```








```{r}


patient_ye_and_3d <- left_join(sig_zscores_3d, sig_zscores_hyp_signature, by="patient") %>% 
  dplyr::rename("hypoxia_z_score"="z_score") %>% 
   dplyr::rename("eIF3d_z_score"="total_z_score")

```


## Divide patients into 3d sig high/low and hypoxia signature high/low

Make four groups based on quandrants -- 3d levels in hypoxia
```{r}

patient_ye_and_3d <- patient_ye_and_3d %>% 
  mutate(quadrant = ifelse(hypoxia_z_score < 0 & eIF3d_z_score < 0, "low hypoxia and low 3d",
                           ifelse(hypoxia_z_score > 0 & eIF3d_z_score < 0, "high hypoxia and low 3d",
                                  ifelse(hypoxia_z_score > 0 & eIF3d_z_score > 0, "high hypoxia and high 3d",
                                         ifelse(hypoxia_z_score < 0 & eIF3d_z_score > 0, "low hypoxia and high 3d", "other")))))


patient_ye_and_3d$quadrant <- factor(patient_ye_and_3d$quadrant, levels=c("low hypoxia and low 3d","high hypoxia and low 3d","low hypoxia and high 3d", "high hypoxia and high 3d"))


```

Determine correlation and color by group
```{r}
ggplot(patient_ye_and_3d, aes(x=hypoxia_z_score, y= eIF3d_z_score, color=quadrant))+
  geom_point()+
  xlab("Ye hypoxia signature z score")+
  ylab("eIF3d-hyp z score")+
  theme_prism()+ 
  theme(text = element_text(size = 14))+
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0)+
  scale_color_manual(values=c("high hypoxia and high 3d"="#9B4D96", 
                       "high hypoxia and low 3d"="red", 
                       "low hypoxia and low 3d"="grey", 
                       "low hypoxia and high 3d"= "#56B4E9"))


ggsave("intersect_3d_hyp_and_ye_tcga.pdf", path=here("plots/plots_231"),width = 5.5, height=3)

cor.test(patient_ye_and_3d$hypoxia_z_score, patient_ye_and_3d$eIF3d_z_score, method="pearson")
```

Plot survival curve
```{r}
#bind with clin data
patient_ye_and_3d_hyp_clin <- left_join(clin_small, patient_ye_and_3d, by=c("PATIENT_ID"="patient"))

#plot

fit_all_hypoxia_3d <- survfit(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = patient_ye_and_3d_hyp_clin)


coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, patient_ye_and_3d_hyp_clin)
summary_coxfit <- summary(coxfit)


hr <- round(summary_coxfit$coefficients[1, "exp(coef)"], 2)
pval <- format.pval(summary_coxfit$coefficients[1, "Pr(>|z|)"], digits = 2, eps = 0.001)


levels(patient_ye_and_3d_hyp_clin$quadrant)

surv_plot_3d_and_hypoxia_sig<-
ggsurvplot(fit_all_hypoxia_3d,
           data = patient_ye_and_3d_hyp_clin,
           size = 1,
           palette = c("grey", "red", "#56B4E9", "#9B4D96"), 
           break.time.by = 1,
           ylab="Overall Survival",
           xlab = "Time in years",
           xlim = c(0,10),
           font.x = 30,      
           font.y = 30,      
           font.tickslab = 25,
           font.legend = 7,
           conf.int = FALSE,
           pval = F,
           risk.table = FALSE,
           risk.table.col = "strata",
           risk.table.height = 0.25,
           ggtheme = theme_prism())

surv_plot_3d_and_hypoxia_sig$plot +
  annotate(
    "text", 
    x = 3, y = 0.2, 
    label = paste("HR =", hr, "\np =", pval), 
    size = 12, 
    color = "black"
  )
ggsave("intersect_surv_plot_3d_and_hypoxia_sig_tcga.pdf", path=here("plots/plots_231"),width = 11.5, height=8)

#compare middle two quadrants

levels(patient_ye_and_3d_hyp_clin$quadrant)

# Relevel the 'quadrant' factor to set the desired reference group
patient_ye_and_3d_hyp_clin$quadrant <- relevel(patient_ye_and_3d_hyp_clin$quadrant, ref = "low hypoxia and low 3d")

# Fit the Cox model
coxfit <- coxph(Surv(OS_YEARS, OS_STATUS) ~ quadrant, data = patient_ye_and_3d_hyp_clin)
summary_coxfit <- summary(coxfit)
print(summary_coxfit)

summary(patient_ye_and_3d_hyp_clin$quadrant)

```