---
title: "QC"
author: "Kate"
date: "2023-11-01"
output: html_document
editor_options: 
  chunk_output_type: console
---

Quality Control

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r install packages, include=FALSE}

library(here)
library(tximport)
library(tidyverse)
library(EnhancedVolcano)
library(janitor)
library(ggthemes)
library(DESeq2)
library(plotly)
```

```{r}
a209_col = "#CC79A7"
si3e_col = "#56B4E9"
si3d_col = "#E69F00"
```



## Load transcript-level data

```{r meta data}

metadata <- read_csv(here("accessories","metadata_NM.csv"), skip = 0) %>% 
  clean_names()

metadata <- metadata %>% 
  mutate(library = ifelse(library_prep_kit=="Qiagen_miRNA", "riboseq", "rnaseq"))

#ribosome lysate = ribosome seq
#rRNA depletion = RNA seq

```

```{r load alignment file}
#import key on RNA type, remove premRNA
geneInfo <- read_csv(here("accessories","gencode.v26.primary.info.csv.gz"), col_names = F) %>%
  filter(!str_detect(X2, "pre_"))

colnames(geneInfo) <- c("gene_id","transcript_id","biotype","symbol")

#import transcripts for gene to transcript mapping
tx2gene <- geneInfo[,c(2,1)]
gene_symbol <- geneInfo %>% dplyr::select(gene_id, symbol) %>% unique()

```


```{r load in salmon files}
#set up names of quant files based on metadata names
myquantfiles <- paste(here(), "/salmon/",
                      metadata$sample_id,
                      "/quant.sf.gz",
                      sep = "")

names(myquantfiles) <- paste(metadata$genotype, metadata$treatment1, metadata$treatment2, metadata$library, sep = "_")

```


```{r txi import, include=FALSE}
#import transcripts
myTxi <- tximport(myquantfiles, type = "salmon", tx2gene = tx2gene)
# Look at distribution of sum of counts

#distribution of counts histogram
hist_count <- hist(log2(rowSums(myTxi$counts)), breaks = 50)
#determine threshold

#distribution of abundance histogram
hist_abundance <- hist(log2(rowSums(myTxi$abundance)), breaks = 50)

#only keeping rows with counts  log2> 10 across all samples - want around 10-15k genes - want genes that are highly expressed
keepGenes <- rownames(myTxi$counts[log2(rowSums(myTxi$counts)) > 10 ,])

#removes noise
# results show bimodal distribution
#1 RNA per cell = 1 TPM

```


```{r standard deviation filtering}
# log transforming with pseudocount to avoid undefined numbers
qcinput <- log2(myTxi$abundance[keepGenes,] + 1) 

qcinput <- qcinput[rowSds(qcinput) > 1,] 

dim(qcinput)
```
Split normoxia and hypoxia
```{r}

qc_input_normoxia <- qcinput[, grep("Normoxic", colnames(qcinput))]

qc_input_hypoxia <- qcinput[, grep("Hypoxic", colnames(qcinput))]

```


## Heatmap of total samples
### We identified an outlier: hypoxia, si3e, repB, rna seq. That RNA sample also has a very low qubit and yield as well. It will be excluded in further analysis.

```{r correlation coefficient, fig.height=10, fig.width=11}

mycor <- cor(qcinput, method = "pearson")

heatmap <- pheatmap::pheatmap(mat = mycor)
heatmap
ggsave(heatmap, filename=here("plots/fig1/qc/heatmap_with_outlier.pdf"), width=11.5, height=11)

#Normoxia only



mycor_norm <- cor(qc_input_normoxia, method = "pearson")
heatmap <- pheatmap::pheatmap(mat = mycor_norm)
heatmap
ggsave(heatmap, filename=here("plots/fig1/qc/heatmap_with_outlier_norm.pdf"), width=11.5, height=11)

#KDs
qc_input_normoxia_kds <- qc_input_normoxia[, grep("si", colnames(qc_input_normoxia))]

mycor_norm_kds <- cor(qc_input_normoxia_kds, method = "pearson")
heatmap <- pheatmap::pheatmap(mat = mycor_norm_kds)
heatmap
ggsave(heatmap, filename=here("plots/fig1/qc/heatmap_norm_kds.pdf"), width=11.5, height=11)


#kds only pca


pca_qc_input_normoxia_kds <- prcomp(scale(x = qc_input_normoxia_kds,center = T, scale = T))

pca_qc_input_normoxia_kds <- 
  pca_qc_input_normoxia_kds$rotation %>%
  as.data.frame() 

pca_qc_input_normoxia_kds <- pca_qc_input_normoxia_kds %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)

pca <- ggplot(data = pca_qc_input_normoxia_kds, aes(x=PC1, y = PC2, color = treatment, shape = type))+ 
  geom_point(size=3) +
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/pca_with_outlier_normoxia_kds_only.pdf"), width= 4, height=3)



#ribo only pca

qc_input_normoxia_kds_ribo <- qc_input_normoxia_kds[, grep("ribo", colnames(qc_input_normoxia_kds))]

pca_qc_input_normoxia_kds_ribo <- prcomp(scale(x = qc_input_normoxia_kds_ribo,center = T, scale = T))

pca_qc_input_normoxia_kds_ribo <- 
  pca_qc_input_normoxia_kds_ribo$rotation %>%
  as.data.frame() 

pca_qc_input_normoxia_kds_ribo <- pca_qc_input_normoxia_kds_ribo %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)

pca <- ggplot(data = pca_qc_input_normoxia_kds_ribo, aes(x=PC1, y = PC2, color = treatment, shape = type))+ 
  geom_point(size=3) +
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/pca_with_outlier_normoxia_kds_RIBO.pdf"), width= 4, height=3)


#rna only pca

#with outlier


qc_input_normoxia_kds_rna <- qc_input_normoxia_kds[, grep("rna", colnames(qc_input_normoxia_kds))]

pca_qc_input_normoxia_kds_rna <- prcomp(scale(x = qc_input_normoxia_kds_rna,center = T, scale = T))

pca_qc_input_normoxia_kds_rna <- 
  pca_qc_input_normoxia_kds_rna$rotation %>%
  as.data.frame() 

pca_qc_input_normoxia_kds_rna <- pca_qc_input_normoxia_kds_rna %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)

pca <- ggplot(data = pca_qc_input_normoxia_kds_rna, aes(x=PC1, y = PC2, color = treatment, shape = type))+ 
  geom_point(size=3) +
  scale_shape_manual(values = c(17))+
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/pca_with_outlier_normoxia_kds_rna.pdf"), width= 4, height=3)




#drugs
qc_input_normoxia_comp <- qc_input_normoxia[, grep("si", colnames(qc_input_normoxia), invert = TRUE)]

mycor_norm_comp <- cor(qc_input_normoxia_comp, method = "pearson")
heatmap <- pheatmap::pheatmap(mat = mycor_norm_comp)
heatmap
ggsave(heatmap, filename=here("plots/fig1/qc/heatmap_norm_comp.pdf"), width=11.5, height=11)


#ribo only

qc_input_normoxia_comp_ribo <- qc_input_normoxia_comp[, grep("ribo", colnames(qc_input_normoxia_comp))]

pca_qc_input_normoxia_comp_ribo <- prcomp(scale(x = qc_input_normoxia_comp_ribo,center = T, scale = T))

pca_qc_input_normoxia_comp_ribo <- 
  pca_qc_input_normoxia_comp_ribo$rotation %>%
  as.data.frame() 

pca_qc_input_normoxia_comp_ribo <- pca_qc_input_normoxia_comp_ribo %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)

pca <- ggplot(data = pca_qc_input_normoxia_comp_ribo, aes(x=PC1, y = PC2, color = treatment, shape = type))+ 
  geom_point(size=3) +
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/pca_with_outlier_normoxia_comp_ribo.pdf"), width= 4, height=3)




#rna only

qc_input_normoxia_comp_rna <- qc_input_normoxia_comp[, grep("rna", colnames(qc_input_normoxia_comp))]

pca_qc_input_normoxia_comp_rna <- prcomp(scale(x = qc_input_normoxia_comp_rna,center = T, scale = T))

pca_qc_input_normoxia_comp_rna <- 
  pca_qc_input_normoxia_comp_rna$rotation %>%
  as.data.frame() 

pca_qc_input_normoxia_comp_rna <- pca_qc_input_normoxia_comp_rna %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)

pca <- ggplot(data = pca_qc_input_normoxia_comp_rna, aes(x=PC1, y = PC2, color = treatment, shape = type))+ 
  geom_point(size=3) +
  scale_shape_manual(values = c(17))+
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/pca_with_outlier_normoxia_comp_rna.pdf"), width= 4, height=3)



#Hypoxia only

mycor_hyp <- cor(qc_input_hypoxia, method = "pearson")

heatmap <- pheatmap::pheatmap(mat = mycor_hyp)
heatmap
ggsave(heatmap, filename=here("plots/fig1/qc/heatmap_with_outlier_hyp.pdf"), width=11.5, height=11)



#KDs -- with outlier
qc_input_hypoxia_kds <- qc_input_hypoxia[, grep("si", colnames(qc_input_hypoxia))]

mycor_hyp_kds <- cor(qc_input_hypoxia_kds, method = "pearson")
heatmap <- pheatmap::pheatmap(mat = mycor_hyp_kds)
heatmap
ggsave(heatmap, filename=here("plots/fig1/qc/heatmap_hyp_kds.pdf"), width=11.5, height=11)




#KDs -- without outlier
qc_input_hypoxia_kds_no_out <- qc_input_hypoxia_kds[, grep("Hypoxic_si3e_RepB_rnaseq", colnames(qc_input_hypoxia_kds), invert=TRUE)]

mycor_hyp_kds_no_out <- cor(qc_input_hypoxia_kds_no_out, method = "pearson")
heatmap <- pheatmap::pheatmap(mat = mycor_hyp_kds_no_out)
heatmap
ggsave(heatmap, filename=here("plots/fig1/qc/heatmap_hyp_kds_no_outlier.pdf"), width=11.5, height=11)





#drugs
qc_input_hypoxia_comp <- qc_input_hypoxia[, grep("si", colnames(qc_input_hypoxia), invert = TRUE)]

mycor_hyp_comp <- cor(qc_input_hypoxia_comp, method = "pearson")
heatmap <- pheatmap::pheatmap(mat = mycor_hyp_comp)
heatmap
ggsave(heatmap, filename=here("plots/fig1/qc/heatmap_hyp_comp.pdf"), width=11.5, height=11)




#pca

#ribo only

qc_input_hypoxia_comp_ribo <- qc_input_hypoxia_comp[, grep("ribo", colnames(qc_input_hypoxia_comp))]

pca_qc_input_hypoxia_comp_ribo <- prcomp(scale(x = qc_input_hypoxia_comp_ribo,center = T, scale = T))

pca_qc_input_hypoxia_comp_ribo <- 
  pca_qc_input_hypoxia_comp_ribo$rotation %>%
  as.data.frame() 

pca_qc_input_hypoxia_comp_ribo <- pca_qc_input_hypoxia_comp_ribo %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)

pca <- ggplot(data = pca_qc_input_hypoxia_comp_ribo, aes(x=PC1, y = PC2, color = treatment, shape = type))+ 
  geom_point(size=3) +
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/pca_with_outlier_hypoxia_comp_ribo.pdf"), width= 4, height=3)




#rna only

qc_input_hypoxia_comp_rna <- qc_input_hypoxia_comp[, grep("rna", colnames(qc_input_hypoxia_comp))]

pca_qc_input_hypoxia_comp_rna <- prcomp(scale(x = qc_input_hypoxia_comp_rna,center = T, scale = T))

pca_qc_input_hypoxia_comp_rna <- 
  pca_qc_input_hypoxia_comp_rna$rotation %>%
  as.data.frame() 

pca_qc_input_hypoxia_comp_rna <- pca_qc_input_hypoxia_comp_rna %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)

pca <- ggplot(data = pca_qc_input_hypoxia_comp_rna, aes(x=PC1, y = PC2, color = treatment, shape = type))+ 
  geom_point(size=3) +
  scale_shape_manual(values = c(17))+
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/pca_with_outlier_hypoxia_comp_rna.pdf"), width= 4, height=3)


```


# PCA plot with outlier

```{r pca}

pca_res_orig <- prcomp(scale(x = qcinput,center = T, scale = T))

pca_res_data <- 
  pca_res_orig$rotation %>%
  as.data.frame() 


pca_res_data <- pca_res_data %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)


pca <- ggplot(data = pca_res_data, aes(x=PC1, y = PC2, color = treatment, shape = hypoxia))+ 
  geom_point(size=3) +
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/pca_with_outlier.pdf"), width= 4, height=3)






# Normoxia only


pca_res_norm <- prcomp(scale(x = qc_input_normoxia,center = T, scale = T))

pca_res_norm <- 
  pca_res_norm$rotation %>%
  as.data.frame() 


pca_res_norm <- pca_res_norm %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)

pca <- ggplot(data = pca_res_norm, aes(x=PC1, y = PC2, color = treatment, shape = type))+ 
  geom_point(size=3) +
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/pca_with_outlier_normoxia.pdf"), width= 4, height=3)



# separate ribo vs rna

#ribo

qc_input_norm_ribo <- qc_input_normoxia[, grep("ribo", colnames(qc_input_normoxia))]


pca_res_norm_ribo <- prcomp(scale(x = qc_input_norm_ribo,center = T, scale = T))

pca_res_norm_ribo <- 
  pca_res_norm_ribo$rotation %>%
  as.data.frame() 


pca_res_norm_ribo <- pca_res_norm_ribo %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)

pca <- ggplot(data = pca_res_norm_ribo, aes(x=PC1, y = PC2, color = treatment, shape = type))+ 
  geom_point(size=3) +
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/pca_with_outlier_normoxia_RIBO.pdf"), width= 4, height=3)


#ribo only KDS




#ribo only drugs













#RNA

qc_input_norm_rna <- qc_input_normoxia[, grep("rna", colnames(qc_input_normoxia))]


pca_res_norm_rna <- prcomp(scale(x = qc_input_norm_rna,center = T, scale = T))

pca_res_norm_rna <- 
  pca_res_norm_rna$rotation %>%
  as.data.frame() 


pca_res_norm_rna <- pca_res_norm_rna %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)

pca <- ggplot(data = pca_res_norm_rna, aes(x=PC1, y = PC2, color = treatment, shape = type))+ 
  geom_point(size=3) +
  scale_shape_manual(values = c(17)) +
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/pca_with_outlier_normoxia_rna.pdf"), width= 4, height=3)

# rna only kds






#rna only drugs




#Hypoxia only
pca_res_hyp <- prcomp(scale(x = qc_input_hypoxia,center = T, scale = T))

pca_res_hyp <- 
  pca_res_hyp$rotation %>%
  as.data.frame() 


pca_res_hyp <- pca_res_hyp %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)

pca <- ggplot(data = pca_res_hyp, aes(x=PC1, y = PC2, color = treatment, shape = type))+ 
  geom_point(size=3) +
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/pca_with_outlier_hypoxia.pdf"), width= 4, height=3)


#hypoxia without outlier

qcinput_hyp_no_outlier <- qc_input_hypoxia[,colnames(qc_input_hypoxia) != "Hypoxic_si3e_RepB_rnaseq"] 


pca_res_hyp <- prcomp(scale(x = qcinput_hyp_no_outlier,center = T, scale = T))

pca_res_hyp <- 
  pca_res_hyp$rotation %>%
  as.data.frame() 


pca_res_hyp <- pca_res_hyp %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)

pca <- ggplot(data = pca_res_hyp, aes(x=PC1, y = PC2, color = treatment, shape = type))+ 
  geom_point(size=3) +
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/pca_without_outlier_hypoxia.pdf"), width= 4, height=3)





```


### Heatmap without outlier
We removed the outlier and reran the correlation plot

```{r remove outlier from qcinput2}

#remake qcinput as qcinput2 without the outlier

qcinput2 <- qcinput[,colnames(qcinput) != "Hypoxic_si3e_RepB_rnaseq"] 


qcinput2 <- qcinput2[rowSds(qcinput2) > 1,] 

dim(qcinput2)

mycor2 <- cor(qcinput2, method = "pearson")

heatmap_2 <- pheatmap::pheatmap(mat = mycor2)

ggsave(heatmap_2, filename=here("plots/fig1/qc/heatmap_without_outlier.pdf"), width=11.5, height=11)

```



## PCA plot without the outlier.

```{r pca}

pca_res <- prcomp(scale(x = qcinput2,center = T, scale = T))

#pca plot by hypoxia and rep

pca_res_data <- 
  pca_res$rotation %>%
  as.data.frame() 


pca_res_data <- pca_res_data %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)


#first by hypoxia and rep

pca <- ggplot(data = pca_res_data, aes(x=PC1, y = PC2, color = treatment, shape = hypoxia))+ 
  geom_point(size=5) +
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/pca_without_outlier.pdf"), width= 4, height=3)


```


## Heatmap of ribosome profiling data
```{r ribosome heatmap}

ribonames <- qcinput2[ , grep("riboseq", colnames(qcinput2))]

ribocolnames <- colnames(ribonames)

qcinput_ribo <- qcinput2[,colnames(qcinput2) %in% ribocolnames] 


qcinput_ribo <- qcinput_ribo[rowSds(qcinput_ribo) > 1,] 

dim(qcinput_ribo)

#heatmap sans outlier
mycor_ribo <- cor(qcinput_ribo, method = "pearson")

ribo_heatmap <- pheatmap::pheatmap(mat = mycor_ribo)

ggsave(ribo_heatmap, filename=here("plots/fig1/qc/ribo_heatmap.pdf"), width=11.5, height=11)


```



## PCA plot for ribosome profiling data
```{r ribo pca plot}

pca_res_ribo <- prcomp(scale(x = qcinput_ribo,center = T, scale = T))

#pca plot by hypoxia and rep

pca_res_data_ribo <- 
  pca_res_ribo$rotation %>%
  as.data.frame() 


pca_res_data_ribo <- pca_res_data_ribo %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)


pca <- ggplot(data = pca_res_data_ribo, aes(x=PC1, y = PC2))+ 
  geom_point(aes(color = treatment, shape=hypoxia)) +
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/ribo_pca.pdf"), width= 4, height=3)


```

## Heatmap for RNA sequencing data
```{r rna heatmap setup}

rnanames <- qcinput2[ , grep("rnaseq", colnames(qcinput2))]

rnacolnames <- colnames(rnanames)

qcinput_rna <- qcinput2[,colnames(qcinput2) %in% rnacolnames] 


qcinput_rna <- qcinput_rna[rowSds(qcinput_rna) > 1,] 

dim(qcinput_rna)


mycor_rna <- cor(qcinput_rna, method = "pearson")

rna_heatmap <- pheatmap::pheatmap(mat = mycor_rna)

ggsave(rna_heatmap, filename=here("plots/fig1/qc/rna_heatmap.pdf"), width=11.5, height=11)



```


## PCA plot for RNA sequencing data
```{r rna pca plot}

pca_res_rna <- prcomp(scale(x = qcinput_rna,center = T, scale = T))


#pca plot by hypoxia and rep

pca_res_data_rna <- 
  pca_res_rna$rotation %>%
  as.data.frame() 


pca_res_data_rna <- pca_res_data_rna %>% 
  rownames_to_column("sample") %>%
  separate(col = sample, into = c("hypoxia","treatment","rep","type"), sep = "_", remove = T)


pca <- ggplot(data = pca_res_data_rna, aes(x=PC1, y = PC2))+ 
  geom_point(aes(color = treatment, shape=hypoxia)) +
  theme_few()

ggplotly(pca)

ggsave(pca, filename=here("plots/fig1/qc/rna_pca.pdf"), width= 4, height=3)


```

## Confirm knockdowns
```{r blacklist set up, echo=TRUE, message=FALSE, warning=FALSE}
nonpolyA <- readxl::read_xlsx(path = here("accessories","polyAminus.xlsx"), sheet = 1) %>% 
   pull(Gene)

nonpolyA_ID <- geneInfo %>%
  filter(symbol %in% nonpolyA) %>% 
  pull(gene_id) %>% 
  unique()


# we want to remove any genes that match these annotation categories # before we do DE analysis
blacklist_smRNA <- c("snoRNA","miRNA","snRNA","rRNA","Mt_tRNA","scaRNA","Mt_tRNA","Mt_rRNA")

# there are 5,355 genes we want to exclude
blacklist_smRNA_ID <-geneInfo %>%
  filter(biotype %in% blacklist_smRNA) %>% 
  pull(gene_id) %>% 
  unique()


genes_to_kill <- unique(c(blacklist_smRNA_ID,nonpolyA_ID))
head(keepGenes)
head(genes_to_kill)

```


### Set up translational efficiency analysis
RNA and ribo counts are separated out

```{r deltaTE count setup, message=FALSE, warning=FALSE}
# separate out RNA counts versus ribo count
# from https://currentprotocols.onlinelibrary.wiley.com/doi/full/10.1002/cpmb.108

allCounts <- myTxi$counts %>%
  as.data.frame() %>% 
  rownames_to_column(var = "gene") %>% 
  filter(gene %in% keepGenes) %>%
  filter(!gene %in% genes_to_kill) %>%
  column_to_rownames(var = "gene") %>% 
  as.matrix() %>%
  round(.,0)

allCountsNorm <- edgeR::cpm(allCounts)



ribo_counts_norm <- as.matrix(allCounts[,grep(pattern = "ribo",colnames(allCountsNorm))])
rna_counts_norm <- as.matrix(allCounts[,grep(pattern = "rna",colnames(allCountsNorm))])
```

```{r gene-specific, fig.height=11, fig.width=10, warning=FALSE}

myGs <- c("EIF3E", "EIF3D")

myGtable <- geneInfo %>% 
  filter(symbol %in% myGs) %>%
  dplyr::select(gene_id,symbol) %>% 
  unique()

myGdata <- allCountsNorm[myGtable$gene_id,] %>%
  reshape2::melt() %>% 
  filter(Var2 != "Hypoxic_si3e_RepB_rnaseq") %>% 
  separate(col = Var2, into = c("condition", "treatment", "rep", "library")) 
  
colnames(myGdata)[1] <- "gene_id"
colnames(myGdata)[6] <- "cpm"
myGdata$cpm <- log2(myGdata$cpm + 1)

myGdata <- left_join(myGdata, myGtable)

genestoplot <- myGdata %>% 
  filter(treatment== "sictrl" |
           treatment== "si3e" |
           treatment == "si3d") %>% 
  filter(library =="rnaseq") %>% 
  filter(condition=="Hypoxic")


# myGdata %>%
#   filter(treatment != "DMSO" & treatment != "a209") %>% 
#   filter(condition=="Normoxic") %>% 
#   ggpubr::ggbarplot(x="treatment", y="cpm")+
#   facet_wrap(~gene_id+library)





# ggpubr::ggdotplot(data = genestoplot, x = "symbol", y = "cpm", fill = "symbol", facet.by = "treatment")

ggsave(filename=here("plots/fig1/qc/confirm_KD.pdf"), width= 4, height=3)
```
New bars

Normoxia
```{r}
myGdata$treatment <- relevel(factor(myGdata$treatment), ref = "sictrl")
myGdata$library <- relevel(factor(myGdata$library), ref = "rnaseq")

custom_colors <- c("sictrl" = "grey", "si3d" = si3d_col, "si3e" = si3e_col)



myGdata %>%
  filter(treatment != "DMSO" & treatment != "a209") %>%
  filter(condition == "Normoxic") %>%
  group_by(treatment, symbol, library) %>%
  summarise(mean_cpm = mean(cpm, na.rm = TRUE),
            sd_cpm = sd(cpm, na.rm = TRUE)) %>%
  ggplot(aes(x = treatment, y = mean_cpm, fill = treatment)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_errorbar(aes(ymin = mean_cpm - sd_cpm, ymax = mean_cpm + sd_cpm), 
                width = 0.2, position = position_dodge(0.9)) +
  facet_wrap(~symbol + library, scales = "free_y") +
  labs(y = "Mean CPM", x = "Treatment") +
  scale_fill_manual(values = custom_colors)+
  theme_cowplot()

ggsave(here("plots/3eanddbars.pdf"))
```
hypoxia
```{r}
#need to get rid of the outlier here

myGdata %>%
  filter(treatment != "DMSO" & treatment != "a209") %>%
  filter(condition == "Hypoxic") %>%
  group_by(treatment, symbol, library) %>%
  summarise(mean_cpm = mean(cpm, na.rm = TRUE),
            sd_cpm = sd(cpm, na.rm = TRUE)) %>%
  ggplot(aes(x = treatment, y = mean_cpm, fill = treatment)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_errorbar(aes(ymin = mean_cpm - sd_cpm, ymax = mean_cpm + sd_cpm), 
                width = 0.2, position = position_dodge(0.9)) +
  facet_wrap(~symbol + library, scales = "free_y") +
  labs(y = "Mean CPM", x = "Treatment") +
  scale_fill_manual(values = custom_colors)+
  theme_cowplot()

ggsave(here("plots/3eanddbars_hypoxia.pdf"))
```

HIF1alpha in hypoxia

S3 - ribo and rna do not change with KD. Bring in TE later.

Make same for a209 and DMSO

```{r}


myGtable <- geneInfo %>% 
  filter(symbol =="HIF1A") %>%
  dplyr::select(gene_id,symbol) %>% 
  unique()



myGdata <- allCountsNorm[myGtable$gene_id,] %>%
  reshape2::melt() %>% 
  mutate(gene_id=myGtable$gene_id) %>% 
    mutate(symbol=myGtable$symbol) %>% 
  rownames_to_column("Var2") %>% 
  filter(Var2 != "Hypoxic_si3e_RepB_rnaseq") %>% 
  separate(col = Var2, into = c("condition", "treatment", "rep", "library")) 
  

colnames(myGdata)[5] <- "cpm"
myGdata$cpm <- log2(myGdata$cpm + 1)


myGdata$treatment <- relevel(factor(myGdata$treatment), ref = "sictrl")
myGdata$library <- relevel(factor(myGdata$library), ref = "rnaseq")

custom_colors <- c("sictrl" = "grey", "si3d" = si3d_col, "si3e" = si3e_col)


myGdata %>%
  filter(treatment != "DMSO" & treatment != "a209") %>%
  filter(condition == "Hypoxic") %>%
  group_by(treatment, symbol, library) %>%
  summarise(mean_cpm = mean(cpm, na.rm = TRUE),
            sd_cpm = sd(cpm, na.rm = TRUE)) %>%
  ggplot(aes(x = treatment, y = mean_cpm, fill = treatment)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_errorbar(aes(ymin = mean_cpm - sd_cpm, ymax = mean_cpm + sd_cpm), 
                width = 0.2, position = position_dodge(0.9)) +
  facet_wrap(~symbol + library, scales = "free_y") +
  labs(y = "Mean CPM", x = "Treatment") +
  scale_fill_manual(values = custom_colors)+
  theme_cowplot()

ggsave(here("plots/hif1a.pdf"))




```

DMSO and a209
```{r}
custom_colors <- c("DMSO" = "grey", "a209" = a209_col)

myGdata$treatment <- relevel(factor(myGdata$treatment), ref = "DMSO")


myGdata %>%
  filter(treatment == "DMSO" | treatment == "a209") %>%
  filter(condition == "Hypoxic") %>%
  group_by(treatment, symbol, library) %>%
  summarise(mean_cpm = mean(cpm, na.rm = TRUE),
            sd_cpm = sd(cpm, na.rm = TRUE)) %>%
  ggplot(aes(x = treatment, y = mean_cpm, fill = treatment)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_errorbar(aes(ymin = mean_cpm - sd_cpm, ymax = mean_cpm + sd_cpm), 
                width = 0.2, position = position_dodge(0.9)) +
  facet_wrap(~symbol + library, scales = "free_y") +
  labs(y = "Mean CPM", x = "Treatment") +
  scale_fill_manual(values = custom_colors)+
  theme_cowplot()

ggsave(here("plots/hif1a_a209.pdf"))


```


















```{r filter out outlier in analysis for part 2, eval=FALSE, warning=FALSE, include=FALSE}

#remove outlier from metadata

metadata2 <- metadata %>% 
  filter(metadata$sample_id != "NM2023_0113")
```


# RIBOCODE ----------------------------------------------------------------------
# load bam files

