---
title: "QC"
author: "Kate"
date: "2023-11-01"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r install packages, include=FALSE}
library(here)
library(tximport)
library(tidyverse)
library(EnhancedVolcano)
library(janitor)
library(ggthemes)
library(DESeq2)
library(plotly)
library(ggpubr)
library(reshape2)


library(here)
library(janitor)
library(tximport)
library(tidyverse)
library(EnhancedVolcano)
library(msigdbr)
library(viridis)
library(gridExtra)
library(ggthemes)
library(ggrepel)
library(scales)
library(DESeq2)
library(limma)
library(clusterProfiler)
library(plotly)
library(ggokabeito)
library(ggvenn)
library(ggpubr)
library(DT)
library(edgeR)
library(cowplot)
library(ggExtra)
library(ggprism)
library(GeneOverlap)
library(writexl)
library(reshape2)
```

```{r}
a209_col = "#CC79A7"
si3e_col = "#56B4E9"
si3d_col = "#E69F00"
c8430_col = "#009E73"
```

## Load transcript-level data

Importing metadata and all salmon counts
```{r meta data}
metadata <- read_csv(here("accessories","metadata_NM_8430.csv"), skip = 0) %>% 
  clean_names()

metadata <- metadata %>% 
  mutate(library = ifelse(library_prep_kit=="Qiagen_miRNA", "ribo", "rna"))

#ribosome lysate = ribosome seq
#rRNA depletion = RNA seq
#rename variables remove outlier

metadata <- metadata %>% 
  dplyr::rename("treatment" = "treatment1") %>% 
  dplyr::rename("rep" = "treatment2") %>% 
  dplyr::rename("condition" = "genotype")

#edit factor levels
metadata$library <- relevel(x = factor(metadata$library), ref = "rna")
metadata$library <- recode(metadata$library, rnaseq = "rna", riboseq = "ribo")

metadata$treatment <- relevel(x = factor(metadata$treatment), ref = "DMSO")
metadata$condition <- relevel(x = factor(metadata$condition), ref = "normoxic")
metadata$condition <- recode(metadata$condition, normoxic = "normoxia", hypoxic = "hypoxia")
metadata$rep <- recode(metadata$rep, RepA = "a", RepB = "b", RepC = "c")

metadata <- metadata %>%
dplyr::select(sample_id, library, treatment, condition, rep) 

```

```{r import gene info}

#import key on RNA type, remove premRNA
geneInfo <- read_csv(here("accessories","gencode.v26.primary.info.csv.gz"), col_names = F) %>%
  filter(!str_detect(X2, "pre_"))

colnames(geneInfo) <- c("gene_id","transcript_id","biotype","symbol")

#import transcripts for gene to transcript mapping
tx2gene <- geneInfo[,c(2,1)]
gene_symbol <- geneInfo %>% dplyr::select(gene_id, symbol) %>% unique()

```


@Neel, here, I am not filtering for protein coding only as I do in calculate_te.Rmd.  Is that okay for these qc steps?

```{r txi import, include=FALSE}
#name files to import
myquantfiles <- paste(here(), "/salmon/",
                      metadata$sample_id,
                      "/quant.sf.gz",
                      sep = "")

names(myquantfiles) <- paste(metadata$condition, metadata$treatment, metadata$rep, metadata$library, sep = "_")

#import salmon files
myTxi <- tximport(myquantfiles, type = "salmon", tx2gene = tx2gene)

view(myTxi$counts)

```


No spike ins because i aligned only to the human

```{r}
# visualize distribution of counts
hist_count <- hist(log2(rowSums(myTxi$counts)), breaks = 50)
#determine threshold

#distribution of abundance histogram
#hist_abundance <- hist(log2(rowSums(myTxi$abundance)), breaks = 50)

#only keeping rows with counts  log2> 10 across all samples - want around 10-15k genes - want genes that are highly expressed
keepGenes <- rownames(myTxi$counts[log2(rowSums(myTxi$counts)) > 10 ,])

#removes noise and keeps only highly expressed genes
# results show bimodal distribution
#1 RNA per cell = 1 TPM
```




```{r standard deviation filtering}
# log transforming with pseudocount to avoid undefined numbers
qcinput <- log2(myTxi$abundance[keepGenes,] + 1) 

qcinput <- qcinput[rowSds(qcinput) > 1,] 

dim(qcinput)
```



```{r}

mycor <- cor(qcinput, method = "pearson")

heatmap <- pheatmap::pheatmap(
  mat = mycor,
  border_color = "black",  # Black borders around cells
   #color = colorRampPalette(c("blue", "white", "red"))(100),  # Custom color gradient
   scale = "none",  
  cluster_rows = TRUE,  
  cluster_cols = TRUE,  
  fontsize = 12 
)

heatmap

ggsave(here("plots/plots_8430/qc.pdf"), heatmap)
```


Split normoxia and hypoxia
```{r}

qc_input_normoxia <- qcinput[, grep("normoxia", colnames(qcinput))]

qc_input_hypoxia <- qcinput[, grep("hypoxia", colnames(qcinput))]

```

# Visualizations

# Figure 1: Normoxia

```{r}
#KDs
mycor_norm <- cor(qc_input_normoxia, method = "pearson")

colnames(mycor_norm) <- gsub("^normoxia_", "", colnames(mycor_norm))  # Remove "normoxia_" prefix
rownames(mycor_norm) <- gsub("^normoxia_", "", rownames(mycor_norm))  # Same for row names

# Replace underscores with spaces
colnames(mycor_norm) <- gsub("_", " ", colnames(mycor_norm))
rownames(mycor_norm) <- gsub("_", " ", rownames(mycor_norm))

# Create the heatmap with the cleaned-up names
heatmap_normoxia <- pheatmap::pheatmap(
  mat = mycor_norm,
  border_color = "black",  # Black borders around cells
   #color = colorRampPalette(c("blue", "white", "red"))(100),  # Custom color gradient
   scale = "none",  
  cluster_rows = TRUE,  
  cluster_cols = TRUE,  
  fontsize = 12 
)


heatmap_normoxia
ggsave(heatmap_normoxia, filename=here("plots/qc/heatmap_normoxia_kds.pdf"), width=4.5, height=4)

# RIBO average correlation coefficient

qc_input_normoxia_ribo <- qc_input_normoxia[, grep("ribo", colnames(qc_input_normoxia))]

mycor_ribo_norm <- cor(qc_input_normoxia_ribo, method = "pearson")

# Extract the lower triangular part of the matrix without the diagonal
lower_tri <- mycor_ribo_norm[lower.tri(mycor_ribo_norm, diag=FALSE)]

# Calculate the average correlation coefficient (excluding diagonal)
average_correlation <- mean(lower_tri)

# Print the result
average_correlation


# rna average correlation coefficient

qc_input_normoxia_rna <- qc_input_normoxia[, grep("rna", colnames(qc_input_normoxia))]

mycor_rna_norm <- cor(qc_input_normoxia_rna, method = "pearson")


# Extract the lower triangular part of the matrix without the diagonal
lower_tri <- mycor_rna_norm[lower.tri(mycor_rna_norm, diag=FALSE)]

# Calculate the average correlation coefficient (excluding diagonal)
average_correlation <- mean(lower_tri)

# Print the result
average_correlation

```


# Figure 2: Hypoxia, knockdowns


Heatmap

```{r}
mycor_hyp_kds <- cor(qc_input_hypoxia, method = "pearson")

colnames(mycor_hyp_kds) <- gsub("^hypoxia_", "", colnames(mycor_hyp_kds))  # Remove "hypoxia_" prefix
rownames(mycor_hyp_kds) <- gsub("^hypoxia_", "", rownames(mycor_hyp_kds))  # Same for row names

# Replace underscores with spaces
colnames(mycor_hyp_kds) <- gsub("_", " ", colnames(mycor_hyp_kds))
rownames(mycor_hyp_kds) <- gsub("_", " ", rownames(mycor_hyp_kds))

# Create the heatmap with the cleaned-up names
heatmap_hypoxia <- pheatmap::pheatmap(
  mat = mycor_hyp_kds,
  border_color = "black",  # Black borders around cells
   #color = colorRampPalette(c("blue", "white", "red"))(100),  # Custom color gradient
   scale = "none",  
  cluster_rows = TRUE,  
  cluster_cols = TRUE,  
  fontsize = 12 
)





heatmap_hypoxia
ggsave(heatmap_hypoxia, filename=here("plots/qc/heatmap_hyp_kds_w_outlier.pdf"), width=4.5, height=4)

```

```{r}

# RIBO average correlation coefficient

qc_input_hypoxia_ribo <- qc_input_hypoxia[, grep("ribo", colnames(qc_input_hypoxia))]

mycor_ribo_hyp <- cor(qc_input_hypoxia_ribo, method = "pearson")


# Extract the lower triangular part of the matrix without the diagonal
lower_tri <- mycor_ribo_hyp[lower.tri(mycor_ribo_hyp, diag=FALSE)]

# Calculate the average correlation coefficient (excluding diagonal)
average_correlation <- mean(lower_tri)

# Print the result
average_correlation



# rna average correlation coefficient

qc_input_hypoxia_rna <- qc_input_hypoxia[, grep("rna", colnames(qc_input_hypoxia))]

mycor_rna_hyp <- cor(qc_input_hypoxia_rna, method = "pearson")


# Extract the lower triangular part of the matrix without the diagonal
lower_tri <- mycor_rna_hyp[lower.tri(mycor_rna_hyp, diag=FALSE)]

# Calculate the average correlation coefficient (excluding diagonal)
average_correlation <- mean(lower_tri)

# Print the result
average_correlation


```

qc looks great.

# Calculate TE


### Filter
```{r keep only protein coding genes}

pc <- geneInfo %>%
  filter(biotype =="protein_coding") %>%
  pull(gene_id) %>%
  unique()

pc_keep <- pc[pc %in% keepGenes]

# remove non-polyadenylated mRNAs
nonpolyA <- readxl::read_xlsx(path = here("accessories","polyAminus.xlsx"), sheet = 1) %>% 
   pull(Gene)

nonpolyA_ID <- geneInfo %>%
  filter(symbol %in% nonpolyA) %>% 
  pull(gene_id) %>% 
  unique()

# we want to remove any genes that match these annotation categories # before we do DE analysis
blacklist_smRNA <- c("snoRNA","miRNA","snRNA","rRNA","Mt_tRNA","scaRNA","Mt_tRNA","Mt_rRNA")

blacklist_smRNA_ID <- geneInfo %>%
  filter(biotype %in% blacklist_smRNA) %>% 
  pull(gene_id) %>% 
  unique()

genes_to_kill <- unique(c(blacklist_smRNA_ID,nonpolyA_ID))
```


```{r filter and extract counts}

allCounts <- myTxi$counts %>%
  as.data.frame() %>% 
  rownames_to_column(var = "gene") %>% 
  filter(gene %in% pc_keep) %>%
  filter(!gene %in% genes_to_kill) %>%
  column_to_rownames(var = "gene") %>% 
  as.matrix() %>%
  round(.,0)

all_raw_filtered_counts <- allCounts %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id")

#export allCounts for DeSeq analysis
write_csv(all_raw_filtered_counts, here("counts/calculate_te/raw_filtered_allCounts_8430.csv"))


allCountsNorm <- edgeR::cpm(allCounts[,order(colnames(allCounts))])

riboN <- log2(1+ as.matrix(allCountsNorm[,grep(pattern = "ribo",colnames(allCountsNorm))]))
rnaN <- log2(1 + as.matrix(allCountsNorm[,grep(pattern = "rna",colnames(allCountsNorm))])) 


TE_original <- riboN - rnaN

hist(TE_original)

median(as.data.frame(TE_original) %>% melt() %>% pull(value))

#center data --

TE <- (riboN - rnaN) + 0.5275673
# this is why im using 0.5275673! -- all changes are relative


hist(TE)

colnames(TE) <- gsub(pattern = "ribo", replacement = "TE", x = colnames(TE))


gghistogram(as.data.frame(TE) %>% melt(), x="value")
ribo_counts <- as.matrix(allCounts[,grep(pattern = "ribo",colnames(allCounts))])
rna_counts <- as.matrix(allCounts[,grep(pattern = "rna",colnames(allCounts))])

```


### calculation
```{r}

full_te_list <- TE %>% 
  as.data.frame() %>%
  rownames_to_column("gene_id") %>% 
  left_join(gene_symbol, by="gene_id") %>% 
  dplyr::select(symbol, everything())


full_rna_list <- rnaN %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id")


full_ribo_list <- riboN %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id")

full_te_list <- full_te_list %>% 
  left_join(full_rna_list, by="gene_id")

complete_list <- full_te_list %>% 
  left_join(full_ribo_list, by= "gene_id")


write_csv(complete_list, here("counts/calculate_te/complete_te_rna_ribo_list_8430.csv"))


```



```{r reformat data}
all_long <- complete_list %>% 
  melt() %>% 
    separate(variable, into=(c("condition", "treatment", "rep", "type")))

```

Sanity check:

Look at RNA for eIF3e and eIF3d KD compared to control


```{r eIF3e RNA}

# rna_3e_formatted <- all_long %>% 
#   filter(symbol=="EIF3E") %>% 
#   filter(condition=="normoxia") %>% 
#   filter(type=="rna")
#   
# 
# ggplot(rna_3e_formatted, aes(x=treatment, y=value))+
#   geom_point()

```


```{r eIF3d RNA}

# rna_3d_formatted <- all_long %>% 
#   filter(symbol=="EIF3D") %>% 
#   filter(condition=="normoxia") %>% 
#   filter(type=="rna")
#   
# 
# ggplot(rna_3d_formatted, aes(x=treatment, y=value))+
#   geom_point()

```


Example of TE: Analyze ATF4 rna ribo and TE in control compared to KD

Example- ATF4
```{r}
atf4_data <- all_long %>% 
  filter(symbol=="ATF4") %>% 
  filter(condition=="normoxia")


ggplot(atf4_data, aes(x=treatment, y=value))+
  geom_point()+
  facet_wrap(~type)
```

```{r}
eif3e_data <- all_long %>% 
  filter(symbol=="EIF3E") %>% 
  filter(condition=="hypoxia")


ggplot(eif3e_data, aes(x=treatment, y=value))+
  geom_point()+
  facet_wrap(~type)
```

Edit metadata names relevel factors in before exporting for analysis

```{r}

#change names
metadata$library <- recode(metadata$library, rnaseq = "rna", riboseq = "ribo", teseq="te")
metadata$rep <- recode(metadata$rep, RepA = "a", RepB = "b", RepC = "c")
metadata$condition <- recode(metadata$condition, Normoxic = "normoxia", Hypoxic = "hypoxia")

#update factor levels
metadata$library <- relevel(x = factor(metadata$library), ref = "rna")
metadata$treatment <- factor(metadata$treatment, levels= c("DMSO", "8430"))
metadata$condition <- relevel(x = factor(metadata$condition), ref = "normoxia")

# write metadata for use in future analyses

write_csv(metadata, here("counts/calculate_te/metadata_8430.csv"))

```

In the following analyses, pull TE and metadata from this file
# ###############################################################################
# just doing TE here

## HIF1alpha levels after compound treatment

```{r}

all_gene_te <- complete_list

all_gene_te_long <- all_gene_te %>% 
  melt() %>% 
tidyr::extract(variable, into = c("condition", "treatment", "rep", "type"), regex = "([^_]+)_([^_]+)_([^_]+)_([^_]+)") %>% 
  filter(condition=="hypoxia")

all_gene_te_long$treatment <- relevel(factor(all_gene_te_long$treatment), ref = "DMSO")


hif <-all_gene_te_long %>% 
  filter(symbol=="HIF1A") %>% 
  filter(type != "TE")

hif_summary <- hif %>%
  group_by(symbol, type, treatment) %>%
  summarise(
    mean_value = mean(value),
    sd = sd(value)  # Standard deviation
  )

hif_summary$type <- factor(hif_summary$type, levels=c("rna","ribo"))

ggplot(hif_summary, aes(x = treatment, y = mean_value, fill = treatment)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  geom_errorbar(aes(ymin = mean_value - sd, ymax = mean_value + sd), 
                width = 0.2, position = position_dodge(0.9)) +
  facet_wrap(~ symbol + type) +
  scale_fill_manual(values = c("grey", c8430_col)) +
  theme_cowplot()+
    theme(legend.position = "none")

ggsave(here("plots/plots_8430/hif_8430_hypoxia.pdf"), width = 4, height = 3)

```

## Acute hypoxic response: DMSO normoxia to hypoxia.

### TE changes
```{r}

#organize metadata
metadata_acute_DMSO <- metadata %>% 
  filter(treatment== "DMSO") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_acute_DMSO$library <- relevel(factor(metadata_acute_DMSO$library), ref = "rna")
metadata_acute_DMSO$condition <- relevel(factor(metadata_acute_DMSO$condition), ref = "normoxia")

levels(metadata_acute_DMSO$library)
levels(metadata_acute_DMSO$condition)

names <- as.character(metadata_acute_DMSO$name)

all_raw <- all_raw_filtered_counts %>% 
  column_to_rownames("gene_id")

acute_DMSO <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_DMSO <- DESeqDataSetFromMatrix(
  countData = acute_DMSO,
  colData = metadata_acute_DMSO,
  design = ~condition + library + condition:library
  )

ddsMat_acute_DMSO <- DESeq(ddsMat_acute_DMSO)
resultsNames(ddsMat_acute_DMSO)
res_acute_DMSO <- results(ddsMat_acute_DMSO, name=resultsNames(ddsMat_acute_DMSO)[4], cooksCutoff=F, independentFiltering=F)


res_acute_DMSO <- lfcShrink(dds = ddsMat_acute_DMSO, coef = resultsNames(ddsMat_acute_DMSO)[4], res = res_acute_DMSO) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_DMSO %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_DMSO %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

all_counts_acute_DMSO <- data.frame(upreg, downreg)

```


### RNA changes
```{r}

#organize metadata
metadata_acute_DMSO_rna <- metadata %>% 
  filter(treatment== "DMSO") %>% 
  filter(library=="rna") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_DMSO_rna$condition <- relevel(factor(metadata_acute_DMSO_rna$condition), ref = "normoxia")

levels(metadata_acute_DMSO_rna$condition)

names <- as.character(metadata_acute_DMSO_rna$name)

acute_DMSO_rna <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_DMSO_rna <- DESeqDataSetFromMatrix(
  countData = acute_DMSO_rna,
  colData = metadata_acute_DMSO_rna,
  design = ~condition
  )

ddsMat_acute_DMSO_rna <- DESeq(ddsMat_acute_DMSO_rna)
resultsNames(ddsMat_acute_DMSO_rna)
res_acute_DMSO_rna <- results(ddsMat_acute_DMSO_rna, name=resultsNames(ddsMat_acute_DMSO_rna)[2], cooksCutoff=F, independentFiltering=F)


res_acute_DMSO_rna <- lfcShrink(dds = ddsMat_acute_DMSO_rna, coef = resultsNames(ddsMat_acute_DMSO_rna)[2], res = res_acute_DMSO_rna) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_DMSO_rna %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_DMSO_rna %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="rna"

all_counts_acute_DMSO_rna <- data.frame(dataset, upreg, downreg)

```


### RIBO changes

```{r}

#organize metadata
metadata_acute_DMSO_ribo <- metadata %>% 
  filter(treatment== "DMSO") %>% 
  filter(library=="ribo") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_DMSO_ribo$condition <- relevel(factor(metadata_acute_DMSO_ribo$condition), ref = "normoxia")

levels(metadata_acute_DMSO_ribo$condition)

names <- as.character(metadata_acute_DMSO_ribo$name)

acute_DMSO_ribo <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_DMSO_ribo <- DESeqDataSetFromMatrix(
  countData = acute_DMSO_ribo,
  colData = metadata_acute_DMSO_ribo,
  design = ~condition
  )

ddsMat_acute_DMSO_ribo <- DESeq(ddsMat_acute_DMSO_ribo)
resultsNames(ddsMat_acute_DMSO_ribo)
res_acute_DMSO_ribo <- results(ddsMat_acute_DMSO_ribo, name=resultsNames(ddsMat_acute_DMSO_ribo)[2], cooksCutoff=F, independentFiltering=F)


res_acute_DMSO_ribo <- lfcShrink(dds = ddsMat_acute_DMSO_ribo, coef = resultsNames(ddsMat_acute_DMSO_ribo)[2], res = res_acute_DMSO_ribo) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_DMSO_ribo %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_DMSO_ribo %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="ribo"

all_counts_acute_DMSO_ribo <- data.frame(dataset, upreg, downreg)

```

## Visualize DMSO acute hypoxic changes
```{r}

acute_ctrl_counts <- rbind(all_counts_acute_DMSO_rna, all_counts_acute_DMSO_ribo) %>% melt()

acute_ctrl_counts$dataset <- as.factor(acute_ctrl_counts$dataset)
acute_ctrl_counts$dataset <- relevel(acute_ctrl_counts$dataset, ref = "rna")

ggbarplot(acute_ctrl_counts, x="variable", y="value")+
  facet_wrap(~dataset)+
    geom_text(aes(label = value), vjust = -0.3)+
  theme_cowplot()

ggsave(here("plots/plots_8430/acute_DMSO_counts.pdf"), width=4, height=4)


```


## Acute hypoxic response: 8430 normoxia to hypoxia.

### TE changes
```{r}

#organize metadata
metadata_acute_8430 <- metadata %>% 
  filter(treatment== "8430") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_acute_8430$library <- relevel(factor(metadata_acute_8430$library), ref = "rna")
metadata_acute_8430$condition <- relevel(factor(metadata_acute_8430$condition), ref = "normoxia")

levels(metadata_acute_8430$library)
levels(metadata_acute_8430$condition)

names <- as.character(metadata_acute_8430$name)

acute_8430 <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_8430 <- DESeqDataSetFromMatrix(
  countData = acute_8430,
  colData = metadata_acute_8430,
  design = ~condition + library + condition:library
  )

ddsMat_acute_8430 <- DESeq(ddsMat_acute_8430)
resultsNames(ddsMat_acute_8430)
res_acute_8430 <- results(ddsMat_acute_8430, name=resultsNames(ddsMat_acute_8430)[4], cooksCutoff=F, independentFiltering=F)


res_acute_8430 <- lfcShrink(dds = ddsMat_acute_8430, coef = resultsNames(ddsMat_acute_8430)[4], res = res_acute_8430) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_8430 %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_8430 %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

all_counts_acute_8430 <- data.frame(upreg, downreg)
all_counts_acute_8430
```

### RNA changes
```{r}

#organize metadata
metadata_acute_8430_rna <- metadata %>% 
  filter(treatment== "8430") %>% 
  filter(library=="rna") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_8430_rna$condition <- relevel(factor(metadata_acute_8430_rna$condition), ref = "normoxia")

levels(metadata_acute_8430_rna$condition)

names <- as.character(metadata_acute_8430_rna$name)

acute_8430_rna <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_8430_rna <- DESeqDataSetFromMatrix(
  countData = acute_8430_rna,
  colData = metadata_acute_8430_rna,
  design = ~condition
  )

ddsMat_acute_8430_rna <- DESeq(ddsMat_acute_8430_rna)
resultsNames(ddsMat_acute_8430_rna)
res_acute_8430_rna <- results(ddsMat_acute_8430_rna, name=resultsNames(ddsMat_acute_8430_rna)[2], cooksCutoff=F, independentFiltering=F)


res_acute_8430_rna <- lfcShrink(dds = ddsMat_acute_8430_rna, coef = resultsNames(ddsMat_acute_8430_rna)[2], res = res_acute_8430_rna) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_8430_rna %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_8430_rna %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="rna"

all_counts_acute_8430_rna <- data.frame(dataset, upreg, downreg)

```


### RIBO changes

```{r}

#organize metadata
metadata_acute_8430_ribo <- metadata%>% 
  filter(treatment== "8430") %>% 
  filter(library=="ribo") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_8430_ribo$condition <- relevel(factor(metadata_acute_8430_ribo$condition), ref = "normoxia")

levels(metadata_acute_8430_ribo$condition)

names <- as.character(metadata_acute_8430_ribo$name)

acute_8430_ribo <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_8430_ribo <- DESeqDataSetFromMatrix(
  countData = acute_8430_ribo,
  colData = metadata_acute_8430_ribo,
  design = ~condition
  )

ddsMat_acute_8430_ribo <- DESeq(ddsMat_acute_8430_ribo)
resultsNames(ddsMat_acute_8430_ribo)
res_acute_8430_ribo <- results(ddsMat_acute_8430_ribo, name=resultsNames(ddsMat_acute_8430_ribo)[2], cooksCutoff=F, independentFiltering=F)


res_acute_8430_ribo <- lfcShrink(dds = ddsMat_acute_8430_ribo, coef = resultsNames(ddsMat_acute_8430_ribo)[2], res = res_acute_8430_ribo) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_8430_ribo %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_8430_ribo %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="ribo"

all_counts_acute_8430_ribo <- data.frame(dataset, upreg, downreg)

```


## Visualize acute hypoxic response in DMSO
Export all acute changes as a single table

```{r}

DMSO_te <- res_acute_DMSO[,c(1,2,4,7)]
colnames(DMSO_te)[3] ="LFC_te_DMSO"
colnames(DMSO_te)[4] ="padj_te_DMSO"

DMSO_ribo <- res_acute_DMSO_ribo[,c(1,4,7)]
colnames(DMSO_ribo)[2] ="LFC_ribo_DMSO"
colnames(DMSO_ribo)[3] ="padj_ribo_DMSO"

DMSO_rna <- res_acute_DMSO_rna[,c(1,4,7)]
colnames(DMSO_rna)[2] ="LFC_rna_DMSO"
colnames(DMSO_rna)[3] ="padj_rna_DMSO"

c8430_te <- res_acute_8430[,c(1,4,7)]
colnames(c8430_te)[2] ="LFC_te_8430"
colnames(c8430_te)[3] ="padj_te_8430"

c8430_ribo <- res_acute_8430_ribo[,c(1,4,7)]
colnames(c8430_ribo)[2] ="LFC_ribo_8430"
colnames(c8430_ribo)[3] ="padj_ribo_8430"

c8430_rna <- res_acute_8430_rna[,c(1,4,7)]
colnames(c8430_rna)[2] ="LFC_rna_8430"
colnames(c8430_rna)[3] ="padj_rna_8430"

#bind cols ----------------------------------------------------------------------

all_changes_dmso_8430 <- purrr::reduce(list(DMSO_te,DMSO_ribo,DMSO_rna,c8430_te,c8430_ribo,c8430_rna), dplyr::left_join, by = 'gene_id') %>% 
  as.data.frame()

write_csv(all_changes_dmso_8430, here("counts/all_norm_to_hyp_changes_8430.csv"))
```



## Visualize acute hypoxic response with histograms

## Acute hypoxic changes

```{r}

DMSO <- all_changes_dmso_8430 %>% 
  filter(padj_rna_DMSO < 0.05 | padj_ribo_DMSO < 0.05)

DMSO_acute <- ggplot(data = DMSO, aes(x = LFC_rna_DMSO, y = LFC_ribo_DMSO)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote(Log[2]~"DMSO LFC"~"RIBO: hypoxia vs normoxia")) +
  xlab(bquote(Log[2]~"DMSO LFC"~"RNA: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 5)+
  theme_prism()

DMSO_acute <- ggMarginal(DMSO_acute, type = "histogram",
           xparams = list(binwidth = .1, fill = "grey"),
           yparams = list(binwidth = .1, fill = "red")
           )

ggsave(filename = here("plots/plots_8430/rna_ribo_DMSO_acute.pdf"), DMSO_acute, width = 4, height = 4)

DMSO_acute
```


## TE changes - 8430

```{r}

sig_DMSO <- all_changes_dmso_8430 %>% 
  filter(padj_te_DMSO <0.05)

#counts DMSO
sig_DMSO %>% 
  filter(LFC_te_DMSO < 0) %>% 
  filter(padj_te_DMSO < 0.05) %>% 
  nrow()

sig_DMSO %>% 
  filter(LFC_te_DMSO > 0) %>% 
  filter(padj_te_DMSO < 0.05) %>% 
  nrow()

#counts 8430
sig_DMSO %>% 
  filter(LFC_te_8430 < 0) %>% 
  filter(padj_te_8430 < 0.05) %>% 
  nrow()

sig_DMSO %>% 
  filter(LFC_te_8430 > 0) %>% 
  filter(padj_te_8430 < 0.05) %>% 
  nrow()



scatter_acute_dmso_8430 <- ggplot(data = sig_DMSO, aes(x = LFC_te_DMSO, y = LFC_te_8430)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote(Log[2]~"8430 LFC"~"TE: hypoxia vs normoxia")) +
  xlab(bquote(Log[2]~"DMSO LFC"~"TE: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 9)+
  theme_prism()

scatter_acute_dmso_8430 <- ggMarginal(scatter_acute_dmso_8430, type = "histogram",
           xparams = list(binwidth = .1, fill = "grey"),
           yparams = list(binwidth = .1, fill = "#009E73")
           )

scatter_acute_dmso_8430

ggsave(filename= here("plots/plots_8430/acute_dmso_8430.pdf"), scatter_acute_dmso_8430, height=4, width=4)

```


# Differential gene expression for 8430 compared to DMSO in normoxia

```{r}

#organize metadata
metadata_normox_8430 <- metadata %>% 
  filter(condition=="normoxia") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_normox_8430$library <- relevel(factor(metadata_normox_8430$library), ref = "rna")
metadata_normox_8430$treatment <- relevel(factor(metadata_normox_8430$treatment), ref = "DMSO")

levels(metadata_normox_8430$library)
levels(metadata_normox_8430$treatment)

names <- as.character(metadata_normox_8430$name)

names

normox_8430 <- all_raw[, colnames(all_raw) %in% names]


ddsMat_normox_8430 <- DESeqDataSetFromMatrix(
  countData = normox_8430,
  colData = metadata_normox_8430,
  design = ~treatment + library + treatment:library
  )

ddsMat_normox_8430 <- DESeq(ddsMat_normox_8430)
resultsNames(ddsMat_normox_8430)
res_normox_8430 <- results(ddsMat_normox_8430, name=resultsNames(ddsMat_normox_8430)[4], cooksCutoff=F, independentFiltering=F)


res_normox_8430 <- lfcShrink(dds = ddsMat_normox_8430, coef = resultsNames(ddsMat_normox_8430)[4], res = res_normox_8430) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_normox_8430 %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_normox_8430 %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset= "normoxia_8430"

all_counts_normox_8430 <- data.frame(dataset, upreg, downreg)
all_counts_normox_8430

write_csv(res_normox_8430, here("counts/normoxia_dmso_8430.csv"))



```


# Differential gene expression for 8430 compared to DMSO in hypoxia

```{r}

#organize metadata
metadata_hypox_8430 <- metadata %>% 
  filter(condition=="hypoxia") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_hypox_8430$library <- relevel(factor(metadata_hypox_8430$library), ref = "rna")
metadata_hypox_8430$treatment <- relevel(factor(metadata_hypox_8430$treatment), ref = "DMSO")

levels(metadata_hypox_8430$library)
levels(metadata_hypox_8430$treatment)

names <- as.character(metadata_hypox_8430$name)

names

hypox_8430 <- all_raw[, colnames(all_raw) %in% names]


ddsMat_hypox_8430 <- DESeqDataSetFromMatrix(
  countData = hypox_8430,
  colData = metadata_hypox_8430,
  design = ~treatment + library + treatment:library
  )

ddsMat_hypox_8430 <- DESeq(ddsMat_hypox_8430)
resultsNames(ddsMat_hypox_8430)
res_hypox_8430 <- results(ddsMat_hypox_8430, name=resultsNames(ddsMat_hypox_8430)[4], cooksCutoff=F, independentFiltering=F)


res_hypox_8430 <- lfcShrink(dds = ddsMat_hypox_8430, coef = resultsNames(ddsMat_hypox_8430)[4], res = res_hypox_8430) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_hypox_8430 %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_hypox_8430 %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset= "hypoxia_8430"

all_counts_hypox_8430 <- data.frame(dataset, upreg, downreg)
all_counts_hypox_8430

write_csv(res_hypox_8430, here("counts/hypoxia_dmso_8430.csv"))



```

### Visualize normoxic vs hyoxic response for 8430
```{r}
hyp_vs_norm_8430 <- rbind(all_counts_normox_8430, all_counts_hypox_8430) %>% 
  melt() %>% 
extract(dataset, into = c("condition", "sample"), regex = "([^_]+)_([^_]+)")
```

# effect of si3e KD norm and hypoxia
```{r}

c8430_bar <- ggbarplot(data = hyp_vs_norm_8430, x = "condition", y = "value", fill = "variable", palette = c("#37de5e",c8430_col), position = position_dodge(0.75), label = T, title= "8430")+
  ylab("Transcript counts")+
  xlab("")+
  theme_prism()+
  theme_cowplot(font_size = 12)+
  scale_y_continuous(limits = c(0, 300))

c8430_bar

ggsave(here("plots/plots_8430/8430bars.pdf"), c8430_bar, width = 3, height = 4)

```

What are those genes?

```{r}

res_normox_8430 %>% 
  filter(padj < 0.05) %>% 
  dplyr::select(symbol, log2FoldChange, padj) %>% 
  view()

res_hypox_8430 %>% 
  filter(padj < 0.05) %>% 
  dplyr::select(symbol, log2FoldChange, padj) %>% 
  view()

```





# Compare c8430 to eIF3d and eIF3e
## Import TE changes for eIF3d and eIF3e in normoxia nad hypoxia
Pull normoxia data from figure 1
```{r}

#3d

#up
res_normox_si3d <-read_csv(here("counts/fig1/normoxia_sictrl_3d.csv"))

si3d_normoxia_up <- res_normox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

si3d_normoxia_up$gs_name <- rep("si3d_normoxia_up")


#down

si3d_normoxia_down <- res_normox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

si3d_normoxia_down$gs_name <- rep("si3d_normoxia_down")


#3e
#up
res_normox_si3e <-read_csv(here("counts/fig1/normoxia_sictrl_3e.csv"))


si3e_normoxia_up <- res_normox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

si3e_normoxia_up$gs_name <- rep("si3e_normoxia_up")

#down

si3e_normoxia_down <- res_normox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

si3e_normoxia_down$gs_name <- rep("si3e_normoxia_down")


```

Import hypoxia data from figure 3
```{r}

#3d
res_hypox_si3d <-read_csv(here("counts/fig3/hypoxia_si_ctrl_3d.csv"))


si3d_hypoxia_up <- res_hypox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

si3d_hypoxia_up$gs_name <- rep("si3d_hypoxia_up")


#down

si3d_hypoxia_down <- res_hypox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

si3d_hypoxia_down$gs_name <- rep("si3d_hypoxia_down")



#3e
res_hypox_si3e <-read_csv(here("counts/fig3/hypoxia_si_ctrl_3e.csv"))


si3e_hypoxia_up <- res_hypox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

si3e_hypoxia_up$gs_name <- rep("si3e_hypoxia_up")

#down

si3e_hypoxia_down <- res_hypox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

si3e_hypoxia_down$gs_name <- rep("si3e_hypoxia_down")


```

## Gene lists for a209 in normoxia and hypoxia

```{r}

#normoxia 

#up
a209_normoxia_up <- res_normox_a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

a209_normoxia_up$gs_name <- rep("a209_normoxia_up")

#down

a209_normoxia_down <- res_normox_a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

a209_normoxia_down$gs_name <- rep("a209_normoxia_down")


#hypoxia

#up

a209_hypoxia_up <- res_hypox_a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

a209_hypoxia_up$gs_name <- rep("a209_hypoxia_up")

#down

a209_hypoxia_down <- res_hypox_a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

a209_hypoxia_down$gs_name <- rep("a209_hypoxia_down")


```




# Odds ratios for a209 3e and 3d overlap in normoxia

The following analysis determines the number of genes that overlap in each condition and the statistical significance (p-value) of their overlap (odds ratio).


Create a matrix that compares each condition for gene overlap analysis.
```{r}
all_lists <- bind_rows(si3e_normoxia_up, si3d_normoxia_up, a209_normoxia_up, si3e_normoxia_down, si3d_normoxia_down, a209_normoxia_down, 
                       si3e_hypoxia_up, si3d_hypoxia_up, a209_hypoxia_up, si3e_hypoxia_down, si3d_hypoxia_down, a209_hypoxia_down) %>% 
extract(gs_name,
        into = c("sample", "condition", "direction"),
        regex = "([^_]+)_([^_]+)_([^_]+)")
```


### get number of expressed genes for universe value
```{r}
all_209_symbols_expressed <- rbind(
res_normox_a209,
res_hypox_a209)

universe <- unique(all_209_symbols_expressed$symbol)

```


### Normoxia analysis
```{r}

norm_matrix <- all_lists %>%
  filter(condition=="normoxia") %>% 
  dplyr::select(-condition)


norm_matrix_comb <- norm_matrix %>% 
  tidyr::unite("description", 2:3, sep=', ')

gom.allsamples <- split(norm_matrix_comb$symbol, norm_matrix_comb$description)

#double check genome size here

gom.all <- newGOM(gom.allsamples, gom.allsamples, genome.size = 11629)

norm_all.odds.matrix <- getMatrix(gom.all, "odds")
norm_all.pval.matrix <- getMatrix(gom.all, "pval")


```

# plot odds ratios as bars with p values
```{r}

#merge odds and pval matricies

odds <- norm_all.odds.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:7, names_to = "comparison", values_to="odds_value")

pval <- norm_all.pval.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:7, names_to = "comparison", values_to="pval_value")


odds_pval <- left_join(odds, pval, by=c("sample", "comparison")) %>% 
  tidyr::unite("comparison", 1:2, sep="-") %>% 
  filter(comparison == "a209, up-si3e, up"| 
         comparison =="a209, down-si3e, down"|
           comparison == "a209, up-si3d, up"|
           comparison == "a209, down-si3d, down") %>% 
  mutate(comparison_with_a209 = sub(".*-", "", comparison)) %>% 
  mutate(pvalue_plot = signif(pval_value, digits=3)) %>% 
  mutate(label = paste0("p = ", pvalue_plot, sep=" "))

odds_pval$comparison_with_a209 <- factor(odds_pval$comparison_with_a209, levels = c("si3e, up", "si3e, down", "si3d, up", "si3d, down"))

odds_pval <- odds_pval %>% 
  mutate(group = ifelse(comparison== "a209, down-si3d, down" | comparison== "a209, up-si3d, up", "si3d", "si3e"))


ggpubr::ggbarplot(odds_pval, x="comparison_with_a209", y="odds_value", fill="group", label = odds_pval$label, title="a209 overlaps in normoxia", palette = c(si3d_col, si3e_col), ylim=(c(0,7)))+
  geom_hline(aes(yintercept=1))


ggsave(here("plots/fig6/normoverlap.pdf"), width=4, height=4)

```

### Hypoxia - double check genome size
```{r}

hyp_matrix <- all_lists %>%
  filter(condition=="hypoxia") %>% 
  dplyr::select(-condition)


hyp_matrix_comb <- hyp_matrix %>% 
  tidyr::unite("description", 2:3, sep=', ')

gom.allsamples <- split(hyp_matrix_comb$symbol, hyp_matrix_comb$description)

gom.all <- newGOM(gom.allsamples, gom.allsamples, genome.size = 11629)

hyp_all.odds.matrix <- getMatrix(gom.all, "odds")
hyp_all.pval.matrix <- getMatrix(gom.all, "pval")


```

# plot odds ratios as bars with p values

```{r}

#merge odds and pval matricies

odds <- hyp_all.odds.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:7, names_to = "comparison", values_to="odds_value")

pval <- hyp_all.pval.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:7, names_to = "comparison", values_to="pval_value")


odds_pval <- left_join(odds, pval, by=c("sample", "comparison")) %>% 
  tidyr::unite("comparison", 1:2, sep="-") %>% 
  filter(comparison == "a209, up-si3e, up"| 
         comparison =="a209, down-si3e, down"|
           comparison == "a209, up-si3d, up"|
           comparison == "a209, down-si3d, down") %>% 
  mutate(comparison_with_a209 = sub(".*-", "", comparison)) %>% 
  mutate(pvalue_plot = signif(pval_value, digits=3)) %>% 
  mutate(label = paste0("p = ", pvalue_plot, sep=" "))

odds_pval$comparison_with_a209 <- factor(odds_pval$comparison_with_a209, levels = c("si3e, up", "si3e, down", "si3d, up", "si3d, down"))

odds_pval <- odds_pval %>% 
  mutate(group = ifelse(comparison== "a209, down-si3d, down" | comparison== "a209, up-si3d, up", "si3d", "si3e"))

ggpubr::ggbarplot(odds_pval, x="comparison_with_a209", y="odds_value", fill="group", label = odds_pval$label, title="a209 overlaps in hypoxia", palette = c(si3d_col, si3e_col), ylim=(c(0,7)))+
  geom_hline(aes(yintercept=1))


ggsave(filename=here("plots/fig6/hypoverlap.pdf"), width=4, height=4)



```







Pathway analysis
For Connor's thesis


a209 TE

make lists of genes


## Pathway analysis gene list setup
```{r prep GSEA analysis}

#norm lists

a209_norm <- res_normox_a209$log2FoldChange
names(a209_norm) <- res_normox_a209$symbol

a209_norm <- sort(a209_norm, decreasing = TRUE)

a209_norm <- a209_norm[!duplicated(names(a209_norm))]

#hyp lists

a209_hyp <- res_hypox_a209$log2FoldChange
names(a209_hyp) <- res_hypox_a209$symbol

a209_hyp <- sort(a209_hyp, decreasing = TRUE)

a209_hyp <- a209_hyp[!duplicated(names(a209_hyp))]

```



```{r}
m_df_h <- msigdbr(species = "Homo sapiens") %>%
  filter(gs_cat == "H") %>% 
  dplyr::select(gs_name, gene_symbol)

m_df_c2 <- msigdbr(species = "Homo sapiens") %>%
  filter(gs_cat == "C2" & gs_subcat != "CGP") %>% 
  dplyr::select(gs_name, gene_symbol)
```


```{r}
a209_norm_h <- GSEA(
  geneList = a209_norm,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )

a209_hyp_h <- GSEA(
  geneList = a209_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_h
  )



a209_norm_h <-
  a209_norm_h@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) 

a209_hyp_h <- 
  a209_hyp_h@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) 


```

```{r GSEA KEGG}
a209_norm_kegg <- GSEA(
  geneList = a209_norm,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^KEGG', gs_name))
  )

a209_hyp_kegg <- GSEA(
  geneList = a209_hyp,
  eps = 0,
  pAdjustMethod = "fdr",
  pvalueCutoff = 1,
  minGSSize = 20,
  maxGSSize = 1000,
  TERM2GENE = m_df_c2 %>%
  filter(grepl('^KEGG', gs_name))
  )

a209_norm_kegg <-
  a209_norm_kegg@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) 

a209_hyp_kegg <- 
  a209_hyp_kegg@result %>%
  dplyr::select(ID,NES,p.adjust,core_enrichment) 



a209_pathways <- list("normoxia_hallmark_a209"=a209_norm_h,
            "hypoxia_hallmark_a209"=a209_hyp_h,
            "normoxia_kegg_a209"=a209_norm_kegg,
            "hypoxia_kegg_a209"=a209_hyp_kegg)

writexl::write_xlsx(a209_pathways, here("pathway_tables/a209_normoxia_and_hypoxia.xlsx"))

```

# Invasion
```{r}
go_bp <- msigdbr(species = "Homo sapiens", category = "C5")

m_df_c2 <- msigdbr(species = "Homo sapiens") %>%
  filter(gs_cat == "C2" & gs_subcat != "CGP") %>% 
  dplyr::select(gs_name, gene_symbol)


go_invasion <- go_bp %>% 
  filter(gs_name== "GO_CELL_INVASION")

go_invasion <- go_bp %>%
  filter(grepl("invasion", gs_name, ignore.case = TRUE) |
         grepl("invasion", gs_description, ignore.case = TRUE))
```

GO_CELL_INVASION

GO_POSITIVE_REGULATION_OF_CELL_INVASION

GO_NEGATIVE_REGULATION_OF_CELL_INVASION

GO_INVASION_OF_TUMOR_CELLS_INTO_EXTRACELLULAR_MATRIX


import invasion pathway
```{r}
invasion_genes <- read_csv(here("other_data/melanoma_invasion.csv")) %>% 
  rename("FC_invasion_score"="Fold Change") %>% 
  rename("symbol"="Symbol")
```

cytoskelatal adhesion emt
merge invasion genes
```{r}

res_3d_invasion <-
res_hypox_si3d %>% 
  left_join(invasion_genes) %>% 
  filter(padj < 0.05)
```

down with 3dKD = 3d upregulates

Look at stuff down with 3d KD and up with invasion

```{r}
invasion_interest_3d_hypox <- 
res_3d_invasion %>% 
  filter(log2FoldChange < 0 & FC_invasion_score > 0)

print(invasion_interest_3d_hypox)
```



hypox 3e

```{r}

res_3e_invasion <-
res_hypox_si3e %>% 
  left_join(invasion_genes) %>% 
  filter(padj < 0.05)

invasion_interest_3e_hypox <- 
res_3e_invasion %>% 
  filter(log2FoldChange < 0 & FC_invasion_score > 0)

print(invasion_interest_3e_hypox)

```

Same two genes come up.

Now try normoxia


```{r}

res_3d_invasion <-
res_normox_si3d %>% 
  left_join(invasion_genes) %>% 
  filter(padj < 0.05)

invasion_interest_3d_normox <- 
res_3d_invasion %>% 
  filter(log2FoldChange < 0 & FC_invasion_score > 0)

print(invasion_interest_3d_normox)

res_3e_invasion <-
res_normox_si3e %>% 
  left_join(invasion_genes) %>% 
  filter(padj < 0.05)

invasion_interest_3e_normox <- 
res_3e_invasion %>% 
  filter(log2FoldChange < 0 & FC_invasion_score > 0)

print(invasion_interest_3e_normox)

```

Also the same!!!!!

Two invasion genes are signficantly upregualted by 3d (down with 3d KD). These genes promote invasion

Ectopic overexpression of UQCRH in KMRC2 restored mitochondrial membrane potential, increased oxygen consumption, and attenuated the Warburg effect at the cellular level.

what is the ribo and rna counts for these?
```{r}

```



