---
title: "231 riboseq"
author: "Kate"
date: "2025-08-07"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}

library(here)
library(janitor)
library(tximport)
library(tidyverse)
library(EnhancedVolcano)
library(msigdbr)
library(viridis)
library(gridExtra)
library(ggthemes)
library(ggrepel)
library(scales)
library(DESeq2)
library(limma)
library(clusterProfiler)
library(plotly)
library(ggokabeito)
library(ggvenn)
library(ggpubr)
library(DT)
library(edgeR)
library(cowplot)
library(ggExtra)
library(ggprism)
library(GeneOverlap)
library(writexl)
library(reshape2)
```

```{r}
a209_col = "#CC79A7"
si3e_col = "#56B4E9"
si3d_col = "#E69F00"
```

## Load transcript-level data

Importing metadata and all salmon counts
```{r meta data}
metadata <- read_csv(here("accessories","metadata_NM_231.csv"), skip = 0) %>% 
  clean_names()

metadata <- metadata %>% 
  mutate(library = ifelse(type=="Qiagen_miRNA", "ribo", "rna"))

#ribosome lysate = ribosome seq
#rRNA depletion = RNA seq
#rename variables remove outlier

#edit factor levels
metadata$library <- relevel(x = factor(metadata$library), ref = "rna")

metadata$treatment <- relevel(x = factor(metadata$treatment), ref = "siCTRL")
metadata$treatment <- recode(metadata$treatment, "siCTRL" = "sictrl")
metadata$condition <- relevel(x = factor(metadata$condition), ref = "normoxia")
metadata$rep <- recode(metadata$rep, RepA = "a", RepB = "b", RepC = "c")

metadata <- metadata %>%
dplyr::select(id, library, treatment, condition, rep) 

```

```{r import gene info}

#import key on RNA type, remove premRNA
geneInfo <- read_csv(here("accessories","gencode.v26.primary.info.csv.gz"), col_names = F) %>%
  filter(!str_detect(X2, "pre_"))

colnames(geneInfo) <- c("gene_id","transcript_id","biotype","symbol")

#import transcripts for gene to transcript mapping
tx2gene <- geneInfo[,c(2,1)]
gene_symbol <- geneInfo %>% dplyr::select(gene_id, symbol) %>% unique()

```

```{r txi import, include=FALSE}
#name files to import
myquantfiles <- paste(here(), "/salmon/",
                      metadata$id,
                      "/quant.sf.gz",
                      sep = "")

names(myquantfiles) <- paste(metadata$condition, metadata$treatment, metadata$rep, metadata$library, sep = "_")

#import salmon files
myTxi <- tximport(myquantfiles, type = "salmon", tx2gene = tx2gene)
```


No spike ins because i aligned only to the human

```{r}
# visualize distribution of counts
hist_count <- hist(log2(rowSums(myTxi$counts)), breaks = 50)
#determine threshold

#distribution of abundance histogram
#hist_abundance <- hist(log2(rowSums(myTxi$abundance)), breaks = 50)

#only keeping rows with counts  log2> 10 across all samples - want around 10-15k genes - want genes that are highly expressed
keepGenes <- rownames(myTxi$counts[log2(rowSums(myTxi$counts)) > 10 ,])

#removes noise and keeps only highly expressed genes
# results show bimodal distribution
#1 RNA per cell = 1 TPM
```




```{r standard deviation filtering}
# log transforming with pseudocount to avoid undefined numbers
qcinput <- log2(myTxi$abundance[keepGenes,] + 1) 

qcinput <- qcinput[rowSds(qcinput) > 1,] 

dim(qcinput)
```



```{r}

mycor <- cor(qcinput, method = "pearson")

heatmap <- pheatmap::pheatmap(
  mat = mycor,
  border_color = "black",  # Black borders around cells
   #color = colorRampPalette(c("blue", "white", "red"))(100),  # Custom color gradient
   scale = "none",  
  cluster_rows = TRUE,  
  cluster_cols = TRUE,  
  fontsize = 8 
)

heatmap

ggsave(here("plots/plots_231/qc.pdf"), heatmap, width=6.5, height = 6)
```


Split normoxia and hypoxia
```{r}

qc_input_normoxia <- qcinput[, grep("normoxia", colnames(qcinput))]

qc_input_hypoxia <- qcinput[, grep("hypoxia", colnames(qcinput))]

```


# Visualizations

# Figure 1: Normoxia

```{r}
#KDs
mycor_norm <- cor(qc_input_normoxia, method = "pearson")

colnames(mycor_norm) <- gsub("^normoxia_", "", colnames(mycor_norm))  # Remove "normoxia_" prefix
rownames(mycor_norm) <- gsub("^normoxia_", "", rownames(mycor_norm))  # Same for row names

# Replace underscores with spaces
colnames(mycor_norm) <- gsub("_", " ", colnames(mycor_norm))
rownames(mycor_norm) <- gsub("_", " ", rownames(mycor_norm))

# Create the heatmap with the cleaned-up names
heatmap_normoxia <- pheatmap::pheatmap(
  mat = mycor_norm,
  border_color = "black",  # Black borders around cells
   #color = colorRampPalette(c("blue", "white", "red"))(100),  # Custom color gradient
   scale = "none",  
  cluster_rows = TRUE,  
  cluster_cols = TRUE,  
  fontsize = 12 
)


heatmap_normoxia
ggsave(heatmap_normoxia, filename=here("plots/plots_231/heatmap_normoxia.pdf"), width=4.5, height=4)

# RIBO average correlation coefficient

qc_input_normoxia_ribo <- qc_input_normoxia[, grep("ribo", colnames(qc_input_normoxia))]

mycor_ribo_norm <- cor(qc_input_normoxia_ribo, method = "pearson")

# Extract the lower triangular part of the matrix without the diagonal
lower_tri <- mycor_ribo_norm[lower.tri(mycor_ribo_norm, diag=FALSE)]

# Calculate the average correlation coefficient (excluding diagonal)
average_correlation <- mean(lower_tri)

# Print the result
average_correlation


# rna average correlation coefficient

qc_input_normoxia_rna <- qc_input_normoxia[, grep("rna", colnames(qc_input_normoxia))]

mycor_rna_norm <- cor(qc_input_normoxia_rna, method = "pearson")


# Extract the lower triangular part of the matrix without the diagonal
lower_tri <- mycor_rna_norm[lower.tri(mycor_rna_norm, diag=FALSE)]

# Calculate the average correlation coefficient (excluding diagonal)
average_correlation <- mean(lower_tri)

# Print the result
average_correlation

```


# Figure 2: Hypoxia, knockdowns


Heatmap

```{r}
mycor_hyp_kds <- cor(qc_input_hypoxia, method = "pearson")

colnames(mycor_hyp_kds) <- gsub("^hypoxia_", "", colnames(mycor_hyp_kds))  # Remove "hypoxia_" prefix
rownames(mycor_hyp_kds) <- gsub("^hypoxia_", "", rownames(mycor_hyp_kds))  # Same for row names

# Replace underscores with spaces
colnames(mycor_hyp_kds) <- gsub("_", " ", colnames(mycor_hyp_kds))
rownames(mycor_hyp_kds) <- gsub("_", " ", rownames(mycor_hyp_kds))

# Create the heatmap with the cleaned-up names
heatmap_hypoxia <- pheatmap::pheatmap(
  mat = mycor_hyp_kds,
  border_color = "black",  # Black borders around cells
   #color = colorRampPalette(c("blue", "white", "red"))(100),  # Custom color gradient
   scale = "none",  
  cluster_rows = TRUE,  
  cluster_cols = TRUE,  
  fontsize = 12 
)





heatmap_hypoxia
ggsave(heatmap_hypoxia, filename=here("plots/plots_231/heatmap_hyp.pdf"), width=4.5, height=4)

```

```{r}

# RIBO average correlation coefficient

qc_input_hypoxia_ribo <- qc_input_hypoxia[, grep("ribo", colnames(qc_input_hypoxia))]

mycor_ribo_hyp <- cor(qc_input_hypoxia_ribo, method = "pearson")


# Extract the lower triangular part of the matrix without the diagonal
lower_tri <- mycor_ribo_hyp[lower.tri(mycor_ribo_hyp, diag=FALSE)]

# Calculate the average correlation coefficient (excluding diagonal)
average_correlation <- mean(lower_tri)

# Print the result
average_correlation



# rna average correlation coefficient

qc_input_hypoxia_rna <- qc_input_hypoxia[, grep("rna", colnames(qc_input_hypoxia))]

mycor_rna_hyp <- cor(qc_input_hypoxia_rna, method = "pearson")


# Extract the lower triangular part of the matrix without the diagonal
lower_tri <- mycor_rna_hyp[lower.tri(mycor_rna_hyp, diag=FALSE)]

# Calculate the average correlation coefficient (excluding diagonal)
average_correlation <- mean(lower_tri)

# Print the result
average_correlation


```

qc looks great.




# Calculate TE


### Filter
```{r keep only protein coding genes}

pc <- geneInfo %>%
  filter(biotype =="protein_coding") %>%
  pull(gene_id) %>%
  unique()

pc_keep <- pc[pc %in% keepGenes]

# remove non-polyadenylated mRNAs
nonpolyA <- readxl::read_xlsx(path = here("accessories","polyAminus.xlsx"), sheet = 1) %>% 
   pull(Gene)

nonpolyA_ID <- geneInfo %>%
  filter(symbol %in% nonpolyA) %>% 
  pull(gene_id) %>% 
  unique()

# we want to remove any genes that match these annotation categories # before we do DE analysis
blacklist_smRNA <- c("snoRNA","miRNA","snRNA","rRNA","Mt_tRNA","scaRNA","Mt_tRNA","Mt_rRNA")

blacklist_smRNA_ID <- geneInfo %>%
  filter(biotype %in% blacklist_smRNA) %>% 
  pull(gene_id) %>% 
  unique()

genes_to_kill <- unique(c(blacklist_smRNA_ID,nonpolyA_ID))
```


```{r filter and extract counts}

allCounts <- myTxi$counts %>%
  as.data.frame() %>% 
  rownames_to_column(var = "gene") %>% 
  filter(gene %in% pc_keep) %>%
  filter(!gene %in% genes_to_kill) %>%
  column_to_rownames(var = "gene") %>% 
  as.matrix() %>%
  round(.,0)

all_raw_filtered_counts <- allCounts %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id")

#export allCounts for DeSeq analysis
write_csv(all_raw_filtered_counts, here("counts/calculate_te/raw_filtered_allCounts_231.csv"))


allCountsNorm <- edgeR::cpm(allCounts[,order(colnames(allCounts))])

riboN <- log2(1+ as.matrix(allCountsNorm[,grep(pattern = "ribo",colnames(allCountsNorm))]))
rnaN <- log2(1 + as.matrix(allCountsNorm[,grep(pattern = "rna",colnames(allCountsNorm))])) 


TE_original <- riboN - rnaN

hist(TE_original)

median(as.data.frame(TE_original) %>% melt() %>% pull(value))

#center data --

TE <- (riboN - rnaN) + 0.5749442
# this is why im using 0.5749442! -- all changes are relative


hist(TE)

colnames(TE) <- gsub(pattern = "ribo", replacement = "TE", x = colnames(TE))


gghistogram(as.data.frame(TE) %>% melt(), x="value")
ribo_counts <- as.matrix(allCounts[,grep(pattern = "ribo",colnames(allCounts))])
rna_counts <- as.matrix(allCounts[,grep(pattern = "rna",colnames(allCounts))])

```


### calculation
```{r}

full_te_list <- TE %>% 
  as.data.frame() %>%
  rownames_to_column("gene_id") %>% 
  left_join(gene_symbol, by="gene_id") %>% 
  dplyr::select(symbol, everything())


full_rna_list <- rnaN %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id")


full_ribo_list <- riboN %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id")

full_te_list <- full_te_list %>% 
  left_join(full_rna_list, by="gene_id")

complete_list <- full_te_list %>% 
  left_join(full_ribo_list, by= "gene_id")


write_csv(complete_list, here("counts/calculate_te/complete_te_rna_ribo_list_231.csv"))


```



```{r reformat data}
all_long <- complete_list %>% 
  melt() %>% 
    separate(variable, into=(c("condition", "treatment", "rep", "type")))

```

Sanity check:

Look at RNA for eIF3e and eIF3d KD compared to control


```{r eIF3e RNA}

rna_3e_formatted <- all_long %>%
  filter(symbol=="EIF3E") %>%
  filter(condition=="normoxia") %>%
  filter(type=="rna")


ggplot(rna_3e_formatted, aes(x=treatment, y=value))+
  geom_point()+
  theme_cowplot()

```


```{r eIF3d RNA}

rna_3d_formatted <- all_long %>%
  filter(symbol=="EIF3D") %>%
  filter(condition=="normoxia") %>%
  filter(type=="rna")


ggplot(rna_3d_formatted, aes(x=treatment, y=value))+
  geom_point()+
  theme_cowplot()

```


Example of TE: Analyze ATF4 rna ribo and TE in control compared to KD

Example- ATF4
```{r}
atf4_data <- all_long %>% 
  filter(symbol=="ATF4") %>% 
  filter(condition=="normoxia")


ggplot(atf4_data, aes(x=treatment, y=value))+
  geom_point()+
  facet_wrap(~type)
```

```{r}
eif3e_data <- all_long %>% 
  filter(symbol=="EIF3E") %>% 
  filter(condition=="hypoxia")


ggplot(eif3e_data, aes(x=treatment, y=value))+
  geom_point()+
  facet_wrap(~type)
```

# ###############################################################################
# just doing TE here






## HIF1alpha levels after compound treatment

```{r}

all_gene_te <- complete_list

all_gene_te_long <- all_gene_te %>% 
  melt() %>% 
tidyr::extract(variable, into = c("condition", "treatment", "rep", "type"), regex = "([^_]+)_([^_]+)_([^_]+)_([^_]+)") %>% 
  filter(condition=="hypoxia")

all_gene_te_long$treatment <- relevel(factor(all_gene_te_long$treatment), ref = "sictrl")


hif <-all_gene_te_long %>% 
  filter(symbol=="HIF1A") %>% 
  filter(type != "TE")

hif_summary <- hif %>%
  group_by(symbol, type, treatment) %>%
  summarise(
    mean_value = mean(value),
    sd = sd(value)  # Standard deviation
  )

hif_summary$type <- factor(hif_summary$type, levels=c("rna","ribo"))

ggplot(hif_summary, aes(x = treatment, y = mean_value, fill = treatment)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  geom_errorbar(aes(ymin = mean_value - sd, ymax = mean_value + sd), 
                width = 0.2, position = position_dodge(0.9)) +
  facet_wrap(~ symbol + type) +
  scale_fill_manual(values = c("grey", si3d_col, si3e_col)) +
  theme_cowplot()+
    theme(legend.position = "none")

ggsave(here("plots/plots_231/hif_231_hypoxia.pdf"), width = 4, height = 3)

```

## Acute hypoxic response: DMSO normoxia to hypoxia.

### TE changes
```{r}

#organize metadata
metadata_acute_sictrl <- metadata %>% 
  filter(treatment== "sictrl") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_acute_sictrl$library <- relevel(factor(metadata_acute_sictrl$library), ref = "rna")
metadata_acute_sictrl$condition <- relevel(factor(metadata_acute_sictrl$condition), ref = "normoxia")

levels(metadata_acute_sictrl$library)
levels(metadata_acute_sictrl$condition)

names <- as.character(metadata_acute_sictrl$name)

all_raw <- all_raw_filtered_counts %>% 
  column_to_rownames("gene_id")

acute_sictrl <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_sictrl <- DESeqDataSetFromMatrix(
  countData = acute_sictrl,
  colData = metadata_acute_sictrl,
  design = ~condition + library + condition:library
  )

ddsMat_acute_sictrl <- DESeq(ddsMat_acute_sictrl)
resultsNames(ddsMat_acute_sictrl)
res_acute_sictrl <- results(ddsMat_acute_sictrl, name=resultsNames(ddsMat_acute_sictrl)[4], cooksCutoff=F, independentFiltering=F)


res_acute_sictrl <- lfcShrink(dds = ddsMat_acute_sictrl, coef = resultsNames(ddsMat_acute_sictrl)[4], res = res_acute_sictrl) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_sictrl %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_sictrl %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

all_counts_acute_sictrl <- data.frame(upreg, downreg)

```


### RNA changes
```{r}

#organize metadata
metadata_acute_sictrl_rna <- metadata %>% 
  filter(treatment== "sictrl") %>% 
  filter(library=="rna") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_sictrl_rna$condition <- relevel(factor(metadata_acute_sictrl_rna$condition), ref = "normoxia")

levels(metadata_acute_sictrl_rna$condition)

names <- as.character(metadata_acute_sictrl_rna$name)

acute_sictrl_rna <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_sictrl_rna <- DESeqDataSetFromMatrix(
  countData = acute_sictrl_rna,
  colData = metadata_acute_sictrl_rna,
  design = ~condition
  )

ddsMat_acute_sictrl_rna <- DESeq(ddsMat_acute_sictrl_rna)
resultsNames(ddsMat_acute_sictrl_rna)
res_acute_sictrl_rna <- results(ddsMat_acute_sictrl_rna, name=resultsNames(ddsMat_acute_sictrl_rna)[2], cooksCutoff=F, independentFiltering=F)


res_acute_sictrl_rna <- lfcShrink(dds = ddsMat_acute_sictrl_rna, coef = resultsNames(ddsMat_acute_sictrl_rna)[2], res = res_acute_sictrl_rna) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_sictrl_rna %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_sictrl_rna %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="rna"

all_counts_acute_sictrl_rna <- data.frame(dataset, upreg, downreg)

```


### RIBO changes

```{r}

#organize metadata
metadata_acute_sictrl_ribo <- metadata %>% 
  filter(treatment== "sictrl") %>% 
  filter(library=="ribo") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_sictrl_ribo$condition <- relevel(factor(metadata_acute_sictrl_ribo$condition), ref = "normoxia")

levels(metadata_acute_sictrl_ribo$condition)

names <- as.character(metadata_acute_sictrl_ribo$name)

acute_sictrl_ribo <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_sictrl_ribo <- DESeqDataSetFromMatrix(
  countData = acute_sictrl_ribo,
  colData = metadata_acute_sictrl_ribo,
  design = ~condition
  )

ddsMat_acute_sictrl_ribo <- DESeq(ddsMat_acute_sictrl_ribo)
resultsNames(ddsMat_acute_sictrl_ribo)
res_acute_sictrl_ribo <- results(ddsMat_acute_sictrl_ribo, name=resultsNames(ddsMat_acute_sictrl_ribo)[2], cooksCutoff=F, independentFiltering=F)


res_acute_sictrl_ribo <- lfcShrink(dds = ddsMat_acute_sictrl_ribo, coef = resultsNames(ddsMat_acute_sictrl_ribo)[2], res = res_acute_sictrl_ribo) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_sictrl_ribo %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_sictrl_ribo %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="ribo"

all_counts_acute_sictrl_ribo <- data.frame(dataset, upreg, downreg)

```

## Visualize sictrl acute hypoxic changes
```{r}

acute_ctrl_counts <- rbind(all_counts_acute_sictrl_rna, all_counts_acute_sictrl_ribo) %>% melt()

acute_ctrl_counts$dataset <- as.factor(acute_ctrl_counts$dataset)
acute_ctrl_counts$dataset <- relevel(acute_ctrl_counts$dataset, ref = "rna")

ggbarplot(acute_ctrl_counts, x="variable", y="value")+
  facet_wrap(~dataset)+
    geom_text(aes(label = value), vjust = -0.3)+
  theme_cowplot()

ggsave(here("plots/plots_231/acute_sictrl_counts.pdf"), width=4, height=4)


```


## Acute hypoxic response: 231 normoxia to hypoxia.

### TE changes
```{r}

#organize metadata
metadata_acute_si3e <- metadata %>% 
  filter(treatment== "si3e") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_acute_si3e$library <- relevel(factor(metadata_acute_si3e$library), ref = "rna")
metadata_acute_si3e$condition <- relevel(factor(metadata_acute_si3e$condition), ref = "normoxia")

levels(metadata_acute_si3e$library)
levels(metadata_acute_si3e$condition)

names <- as.character(metadata_acute_si3e$name)

acute_si3e <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_si3e <- DESeqDataSetFromMatrix(
  countData = acute_si3e,
  colData = metadata_acute_si3e,
  design = ~condition + library + condition:library
  )

ddsMat_acute_si3e <- DESeq(ddsMat_acute_si3e)
resultsNames(ddsMat_acute_si3e)
res_acute_si3e <- results(ddsMat_acute_si3e, name=resultsNames(ddsMat_acute_si3e)[4], cooksCutoff=F, independentFiltering=F)


res_acute_si3e <- lfcShrink(dds = ddsMat_acute_si3e, coef = resultsNames(ddsMat_acute_si3e)[4], res = res_acute_si3e) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_si3e %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_si3e %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

all_counts_acute_si3e <- data.frame(upreg, downreg)
all_counts_acute_si3e
```

### RNA changes
```{r}

#organize metadata
metadata_acute_si3e_rna <- metadata %>% 
  filter(treatment== "si3e") %>% 
  filter(library=="rna") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_si3e_rna$condition <- relevel(factor(metadata_acute_si3e_rna$condition), ref = "normoxia")

levels(metadata_acute_si3e_rna$condition)

names <- as.character(metadata_acute_si3e_rna$name)

acute_si3e_rna <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_si3e_rna <- DESeqDataSetFromMatrix(
  countData = acute_si3e_rna,
  colData = metadata_acute_si3e_rna,
  design = ~condition
  )

ddsMat_acute_si3e_rna <- DESeq(ddsMat_acute_si3e_rna)
resultsNames(ddsMat_acute_si3e_rna)
res_acute_si3e_rna <- results(ddsMat_acute_si3e_rna, name=resultsNames(ddsMat_acute_si3e_rna)[2], cooksCutoff=F, independentFiltering=F)


res_acute_si3e_rna <- lfcShrink(dds = ddsMat_acute_si3e_rna, coef = resultsNames(ddsMat_acute_si3e_rna)[2], res = res_acute_si3e_rna) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_si3e_rna %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_si3e_rna %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="rna"

all_counts_acute_si3e_rna <- data.frame(dataset, upreg, downreg)

```


### RIBO changes

```{r}

#organize metadata
metadata_acute_si3e_ribo <- metadata%>% 
  filter(treatment== "si3e") %>% 
  filter(library=="ribo") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_si3e_ribo$condition <- relevel(factor(metadata_acute_si3e_ribo$condition), ref = "normoxia")

levels(metadata_acute_si3e_ribo$condition)

names <- as.character(metadata_acute_si3e_ribo$name)

acute_si3e_ribo <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_si3e_ribo <- DESeqDataSetFromMatrix(
  countData = acute_si3e_ribo,
  colData = metadata_acute_si3e_ribo,
  design = ~condition
  )

ddsMat_acute_si3e_ribo <- DESeq(ddsMat_acute_si3e_ribo)
resultsNames(ddsMat_acute_si3e_ribo)
res_acute_si3e_ribo <- results(ddsMat_acute_si3e_ribo, name=resultsNames(ddsMat_acute_si3e_ribo)[2], cooksCutoff=F, independentFiltering=F)


res_acute_si3e_ribo <- lfcShrink(dds = ddsMat_acute_si3e_ribo, coef = resultsNames(ddsMat_acute_si3e_ribo)[2], res = res_acute_si3e_ribo) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_si3e_ribo %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_si3e_ribo %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="ribo"

all_counts_acute_si3e_ribo <- data.frame(dataset, upreg, downreg)

```

# 3d

### TE changes
```{r}

#organize metadata
metadata_acute_si3d <- metadata %>% 
  filter(treatment== "si3d") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_acute_si3d$library <- relevel(factor(metadata_acute_si3d$library), ref = "rna")
metadata_acute_si3d$condition <- relevel(factor(metadata_acute_si3d$condition), ref = "normoxia")

levels(metadata_acute_si3d$library)
levels(metadata_acute_si3d$condition)

names <- as.character(metadata_acute_si3d$name)

acute_si3d <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_si3d <- DESeqDataSetFromMatrix(
  countData = acute_si3d,
  colData = metadata_acute_si3d,
  design = ~condition + library + condition:library
  )

ddsMat_acute_si3d <- DESeq(ddsMat_acute_si3d)
resultsNames(ddsMat_acute_si3d)
res_acute_si3d <- results(ddsMat_acute_si3d, name=resultsNames(ddsMat_acute_si3d)[4], cooksCutoff=F, independentFiltering=F)


res_acute_si3d <- lfcShrink(dds = ddsMat_acute_si3d, coef = resultsNames(ddsMat_acute_si3d)[4], res = res_acute_si3d) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_si3d %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_si3d %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

all_counts_acute_si3d <- data.frame(upreg, downreg)
all_counts_acute_si3d
```

### RNA changes
```{r}

#organize metadata
metadata_acute_si3d_rna <- metadata %>% 
  filter(treatment== "si3d") %>% 
  filter(library=="rna") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_si3d_rna$condition <- relevel(factor(metadata_acute_si3d_rna$condition), ref = "normoxia")

levels(metadata_acute_si3d_rna$condition)

names <- as.character(metadata_acute_si3d_rna$name)

acute_si3d_rna <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_si3d_rna <- DESeqDataSetFromMatrix(
  countData = acute_si3d_rna,
  colData = metadata_acute_si3d_rna,
  design = ~condition
  )

ddsMat_acute_si3d_rna <- DESeq(ddsMat_acute_si3d_rna)
resultsNames(ddsMat_acute_si3d_rna)
res_acute_si3d_rna <- results(ddsMat_acute_si3d_rna, name=resultsNames(ddsMat_acute_si3d_rna)[2], cooksCutoff=F, independentFiltering=F)


res_acute_si3d_rna <- lfcShrink(dds = ddsMat_acute_si3d_rna, coef = resultsNames(ddsMat_acute_si3d_rna)[2], res = res_acute_si3d_rna) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_si3d_rna %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_si3d_rna %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="rna"

all_counts_acute_si3d_rna <- data.frame(dataset, upreg, downreg)

```


### RIBO changes

```{r}

#organize metadata
metadata_acute_si3d_ribo <- metadata%>% 
  filter(treatment== "si3d") %>% 
  filter(library=="ribo") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))


metadata_acute_si3d_ribo$condition <- relevel(factor(metadata_acute_si3d_ribo$condition), ref = "normoxia")

levels(metadata_acute_si3d_ribo$condition)

names <- as.character(metadata_acute_si3d_ribo$name)

acute_si3d_ribo <- all_raw[, colnames(all_raw) %in% names]


ddsMat_acute_si3d_ribo <- DESeqDataSetFromMatrix(
  countData = acute_si3d_ribo,
  colData = metadata_acute_si3d_ribo,
  design = ~condition
  )

ddsMat_acute_si3d_ribo <- DESeq(ddsMat_acute_si3d_ribo)
resultsNames(ddsMat_acute_si3d_ribo)
res_acute_si3d_ribo <- results(ddsMat_acute_si3d_ribo, name=resultsNames(ddsMat_acute_si3d_ribo)[2], cooksCutoff=F, independentFiltering=F)


res_acute_si3d_ribo <- lfcShrink(dds = ddsMat_acute_si3d_ribo, coef = resultsNames(ddsMat_acute_si3d_ribo)[2], res = res_acute_si3d_ribo) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)


# count table -------------------------------------------------------------------

upreg <- res_acute_si3d_ribo %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_acute_si3d_ribo %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset="ribo"

all_counts_acute_si3d_ribo <- data.frame(dataset, upreg, downreg)

```






## Visualize acute hypoxic response 
Export all acute changes as a single table

```{r}


sictrl_te <- res_acute_sictrl[,c(1,2,4,7)]
colnames(sictrl_te)[3] ="LFC_te_sictrl"
colnames(sictrl_te)[4] ="padj_te_sictrl"

sictrl_ribo <- res_acute_sictrl_ribo[,c(1,4,7)]
colnames(sictrl_ribo)[2] ="LFC_ribo_sictrl"
colnames(sictrl_ribo)[3] ="padj_ribo_sictrl"

sictrl_rna <- res_acute_sictrl_rna[,c(1,4,7)]
colnames(sictrl_rna)[2] ="LFC_rna_sictrl"
colnames(sictrl_rna)[3] ="padj_rna_sictrl"

si3e_te <- res_acute_si3e[,c(1,4,7)]
colnames(si3e_te)[2] ="LFC_te_si3e"
colnames(si3e_te)[3] ="padj_te_3e"

si3e_ribo <- res_acute_si3e_ribo[,c(1,4,7)]
colnames(si3e_ribo)[2] ="LFC_ribo_si3e"
colnames(si3e_ribo)[3] ="padj_ribo_si3e"

si3e_rna <- res_acute_si3e_rna[,c(1,4,7)]
colnames(si3e_rna)[2] ="LFC_rna_si3e"
colnames(si3e_rna)[3] ="padj_rna_si3e"

si3d_te <- res_acute_si3d[,c(1,4,7)]
colnames(si3d_te)[2] ="LFC_te_si3d"
colnames(si3d_te)[3] ="padj_te_si3d"

si3d_ribo <- res_acute_si3d_ribo[,c(1,4,7)]
colnames(si3d_ribo)[2] ="LFC_ribo_si3d"
colnames(si3d_ribo)[3] ="padj_ribo_si3d"

si3d_rna <- res_acute_si3d_rna[,c(1,4,7)]
colnames(si3d_rna)[2] ="LFC_rna_si3d"
colnames(si3d_rna)[3] ="padj_rna_si3d"

#bind cols ----------------------------------------------------------------------

all_changes <- purrr::reduce(list(sictrl_te,sictrl_ribo,sictrl_rna,si3e_te,si3e_ribo,si3e_rna,si3d_te,si3d_ribo,si3d_rna), dplyr::left_join, by = 'gene_id') %>% 
  as.data.frame()

write_csv(all_changes, here("counts/all_norm_to_hyp_changes_231.csv"))
```



## Visualize acute hypoxic response with histograms

## Acute hypoxic changes

```{r}

sictrl <- all_changes %>% 
  filter(padj_rna_sictrl < 0.05 | padj_ribo_sictrl < 0.05)

sictrl_acute <- ggplot(data = sictrl, aes(x = LFC_rna_sictrl, y = LFC_ribo_sictrl)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote(Log[2]~"sictrl LFC"~"RIBO: hypoxia vs normoxia")) +
  xlab(bquote(Log[2]~"sictrl LFC"~"RNA: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 5)+
  theme_prism()

sictrl_acute <- ggMarginal(sictrl_acute, type = "histogram",
           xparams = list(binwidth = .1, fill = "grey"),
           yparams = list(binwidth = .1, fill = "red")
           )

ggsave(filename = here("plots/plots_231/rna_ribo_sictrl_acute.pdf"), sictrl_acute, width = 4, height = 4)

sictrl_acute
```

# acute hypoxic response with 3e KD


```{r}

sig_sictrl <- all_changes %>% 
  filter(padj_te_sictrl <0.05)

#counts sictrl
sig_sictrl %>% 
  filter(LFC_te_sictrl < 0) %>% 
  filter(padj_te_sictrl < 0.05) %>% 
  nrow()

sig_sictrl %>% 
  filter(LFC_te_sictrl > 0) %>% 
  filter(padj_te_sictrl < 0.05) %>% 
  nrow()

#counts 3eKD
sig_sictrl %>% 
  filter(LFC_te_si3e < 0) %>% 
  filter(padj_te_3e < 0.05) %>% 
  nrow()

sig_sictrl %>% 
  filter(LFC_te_si3e > 0) %>% 
  filter(padj_te_3e < 0.05) %>% 
  nrow()



scatter_acute_sictrl_3e <- ggplot(data = sig_sictrl, aes(x = LFC_te_sictrl, y = LFC_te_si3e)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote(Log[2]~"si-eIF3e LFC"~"TE: hypoxia vs normoxia")) +
  xlab(bquote(Log[2]~"sictrl LFC"~"TE: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 9)+
  theme_prism()

scatter_acute_sictrl_3e <- ggMarginal(scatter_acute_sictrl_3e, type = "histogram",
           xparams = list(binwidth = .1, fill = "grey"),
           yparams = list(binwidth = .1, fill = si3e_col)
           )

scatter_acute_sictrl_3e

ggsave(filename= here("plots/plots_231/acute_sictrl_si3e.pdf"), scatter_acute_sictrl_3e, height=4, width=4)

```



si3d

```{r}

sig_sictrl <- all_changes %>% 
  filter(padj_te_sictrl <0.05)

#counts sictrl
sig_sictrl %>% 
  filter(LFC_te_sictrl < 0) %>% 
  filter(padj_te_sictrl < 0.05) %>% 
  nrow()

sig_sictrl %>% 
  filter(LFC_te_sictrl > 0) %>% 
  filter(padj_te_sictrl < 0.05) %>% 
  nrow()

#counts 3dKD
sig_sictrl %>% 
  filter(LFC_te_si3d < 0) %>% 
  filter(padj_te_si3d < 0.05) %>% 
  nrow()

sig_sictrl %>% 
  filter(LFC_te_si3d > 0) %>% 
  filter(padj_te_si3d < 0.05) %>% 
  nrow()



scatter_acute_sictrl_3d <- ggplot(data = sig_sictrl, aes(x = LFC_te_sictrl, y = LFC_te_si3d)) + 
  ylim(-2,2) + 
  xlim(-2,2) +
  geom_hline(yintercept = 0, color = "grey") + 
  geom_vline(xintercept = 0, color = "grey") +
  geom_point(alpha=.25) +
  ylab(bquote(Log[2]~"si-eIF3d LFC"~"TE: hypoxia vs normoxia")) +
  xlab(bquote(Log[2]~"sictrl LFC"~"TE: hypoxia vs normoxia")) +
    theme_cowplot(font_size = 9)+
  theme_prism()

scatter_acute_sictrl_3d <- ggMarginal(scatter_acute_sictrl_3d, type = "histogram",
           xparams = list(binwidth = .1, fill = "grey"),
           yparams = list(binwidth = .1, fill = si3d_col)
           )

scatter_acute_sictrl_3d

ggsave(filename= here("plots/plots_231/acute_sictrl_si3d.pdf"), scatter_acute_sictrl_3d, height=4, width=4)

```



# Normoxia
##3e
```{r}

#organize metadata
metadata_normox_si3e <- metadata %>% 
  filter(treatment== "si3e"| treatment=="sictrl") %>% 
  filter(condition=="normoxia") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_normox_si3e$library <- relevel(factor(metadata_normox_si3e$library), ref = "rna")
metadata_normox_si3e$treatment <- relevel(factor(metadata_normox_si3e$treatment), ref = "sictrl")

levels(metadata_normox_si3e$library)
levels(metadata_normox_si3e$treatment)

names <- as.character(metadata_normox_si3e$name)

normox_si3e <- all_raw[, colnames(all_raw) %in% names]


ddsMat_normox_si3e <- DESeqDataSetFromMatrix(
  countData = normox_si3e,
  colData = metadata_normox_si3e,
  design = ~treatment + library + treatment:library
  )

ddsMat_normox_si3e <- DESeq(ddsMat_normox_si3e)
resultsNames(ddsMat_normox_si3e)
res_normox_si3e <- results(ddsMat_normox_si3e, name=resultsNames(ddsMat_normox_si3e)[4], cooksCutoff=F, independentFiltering=F)


res_normox_si3e <- lfcShrink(dds = ddsMat_normox_si3e, coef = resultsNames(ddsMat_normox_si3e)[4], res = res_normox_si3e) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)



# count table -------------------------------------------------------------------

upreg <- res_normox_si3e %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_normox_si3e %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset= "normoxia_3e"

all_counts_normox_si3e <- data.frame(dataset, upreg, downreg)
all_counts_normox_si3e

write_csv(res_normox_si3e, here("counts/231/normoxia_si_ctrl_3e.csv"))

```

##3d

```{r}

#organize metadata
metadata_normox_si3d <- metadata %>% 
  filter(treatment== "si3d"| treatment=="sictrl") %>% 
  filter(condition=="normoxia") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_normox_si3d$library <- relevel(factor(metadata_normox_si3d$library), ref = "rna")
metadata_normox_si3d$treatment <- relevel(factor(metadata_normox_si3d$treatment), ref = "sictrl")

levels(metadata_normox_si3d$library)
levels(metadata_normox_si3d$treatment)

names <- as.character(metadata_normox_si3d$name)

normox_si3d <- all_raw[, colnames(all_raw) %in% names]


ddsMat_normox_si3d <- DESeqDataSetFromMatrix(
  countData = normox_si3d,
  colData = metadata_normox_si3d,
  design = ~treatment + library + treatment:library
  )

ddsMat_normox_si3d <- DESeq(ddsMat_normox_si3d)
resultsNames(ddsMat_normox_si3d)
res_normox_si3d <- results(ddsMat_normox_si3d, name=resultsNames(ddsMat_normox_si3d)[4], cooksCutoff=F, independentFiltering=F)


res_normox_si3d <- lfcShrink(dds = ddsMat_normox_si3d, coef = resultsNames(ddsMat_normox_si3d)[4], res = res_normox_si3d) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)



# count table -------------------------------------------------------------------

upreg <- res_normox_si3d %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_normox_si3d %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset= "normoxia_3d"

all_counts_normox_si3d <- data.frame(dataset, upreg, downreg)
all_counts_normox_si3d

write_csv(res_normox_si3d, here("counts/231/normoxia_si_ctrl_3d.csv"))

```


# Hypoxia
##3e
```{r}

#organize metadata
metadata_hypox_si3e <- metadata %>% 
  filter(treatment== "si3e"| treatment=="sictrl") %>% 
  filter(condition=="hypoxia") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_hypox_si3e$library <- relevel(factor(metadata_hypox_si3e$library), ref = "rna")
metadata_hypox_si3e$treatment <- relevel(factor(metadata_hypox_si3e$treatment), ref = "sictrl")

levels(metadata_hypox_si3e$library)
levels(metadata_hypox_si3e$treatment)

names <- as.character(metadata_hypox_si3e$name)

hypox_si3e <- all_raw[, colnames(all_raw) %in% names]


ddsMat_hypox_si3e <- DESeqDataSetFromMatrix(
  countData = hypox_si3e,
  colData = metadata_hypox_si3e,
  design = ~treatment + library + treatment:library
  )

ddsMat_hypox_si3e <- DESeq(ddsMat_hypox_si3e)
resultsNames(ddsMat_hypox_si3e)
res_hypox_si3e <- results(ddsMat_hypox_si3e, name=resultsNames(ddsMat_hypox_si3e)[4], cooksCutoff=F, independentFiltering=F)


res_hypox_si3e <- lfcShrink(dds = ddsMat_hypox_si3e, coef = resultsNames(ddsMat_hypox_si3e)[4], res = res_hypox_si3e) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)



# count table -------------------------------------------------------------------

upreg <- res_hypox_si3e %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_hypox_si3e %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset= "hypoxia_3e"

all_counts_hypox_si3e <- data.frame(dataset, upreg, downreg)
all_counts_hypox_si3e

write_csv(res_hypox_si3e, here("counts/231/hypoxia_si_ctrl_3e.csv"))

```

##3d

```{r}

#organize metadata
metadata_hypox_si3d <- metadata %>% 
  filter(treatment== "si3d"| treatment=="sictrl") %>% 
  filter(condition=="hypoxia") %>% 
  mutate("name" = paste(condition, treatment, rep, library, sep = "_"))

metadata_hypox_si3d$library <- relevel(factor(metadata_hypox_si3d$library), ref = "rna")
metadata_hypox_si3d$treatment <- relevel(factor(metadata_hypox_si3d$treatment), ref = "sictrl")

levels(metadata_hypox_si3d$library)
levels(metadata_hypox_si3d$treatment)

names <- as.character(metadata_hypox_si3d$name)

hypox_si3d <- all_raw[, colnames(all_raw) %in% names]


ddsMat_hypox_si3d <- DESeqDataSetFromMatrix(
  countData = hypox_si3d,
  colData = metadata_hypox_si3d,
  design = ~treatment + library + treatment:library
  )

ddsMat_hypox_si3d <- DESeq(ddsMat_hypox_si3d)
resultsNames(ddsMat_hypox_si3d)
res_hypox_si3d <- results(ddsMat_hypox_si3d, name=resultsNames(ddsMat_hypox_si3d)[4], cooksCutoff=F, independentFiltering=F)


res_hypox_si3d <- lfcShrink(dds = ddsMat_hypox_si3d, coef = resultsNames(ddsMat_hypox_si3d)[4], res = res_hypox_si3d) %>% 
  as.data.frame %>% 
  rownames_to_column(var = "gene_id") %>%
  inner_join(gene_symbol, .)



# count table -------------------------------------------------------------------

upreg <- res_hypox_si3d %>%
  filter(padj < .05 & log2FoldChange >0) %>%
  nrow()

downreg <- res_hypox_si3d %>%
  filter(padj < .05 & log2FoldChange <0) %>%
  nrow()

dataset= "hypoxia_3d"

all_counts_hypox_si3d <- data.frame(dataset, upreg, downreg)
all_counts_hypox_si3d

write_csv(res_hypox_si3d, here("counts/231/hypoxia_si_ctrl_3d.csv"))

```


### Visualize normoxic vs hyoxic response
```{r}
hyp_vs_norm_kds <- rbind(all_counts_normox_si3e, all_counts_normox_si3d,
                         all_counts_hypox_si3e, all_counts_hypox_si3d) %>%
  melt() %>%
  mutate(
    condition = str_extract(dataset, "^(normox|hypox)"),
    sample = str_extract(dataset, "3[ed]")
  )
  
```

# effect of si3e KD norm and hypoxia
```{r}

eif3e_counts <- hyp_vs_norm_kds %>% 
  filter(sample=="3e")

eif3e_bar <- ggbarplot(data = eif3e_counts, x = "condition", y = "value", fill = "variable", palette = c("#2492a6",si3e_col), position = position_dodge(0.75), label = T, title= "eIF3e")+
  ylab("Transcript counts")+
  xlab("")+
  theme_prism()+
  # theme_cowplot(font_size = 12)+
  scale_y_continuous(limits = c(0, 1600))

eif3e_bar

ggsave(here("plots/plots_231/eIF3ebars.pdf"), eif3e_bar, width = 4, height = 4)

```


# effect of si3d KD norm and hypoxia

```{r}

eif3d_counts <- hyp_vs_norm_kds %>% 
  filter(sample=="3d")

eif3d_bar <- ggbarplot(data = eif3d_counts, x = "condition", y = "value", fill = "variable", palette = c("#ffd412",si3d_col), position = position_dodge(0.75), label = T, title= "eIF3d")+
  ylab("Transcript counts")+
  xlab("")+
  theme_prism()+
  # theme_cowplot(font_size = 12)+
  scale_y_continuous(limits = c(0, 1600))

eif3d_bar

ggsave(here("plots/plots_231/eIF3dbars.pdf"), eif3d_bar, width = 4, height = 4)

```



# Do targets overlap with data from MCF7-SIX1s?

pull in mcf7 six q data and label everything by cell line


## Lists for this data
231s

normoxia
```{r}

#3d

#up

mb231_si3d_normoxia_up <- res_normox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mb231_si3d_normoxia_up$gs_name <- rep("mb231_si3d_normoxia_up")


#down

mb231_si3d_normoxia_down <- res_normox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mb231_si3d_normoxia_down$gs_name <- rep("mb231_si3d_normoxia_down")


#3e
#up

mb231_si3e_normoxia_up <- res_normox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mb231_si3e_normoxia_up$gs_name <- rep("mb231_si3e_normoxia_up")

#down

mb231_si3e_normoxia_down <- res_normox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mb231_si3e_normoxia_down$gs_name <- rep("mb231_si3e_normoxia_down")


```


hypoxia
```{r}

#3d

#up

mb231_si3d_hypoxia_up <- res_hypox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mb231_si3d_hypoxia_up$gs_name <- rep("mb231_si3d_hypoxia_up")


#down

mb231_si3d_hypoxia_down <- res_hypox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mb231_si3d_hypoxia_down$gs_name <- rep("mb231_si3d_hypoxia_down")


#3e
#up

mb231_si3e_hypoxia_up <- res_hypox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mb231_si3e_hypoxia_up$gs_name <- rep("mb231_si3e_hypoxia_up")

#down

mb231_si3e_hypoxia_down <- res_hypox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mb231_si3e_hypoxia_down$gs_name <- rep("mb231_si3e_hypoxia_down")


```


Pull in data from mcf7-six1

```{r}

#3d

#up
mcf7six1_res_normox_si3d <-read_csv(here("counts/fig1/normoxia_sictrl_3d.csv"))

mcf7six1_si3d_normoxia_up <- mcf7six1_res_normox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mcf7six1_si3d_normoxia_up$gs_name <- rep("mcf7six1_si3d_normoxia_up")


#down

mcf7six1_si3d_normoxia_down <- mcf7six1_res_normox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mcf7six1_si3d_normoxia_down$gs_name <- rep("mcf7six1_si3d_normoxia_down")


#3e
#up
mcf7six1_res_normox_si3e <-read_csv(here("counts/fig1/normoxia_sictrl_3e.csv"))


mcf7six1_si3e_normoxia_up <- mcf7six1_res_normox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mcf7six1_si3e_normoxia_up$gs_name <- rep("mcf7six1_si3e_normoxia_up")

#down

mcf7six1_si3e_normoxia_down <- mcf7six1_res_normox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mcf7six1_si3e_normoxia_down$gs_name <- rep("mcf7six1_si3e_normoxia_down")


```

Import hypoxia data from figure 3
```{r}

#3d
mcf7six1_res_hypox_si3d <-read_csv(here("counts/fig3/hypoxia_si_ctrl_3d.csv"))


mcf7six1_si3d_hypoxia_up <- mcf7six1_res_hypox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mcf7six1_si3d_hypoxia_up$gs_name <- rep("mcf7six1_si3d_hypoxia_up")


#down

mcf7six1_si3d_hypoxia_down <- mcf7six1_res_hypox_si3d %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mcf7six1_si3d_hypoxia_down$gs_name <- rep("mcf7six1_si3d_hypoxia_down")



#3e
mcf7six1_res_hypox_si3e <-read_csv(here("counts/fig3/hypoxia_si_ctrl_3e.csv"))


mcf7six1_si3e_hypoxia_up <- mcf7six1_res_hypox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

mcf7six1_si3e_hypoxia_up$gs_name <- rep("mcf7six1_si3e_hypoxia_up")

#down

mcf7six1_si3e_hypoxia_down <- mcf7six1_res_hypox_si3e %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

mcf7six1_si3e_hypoxia_down$gs_name <- rep("mcf7six1_si3e_hypoxia_down")


```



## Gene lists for a209 in normoxia and hypoxia

```{r}

#normoxia a209


res_normox_a209 <-read_csv(here("counts/fig6/normoxia_dmso_a209.csv"))


#up
a209_normoxia_up <- res_normox_a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

a209_normoxia_up$gs_name <- rep("mcf7six1_a209_normoxia_up")

#down

a209_normoxia_down <- res_normox_a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

a209_normoxia_down$gs_name <- rep("mcf7six1_a209_normoxia_down")


#hypoxia a209
res_hypox_a209 <-read_csv(here("counts/fig6/hypoxia_dmso_a209.csv"))

#up

a209_hypoxia_up <- res_hypox_a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange > 0) %>%
dplyr::select(symbol)

a209_hypoxia_up$gs_name <- rep("mcf7six1_a209_hypoxia_up")

#down

a209_hypoxia_down <- res_hypox_a209 %>% 
  as.data.frame() %>% 
  filter(padj < .05 & log2FoldChange < 0) %>%
dplyr::select(symbol)

a209_hypoxia_down$gs_name <- rep("mcf7six1_a209_hypoxia_down")


```

# Overlap of 3e targets in MCF7SIX1 and MD-MBA-231


# Odds ratios for a209 3e and 3d overlap in normoxia

The following analysis determines the number of genes that overlap in each condition and the statistical significance (p-value) of their overlap (odds ratio).


Create a matrix that compares each condition for gene overlap analysis.
```{r}
all_lists <- bind_rows(mcf7six1_si3e_normoxia_up, mcf7six1_si3d_normoxia_up, a209_normoxia_up, 
                       mcf7six1_si3e_normoxia_down, mcf7six1_si3d_normoxia_down, a209_normoxia_down, 
                       mcf7six1_si3e_hypoxia_up, mcf7six1_si3d_hypoxia_up, a209_hypoxia_up, 
                       mcf7six1_si3e_hypoxia_down, mcf7six1_si3d_hypoxia_down, a209_hypoxia_down,
                       mb231_si3e_normoxia_up, mb231_si3d_normoxia_up, 
                       mb231_si3e_normoxia_down, mb231_si3d_normoxia_down, 
                       mb231_si3e_hypoxia_up, mb231_si3d_hypoxia_up, 
                       mb231_si3e_hypoxia_down, mb231_si3d_hypoxia_down) %>% 
extract(gs_name,
        into = c("cell_line", "treatment", "condition", "direction"),
        regex = "([^_]+)_([^_]+)_([^_]+)_([^_]+)")
```


### get number of expressed genes for universe value for mcf7six1 and md-mba-231
```{r}
all_3e_symbols_expressed <- rbind(
res_normox_si3e,
res_hypox_si3e,
mcf7six1_res_hypox_si3e,
mcf7six1_res_hypox_si3e)

universe <- unique(all_3e_symbols_expressed$symbol)

```






### Normoxia analysis
```{r}

norm_matrix <- all_lists %>%
  filter(condition=="normoxia") %>% 
  dplyr::select(-condition)


norm_matrix_comb <- norm_matrix %>% 
  tidyr::unite("description", 2:4, sep=', ')

gom.allsamples <- split(norm_matrix_comb$symbol, norm_matrix_comb$description)

#double check genome size here

gom.all <- newGOM(gom.allsamples, gom.allsamples, genome.size = 12495)

norm_all.odds.matrix <- getMatrix(gom.all, "odds")
norm_all.pval.matrix <- getMatrix(gom.all, "pval")


```








# plot odds ratios as bars with p values
```{r}

#merge odds and pval matricies

odds <- norm_all.odds.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:11, names_to = "comparison", values_to="odds_value")

pval <- norm_all.pval.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:11, names_to = "comparison", values_to="pval_value")


odds_pval <- left_join(odds, pval, by=c("sample", "comparison")) %>% 
  tidyr::unite("comparison", 1:2, sep="-") 
```




```{r}
odds_pval_3e<-odds_pval %>% 
  filter(comparison == "mcf7six1, si3e, down-mb231, si3e, down"| 
         comparison =="mcf7six1, si3e, up-mb231, si3e, up"|
           comparison == "mcf7six1, si3d, down-mb231, si3d, down"|
           comparison == "mcf7six1, si3d, up-mb231, si3d, up")


ggpubr::ggbarplot(odds_pval_3e, x="comparison", y="odds_value", label = formatC(odds_pval_3e$pval_value, format = "e", digits = 2), title="Overlap of 3e/3d targets in normoxia", palette = c(si3d_col, si3e_col), ylim=c(0,16))+
  geom_hline(aes(yintercept=1))+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))



ggsave(here("plots/plots_231/normoverlap.pdf"), width=4, height=8)

```

### Hypoxia

```{r}

hyp_matrix <- all_lists %>%
  filter(condition=="hypoxia") %>% 
  dplyr::select(-condition)


hyp_matrix_comb <- hyp_matrix %>% 
  tidyr::unite("description", 2:4, sep=', ')

gom.allsamples <- split(hyp_matrix_comb$symbol, hyp_matrix_comb$description)

#double check genome size here

gom.all <- newGOM(gom.allsamples, gom.allsamples, genome.size = 12495)

hyp_all.odds.matrix <- getMatrix(gom.all, "odds")
hyp_all.pval.matrix <- getMatrix(gom.all, "pval")


```








# plot odds ratios as bars with p values
```{r}

#merge odds and pval matricies

odds <- hyp_all.odds.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:11, names_to = "comparison", values_to="odds_value")

pval <- hyp_all.pval.matrix %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  pivot_longer(2:11, names_to = "comparison", values_to="pval_value")


odds_pval <- left_join(odds, pval, by=c("sample", "comparison")) %>% 
  tidyr::unite("comparison", 1:2, sep="-") 
```




```{r}
odds_pval_3e<-odds_pval %>% 
  filter(comparison == "mcf7six1, si3e, down-mb231, si3e, down"| 
         comparison =="mcf7six1, si3e, up-mb231, si3e, up"|
           comparison == "mcf7six1, si3d, down-mb231, si3d, down"|
           comparison == "mcf7six1, si3d, up-mb231, si3d, up")


ggpubr::ggbarplot(odds_pval_3e, x="comparison", y="odds_value", label = formatC(odds_pval_3e$pval_value, format = "e", digits = 2), title="Overlap of 3e/3d targets in hypoxia", palette = c(si3d_col, si3e_col), ylim=c(0,16))+
  geom_hline(aes(yintercept=1))+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))



ggsave(here("plots/plots_231/hypoverlap.pdf"), width=4, height=8)

```



# Venn diagrams

```{r}

all_lists <- bind_rows(mcf7six1_si3e_normoxia_up, mcf7six1_si3d_normoxia_up, a209_normoxia_up, 
                       mcf7six1_si3e_normoxia_down, mcf7six1_si3d_normoxia_down, a209_normoxia_down, 
                       mcf7six1_si3e_hypoxia_up, mcf7six1_si3d_hypoxia_up, a209_hypoxia_up, 
                       mcf7six1_si3e_hypoxia_down, mcf7six1_si3d_hypoxia_down, a209_hypoxia_down,
                       mb231_si3e_normoxia_up, mb231_si3d_normoxia_up, 
                       mb231_si3e_normoxia_down, mb231_si3d_normoxia_down, 
                       mb231_si3e_hypoxia_up, mb231_si3d_hypoxia_up, 
                       mb231_si3e_hypoxia_down, mb231_si3d_hypoxia_down) %>% 
extract(gs_name,
        into = c("cell_line", "treatment", "condition", "direction"),
        regex = "([^_]+)_([^_]+)_([^_]+)_([^_]+)")

```

Normoxia
## Venn diagram of 3e/d gene overlaps in normoxia
```{r compare genes}
mcf7six1_si3e_normoxia_up <- mcf7six1_si3e_normoxia_up %>% pull(symbol)
mcf7six1_si3d_normoxia_up <- mcf7six1_si3d_normoxia_up %>% pull(symbol)

mcf7six1_si3e_normoxia_down <- mcf7six1_si3e_normoxia_down %>% pull(symbol)
mcf7six1_si3d_normoxia_down <- mcf7six1_si3d_normoxia_down %>% pull(symbol)

mb231_si3e_normoxia_up <- mb231_si3e_normoxia_up %>% pull(symbol)
mb231_si3d_normoxia_up <- mb231_si3d_normoxia_up %>% pull(symbol)

mb231_si3e_normoxia_down <- mb231_si3e_normoxia_down %>% pull(symbol)
mb231_si3d_normoxia_down <- mb231_si3d_normoxia_down %>% pull(symbol)

#3e

#up

norm_te_up_3e <- list(
  MCF7SIX1 = mcf7six1_si3e_normoxia_up, 
  MDA_MB_231 = mb231_si3e_normoxia_up
)

ggvenn(
  norm_te_up_3e,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3e_col, "blue")
) +
  ggtitle("Up with 3e KD - Normoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3e_norm_vennup.pdf"))


#down

norm_te_down_3e <- list(
  MCF7SIX1 = mcf7six1_si3e_normoxia_down, 
  MDA_MB_231 = mb231_si3e_normoxia_down
)

ggvenn(
  norm_te_down_3e,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3e_col, "blue")
) +
  ggtitle("down with 3e KD - Normoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3e_norm_venndown.pdf"))






#3d

#up

norm_te_up_3d <- list(
  MCF7SIX1 = mcf7six1_si3d_normoxia_up, 
  MDA_MB_231 = mb231_si3d_normoxia_up
)

ggvenn(
  norm_te_up_3d,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3d_col, "red"))+
  ggtitle("Up with 3d KD - Normoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3d_norm_vennup.pdf"))


#down

norm_te_down_3d <- list(
  MCF7SIX1 = mcf7six1_si3d_normoxia_down, 
  MDA_MB_231 = mb231_si3d_normoxia_down
)

ggvenn(
  norm_te_down_3d,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3d_col, "red")
) +
  ggtitle("down with 3d KD - Normoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3d_norm_venndown.pdf"))


```




hypoxia
## Venn diagram of 3e/d gene overlaps in hypoxia
```{r compare genes}
mcf7six1_si3e_hypoxia_up <- mcf7six1_si3e_hypoxia_up %>% pull(symbol)
mcf7six1_si3d_hypoxia_up <- mcf7six1_si3d_hypoxia_up %>% pull(symbol)

mcf7six1_si3e_hypoxia_down <- mcf7six1_si3e_hypoxia_down %>% pull(symbol)
mcf7six1_si3d_hypoxia_down <- mcf7six1_si3d_hypoxia_down %>% pull(symbol)

mb231_si3e_hypoxia_up <- mb231_si3e_hypoxia_up %>% pull(symbol)
mb231_si3d_hypoxia_up <- mb231_si3d_hypoxia_up %>% pull(symbol)

mb231_si3e_hypoxia_down <- mb231_si3e_hypoxia_down %>% pull(symbol)
mb231_si3d_hypoxia_down <- mb231_si3d_hypoxia_down %>% pull(symbol)

#3e

#up

hyp_te_up_3e <- list(
  MCF7SIX1 = mcf7six1_si3e_hypoxia_up, 
  MDA_MB_231 = mb231_si3e_hypoxia_up
)

ggvenn(
  hyp_te_up_3e,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3e_col, "blue")
) +
  ggtitle("Up with 3e KD - hypoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3e_hyp_vennup.pdf"))


#down

hyp_te_down_3e <- list(
  MCF7SIX1 = mcf7six1_si3e_hypoxia_down, 
  MDA_MB_231 = mb231_si3e_hypoxia_down
)

ggvenn(
  hyp_te_down_3e,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3e_col, "blue")
) +
  ggtitle("down with 3e KD - hypoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3e_hyp_venndown.pdf"))






#3d

#up

hyp_te_up_3d <- list(
  MCF7SIX1 = mcf7six1_si3d_hypoxia_up, 
  MDA_MB_231 = mb231_si3d_hypoxia_up
)

ggvenn(
  hyp_te_up_3d,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3d_col, "red"))+
  ggtitle("Up with 3d KD - hypoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3d_hyp_vennup.pdf"))


#down

hyp_te_down_3d <- list(
  MCF7SIX1 = mcf7six1_si3d_hypoxia_down, 
  MDA_MB_231 = mb231_si3d_hypoxia_down
)

ggvenn(
  hyp_te_down_3d,
  auto_scale = TRUE,
  text_size = 6,
  fill_color = c(si3d_col, "red")
) +
  ggtitle("down with 3d KD - hypoxia") +
  theme(
    plot.title = element_text(hjust = 0.5, size = 18) # center + bigger font
  )

ggsave(width = 6, height = 6, filename = here("plots/plots_231/3d_hyp_venndown.pdf"))


```





# Acute hypoxic response

Genes collapsed by 3e KD in acute hypoxic response:
```{r}
mb231_acute_3e <- all_changes %>%
  filter(padj_te_sictrl < 0.05 & padj_te_3e > 0.05)

nrow(mb231_acute_3e)

mb231_acute_3d <- all_changes %>%
  filter(padj_te_sictrl < 0.05 & padj_te_si3d > 0.05)

nrow(mb231_acute_3d)
```


Gather same lists from MCF7SIX1 data
```{r}

mcf7six1_acute <- read_csv(here("counts/fig3/all_norm_to_hyp_changes.csv"))

mcf7six1_acute_3e <- mcf7six1_acute %>%
  filter(padj_te_sictrl < 0.05 & padj_te_3e > 0.05)

nrow(mcf7six1_acute_3e)



mcf7six1_acute_3d <- mcf7six1_acute %>%
  filter(padj_te_sictrl < 0.05 & padj_te_si3d > 0.05)

nrow(mcf7six1_acute_3d)


```

3e overlap between cell lines
```{r}
acute_3e_sig_down_MCF7SIX1 <- mcf7six1_acute_3e %>% 
  filter(LFC_te_sictrl > 0) %>% 
  pull(symbol)

acute_3e_sig_up_MCF7SIX1 <- mcf7six1_acute_3e %>% 
  filter(LFC_te_sictrl < 0) %>% 
  pull(symbol)


acute_3e_sig_down_MB231 <- mb231_acute_3e %>% 
  filter(LFC_te_sictrl > 0) %>% 
  pull(symbol)

acute_3e_sig_up_MB231 <- mb231_acute_3e %>% 
  filter(LFC_te_sictrl < 0) %>% 
  pull(symbol)



acute_down <- list(
  "MCF7-SIX1" = acute_3e_sig_down_MCF7SIX1,
  "MDA-MB-231" = acute_3e_sig_down_MB231
)


ggvenn(acute_down)

view(intersect(acute_3e_sig_down_MCF7SIX1, acute_3e_sig_down_MB231))

ggsave(here("plots/plots_231/acute_down.pdf"))






acute_up <- list(
  "MCF7-SIX1" = acute_3e_sig_up_MCF7SIX1,
  "MDA-MB-231" = acute_3e_sig_up_MB231
)


ggvenn(acute_up)

view(intersect(acute_3e_sig_up_MCF7SIX1, acute_3e_sig_up_MB231))

ggsave(here("plots/plots_231/acute_up.pdf"))


```


3d overlap between cell lines
```{r}
acute_3d_sig_down_MCF7SIX1 <- mcf7six1_acute_3d %>% 
  filter(LFC_te_sictrl > 0) %>% 
  pull(symbol)

acute_3d_sig_up_MCF7SIX1 <- mcf7six1_acute_3d %>% 
  filter(LFC_te_sictrl < 0) %>% 
  pull(symbol)


acute_3d_sig_down_MB231 <- mb231_acute_3d %>% 
  filter(LFC_te_sictrl > 0) %>% 
  pull(symbol)

acute_3d_sig_up_MB231 <- mb231_acute_3d %>% 
  filter(LFC_te_sictrl < 0) %>% 
  pull(symbol)



acute_down <- list(
  "MCF7-SIX1" = acute_3d_sig_down_MCF7SIX1,
  "MDA-MB-231" = acute_3d_sig_down_MB231
)


ggvenn(acute_down)

view(intersect(acute_3d_sig_down_MCF7SIX1, acute_3d_sig_down_MB231))

ggsave(here("plots/plots_231/acute_down_3d.pdf"))






acute_up <- list(
  "MCF7-SIX1" = acute_3d_sig_up_MCF7SIX1,
  "MDA-MB-231" = acute_3d_sig_up_MB231
)


ggvenn(acute_up)

view(intersect(acute_3d_sig_up_MCF7SIX1, acute_3d_sig_up_MB231))

ggsave(here("plots/plots_231/acute_up_d.pdf"))


```




